{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Things to do before...\n",
    "## Align genomes with mauve\n",
    "## Extract orthologs with ID 35% Coverage 51%"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "import numpy as np\n",
    "import os\n",
    "from os import listdir,walk\n",
    "from os.path import isfile, join\n",
    "path_to_files='../mauve_alignments/'\n",
    "path_out=\"/home/maeperez/projects/def-bacc/maeperez/Clams/VesicSymb_Evolution/dowstream_analyses_and_data/\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read and parse ortholog file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "######## read and parse ortholog file #########\n",
    "genomes=['SUP05','Bathy','R.magnifica','R.fausta','R.pacifica','R.phaseoliformis','R.pliocardia','R.rectimargo','R.southwardae','V.diagonalis','V.extenta','V.soyoae1','V.soyoae2','V.gigas1','V.gigas2','V.okutanii','V.marissinica']\n",
    "locus_tag_prefixes=['MS2017','SP60','HUE58','Rmag','Rpac','Rpha','Rpli','Rrec','Rsou','Vdia','Vext','Vsoy1','Vsoy2','Vgig1','Vgig2','COSY','Vmar']\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "#pfx=dict(zip(locus_tag_prefixes,genomes))\n",
    "idx_dict=dict(zip(range(len(genomes)),genomes))\n",
    "\n",
    "ALL_raw=pd.read_csv(path_to_files+'ALL_id35cov51.orthologs',sep='\\t',header=None,names=genomes)\n",
    "ALL=pd.DataFrame([],columns=genomes, index=ALL_raw.index)\n",
    "\n",
    "for idx in ALL_raw.index:\n",
    "    for col in list(ALL_raw):\n",
    "        if type(ALL_raw.loc[idx][col])!=float:\n",
    "            column=ALL_raw.loc[idx][col].split(':')[0]\n",
    "            value=ALL_raw.loc[idx][col].split(':')[1]\n",
    "            ALL.loc[idx][idx_dict[int(column)]]=value\n",
    "            \n",
    "\n",
    "ALL.to_csv(path_to_files+'Orthology_id35cov51.txt',header=True,sep='\\t')  \n",
    "CORE.to_csv(path_to_files+'CoreOrthology_id35cov51.txt',header=True,sep='\\t') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/GenBank/Scanner.py:215: BiopythonParserWarning: Over indented source feature?\n",
      "  BiopythonParserWarning,\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:1398: BiopythonParserWarning: Expected sequence length 1021329, found 1023923 (Contig_1).\n",
      "  BiopythonParserWarning,\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "446\n",
      "727\n",
      "6084\n"
     ]
    }
   ],
   "source": [
    "## remove pseudogenes in refs\n",
    "goodfiles={'Bathy':'../genomes/Bathy_complete_with_CRISPRannot.gbk',\\\n",
    "'R.fausta':'../genomes/R.fausta.gbk',\\\n",
    "'R.magnifica':'../genomes/R.magnifica.gbk',\\\n",
    "'R.pacifica':'../genomes/R.pacifica_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.phaseoliformis':'../genomes/R.phaseoliformis_8_contigs_fully_annotated.gbk',\\\n",
    "'R.pliocardia':'../genomes/R.pliocardia_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.rectimargo':'../genomes/R.rectimargo_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.southwardae':'../genomes/R.southwardae_39_contigs_fully_annotated.gbk',\\\n",
    "'SUP05':'../genomes/SUP05.gbk',\\\n",
    "'V.diagonalis':'../genomes/V.diagonalis_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.extenta':'../genomes/V.extenta_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas1':'../genomes/V.gigas1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas2':'../genomes/V.gigas2_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.marissinica':'../genomes/V.marissinica_withlocustags.gbk',\\\n",
    "'V.okutanii':'../genomes/V.okutanii_with_locus_tags.gbk',\\\n",
    "'V.soyoae1':'../genomes/V.soyoae1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.soyoae2':'../genomes/V.soyoae2_one_contig_circular_fully_annotated.gbk'}\n",
    "pseudogenes=[]\n",
    "for sample in goodfiles.keys():\n",
    "    for record in SeqIO.parse(goodfiles[sample],'genbank'):\n",
    "            for feature in record.features:\n",
    "                if feature.type in ['CDS','gene']:\n",
    "                    locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                    if 'pseudo' in feature.qualifiers: #verify if pseudogene\n",
    "                        pseudogenes+=[locus_tag]\n",
    "                        \n",
    "print(len(pseudogenes))\n",
    "\n",
    "all_pseudogenes=pseudogenes\n",
    "\n",
    "ALL[ALL.isin(pseudogenes)] = np.nan\n",
    "    \n",
    "    \n",
    "ALL['Count']=ALL[genomes].count(axis=1)\n",
    "# ALL.to_csv('Orthology_id35cov51.txt',header=True,sep='\\t')  \n",
    "\n",
    "CORE=ALL[ALL['Count']==17][genomes]\n",
    "\n",
    "# CORE.to_csv('CoreOrthology_id35cov51.txt',header=True,sep='\\t') \n",
    "# pseudogenes\n",
    "print(len(CORE)) # 727 (739 including refs pseudo)\n",
    "print(len(ALL)) # 6084"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           SUP05        Bathy R.magnifica       R.fausta   R.pacifica  \\\n",
      "165   SP60_01355  MS2017_R027  Rmag_R0013  HUE58_RS01145  Rpac_rna_15   \n",
      "345   SP60_03285  MS2017_R007  Rmag_R0035  HUE58_RS05070  Rpac_rna_22   \n",
      "5307         NaN          NaN         NaN            NaN          NaN   \n",
      "\n",
      "     R.phaseoliformis R.pliocardia R.rectimargo R.southwardae V.diagonalis  \\\n",
      "165       Rpha_rna_30  Rpli_rna_12   Rrec_rna_1   Rsou_rna_24  Vdia_rna_25   \n",
      "345       Rpha_rna_29  Rpli_rna_19  Rrec_rna_12   Rsou_rna_33   Vdia_rna_5   \n",
      "5307              NaN          NaN  Rrec_rna_28           NaN          NaN   \n",
      "\n",
      "        V.extenta     V.soyoae1     V.soyoae2      V.gigas1      V.gigas2  \\\n",
      "165   Vext_rna_26  Vsoy1_rna_14  Vsoy2_rna_25  Vgig1_rna_23  Vgig2_rna_19   \n",
      "345    Vext_rna_6  Vsoy1_rna_33   Vsoy2_rna_7  Vgig1_rna_33  Vgig2_rna_29   \n",
      "5307          NaN           NaN           NaN           NaN           NaN   \n",
      "\n",
      "      V.okutanii V.marissinica  Count  \n",
      "165          NaN    Vmar_R0013     16  \n",
      "345   COSY_R0032    Vmar_R0033     17  \n",
      "5307         NaN           NaN      1  \n",
      "3877\n",
      "668\n",
      "3947\n"
     ]
    }
   ],
   "source": [
    "##### remove pseudogenes in our genomes #### (CDS+RNA)\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "## CDS\n",
    "\n",
    "with open(path_to_files+'pseudogenes.txt','r') as f:\n",
    "    lines=f.read().splitlines()\n",
    "\n",
    "# print(len(lines))\n",
    "\n",
    "## RNA\n",
    "\n",
    "gr=pd.read_csv(path_to_files+'Orthogroups_fully_annotated.txt',header=0,sep='\\t', index_col=0)\n",
    "gr['B_thermophilus_symbiont'][0].split(',')\n",
    "l=np.array(gr[gr.apply(lambda col: col.str.contains('_rna_', na=False), axis=1)].values.flatten())\n",
    "l=l[~pd.isna(l)]\n",
    "rna_pseudo=[v[1].replace('_pseudo','') for v in enumerate(l) if 'pseudo' in v[1]]\n",
    "# print(rna_pseudo)\n",
    "lines+=rna_pseudo\n",
    "\n",
    "\n",
    "print(ALL[ALL.apply(lambda r: r.isin(rna_pseudo).any(), axis=1)])\n",
    "# print(len(lines))\n",
    "\n",
    "ALL[ALL.isin(lines)] = np.nan\n",
    "    \n",
    "    \n",
    "ALL['Count']=ALL[genomes].count(axis=1)\n",
    "\n",
    "ALL=ALL[ALL['Count']>0][genomes]\n",
    "ALL['Count']=ALL[genomes].count(axis=1)\n",
    "CORE=ALL[ALL['Count']==17][genomes]\n",
    "\n",
    "\n",
    "\n",
    "all_pseudogenes+=lines\n",
    "\n",
    "print(len(ALL)) #3877 # 3878 with pseudo RNA\n",
    "print(len(CORE)) #668 #669 with pseudo RNA\n",
    "print(len(all_pseudogenes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "ALL.to_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',header=True,sep='\\t')\n",
    "ALL.to_csv(path_out+'Orthology_id35cov51_nopseudo.txt',header=True,sep='\\t')\n",
    "CORE.to_csv(path_to_files+'CoreOrthology_id35cov51_nopseudo.txt',header=True,sep='\\t') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(path_to_files+'ALL_pseudgenes.txt','w') as fa:\n",
    "    fa.write('\\n'.join(all_pseudogenes))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn diagram"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "## no pseudos\n",
    "#http://www.interactivenn.net/\n",
    "df=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',header=0,index_col=0,sep='\\t')\n",
    "\n",
    "df['gene_number']=df.index.values\n",
    "# print( df.columns)\n",
    "dic={}\n",
    "dic['FL']=['Bathy','SUP05']\n",
    "dic['Ruthia']=['R.magnifica','R.fausta', 'R.pacifica','R.phaseoliformis', 'R.pliocardia', 'R.rectimargo', 'R.southwardae']\n",
    "dic['Gigas']=['V.diagonalis', 'V.extenta', 'V.soyoae1','V.soyoae2', 'V.gigas1','V.gigas2','V.okutanii','V.marissinica']\n",
    "\n",
    "fa=open(path_to_files+'input_ivenn_id35cov51_nopseudo.txt','w')\n",
    "fa.close()\n",
    "\n",
    "for k,v in dic.items():\n",
    "    d=df[['gene_number']+v]\n",
    "\n",
    "    pan=df[v].dropna(how='all',axis=0).index\n",
    "    pan_entry=k+':'+','.join([str(v) for v in pan])+';\\n'\n",
    "\n",
    "    core=df[v].dropna(how='any',axis=0).index\n",
    "    core_entry=k+'_core'+':'+','.join([str(v) for v in core])+';\\n'\n",
    "\n",
    "    fa=open(path_to_files+'input_ivenn_id35cov51_nopseudo.txt','a')\n",
    "    fa.write(pan_entry)\n",
    "    fa.write(core_entry)\n",
    "    fa.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tot_FL = 2502 tot_Ruthia = 2376 tot_Gigas = 1096\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[['[FL]', 1336],\n",
       " ['[Ruthia]', 1202],\n",
       " ['[Gigas]', 120],\n",
       " ['[Ruthia] and [Gigas]', 13],\n",
       " ['[Gigas] and [Gigas_core]', 5],\n",
       " ['[FL] and [FL_core]', 36],\n",
       " ['[FL] and [Ruthia]', 94],\n",
       " ['[Ruthia] and [Ruthia_core]', 3],\n",
       " ['[FL] and [Gigas] and [Gigas_core]', 4],\n",
       " ['[FL] and [FL_core] and [Gigas] and [Gigas_core]', 1],\n",
       " ['[FL] and [Ruthia] and [Gigas]', 14],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Gigas]', 33],\n",
       " ['[FL] and [Ruthia] and [Gigas] and [Gigas_core]', 21],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Gigas] and [Gigas_core]', 77],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core] and [Gigas]', 5],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core] and [Gigas]', 49],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]',\n",
       "  669],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]', 52],\n",
       " ['[Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]', 19],\n",
       " ['[Ruthia] and [Gigas] and [Gigas_core]', 14],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core]', 7],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core]', 24],\n",
       " ['[FL] and [FL_core] and [Ruthia]', 80]]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Parse ivenn data\n",
    "with open(path_to_files+'ivenn_id35cov51_nopseudo.txt') as f :\n",
    "    lines=f.read().splitlines()\n",
    "    lines=[[line.split(': ')[0],len(line.split(': ')[1].split(','))] for line in lines]\n",
    "lines\n",
    "tot_FL=sum([category[1] for category in lines if 'FL' in category[0]])\n",
    "tot_Ruthia=sum([category[1] for category in lines if 'Ruthia' in category[0]])\n",
    "tot_Gigas=sum([category[1] for category in lines if 'Gigas' in category[0]])\n",
    "print('tot_FL = '+str(tot_FL),'tot_Ruthia = '+str(tot_Ruthia),'tot_Gigas = '+str(tot_Gigas))\n",
    "lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SUP05                  SP60_06775\n",
       "Bathy                 MS2017_1570\n",
       "R.magnifica                   NaN\n",
       "R.fausta                      NaN\n",
       "R.pacifica                    NaN\n",
       "R.phaseoliformis              NaN\n",
       "R.pliocardia                  NaN\n",
       "R.rectimargo                  NaN\n",
       "R.southwardae                 NaN\n",
       "V.diagonalis         Vdia_peg_320\n",
       "V.extenta            Vext_peg_312\n",
       "V.soyoae1           Vsoy1_peg_685\n",
       "V.soyoae2           Vsoy2_peg_885\n",
       "V.gigas1            Vgig1_peg_274\n",
       "V.gigas2            Vgig2_peg_162\n",
       "V.okutanii              COSY_0649\n",
       "V.marissinica           Vmar_0657\n",
       "Name: 782, dtype: object"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# including pseudos\n",
    "\n",
    "#http://www.interactivenn.net/\n",
    "df=pd.read_csv(path_to_files+'Orthology_id35cov51.txt',header=0,index_col=0,sep='\\t')\n",
    "df.iloc[782]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "dic={}\n",
    "dic['FL']=['Bathy','SUP05']\n",
    "dic['Ruthia']=['R.magnifica','R.fausta', 'R.pacifica','R.phaseoliformis', 'R.pliocardia', 'R.rectimargo', 'R.southwardae']\n",
    "dic['Gigas']=['V.diagonalis', 'V.extenta', 'V.soyoae1','V.soyoae2', 'V.gigas1','V.gigas2','V.okutanii','V.marissinica']\n",
    "\n",
    "fa=open(path_to_files+'input_ivenn_id35cov51.txt','w')\n",
    "fa.close()\n",
    "\n",
    "for k,v in dic.items():\n",
    "\n",
    "    pan=df[v].dropna(how='all',axis=0).index\n",
    "    pan_entry=k+':'+','.join([str(v) for v in pan])+';\\n'\n",
    "\n",
    "    core=df[v].dropna(how='any',axis=0).index\n",
    "    core_entry=k+'_core'+':'+','.join([str(v) for v in core])+';\\n'\n",
    "\n",
    "    fa=open(path_to_files+'input_ivenn_id35cov51.txt','a')\n",
    "    fa.write(pan_entry)\n",
    "    fa.write(core_entry)\n",
    "    fa.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tot_FL = 2805 tot_Ruthia = 4219 tot_Gigas = 1244\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[['[FL]', 1593],\n",
       " ['[Ruthia]', 2966],\n",
       " ['[Gigas]', 229],\n",
       " ['[Ruthia] and [Gigas]', 36],\n",
       " ['[Gigas] and [Gigas_core]', 6],\n",
       " ['[FL] and [FL_core]', 33],\n",
       " ['[FL] and [Ruthia]', 112],\n",
       " ['[Ruthia] and [Ruthia_core]', 6],\n",
       " ['[FL] and [Gigas] and [Gigas_core]', 3],\n",
       " ['[FL] and [FL_core] and [Gigas] and [Gigas_core]', 1],\n",
       " ['[FL] and [Ruthia] and [Gigas]', 4],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Gigas]', 16],\n",
       " ['[FL] and [Ruthia] and [Gigas] and [Gigas_core]', 21],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Gigas] and [Gigas_core]', 59],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core] and [Gigas]', 4],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core] and [Gigas]', 28],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]',\n",
       "  747],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]', 54],\n",
       " ['[Ruthia] and [Ruthia_core] and [Gigas] and [Gigas_core]', 18],\n",
       " ['[Ruthia] and [Gigas] and [Gigas_core]', 18],\n",
       " ['[FL] and [Ruthia] and [Ruthia_core]', 11],\n",
       " ['[FL] and [FL_core] and [Ruthia] and [Ruthia_core]', 34],\n",
       " ['[FL] and [FL_core] and [Ruthia]', 85]]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Parse ivenn data\n",
    "with open(path_to_files+'ivenn_id35cov51.txt') as f :\n",
    "    lines=f.read().splitlines()\n",
    "    lines=[[line.split(': ')[0],len(line.split(': ')[1].split(','))] for line in lines]\n",
    "lines\n",
    "tot_FL=sum([category[1] for category in lines if 'FL' in category[0]])\n",
    "tot_Ruthia=sum([category[1] for category in lines if 'Ruthia' in category[0]])\n",
    "tot_Gigas=sum([category[1] for category in lines if 'Gigas' in category[0]])\n",
    "print('tot_FL = '+str(tot_FL),'tot_Ruthia = '+str(tot_Ruthia),'tot_Gigas = '+str(tot_Gigas))\n",
    "lines"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extract core sequences and align them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:27: BiopythonExperimentalWarning: Bio.codonalign is an experimental module which may undergo significant changes prior to its future official release.\n",
      "  BiopythonExperimentalWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "81 Rmag_0392\n",
      "3825\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "No records found in handle",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-607cde920987>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     96\u001b[0m     \u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfaa\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchild\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdin\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"fasta\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m     \u001b[0mchild\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdin\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m     \u001b[0malign\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mAlignIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchild\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"clustal\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0malphabet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mGapped\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mIUPAC\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextended_protein\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     99\u001b[0m     \u001b[0maln\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMultipleSeqAlignment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msorted\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malign\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m     \u001b[0mcodon_aln\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbuild\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maln\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfna\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/virtualenv/py36/lib/python3.6/site-packages/Bio/AlignIO/__init__.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(handle, format, seq_count, alphabet)\u001b[0m\n\u001b[1;32m    449\u001b[0m         \u001b[0mfirst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    450\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mfirst\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 451\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"No records found in handle\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    452\u001b[0m     \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    453\u001b[0m         \u001b[0msecond\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0miterator\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: No records found in handle"
     ]
    }
   ],
   "source": [
    "\n",
    "########### CORE: retreive sequences from GBK, align aa, backtranslate and output fna; 715 sequences ##########\n",
    "from Bio import AlignIO\n",
    "from Bio.codonalign import build\n",
    "from Bio.Alphabet import Gapped,IUPAC\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Alphabet import generic_dna, generic_protein\n",
    "from Bio.Align.Applications import MuscleCommandline,ClustalwCommandline\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "# os.mkdir('./fna')\n",
    "# os.mkdir('./faa')\n",
    "\n",
    "\n",
    "goodfiles={'Bathy':'../genomes/Bathy_complete_with_CRISPRannot.gbk',\\\n",
    "'R.fausta':'../genomes/R.fausta.gbk',\\\n",
    "'R.magnifica':'../genomes/R.magnifica.gbk',\\\n",
    "'R.pacifica':'../genomes/R.pacifica_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.phaseoliformis':'../genomes/R.phaseoliformis_8_contigs_fully_annotated.gbk',\\\n",
    "'R.pliocardia':'../genomes/R.pliocardia_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.rectimargo':'../genomes/R.rectimargo_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.southwardae':'../genomes/R.southwardae_39_contigs_fully_annotated.gbk',\\\n",
    "'SUP05':'../genomes/SUP05.gbk',\\\n",
    "'V.diagonalis':'../genomes/V.diagonalis_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.extenta':'../genomes/V.extenta_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas1':'../genomes/V.gigas1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas2':'../genomes/V.gigas2_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.marissinica':'../genomes/V.marissinica_withlocustags.gbk',\\\n",
    "'V.okutanii':'../genomes/V.okutanii_with_locus_tags.gbk',\\\n",
    "'V.soyoae1':'../genomes/V.soyoae1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.soyoae2':'../genomes/V.soyoae2_one_contig_circular_fully_annotated.gbk'}\n",
    "\n",
    "CORE=pd.read_csv(path_to_files+'CoreOrthology_id35cov51.txt',header=0,sep='\\t',index_col=0)\n",
    "\n",
    "# for i in range(688,len(CORE)):\n",
    "for i in [81,280,525,526,533,687]:\n",
    "    gene=CORE.iloc[i]\n",
    "    gene_name=CORE.iloc[i]['R.magnifica']\n",
    "    if gene_name.split('_')[1][0]=='R':\n",
    "        continue\n",
    "    fna=[]\n",
    "    faa=[]\n",
    "    print(i,gene_name)\n",
    "    for sample in sorted(gene.index):\n",
    "        for record in SeqIO.parse(goodfiles[sample],'genbank'):\n",
    "            for feature in record.features:\n",
    "                if feature.type=='CDS':\n",
    "                    locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                    if locus_tag == CORE.iloc[i][sample]:\n",
    "#                         if locus_tag == gene_name:\n",
    "#                             product= feature.qualifiers['product'][0]\n",
    "#                             print gene_name, product\n",
    "#                             df=df.append({'gene':gene_name,'product':product}, ignore_index=True)\n",
    "                        \n",
    "                        \n",
    "                        ### create SeqRecord object ####\n",
    "                        DNAseq=feature.location.extract(record).seq\n",
    "                        if locus_tag == gene_name:\n",
    "                            print(len(DNAseq))\n",
    "                        try:\n",
    "                            AAseq=Seq(feature.qualifiers['translation'][0],alphabet=Gapped(IUPAC.extended_protein))\n",
    "                        except KeyError:\n",
    "                            print(feature)\n",
    "#                             AAseq=feature.extract(record).translate(11)\n",
    "#                             AAseq.id=locus_tag\n",
    "#                             print(AAseq.seq)\n",
    "                        header='|'.join([sample,CORE.iloc[i][sample]])\n",
    "\n",
    "                        fna_seq = SeqRecord(DNAseq,id=header)\n",
    "                        faa_seq = SeqRecord(AAseq,id=header)\n",
    "                        \n",
    "                        fna.append(fna_seq)\n",
    "                        faa.append(faa_seq)                  \n",
    "\n",
    "    faa=[f for f in sorted(faa, key=lambda x : x.id)]\n",
    "    SeqIO.write(faa, path_to_files+'faa/'+gene_name+'.faa', \"fasta\")\n",
    "    \n",
    "#     muscle_cline = MuscleCommandline('/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx512/Compiler/gcc7.3/muscle/3.8.31/bin/muscle',input='fna/'+gene_name+'.fasta', out='fna/aligned_'+gene_name+'.faa')\n",
    "#     muscle_cline()\n",
    "#     align = AlignIO.read('fna/aligned_'+gene_name+'.faa', \"fasta\",alphabet=Gapped(IUPAC.extended_protein))\n",
    "\n",
    "#     stdout, stderr = muscle_cline()\n",
    "#     align = AlignIO.read(StringIO(stdout), \"fasta\")\n",
    "#     print(align)\n",
    "\n",
    "    ### run MUSCLE ####\n",
    "    muscle_cline = MuscleCommandline(clwstrict=True)\n",
    "    child = subprocess.Popen(str(muscle_cline),\n",
    "                             stdin=subprocess.PIPE,\n",
    "                             stdout=subprocess.PIPE,\n",
    "                             stderr=subprocess.PIPE,\n",
    "                             universal_newlines=True,\n",
    "                             shell=(sys.platform!=\"win32\"))\n",
    "    SeqIO.write(faa, child.stdin, \"fasta\")\n",
    "    child.stdin.close()\n",
    "    align = AlignIO.read(child.stdout, \"clustal\",alphabet=Gapped(IUPAC.extended_protein))\n",
    "    aln = MultipleSeqAlignment([f for f in sorted(align, key=lambda x : x.id)])\n",
    "    codon_aln = build(aln, fna)\n",
    "\n",
    "#     SeqIO.write(aln, path_to_files+'faa/aligned_'+gene_name+'.faa', \"fasta\")\n",
    "    SeqIO.write(codon_aln, path_to_files+'fna/aligned_'+gene_name+'.fna', \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extra stuff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3877\n",
      "668\n",
      "len core no_pseudo 668\n",
      "len core orthogroup no_pseudo 608\n",
      "in orthogroups only 23\n",
      "in positional orthology only 83\n",
      "585 585\n",
      "0\n",
      "MS2017_0004|SP60_04030|Rmag_0003|Rpac_peg_228|Rpha_peg_1022|Rpli_peg_1590|Rrec_peg_704|Rsou_peg_1464|Vdia_peg_981|Vext_peg_969|COSY_0003|Vgig2_peg_956|Vsoy2_peg_232|Vgig1_peg_12|Vsoy1_peg_28\n",
      "MS2017_0004|SP60_04030|Rmag_0003|Rpac_peg_228|Rpha_peg_1022|Rpli_peg_1590|Rrec_peg_704|Rsou_peg_1464|Vdia_peg_981|Vext_peg_969|COSY_0003|Vgig2_peg_956|Vsoy2_peg_232|Vgig1_peg_12|Vsoy1_peg_28\n"
     ]
    }
   ],
   "source": [
    "### Look for discrepancies between Corinas core orthogroups and this orthology\n",
    "print(len(ALL)) #3877 # 3878 with pseudo RNA\n",
    "print(len(CORE)) #668 #669 with pseudo RNA\n",
    "\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',sep='\\t', index_col=0)\n",
    "ALL=ALL[['R.fausta','V.marissinica','Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1','Count']]\n",
    "# print(ALL[:5])\n",
    "CORE_nopseudo=ALL[ALL['Count']==17]['R.magnifica'].values\n",
    "print('len core no_pseudo',len(CORE_nopseudo))\n",
    "\n",
    "CORE_orthogroup=pd.read_csv(path_to_files+'Core_orthogroups_N0.txt',sep='\\t')\n",
    "CORE_orthogroup.columns=['Bathy', 'R.fausta', 'R.magnifica', 'R.pacifica',\n",
    "       'R.phaseoliformis', 'R.pliocardia', 'R.rectimargo', 'R.southwardae',\n",
    "       'SUP05', 'V.diagonalis', 'V.extenta', 'V.gigas1', 'V.gigas2',\n",
    "       'V.marissinica', 'V.okutanii', 'V.soyoae1', 'V.soyoae2']\n",
    "print('len core orthogroup no_pseudo',len(CORE_orthogroup))\n",
    "\n",
    "print('in orthogroups only',len(set(CORE_orthogroup['R.magnifica'].values)-set(CORE_nopseudo)))\n",
    "print('in positional orthology only',len(set(CORE_nopseudo)-set(CORE_orthogroup['R.magnifica'].values)))\n",
    "\n",
    "\n",
    "# For those that are found in both dataset, is the orthology the same?\n",
    "intersec_core=set(CORE_nopseudo).intersection(set(CORE_orthogroup['R.magnifica'].values))\n",
    "\n",
    "CORE2=CORE_orthogroup[CORE_orthogroup.apply(lambda r: r.isin(intersec_core).any(), axis=1)].sort_values('R.magnifica')\n",
    "CORE2['R.pliocardia'] = CORE2['R.pliocardia'].str.replace('Rple','Rpli')\n",
    "CORE2=CORE2[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "core2_str=['|'.join(CORE2.iloc[i].values) for i in range(len(CORE2))]\n",
    "\n",
    "\n",
    "CORE1=ALL[ALL.apply(lambda r: r.isin(intersec_core).any(), axis=1)].sort_values('R.magnifica')\n",
    "CORE1=CORE1[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "core1_str=['|'.join(CORE1.iloc[i].values) for i in range(len(CORE1))]\n",
    "\n",
    "print(len(core1_str),len(core2_str))\n",
    "\n",
    "print(len(set(core1_str)-set(core2_str)))\n",
    "\n",
    "print(core1_str[2])\n",
    "print(core2_str[2])\n",
    "# the answer is YES (Ouf!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in orthogroups only 23\n",
      "in positional orthology only 83\n",
      "in Orthogroup only\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R.fausta</th>\n",
       "      <th>V.marissinica</th>\n",
       "      <th>Bathy</th>\n",
       "      <th>SUP05</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pliocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>HUE58_RS05470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MS2017_2127</td>\n",
       "      <td>SP60_03800</td>\n",
       "      <td>Rmag_1053</td>\n",
       "      <td>Rpac_peg_860</td>\n",
       "      <td>Rpha_peg_916</td>\n",
       "      <td>Rpli_peg_1108</td>\n",
       "      <td>Rrec_peg_666</td>\n",
       "      <td>Rsou_peg_1601</td>\n",
       "      <td>Vdia_peg_1017</td>\n",
       "      <td>Vext_peg_997</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vsoy1_peg_992</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1962</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vdia_peg_1009</td>\n",
       "      <td>Vext_peg_996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COSY_0953</td>\n",
       "      <td>Vgig2_peg_984</td>\n",
       "      <td>Vsoy2_peg_206</td>\n",
       "      <td>Vgig1_peg_984</td>\n",
       "      <td>Vsoy1_peg_1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           R.fausta V.marissinica        Bathy       SUP05 R.magnifica  \\\n",
       "403   HUE58_RS05470           NaN  MS2017_2127  SP60_03800   Rmag_1053   \n",
       "1962            NaN           NaN          NaN         NaN         NaN   \n",
       "2002            NaN           NaN          NaN         NaN         NaN   \n",
       "\n",
       "        R.pacifica R.phaseoliformis   R.pliocardia  R.rectimargo  \\\n",
       "403   Rpac_peg_860     Rpha_peg_916  Rpli_peg_1108  Rrec_peg_666   \n",
       "1962           NaN              NaN            NaN           NaN   \n",
       "2002           NaN              NaN            NaN           NaN   \n",
       "\n",
       "      R.southwardae   V.diagonalis     V.extenta V.okutanii       V.gigas2  \\\n",
       "403   Rsou_peg_1601  Vdia_peg_1017  Vext_peg_997        NaN            NaN   \n",
       "1962            NaN  Vdia_peg_1009  Vext_peg_996        NaN            NaN   \n",
       "2002            NaN            NaN           NaN  COSY_0953  Vgig2_peg_984   \n",
       "\n",
       "          V.soyoae2       V.gigas1      V.soyoae1  Count  \n",
       "403             NaN            NaN  Vsoy1_peg_992     12  \n",
       "1962            NaN            NaN            NaN      2  \n",
       "2002  Vsoy2_peg_206  Vgig1_peg_984    Vsoy1_peg_1      5  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('in orthogroups only',len(set(CORE_orthogroup['R.magnifica'].values)-set(CORE_nopseudo)))\n",
    "print('in positional orthology only',len(set(CORE_nopseudo)-set(CORE_orthogroup['R.magnifica'].values)))\n",
    "\n",
    "print('in Orthogroup only')\n",
    "set(CORE_orthogroup['R.magnifica'].values)-set(CORE_nopseudo)\n",
    "\n",
    "CORE_orthogroup[CORE_orthogroup.apply(lambda r: r.isin(set(CORE_orthogroup['R.magnifica'].values)-set(CORE_nopseudo)).any(), axis=1)].sort_values('R.magnifica')\n",
    "ALL[ALL.apply(lambda r: r.isin(set(CORE_nopseudo)-set(CORE_orthogroup['R.magnifica'].values)).any(), axis=1)].sort_values('R.magnifica')\n",
    "ALL[ALL.apply(lambda r: r.isin(['Rmag_1053','Rsou_peg_1602','Rsou_peg_1601','Vext_peg_996','Vext_peg_997',\"Vsoy1_peg_992\",\"Vsoy1_peg_1\"]).any(), axis=1)].sort_values('R.magnifica')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CORE_nopseudo\n",
    "# CORE2=ORTH[ORTH.apply(lambda r: r.isin(CORE_nopseudo).any(), axis=1)]\n",
    "# CORE2=CORE2[CORE2['genome_count']==15]\n",
    "\n",
    "# CORE2['R.pliocardia'] = CORE2['R.pleocardia'].str.replace('Rple','Rpli')\n",
    "# CORE2=CORE2[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "# core2_str=['|'.join(CORE2.iloc[i].values) for i in range(len(CORE2))]\n",
    "\n",
    "\n",
    "# CORE1=ALL[ALL['Count']==17]\n",
    "# CORE1=CORE1[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "# core1_str=['|'.join(CORE1.iloc[i].values) for i in range(len(CORE1))]\n",
    "\n",
    "# print('core genes',len(core1_str), len(core2_str))\n",
    "\n",
    "# set(CORE1['R.magnifica'].values)-set(CORE2['R.magnifica'].values)\n",
    "\n",
    "# # CORE1=CORE1[~CORE1['R.magnifica'].isin(['Rmag_0921', 'Rmag_R0017'])]\n",
    "# # CORE2=CORE2[~CORE2['R.magnifica'].isin(['Rmag_0921', 'Rmag_R0017'])]\n",
    "# # core2_str=['|'.join(CORE2.iloc[i].values) for i in range(len(CORE2))]\n",
    "# # core1_str=['|'.join(CORE1.iloc[i].values) for i in range(len(CORE1))]\n",
    "\n",
    "# print(len(set(core1_str)-set(core2_str)))\n",
    "# set(core1_str)-set(core2_str)\n",
    "\n",
    "# # CORE3=CORE1[CORE1.apply(lambda col: col.str.contains('_rna_|_R',regex=True), axis=1)]\n",
    "\n",
    "# CORE1[~(CORE1['R.magnifica'].str.contains(\"_R\"))]['R.magnifica']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in core genes in ALL but not in previous orthology 18\n",
      "in core genes in previous orthology but not in ALL 31\n",
      "['Rmag_0004', 'Rmag_0019', 'Rmag_0022', 'Rmag_0023', 'Rmag_1057', 'Rmag_0235', 'Rmag_0238', 'Rmag_0785', 'Rmag_0701', 'Rmag_0699', 'Rmag_0446', 'Rmag_0624', 'Rmag_0623', 'Rmag_0600', 'Rmag_0570', 'Rmag_0562', 'Rmag_0537', 'Rmag_0483', 'Rmag_0420', 'Rmag_0337', 'Rmag_0304', 'Rmag_0303', 'Rmag_0909', 'Rmag_0162', 'Rmag_0158', 'Rmag_0145', 'Rmag_0074', 'Rmag_0058', 'Rmag_1039', 'Rmag_R0022', 'Rmag_1053']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bathy</th>\n",
       "      <th>SUP05</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pleocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>product</th>\n",
       "      <th>genome_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MS2017_0005</td>\n",
       "      <td>SP60_04035</td>\n",
       "      <td>Rmag_0004</td>\n",
       "      <td>Rpac_peg_227</td>\n",
       "      <td>Rpha_peg_1023</td>\n",
       "      <td>Rple_peg_1589</td>\n",
       "      <td>Rrec_peg_705</td>\n",
       "      <td>Rsou_peg_1465</td>\n",
       "      <td>Vdia_peg_980</td>\n",
       "      <td>Vext_peg_968</td>\n",
       "      <td>COSY_0004</td>\n",
       "      <td>Vgig2_peg_955</td>\n",
       "      <td>Vsoy2_peg_233</td>\n",
       "      <td>Vgig1_peg_13</td>\n",
       "      <td>Vsoy1_peg_29</td>\n",
       "      <td>Phosphoribulokinase (EC 2.7.1.19)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MS2017_0034</td>\n",
       "      <td>SP60_04115</td>\n",
       "      <td>Rmag_0019</td>\n",
       "      <td>Rpac_peg_210</td>\n",
       "      <td>Rpha_peg_1048</td>\n",
       "      <td>Rple_peg_1569</td>\n",
       "      <td>Rrec_peg_722</td>\n",
       "      <td>Rsou_peg_1777</td>\n",
       "      <td>Vdia_peg_966</td>\n",
       "      <td>Vext_peg_954</td>\n",
       "      <td>COSY_0018</td>\n",
       "      <td>Vgig2_peg_940</td>\n",
       "      <td>Vsoy2_peg_248</td>\n",
       "      <td>Vgig1_peg_28</td>\n",
       "      <td>Vsoy1_peg_44</td>\n",
       "      <td>ATPases of the AAA+ class</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MS2017_0038</td>\n",
       "      <td>SP60_04130</td>\n",
       "      <td>Rmag_0022</td>\n",
       "      <td>Rpac_peg_207</td>\n",
       "      <td>Rpha_peg_1051</td>\n",
       "      <td>Rple_peg_1566</td>\n",
       "      <td>Rrec_peg_725</td>\n",
       "      <td>Rsou_peg_468</td>\n",
       "      <td>Vdia_peg_963</td>\n",
       "      <td>Vext_peg_951</td>\n",
       "      <td>COSY_0021</td>\n",
       "      <td>Vgig2_peg_937</td>\n",
       "      <td>Vsoy2_peg_251</td>\n",
       "      <td>Vgig1_peg_31</td>\n",
       "      <td>Vsoy1_peg_47</td>\n",
       "      <td>FIG01199523: hypothetical protein</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MS2017_0062</td>\n",
       "      <td>SP60_04135</td>\n",
       "      <td>Rmag_0023</td>\n",
       "      <td>Rpac_peg_206</td>\n",
       "      <td>Rpha_peg_1052</td>\n",
       "      <td>Rple_peg_1565</td>\n",
       "      <td>Rrec_peg_726</td>\n",
       "      <td>Rsou_peg_469</td>\n",
       "      <td>Vdia_peg_962</td>\n",
       "      <td>Vext_peg_950</td>\n",
       "      <td>COSY_0022</td>\n",
       "      <td>Vgig2_peg_936</td>\n",
       "      <td>Vsoy2_peg_252</td>\n",
       "      <td>Vgig1_peg_32</td>\n",
       "      <td>Vsoy1_peg_48</td>\n",
       "      <td>Phosphoribosylaminoimidazole-succinocarboxamid...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MS2017_0101</td>\n",
       "      <td>SP60_03840</td>\n",
       "      <td>Rmag_1057</td>\n",
       "      <td>Rpac_peg_864</td>\n",
       "      <td>Rpha_peg_959</td>\n",
       "      <td>Rple_peg_1122</td>\n",
       "      <td>Rrec_peg_673</td>\n",
       "      <td>Rsou_peg_536</td>\n",
       "      <td>Vdia_peg_1006</td>\n",
       "      <td>Vext_peg_993</td>\n",
       "      <td>COSY_0956</td>\n",
       "      <td>Vgig2_peg_980</td>\n",
       "      <td>Vsoy2_peg_209</td>\n",
       "      <td>Vgig1_peg_988</td>\n",
       "      <td>Vsoy1_peg_4</td>\n",
       "      <td>Cytochrome c-type biogenesis protein DsbD, pro...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>MS2017_0351</td>\n",
       "      <td>SP60_05605</td>\n",
       "      <td>Rmag_0235</td>\n",
       "      <td>Rpac_peg_360</td>\n",
       "      <td>Rpha_peg_1113</td>\n",
       "      <td>Rple_peg_105</td>\n",
       "      <td>Rrec_peg_220</td>\n",
       "      <td>Rsou_peg_1047</td>\n",
       "      <td>Vdia_peg_758</td>\n",
       "      <td>Vext_peg_747</td>\n",
       "      <td>COSY_0227</td>\n",
       "      <td>Vgig2_peg_590</td>\n",
       "      <td>Vsoy2_peg_455</td>\n",
       "      <td>Vgig1_peg_702</td>\n",
       "      <td>Vsoy1_peg_250</td>\n",
       "      <td>FIG006388: Possible exported protein</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>MS2017_0357</td>\n",
       "      <td>SP60_05620</td>\n",
       "      <td>Rmag_0238</td>\n",
       "      <td>Rpac_peg_363</td>\n",
       "      <td>Rpha_peg_1110</td>\n",
       "      <td>Rple_peg_108</td>\n",
       "      <td>Rrec_peg_217</td>\n",
       "      <td>Rsou_peg_1050</td>\n",
       "      <td>Vdia_peg_755</td>\n",
       "      <td>Vext_peg_744</td>\n",
       "      <td>COSY_0230</td>\n",
       "      <td>Vgig2_peg_587</td>\n",
       "      <td>Vsoy2_peg_458</td>\n",
       "      <td>Vgig1_peg_699</td>\n",
       "      <td>Vsoy1_peg_253</td>\n",
       "      <td>NADH-ubiquinone oxidoreductase chain B (EC 1.6...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>MS2017_0503</td>\n",
       "      <td>SP60_06205</td>\n",
       "      <td>Rmag_0785</td>\n",
       "      <td>Rpac_peg_279</td>\n",
       "      <td>Rpha_peg_2052</td>\n",
       "      <td>Rple_peg_369</td>\n",
       "      <td>Rrec_peg_301</td>\n",
       "      <td>Rsou_peg_1129</td>\n",
       "      <td>Vdia_peg_241</td>\n",
       "      <td>Vext_peg_236</td>\n",
       "      <td>COSY_0715</td>\n",
       "      <td>Vgig2_peg_90</td>\n",
       "      <td>Vsoy2_peg_957</td>\n",
       "      <td>Vgig1_peg_200</td>\n",
       "      <td>Vsoy1_peg_756</td>\n",
       "      <td>Methionyl-tRNA formyltransferase (EC 2.1.2.9)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>MS2017_0658</td>\n",
       "      <td>SP60_06705</td>\n",
       "      <td>Rmag_0701</td>\n",
       "      <td>Rpac_peg_1134</td>\n",
       "      <td>Rpha_peg_642</td>\n",
       "      <td>Rple_peg_234</td>\n",
       "      <td>Rrec_peg_1457</td>\n",
       "      <td>Rsou_peg_212</td>\n",
       "      <td>Vdia_peg_315</td>\n",
       "      <td>Vext_peg_307</td>\n",
       "      <td>COSY_0653</td>\n",
       "      <td>Vgig2_peg_157</td>\n",
       "      <td>Vsoy2_peg_890</td>\n",
       "      <td>Vgig1_peg_269</td>\n",
       "      <td>Vsoy1_peg_690</td>\n",
       "      <td>Ribulose bisphosphate carboxylase (EC 4.1.1.39)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>MS2017_0662</td>\n",
       "      <td>SP60_06715</td>\n",
       "      <td>Rmag_0699</td>\n",
       "      <td>Rpac_peg_1136</td>\n",
       "      <td>Rpha_peg_644</td>\n",
       "      <td>Rple_peg_230</td>\n",
       "      <td>Rrec_peg_1455</td>\n",
       "      <td>Rsou_peg_210</td>\n",
       "      <td>Vdia_peg_317</td>\n",
       "      <td>Vext_peg_309</td>\n",
       "      <td>COSY_0651</td>\n",
       "      <td>Vgig2_peg_159</td>\n",
       "      <td>Vsoy2_peg_888</td>\n",
       "      <td>Vgig1_peg_271</td>\n",
       "      <td>Vsoy1_peg_688</td>\n",
       "      <td>Rubisco activation protein CbbO</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>MS2017_0872</td>\n",
       "      <td>SP60_00510</td>\n",
       "      <td>Rmag_0446</td>\n",
       "      <td>Rpac_peg_984</td>\n",
       "      <td>Rpha_peg_396</td>\n",
       "      <td>Rple_peg_1307</td>\n",
       "      <td>Rrec_peg_1072</td>\n",
       "      <td>Rsou_peg_1404</td>\n",
       "      <td>Vdia_peg_563</td>\n",
       "      <td>Vext_peg_551</td>\n",
       "      <td>COSY_0412</td>\n",
       "      <td>Vgig2_peg_407</td>\n",
       "      <td>Vsoy2_peg_638</td>\n",
       "      <td>Vgig1_peg_520</td>\n",
       "      <td>Vsoy1_peg_439</td>\n",
       "      <td>Acetolactate synthase small subunit (EC 2.2.1.6)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>MS2017_0923</td>\n",
       "      <td>SP60_07275</td>\n",
       "      <td>Rmag_0624</td>\n",
       "      <td>Rpac_peg_1232</td>\n",
       "      <td>Rpha_peg_52</td>\n",
       "      <td>Rple_peg_521</td>\n",
       "      <td>Rrec_peg_1344</td>\n",
       "      <td>Rsou_peg_712</td>\n",
       "      <td>Vdia_peg_397</td>\n",
       "      <td>Vext_peg_385</td>\n",
       "      <td>COSY_0577</td>\n",
       "      <td>Vgig2_peg_236</td>\n",
       "      <td>Vsoy2_peg_812</td>\n",
       "      <td>Vgig1_peg_351</td>\n",
       "      <td>Vsoy1_peg_612</td>\n",
       "      <td>Prephenate and/or arogenate dehydrogenase (unk...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>MS2017_0924</td>\n",
       "      <td>SP60_07280</td>\n",
       "      <td>Rmag_0623</td>\n",
       "      <td>Rpac_peg_1233</td>\n",
       "      <td>Rpha_peg_53</td>\n",
       "      <td>Rple_peg_522</td>\n",
       "      <td>Rrec_peg_1343</td>\n",
       "      <td>Rsou_peg_713</td>\n",
       "      <td>Vdia_peg_398</td>\n",
       "      <td>Vext_peg_386</td>\n",
       "      <td>COSY_0576</td>\n",
       "      <td>Vgig2_peg_237</td>\n",
       "      <td>Vsoy2_peg_811</td>\n",
       "      <td>Vgig1_peg_352</td>\n",
       "      <td>Vsoy1_peg_611</td>\n",
       "      <td>5-Enolpyruvylshikimate-3-phosphate synthase (E...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328</th>\n",
       "      <td>MS2017_0955</td>\n",
       "      <td>SP60_07430</td>\n",
       "      <td>Rmag_0600</td>\n",
       "      <td>Rpac_peg_1261</td>\n",
       "      <td>Rpha_peg_93</td>\n",
       "      <td>Rple_peg_549</td>\n",
       "      <td>Rrec_peg_1315</td>\n",
       "      <td>Rsou_peg_749</td>\n",
       "      <td>Vdia_peg_419</td>\n",
       "      <td>Vext_peg_407</td>\n",
       "      <td>COSY_0554</td>\n",
       "      <td>Vgig2_peg_258</td>\n",
       "      <td>Vsoy2_peg_788</td>\n",
       "      <td>Vgig1_peg_373</td>\n",
       "      <td>Vsoy1_peg_589</td>\n",
       "      <td>UDP-N-acetylglucosamine 1-carboxyvinyltransfer...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>MS2017_0999</td>\n",
       "      <td>SP60_07690</td>\n",
       "      <td>Rmag_0570</td>\n",
       "      <td>Rpac_peg_1307</td>\n",
       "      <td>Rpha_peg_143</td>\n",
       "      <td>Rple_peg_1146</td>\n",
       "      <td>Rrec_peg_1262</td>\n",
       "      <td>Rsou_peg_920</td>\n",
       "      <td>Vdia_peg_450</td>\n",
       "      <td>Vext_peg_437</td>\n",
       "      <td>COSY_0525</td>\n",
       "      <td>Vgig2_peg_289</td>\n",
       "      <td>Vsoy2_peg_754</td>\n",
       "      <td>Vgig1_peg_404</td>\n",
       "      <td>Vsoy1_peg_555</td>\n",
       "      <td>Methionyl-tRNA synthetase (EC 6.1.1.10)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>MS2017_1007</td>\n",
       "      <td>SP60_07855</td>\n",
       "      <td>Rmag_0562</td>\n",
       "      <td>Rpac_peg_1317</td>\n",
       "      <td>Rpha_peg_150</td>\n",
       "      <td>Rple_peg_1153</td>\n",
       "      <td>Rrec_peg_1252</td>\n",
       "      <td>Rsou_peg_913</td>\n",
       "      <td>Vdia_peg_458</td>\n",
       "      <td>Vext_peg_445</td>\n",
       "      <td>COSY_0516</td>\n",
       "      <td>Vgig2_peg_297</td>\n",
       "      <td>Vsoy2_peg_746</td>\n",
       "      <td>Vgig1_peg_412</td>\n",
       "      <td>Vsoy1_peg_547</td>\n",
       "      <td>Heat shock protein 60 family chaperone GroEL</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>MS2017_1237</td>\n",
       "      <td>SP60_07990</td>\n",
       "      <td>Rmag_0537</td>\n",
       "      <td>Rpac_peg_1358</td>\n",
       "      <td>Rpha_peg_192</td>\n",
       "      <td>Rple_peg_1184</td>\n",
       "      <td>Rrec_peg_1215</td>\n",
       "      <td>Rsou_peg_874</td>\n",
       "      <td>Vdia_peg_482</td>\n",
       "      <td>Vext_peg_469</td>\n",
       "      <td>COSY_0493</td>\n",
       "      <td>Vgig2_peg_320</td>\n",
       "      <td>Vsoy2_peg_722</td>\n",
       "      <td>Vgig1_peg_435</td>\n",
       "      <td>Vsoy1_peg_523</td>\n",
       "      <td>23S rRNA (guanosine-2'-O-) -methyltransferase ...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>MS2017_1241</td>\n",
       "      <td>SP60_00260</td>\n",
       "      <td>Rmag_0483</td>\n",
       "      <td>Rpac_peg_915</td>\n",
       "      <td>Rpha_peg_297</td>\n",
       "      <td>Rple_peg_1244</td>\n",
       "      <td>Rrec_peg_1139</td>\n",
       "      <td>Rsou_peg_1873</td>\n",
       "      <td>Vdia_peg_530</td>\n",
       "      <td>Vext_peg_518</td>\n",
       "      <td>COSY_0446</td>\n",
       "      <td>Vgig2_peg_371</td>\n",
       "      <td>Vsoy2_peg_674</td>\n",
       "      <td>Vgig1_peg_485</td>\n",
       "      <td>Vsoy1_peg_475</td>\n",
       "      <td>3-oxoacyl-[acyl-carrier-protein] synthase, KAS...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>MS2017_1396</td>\n",
       "      <td>SP60_00760</td>\n",
       "      <td>Rmag_0420</td>\n",
       "      <td>Rpac_peg_1416</td>\n",
       "      <td>Rpha_peg_461</td>\n",
       "      <td>Rple_peg_1647</td>\n",
       "      <td>Rrec_peg_1026</td>\n",
       "      <td>Rsou_peg_632</td>\n",
       "      <td>Vdia_peg_590</td>\n",
       "      <td>Vext_peg_578</td>\n",
       "      <td>COSY_0389</td>\n",
       "      <td>Vgig2_peg_431</td>\n",
       "      <td>Vsoy2_peg_615</td>\n",
       "      <td>Vgig1_peg_546</td>\n",
       "      <td>Vsoy1_peg_415</td>\n",
       "      <td>C4-type zinc finger protein, DksA/TraR family</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>MS2017_1577</td>\n",
       "      <td>SP60_01295</td>\n",
       "      <td>Rmag_0337</td>\n",
       "      <td>Rpac_peg_505</td>\n",
       "      <td>Rpha_peg_1664</td>\n",
       "      <td>Rple_peg_666</td>\n",
       "      <td>Rrec_peg_90</td>\n",
       "      <td>Rsou_peg_1309</td>\n",
       "      <td>Vdia_peg_660</td>\n",
       "      <td>Vext_peg_649</td>\n",
       "      <td>COSY_0319</td>\n",
       "      <td>Vgig2_peg_500</td>\n",
       "      <td>Vsoy2_peg_547</td>\n",
       "      <td>Vgig1_peg_612</td>\n",
       "      <td>Vsoy1_peg_343</td>\n",
       "      <td>Pantothenate kinase type III, CoaX-like (EC 2....</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>MS2017_1653</td>\n",
       "      <td>SP60_01525</td>\n",
       "      <td>Rmag_0304</td>\n",
       "      <td>Rpac_peg_460</td>\n",
       "      <td>Rpha_peg_1728</td>\n",
       "      <td>Rple_peg_202</td>\n",
       "      <td>Rrec_peg_133</td>\n",
       "      <td>Rsou_peg_1699</td>\n",
       "      <td>Vdia_peg_698</td>\n",
       "      <td>Vext_peg_687</td>\n",
       "      <td>COSY_0286</td>\n",
       "      <td>Vgig2_peg_530</td>\n",
       "      <td>Vsoy2_peg_516</td>\n",
       "      <td>Vgig1_peg_643</td>\n",
       "      <td>Vsoy1_peg_312</td>\n",
       "      <td>GTP cyclohydrolase I (EC 3.5.4.16) type 2</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>501</th>\n",
       "      <td>MS2017_1654</td>\n",
       "      <td>SP60_01530</td>\n",
       "      <td>Rmag_0303</td>\n",
       "      <td>Rpac_peg_459</td>\n",
       "      <td>Rpha_peg_1729</td>\n",
       "      <td>Rple_peg_201</td>\n",
       "      <td>Rrec_peg_134</td>\n",
       "      <td>Rsou_peg_1698</td>\n",
       "      <td>Vdia_peg_699</td>\n",
       "      <td>Vext_peg_688</td>\n",
       "      <td>COSY_0285</td>\n",
       "      <td>Vgig2_peg_531</td>\n",
       "      <td>Vsoy2_peg_515</td>\n",
       "      <td>Vgig1_peg_644</td>\n",
       "      <td>Vsoy1_peg_311</td>\n",
       "      <td>6-pyruvoyl tetrahydrobiopterin synthase (EC 4....</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>MS2017_1779</td>\n",
       "      <td>SP60_02575</td>\n",
       "      <td>Rmag_0909</td>\n",
       "      <td>Rpac_peg_673</td>\n",
       "      <td>Rpha_peg_1431</td>\n",
       "      <td>Rple_peg_880</td>\n",
       "      <td>Rrec_peg_472</td>\n",
       "      <td>Rsou_peg_260</td>\n",
       "      <td>Vdia_peg_139</td>\n",
       "      <td>Vext_peg_134</td>\n",
       "      <td>COSY_0816</td>\n",
       "      <td>Vgig2_peg_725</td>\n",
       "      <td>Vsoy2_peg_72</td>\n",
       "      <td>Vgig1_peg_838</td>\n",
       "      <td>Vsoy1_peg_856</td>\n",
       "      <td>Glutaminyl-tRNA synthetase (EC 6.1.1.18)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>MS2017_1854</td>\n",
       "      <td>SP60_05225</td>\n",
       "      <td>Rmag_0162</td>\n",
       "      <td>Rpac_peg_3</td>\n",
       "      <td>Rpha_peg_1995</td>\n",
       "      <td>Rple_peg_1340</td>\n",
       "      <td>Rrec_peg_923</td>\n",
       "      <td>Rsou_peg_1966</td>\n",
       "      <td>Vdia_peg_819</td>\n",
       "      <td>Vext_peg_808</td>\n",
       "      <td>COSY_0166</td>\n",
       "      <td>Vgig2_peg_60</td>\n",
       "      <td>Vsoy2_peg_392</td>\n",
       "      <td>Vgig1_peg_170</td>\n",
       "      <td>Vsoy1_peg_187</td>\n",
       "      <td>Translation elongation factor G</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>609</th>\n",
       "      <td>MS2017_1858</td>\n",
       "      <td>SP60_05205</td>\n",
       "      <td>Rmag_0158</td>\n",
       "      <td>Rpac_peg_6</td>\n",
       "      <td>Rpha_peg_1991</td>\n",
       "      <td>Rple_peg_1344</td>\n",
       "      <td>Rrec_peg_920</td>\n",
       "      <td>Rsou_peg_1962</td>\n",
       "      <td>Vdia_peg_822</td>\n",
       "      <td>Vext_peg_811</td>\n",
       "      <td>COSY_0163</td>\n",
       "      <td>Vgig2_peg_57</td>\n",
       "      <td>Vsoy2_peg_389</td>\n",
       "      <td>Vgig1_peg_167</td>\n",
       "      <td>Vsoy1_peg_184</td>\n",
       "      <td>Arabinose 5-phosphate isomerase (EC 5.3.1.13)</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616</th>\n",
       "      <td>MS2017_1882</td>\n",
       "      <td>SP60_05115</td>\n",
       "      <td>Rmag_0145</td>\n",
       "      <td>Rpac_peg_24</td>\n",
       "      <td>Rpha_peg_1970</td>\n",
       "      <td>Rple_peg_1368</td>\n",
       "      <td>Rrec_peg_902</td>\n",
       "      <td>Rsou_peg_947</td>\n",
       "      <td>Vdia_peg_834</td>\n",
       "      <td>Vext_peg_823</td>\n",
       "      <td>COSY_0149</td>\n",
       "      <td>Vgig2_peg_45</td>\n",
       "      <td>Vsoy2_peg_378</td>\n",
       "      <td>Vgig1_peg_156</td>\n",
       "      <td>Vsoy1_peg_173</td>\n",
       "      <td>Succinyl-CoA ligase [ADP-forming] alpha chain ...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>MS2017_2071</td>\n",
       "      <td>SP60_04525</td>\n",
       "      <td>Rmag_0074</td>\n",
       "      <td>Rpac_peg_122</td>\n",
       "      <td>Rpha_peg_750</td>\n",
       "      <td>Rple_peg_1473</td>\n",
       "      <td>Rrec_peg_802</td>\n",
       "      <td>Rsou_peg_1652</td>\n",
       "      <td>Vdia_peg_904</td>\n",
       "      <td>Vext_peg_893</td>\n",
       "      <td>COSY_0080</td>\n",
       "      <td>Vgig2_peg_884</td>\n",
       "      <td>Vsoy2_peg_307</td>\n",
       "      <td>Vgig1_peg_86</td>\n",
       "      <td>Vsoy1_peg_103</td>\n",
       "      <td>2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>679</th>\n",
       "      <td>MS2017_2101</td>\n",
       "      <td>SP60_04365</td>\n",
       "      <td>Rmag_0058</td>\n",
       "      <td>Rpac_peg_148</td>\n",
       "      <td>Rpha_peg_768</td>\n",
       "      <td>Rple_peg_1496</td>\n",
       "      <td>Rrec_peg_782</td>\n",
       "      <td>Rsou_peg_575</td>\n",
       "      <td>Vdia_peg_922</td>\n",
       "      <td>Vext_peg_911</td>\n",
       "      <td>COSY_0063</td>\n",
       "      <td>Vgig2_peg_899</td>\n",
       "      <td>Vsoy2_peg_292</td>\n",
       "      <td>Vgig1_peg_70</td>\n",
       "      <td>Vsoy1_peg_88</td>\n",
       "      <td>21 kDa hemolysin precursor</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>MS2017_2145</td>\n",
       "      <td>SP60_03725</td>\n",
       "      <td>Rmag_1039</td>\n",
       "      <td>Rpac_peg_845</td>\n",
       "      <td>Rpha_peg_901</td>\n",
       "      <td>Rple_peg_1091</td>\n",
       "      <td>Rrec_peg_651</td>\n",
       "      <td>Rsou_peg_1586</td>\n",
       "      <td>Vdia_peg_16</td>\n",
       "      <td>Vext_peg_13</td>\n",
       "      <td>COSY_0940</td>\n",
       "      <td>Vgig2_peg_843</td>\n",
       "      <td>Vsoy2_peg_192</td>\n",
       "      <td>Vgig1_peg_967</td>\n",
       "      <td>Vsoy1_peg_978</td>\n",
       "      <td>Protein YidD</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1883</th>\n",
       "      <td>MS2017_R017</td>\n",
       "      <td>SP60_07495</td>\n",
       "      <td>Rmag_R0022</td>\n",
       "      <td>Rpac_rna_34</td>\n",
       "      <td>Rpha_rna_2</td>\n",
       "      <td>Rple_rna_9</td>\n",
       "      <td>Rrec_rna_36</td>\n",
       "      <td>Rsou_rna_14</td>\n",
       "      <td>Vdia_rna_17</td>\n",
       "      <td>Vext_rna_18</td>\n",
       "      <td>COSY_R0020</td>\n",
       "      <td>Vgig2_rna_11</td>\n",
       "      <td>Vsoy2_rna_33</td>\n",
       "      <td>Vgig1_rna_15</td>\n",
       "      <td>Vsoy1_rna_22</td>\n",
       "      <td>tRNA-Lys-TTT</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6089</th>\n",
       "      <td>MS2017_2127</td>\n",
       "      <td>SP60_03800</td>\n",
       "      <td>Rmag_1053</td>\n",
       "      <td>Rpac_peg_860</td>\n",
       "      <td>Rpha_peg_916</td>\n",
       "      <td>Rple_peg_1108</td>\n",
       "      <td>Rrec_peg_666</td>\n",
       "      <td>Rsou_peg_1601</td>\n",
       "      <td>Vdia_peg_1009</td>\n",
       "      <td>Vext_peg_996</td>\n",
       "      <td>COSY_0953</td>\n",
       "      <td>Vgig2_peg_983</td>\n",
       "      <td>Vsoy2_peg_206</td>\n",
       "      <td>Vgig1_peg_985</td>\n",
       "      <td>Vsoy1_peg_1</td>\n",
       "      <td>FAD-dependent pyridine nucleotide-disulphide o...</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Bathy       SUP05 R.magnifica     R.pacifica R.phaseoliformis  \\\n",
       "3     MS2017_0005  SP60_04035   Rmag_0004   Rpac_peg_227    Rpha_peg_1023   \n",
       "15    MS2017_0034  SP60_04115   Rmag_0019   Rpac_peg_210    Rpha_peg_1048   \n",
       "16    MS2017_0038  SP60_04130   Rmag_0022   Rpac_peg_207    Rpha_peg_1051   \n",
       "17    MS2017_0062  SP60_04135   Rmag_0023   Rpac_peg_206    Rpha_peg_1052   \n",
       "28    MS2017_0101  SP60_03840   Rmag_1057   Rpac_peg_864     Rpha_peg_959   \n",
       "128   MS2017_0351  SP60_05605   Rmag_0235   Rpac_peg_360    Rpha_peg_1113   \n",
       "131   MS2017_0357  SP60_05620   Rmag_0238   Rpac_peg_363    Rpha_peg_1110   \n",
       "200   MS2017_0503  SP60_06205   Rmag_0785   Rpac_peg_279    Rpha_peg_2052   \n",
       "234   MS2017_0658  SP60_06705   Rmag_0701  Rpac_peg_1134     Rpha_peg_642   \n",
       "236   MS2017_0662  SP60_06715   Rmag_0699  Rpac_peg_1136     Rpha_peg_644   \n",
       "293   MS2017_0872  SP60_00510   Rmag_0446   Rpac_peg_984     Rpha_peg_396   \n",
       "308   MS2017_0923  SP60_07275   Rmag_0624  Rpac_peg_1232      Rpha_peg_52   \n",
       "309   MS2017_0924  SP60_07280   Rmag_0623  Rpac_peg_1233      Rpha_peg_53   \n",
       "328   MS2017_0955  SP60_07430   Rmag_0600  Rpac_peg_1261      Rpha_peg_93   \n",
       "340   MS2017_0999  SP60_07690   Rmag_0570  Rpac_peg_1307     Rpha_peg_143   \n",
       "346   MS2017_1007  SP60_07855   Rmag_0562  Rpac_peg_1317     Rpha_peg_150   \n",
       "405   MS2017_1237  SP60_07990   Rmag_0537  Rpac_peg_1358     Rpha_peg_192   \n",
       "409   MS2017_1241  SP60_00260   Rmag_0483   Rpac_peg_915     Rpha_peg_297   \n",
       "422   MS2017_1396  SP60_00760   Rmag_0420  Rpac_peg_1416     Rpha_peg_461   \n",
       "482   MS2017_1577  SP60_01295   Rmag_0337   Rpac_peg_505    Rpha_peg_1664   \n",
       "500   MS2017_1653  SP60_01525   Rmag_0304   Rpac_peg_460    Rpha_peg_1728   \n",
       "501   MS2017_1654  SP60_01530   Rmag_0303   Rpac_peg_459    Rpha_peg_1729   \n",
       "557   MS2017_1779  SP60_02575   Rmag_0909   Rpac_peg_673    Rpha_peg_1431   \n",
       "606   MS2017_1854  SP60_05225   Rmag_0162     Rpac_peg_3    Rpha_peg_1995   \n",
       "609   MS2017_1858  SP60_05205   Rmag_0158     Rpac_peg_6    Rpha_peg_1991   \n",
       "616   MS2017_1882  SP60_05115   Rmag_0145    Rpac_peg_24    Rpha_peg_1970   \n",
       "667   MS2017_2071  SP60_04525   Rmag_0074   Rpac_peg_122     Rpha_peg_750   \n",
       "679   MS2017_2101  SP60_04365   Rmag_0058   Rpac_peg_148     Rpha_peg_768   \n",
       "698   MS2017_2145  SP60_03725   Rmag_1039   Rpac_peg_845     Rpha_peg_901   \n",
       "1883  MS2017_R017  SP60_07495  Rmag_R0022    Rpac_rna_34       Rpha_rna_2   \n",
       "6089  MS2017_2127  SP60_03800   Rmag_1053   Rpac_peg_860     Rpha_peg_916   \n",
       "\n",
       "       R.pleocardia   R.rectimargo  R.southwardae   V.diagonalis  \\\n",
       "3     Rple_peg_1589   Rrec_peg_705  Rsou_peg_1465   Vdia_peg_980   \n",
       "15    Rple_peg_1569   Rrec_peg_722  Rsou_peg_1777   Vdia_peg_966   \n",
       "16    Rple_peg_1566   Rrec_peg_725   Rsou_peg_468   Vdia_peg_963   \n",
       "17    Rple_peg_1565   Rrec_peg_726   Rsou_peg_469   Vdia_peg_962   \n",
       "28    Rple_peg_1122   Rrec_peg_673   Rsou_peg_536  Vdia_peg_1006   \n",
       "128    Rple_peg_105   Rrec_peg_220  Rsou_peg_1047   Vdia_peg_758   \n",
       "131    Rple_peg_108   Rrec_peg_217  Rsou_peg_1050   Vdia_peg_755   \n",
       "200    Rple_peg_369   Rrec_peg_301  Rsou_peg_1129   Vdia_peg_241   \n",
       "234    Rple_peg_234  Rrec_peg_1457   Rsou_peg_212   Vdia_peg_315   \n",
       "236    Rple_peg_230  Rrec_peg_1455   Rsou_peg_210   Vdia_peg_317   \n",
       "293   Rple_peg_1307  Rrec_peg_1072  Rsou_peg_1404   Vdia_peg_563   \n",
       "308    Rple_peg_521  Rrec_peg_1344   Rsou_peg_712   Vdia_peg_397   \n",
       "309    Rple_peg_522  Rrec_peg_1343   Rsou_peg_713   Vdia_peg_398   \n",
       "328    Rple_peg_549  Rrec_peg_1315   Rsou_peg_749   Vdia_peg_419   \n",
       "340   Rple_peg_1146  Rrec_peg_1262   Rsou_peg_920   Vdia_peg_450   \n",
       "346   Rple_peg_1153  Rrec_peg_1252   Rsou_peg_913   Vdia_peg_458   \n",
       "405   Rple_peg_1184  Rrec_peg_1215   Rsou_peg_874   Vdia_peg_482   \n",
       "409   Rple_peg_1244  Rrec_peg_1139  Rsou_peg_1873   Vdia_peg_530   \n",
       "422   Rple_peg_1647  Rrec_peg_1026   Rsou_peg_632   Vdia_peg_590   \n",
       "482    Rple_peg_666    Rrec_peg_90  Rsou_peg_1309   Vdia_peg_660   \n",
       "500    Rple_peg_202   Rrec_peg_133  Rsou_peg_1699   Vdia_peg_698   \n",
       "501    Rple_peg_201   Rrec_peg_134  Rsou_peg_1698   Vdia_peg_699   \n",
       "557    Rple_peg_880   Rrec_peg_472   Rsou_peg_260   Vdia_peg_139   \n",
       "606   Rple_peg_1340   Rrec_peg_923  Rsou_peg_1966   Vdia_peg_819   \n",
       "609   Rple_peg_1344   Rrec_peg_920  Rsou_peg_1962   Vdia_peg_822   \n",
       "616   Rple_peg_1368   Rrec_peg_902   Rsou_peg_947   Vdia_peg_834   \n",
       "667   Rple_peg_1473   Rrec_peg_802  Rsou_peg_1652   Vdia_peg_904   \n",
       "679   Rple_peg_1496   Rrec_peg_782   Rsou_peg_575   Vdia_peg_922   \n",
       "698   Rple_peg_1091   Rrec_peg_651  Rsou_peg_1586    Vdia_peg_16   \n",
       "1883     Rple_rna_9    Rrec_rna_36    Rsou_rna_14    Vdia_rna_17   \n",
       "6089  Rple_peg_1108   Rrec_peg_666  Rsou_peg_1601  Vdia_peg_1009   \n",
       "\n",
       "         V.extenta  V.okutanii       V.gigas2      V.soyoae2       V.gigas1  \\\n",
       "3     Vext_peg_968   COSY_0004  Vgig2_peg_955  Vsoy2_peg_233   Vgig1_peg_13   \n",
       "15    Vext_peg_954   COSY_0018  Vgig2_peg_940  Vsoy2_peg_248   Vgig1_peg_28   \n",
       "16    Vext_peg_951   COSY_0021  Vgig2_peg_937  Vsoy2_peg_251   Vgig1_peg_31   \n",
       "17    Vext_peg_950   COSY_0022  Vgig2_peg_936  Vsoy2_peg_252   Vgig1_peg_32   \n",
       "28    Vext_peg_993   COSY_0956  Vgig2_peg_980  Vsoy2_peg_209  Vgig1_peg_988   \n",
       "128   Vext_peg_747   COSY_0227  Vgig2_peg_590  Vsoy2_peg_455  Vgig1_peg_702   \n",
       "131   Vext_peg_744   COSY_0230  Vgig2_peg_587  Vsoy2_peg_458  Vgig1_peg_699   \n",
       "200   Vext_peg_236   COSY_0715   Vgig2_peg_90  Vsoy2_peg_957  Vgig1_peg_200   \n",
       "234   Vext_peg_307   COSY_0653  Vgig2_peg_157  Vsoy2_peg_890  Vgig1_peg_269   \n",
       "236   Vext_peg_309   COSY_0651  Vgig2_peg_159  Vsoy2_peg_888  Vgig1_peg_271   \n",
       "293   Vext_peg_551   COSY_0412  Vgig2_peg_407  Vsoy2_peg_638  Vgig1_peg_520   \n",
       "308   Vext_peg_385   COSY_0577  Vgig2_peg_236  Vsoy2_peg_812  Vgig1_peg_351   \n",
       "309   Vext_peg_386   COSY_0576  Vgig2_peg_237  Vsoy2_peg_811  Vgig1_peg_352   \n",
       "328   Vext_peg_407   COSY_0554  Vgig2_peg_258  Vsoy2_peg_788  Vgig1_peg_373   \n",
       "340   Vext_peg_437   COSY_0525  Vgig2_peg_289  Vsoy2_peg_754  Vgig1_peg_404   \n",
       "346   Vext_peg_445   COSY_0516  Vgig2_peg_297  Vsoy2_peg_746  Vgig1_peg_412   \n",
       "405   Vext_peg_469   COSY_0493  Vgig2_peg_320  Vsoy2_peg_722  Vgig1_peg_435   \n",
       "409   Vext_peg_518   COSY_0446  Vgig2_peg_371  Vsoy2_peg_674  Vgig1_peg_485   \n",
       "422   Vext_peg_578   COSY_0389  Vgig2_peg_431  Vsoy2_peg_615  Vgig1_peg_546   \n",
       "482   Vext_peg_649   COSY_0319  Vgig2_peg_500  Vsoy2_peg_547  Vgig1_peg_612   \n",
       "500   Vext_peg_687   COSY_0286  Vgig2_peg_530  Vsoy2_peg_516  Vgig1_peg_643   \n",
       "501   Vext_peg_688   COSY_0285  Vgig2_peg_531  Vsoy2_peg_515  Vgig1_peg_644   \n",
       "557   Vext_peg_134   COSY_0816  Vgig2_peg_725   Vsoy2_peg_72  Vgig1_peg_838   \n",
       "606   Vext_peg_808   COSY_0166   Vgig2_peg_60  Vsoy2_peg_392  Vgig1_peg_170   \n",
       "609   Vext_peg_811   COSY_0163   Vgig2_peg_57  Vsoy2_peg_389  Vgig1_peg_167   \n",
       "616   Vext_peg_823   COSY_0149   Vgig2_peg_45  Vsoy2_peg_378  Vgig1_peg_156   \n",
       "667   Vext_peg_893   COSY_0080  Vgig2_peg_884  Vsoy2_peg_307   Vgig1_peg_86   \n",
       "679   Vext_peg_911   COSY_0063  Vgig2_peg_899  Vsoy2_peg_292   Vgig1_peg_70   \n",
       "698    Vext_peg_13   COSY_0940  Vgig2_peg_843  Vsoy2_peg_192  Vgig1_peg_967   \n",
       "1883   Vext_rna_18  COSY_R0020   Vgig2_rna_11   Vsoy2_rna_33   Vgig1_rna_15   \n",
       "6089  Vext_peg_996   COSY_0953  Vgig2_peg_983  Vsoy2_peg_206  Vgig1_peg_985   \n",
       "\n",
       "          V.soyoae1                                            product  \\\n",
       "3      Vsoy1_peg_29                  Phosphoribulokinase (EC 2.7.1.19)   \n",
       "15     Vsoy1_peg_44                          ATPases of the AAA+ class   \n",
       "16     Vsoy1_peg_47                  FIG01199523: hypothetical protein   \n",
       "17     Vsoy1_peg_48  Phosphoribosylaminoimidazole-succinocarboxamid...   \n",
       "28      Vsoy1_peg_4  Cytochrome c-type biogenesis protein DsbD, pro...   \n",
       "128   Vsoy1_peg_250               FIG006388: Possible exported protein   \n",
       "131   Vsoy1_peg_253  NADH-ubiquinone oxidoreductase chain B (EC 1.6...   \n",
       "200   Vsoy1_peg_756      Methionyl-tRNA formyltransferase (EC 2.1.2.9)   \n",
       "234   Vsoy1_peg_690    Ribulose bisphosphate carboxylase (EC 4.1.1.39)   \n",
       "236   Vsoy1_peg_688                    Rubisco activation protein CbbO   \n",
       "293   Vsoy1_peg_439   Acetolactate synthase small subunit (EC 2.2.1.6)   \n",
       "308   Vsoy1_peg_612  Prephenate and/or arogenate dehydrogenase (unk...   \n",
       "309   Vsoy1_peg_611  5-Enolpyruvylshikimate-3-phosphate synthase (E...   \n",
       "328   Vsoy1_peg_589  UDP-N-acetylglucosamine 1-carboxyvinyltransfer...   \n",
       "340   Vsoy1_peg_555            Methionyl-tRNA synthetase (EC 6.1.1.10)   \n",
       "346   Vsoy1_peg_547       Heat shock protein 60 family chaperone GroEL   \n",
       "405   Vsoy1_peg_523  23S rRNA (guanosine-2'-O-) -methyltransferase ...   \n",
       "409   Vsoy1_peg_475  3-oxoacyl-[acyl-carrier-protein] synthase, KAS...   \n",
       "422   Vsoy1_peg_415      C4-type zinc finger protein, DksA/TraR family   \n",
       "482   Vsoy1_peg_343  Pantothenate kinase type III, CoaX-like (EC 2....   \n",
       "500   Vsoy1_peg_312          GTP cyclohydrolase I (EC 3.5.4.16) type 2   \n",
       "501   Vsoy1_peg_311  6-pyruvoyl tetrahydrobiopterin synthase (EC 4....   \n",
       "557   Vsoy1_peg_856           Glutaminyl-tRNA synthetase (EC 6.1.1.18)   \n",
       "606   Vsoy1_peg_187                    Translation elongation factor G   \n",
       "609   Vsoy1_peg_184      Arabinose 5-phosphate isomerase (EC 5.3.1.13)   \n",
       "616   Vsoy1_peg_173  Succinyl-CoA ligase [ADP-forming] alpha chain ...   \n",
       "667   Vsoy1_peg_103  2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N...   \n",
       "679    Vsoy1_peg_88                         21 kDa hemolysin precursor   \n",
       "698   Vsoy1_peg_978                                       Protein YidD   \n",
       "1883   Vsoy1_rna_22                                       tRNA-Lys-TTT   \n",
       "6089    Vsoy1_peg_1  FAD-dependent pyridine nucleotide-disulphide o...   \n",
       "\n",
       "      genome_count  \n",
       "3               15  \n",
       "15              15  \n",
       "16              15  \n",
       "17              15  \n",
       "28              15  \n",
       "128             15  \n",
       "131             15  \n",
       "200             15  \n",
       "234             15  \n",
       "236             15  \n",
       "293             15  \n",
       "308             15  \n",
       "309             15  \n",
       "328             15  \n",
       "340             15  \n",
       "346             15  \n",
       "405             15  \n",
       "409             15  \n",
       "422             15  \n",
       "482             15  \n",
       "500             15  \n",
       "501             15  \n",
       "557             15  \n",
       "606             15  \n",
       "609             15  \n",
       "616             15  \n",
       "667             15  \n",
       "679             15  \n",
       "698             15  \n",
       "1883            15  \n",
       "6089            15  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### Look for discrepancy between old and new orthologies ####\n",
    "\n",
    "# in ALL core not in previous orthology\n",
    "ORTH=pd.read_csv(path_to_files+'Ortho_all_28052020.txt',sep='\\t')\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51.txt',sep='\\t', index_col=0)\n",
    "ALL=ALL[['R.fausta','V.marissinica','Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1','Count']]\n",
    "# print(ALL[:5])\n",
    "ALL_only=[l for l in ALL[ALL['Count']==17]['R.magnifica'].values if l not in ORTH[ORTH['genome_count']==15]['R.magnifica'].values]\n",
    "ALL_only\n",
    "print('in core genes in ALL but not in previous orthology',len(ALL_only))\n",
    "# print(ALL_only)\n",
    "lines_only=[l for l in ORTH[ORTH['genome_count']==15]['R.magnifica'].values if l not in ALL[ALL['Count']==17]['R.magnifica'].values]\n",
    "print('in core genes in previous orthology but not in ALL', len(lines_only))\n",
    "print(lines_only)\n",
    "\n",
    "\n",
    "# @id70: 193 core genes missing\n",
    "# @id60: 39 core genes missing \n",
    "# @id45: 34 core genes missing\n",
    "# @id35: 23 core genes missing\n",
    "\n",
    "discr=ORTH[ORTH['R.magnifica'].isin(lines_only)].sort_values(['R.magnifica']) # all locus_tags that are supporsed to be in core\n",
    "discr=discr[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pleocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values.flatten()\n",
    "\n",
    "\n",
    "ALL[ALL.apply(lambda r: r.isin(discr).any(), axis=1)] \n",
    "ORTH[ORTH['R.magnifica'].isin(lines_only)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5423"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## do any of the genes associated with core genes locus_tags in all corresponds to non core?\n",
    "\n",
    "broken_orth=ALL[ALL.apply(lambda r: r.isin(discr).any(), axis=1)][['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values.flatten()\n",
    "\n",
    "df2=ORTH[ORTH.apply(lambda r: r.isin(broken_orth).any(), axis=1)]\n",
    "len(df2[df2['genome_count']<15]['product'])\n",
    "# print(len(df2[df2['genome_count']<15]))\n",
    "## To do possibly\n",
    "#remove rows from ALL\n",
    "# append rows from ORTH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bathy</th>\n",
       "      <th>SUP05</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pleocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>product</th>\n",
       "      <th>genome_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>MS2017_0185</td>\n",
       "      <td>SP60_03255</td>\n",
       "      <td>Rmag_0989</td>\n",
       "      <td>Rpac_peg_771</td>\n",
       "      <td>Rpha_peg_1634</td>\n",
       "      <td>Rple_peg_1007</td>\n",
       "      <td>Rrec_peg_574</td>\n",
       "      <td>Rsou_peg_1741</td>\n",
       "      <td>Vdia_peg_71</td>\n",
       "      <td>Vext_peg_67</td>\n",
       "      <td>COSY_0884</td>\n",
       "      <td>Vgig2_peg_792</td>\n",
       "      <td>Vsoy2_peg_141</td>\n",
       "      <td>Vgig1_peg_905</td>\n",
       "      <td>Vsoy1_peg_925</td>\n",
       "      <td>16S rRNA processing protein RimM</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3775</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rmag_R0017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA-Asn</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4369</th>\n",
       "      <td>MS2017_0317</td>\n",
       "      <td>SP60_02650</td>\n",
       "      <td>Rmag_0921</td>\n",
       "      <td>Rpac_peg_686</td>\n",
       "      <td>Rpha_peg_1413</td>\n",
       "      <td>Rple_peg_898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rsou_peg_242</td>\n",
       "      <td>Vdia_peg_128</td>\n",
       "      <td>Vext_peg_123</td>\n",
       "      <td>COSY_0827</td>\n",
       "      <td>Vgig2_peg_736</td>\n",
       "      <td>Vsoy2_peg_83</td>\n",
       "      <td>Vgig1_peg_849</td>\n",
       "      <td>Vsoy1_peg_867</td>\n",
       "      <td>UDP-2,3-diacylglucosamine diphosphatase (EC 3....</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Bathy       SUP05 R.magnifica    R.pacifica R.phaseoliformis  \\\n",
       "71    MS2017_0185  SP60_03255   Rmag_0989  Rpac_peg_771    Rpha_peg_1634   \n",
       "3775          NaN         NaN  Rmag_R0017           NaN              NaN   \n",
       "4369  MS2017_0317  SP60_02650   Rmag_0921  Rpac_peg_686    Rpha_peg_1413   \n",
       "\n",
       "       R.pleocardia  R.rectimargo  R.southwardae  V.diagonalis     V.extenta  \\\n",
       "71    Rple_peg_1007  Rrec_peg_574  Rsou_peg_1741   Vdia_peg_71   Vext_peg_67   \n",
       "3775            NaN           NaN            NaN           NaN           NaN   \n",
       "4369   Rple_peg_898           NaN   Rsou_peg_242  Vdia_peg_128  Vext_peg_123   \n",
       "\n",
       "     V.okutanii       V.gigas2      V.soyoae2       V.gigas1      V.soyoae1  \\\n",
       "71    COSY_0884  Vgig2_peg_792  Vsoy2_peg_141  Vgig1_peg_905  Vsoy1_peg_925   \n",
       "3775        NaN            NaN            NaN            NaN            NaN   \n",
       "4369  COSY_0827  Vgig2_peg_736   Vsoy2_peg_83  Vgig1_peg_849  Vsoy1_peg_867   \n",
       "\n",
       "                                                product  genome_count  \n",
       "71                     16S rRNA processing protein RimM            15  \n",
       "3775                                           tRNA-Asn             1  \n",
       "4369  UDP-2,3-diacylglucosamine diphosphatase (EC 3....            14  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ORTH[ORTH.apply(lambda r: r.isin(['Vgig2_peg_984','Vsoy2_peg_206','Vsoy1_peg_1','COSY_0953','Vext_peg_997','Vdia_peg_1017','Vsoy2_peg_206','COSY_0953','Vgig2_peg_983','Vgig1_peg_985']).any(), axis=1)]\n",
    "\n",
    "ORTH[ORTH.apply(lambda r: r.isin(['Rmag_0921', 'Rmag_R0017','Rmag_0989']).any(), axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R.fausta</th>\n",
       "      <th>V.marissinica</th>\n",
       "      <th>Bathy</th>\n",
       "      <th>SUP05</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pliocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>HUE58_RS01660</td>\n",
       "      <td>Vmar_R0017</td>\n",
       "      <td>MS2017_R024</td>\n",
       "      <td>SP60_00715</td>\n",
       "      <td>Rmag_R0017</td>\n",
       "      <td>Rpac_rna_29</td>\n",
       "      <td>Rpha_rna_6</td>\n",
       "      <td>Rpli_rna_27</td>\n",
       "      <td>Rrec_rna_32</td>\n",
       "      <td>Rsou_rna_4</td>\n",
       "      <td>Vdia_rna_21</td>\n",
       "      <td>Vext_rna_22</td>\n",
       "      <td>COSY_R0016</td>\n",
       "      <td>Vgig2_rna_15</td>\n",
       "      <td>Vsoy2_rna_29</td>\n",
       "      <td>Vgig1_rna_19</td>\n",
       "      <td>Vsoy1_rna_18</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>HUE58_RS05040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MS2017_0185</td>\n",
       "      <td>SP60_03255</td>\n",
       "      <td>Rmag_0989</td>\n",
       "      <td>Rpac_peg_771</td>\n",
       "      <td>Rpha_peg_1634</td>\n",
       "      <td>Rpli_peg_1007</td>\n",
       "      <td>Rrec_peg_574</td>\n",
       "      <td>Rsou_peg_1741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>HUE58_RS04605</td>\n",
       "      <td>Vmar_0837</td>\n",
       "      <td>MS2017_0317</td>\n",
       "      <td>SP60_02650</td>\n",
       "      <td>Rmag_0921</td>\n",
       "      <td>Rpac_peg_686</td>\n",
       "      <td>Rpha_peg_1413</td>\n",
       "      <td>Rpli_peg_898</td>\n",
       "      <td>Rrec_peg_483</td>\n",
       "      <td>Rsou_peg_242</td>\n",
       "      <td>Vdia_peg_128</td>\n",
       "      <td>Vext_peg_123</td>\n",
       "      <td>COSY_0827</td>\n",
       "      <td>Vgig2_peg_736</td>\n",
       "      <td>Vsoy2_peg_83</td>\n",
       "      <td>Vgig1_peg_849</td>\n",
       "      <td>Vsoy1_peg_867</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          R.fausta V.marissinica        Bathy       SUP05 R.magnifica  \\\n",
       "90   HUE58_RS01660    Vmar_R0017  MS2017_R024  SP60_00715  Rmag_R0017   \n",
       "339  HUE58_RS05040           NaN  MS2017_0185  SP60_03255   Rmag_0989   \n",
       "976  HUE58_RS04605     Vmar_0837  MS2017_0317  SP60_02650   Rmag_0921   \n",
       "\n",
       "       R.pacifica R.phaseoliformis   R.pliocardia  R.rectimargo  \\\n",
       "90    Rpac_rna_29       Rpha_rna_6    Rpli_rna_27   Rrec_rna_32   \n",
       "339  Rpac_peg_771    Rpha_peg_1634  Rpli_peg_1007  Rrec_peg_574   \n",
       "976  Rpac_peg_686    Rpha_peg_1413   Rpli_peg_898  Rrec_peg_483   \n",
       "\n",
       "     R.southwardae  V.diagonalis     V.extenta  V.okutanii       V.gigas2  \\\n",
       "90      Rsou_rna_4   Vdia_rna_21   Vext_rna_22  COSY_R0016   Vgig2_rna_15   \n",
       "339  Rsou_peg_1741           NaN           NaN         NaN            NaN   \n",
       "976   Rsou_peg_242  Vdia_peg_128  Vext_peg_123   COSY_0827  Vgig2_peg_736   \n",
       "\n",
       "        V.soyoae2       V.gigas1      V.soyoae1  Count  \n",
       "90   Vsoy2_rna_29   Vgig1_rna_19   Vsoy1_rna_18     17  \n",
       "339           NaN            NaN            NaN      9  \n",
       "976  Vsoy2_peg_83  Vgig1_peg_849  Vsoy1_peg_867     17  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ALL[ALL.apply(lambda r: r.isin(['Vgig2_peg_984','Vsoy2_peg_206','Vsoy1_peg_1','COSY_0953','Vext_peg_997','Vdia_peg_1017','Vsoy2_peg_206','COSY_0953','Vgig2_peg_983','Vgig1_peg_985']).any(), axis=1)]\n",
    "\n",
    "ALL[ALL.apply(lambda r: r.isin(['Rmag_0921', 'Rmag_R0017','Rmag_0989']).any(), axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "669\n",
      "core genes 669 667\n",
      "2\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0       Rmag_0520\n",
       "1       Rmag_0518\n",
       "4       Rmag_0514\n",
       "5       Rmag_0513\n",
       "6       Rmag_0512\n",
       "          ...    \n",
       "1064    Rmag_0290\n",
       "1099    Rmag_0013\n",
       "1100    Rmag_0232\n",
       "1115    Rmag_0249\n",
       "1116    Rmag_0329\n",
       "Name: R.magnifica, Length: 638, dtype: object"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Are the core genes in no_pseudo dataset the same orthologies than these in the old orthology?\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "ORTH=pd.read_csv(path_to_files+'Ortho_all_28052020.txt',sep='\\t')\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',sep='\\t', index_col=0)\n",
    "ALL=ALL[['R.fausta','V.marissinica','Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1','Count']]\n",
    "# print(ALL[:5])\n",
    "CORE_nopseudo=ALL[ALL['Count']==17]['R.magnifica'].values\n",
    "print(len(CORE_nopseudo))\n",
    "CORE_nopseudo\n",
    "CORE2=ORTH[ORTH.apply(lambda r: r.isin(CORE_nopseudo).any(), axis=1)]\n",
    "CORE2=CORE2[CORE2['genome_count']==15]\n",
    "\n",
    "CORE2['R.pliocardia'] = CORE2['R.pleocardia'].str.replace('Rple','Rpli')\n",
    "CORE2=CORE2[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "core2_str=['|'.join(CORE2.iloc[i].values) for i in range(len(CORE2))]\n",
    "\n",
    "\n",
    "CORE1=ALL[ALL['Count']==17]\n",
    "CORE1=CORE1[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis','R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis','V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1','V.soyoae1']]\n",
    "core1_str=['|'.join(CORE1.iloc[i].values) for i in range(len(CORE1))]\n",
    "\n",
    "print('core genes',len(core1_str), len(core2_str))\n",
    "\n",
    "set(CORE1['R.magnifica'].values)-set(CORE2['R.magnifica'].values)\n",
    "\n",
    "# CORE1=CORE1[~CORE1['R.magnifica'].isin(['Rmag_0921', 'Rmag_R0017'])]\n",
    "# CORE2=CORE2[~CORE2['R.magnifica'].isin(['Rmag_0921', 'Rmag_R0017'])]\n",
    "# core2_str=['|'.join(CORE2.iloc[i].values) for i in range(len(CORE2))]\n",
    "# core1_str=['|'.join(CORE1.iloc[i].values) for i in range(len(CORE1))]\n",
    "\n",
    "print(len(set(core1_str)-set(core2_str)))\n",
    "set(core1_str)-set(core2_str)\n",
    "\n",
    "# CORE3=CORE1[CORE1.apply(lambda col: col.str.contains('_rna_|_R',regex=True), axis=1)]\n",
    "\n",
    "CORE1[~(CORE1['R.magnifica'].str.contains(\"_R\"))]['R.magnifica']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "669\n"
     ]
    }
   ],
   "source": [
    "print(len(CORE))\n",
    "# CORE['R.magnifica'].to_csv('test', index=False,header=False)\n",
    "output=CORE[~(CORE['R.magnifica'].str.contains(\"_R\"))]['R.magnifica'].copy()\n",
    "output.sort_values(inplace=True)\n",
    "output.to_csv(path_to_files+'fna/filelist_nopseudo_669', index=False,header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "58\n",
      "['Rmag_0047', 'Rmag_0082', 'Rmag_0104', 'Rmag_0124', 'Rmag_0125', 'Rmag_0135', 'Rmag_0140', 'Rmag_0152', 'Rmag_0183', 'Rmag_0234', 'Rmag_0260', 'Rmag_0263', 'Rmag_0269', 'Rmag_0312', 'Rmag_0320', 'Rmag_0331', 'Rmag_0350', 'Rmag_0383', 'Rmag_0405', 'Rmag_0455', 'Rmag_0481', 'Rmag_0497', 'Rmag_0525', 'Rmag_0528', 'Rmag_0589', 'Rmag_0591', 'Rmag_0609', 'Rmag_0647', 'Rmag_0664', 'Rmag_0689', 'Rmag_0719', 'Rmag_0737', 'Rmag_0766', 'Rmag_0794', 'Rmag_0795', 'Rmag_0931', 'Rmag_0933', 'Rmag_0976', 'Rmag_0982', 'Rmag_0989', 'Rmag_0990', 'Rmag_0997', 'Rmag_1016', 'Rmag_1018', 'Rmag_1026', 'Rmag_1030', 'Rmag_1033', 'Rmag_1035', 'Rmag_1038', 'Rmag_1040', 'Rmag_1045', 'Rmag_1051', 'Rmag_1052', 'Rmag_1060', 'Rmag_1063', 'Rmag_1064', 'Rmag_1068', 'Rmag_1071']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[24, 'Rmag_0047'],\n",
       " [46, 'Rmag_0082'],\n",
       " [61, 'Rmag_0104'],\n",
       " [77, 'Rmag_0124'],\n",
       " [78, 'Rmag_0125'],\n",
       " [84, 'Rmag_0135'],\n",
       " [89, 'Rmag_0140'],\n",
       " [97, 'Rmag_0152'],\n",
       " [121, 'Rmag_0183'],\n",
       " [153, 'Rmag_0234'],\n",
       " [170, 'Rmag_0260'],\n",
       " [173, 'Rmag_0263'],\n",
       " [177, 'Rmag_0269'],\n",
       " [207, 'Rmag_0312'],\n",
       " [210, 'Rmag_0320'],\n",
       " [217, 'Rmag_0331'],\n",
       " [231, 'Rmag_0350'],\n",
       " [252, 'Rmag_0383'],\n",
       " [264, 'Rmag_0405'],\n",
       " [294, 'Rmag_0455'],\n",
       " [311, 'Rmag_0481'],\n",
       " [323, 'Rmag_0497'],\n",
       " [341, 'Rmag_0525'],\n",
       " [343, 'Rmag_0528'],\n",
       " [386, 'Rmag_0589'],\n",
       " [387, 'Rmag_0591'],\n",
       " [401, 'Rmag_0609'],\n",
       " [420, 'Rmag_0647'],\n",
       " [430, 'Rmag_0664'],\n",
       " [447, 'Rmag_0689'],\n",
       " [458, 'Rmag_0719'],\n",
       " [465, 'Rmag_0737'],\n",
       " [482, 'Rmag_0766'],\n",
       " [495, 'Rmag_0794'],\n",
       " [496, 'Rmag_0795'],\n",
       " [582, 'Rmag_0931'],\n",
       " [584, 'Rmag_0933'],\n",
       " [613, 'Rmag_0976'],\n",
       " [617, 'Rmag_0982'],\n",
       " [620, 'Rmag_0989'],\n",
       " [621, 'Rmag_0990'],\n",
       " [628, 'Rmag_0997'],\n",
       " [646, 'Rmag_1016'],\n",
       " [647, 'Rmag_1018'],\n",
       " [654, 'Rmag_1026'],\n",
       " [656, 'Rmag_1030'],\n",
       " [658, 'Rmag_1033'],\n",
       " [659, 'Rmag_1035'],\n",
       " [661, 'Rmag_1038'],\n",
       " [662, 'Rmag_1040'],\n",
       " [666, 'Rmag_1045'],\n",
       " [672, 'Rmag_1051'],\n",
       " [673, 'Rmag_1052'],\n",
       " [678, 'Rmag_1060'],\n",
       " [681, 'Rmag_1063'],\n",
       " [682, 'Rmag_1064'],\n",
       " [685, 'Rmag_1068'],\n",
       " [687, 'Rmag_1071']]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('./fna/filelist','r') as f:\n",
    "    old=sorted([line[8:-4] for line in f.read().splitlines()])\n",
    "# print(len(old),old)\n",
    "print(len(set(old)-set(CORE['R.magnifica'].values)))\n",
    "toboot=sorted(set(old)-set(CORE['R.magnifica'].values))\n",
    "print(toboot)\n",
    "[[old.index(item)+1, item] for item in toboot]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[1, 'Rmag_0001'],\n",
       " [2, 'Rmag_0002'],\n",
       " [3, 'Rmag_0003'],\n",
       " [4, 'Rmag_0005'],\n",
       " [5, 'Rmag_0006'],\n",
       " [6, 'Rmag_0007'],\n",
       " [7, 'Rmag_0008'],\n",
       " [8, 'Rmag_0009'],\n",
       " [9, 'Rmag_0010'],\n",
       " [10, 'Rmag_0011'],\n",
       " [11, 'Rmag_0012'],\n",
       " [12, 'Rmag_0013'],\n",
       " [13, 'Rmag_0015'],\n",
       " [14, 'Rmag_0016'],\n",
       " [15, 'Rmag_0018'],\n",
       " [16, 'Rmag_0025'],\n",
       " [17, 'Rmag_0026'],\n",
       " [18, 'Rmag_0027'],\n",
       " [19, 'Rmag_0028'],\n",
       " [20, 'Rmag_0043'],\n",
       " [21, 'Rmag_0044'],\n",
       " [22, 'Rmag_0045'],\n",
       " [23, 'Rmag_0046'],\n",
       " [24, 'Rmag_0047'],\n",
       " [25, 'Rmag_0050'],\n",
       " [26, 'Rmag_0051'],\n",
       " [27, 'Rmag_0052'],\n",
       " [28, 'Rmag_0053'],\n",
       " [29, 'Rmag_0054'],\n",
       " [30, 'Rmag_0055'],\n",
       " [31, 'Rmag_0061'],\n",
       " [32, 'Rmag_0062'],\n",
       " [33, 'Rmag_0063'],\n",
       " [34, 'Rmag_0065'],\n",
       " [35, 'Rmag_0066'],\n",
       " [36, 'Rmag_0067'],\n",
       " [37, 'Rmag_0068'],\n",
       " [38, 'Rmag_0070'],\n",
       " [39, 'Rmag_0071'],\n",
       " [40, 'Rmag_0072'],\n",
       " [41, 'Rmag_0073'],\n",
       " [42, 'Rmag_0077'],\n",
       " [43, 'Rmag_0078'],\n",
       " [44, 'Rmag_0079'],\n",
       " [45, 'Rmag_0080'],\n",
       " [46, 'Rmag_0082'],\n",
       " [47, 'Rmag_0083'],\n",
       " [48, 'Rmag_0085'],\n",
       " [49, 'Rmag_0086'],\n",
       " [50, 'Rmag_0087'],\n",
       " [51, 'Rmag_0088'],\n",
       " [52, 'Rmag_0089'],\n",
       " [53, 'Rmag_0090'],\n",
       " [54, 'Rmag_0091'],\n",
       " [55, 'Rmag_0094'],\n",
       " [56, 'Rmag_0097'],\n",
       " [57, 'Rmag_0098'],\n",
       " [58, 'Rmag_0099'],\n",
       " [59, 'Rmag_0101'],\n",
       " [60, 'Rmag_0103'],\n",
       " [61, 'Rmag_0104'],\n",
       " [62, 'Rmag_0105'],\n",
       " [63, 'Rmag_0106'],\n",
       " [64, 'Rmag_0107'],\n",
       " [65, 'Rmag_0108'],\n",
       " [66, 'Rmag_0109'],\n",
       " [67, 'Rmag_0111'],\n",
       " [68, 'Rmag_0112'],\n",
       " [69, 'Rmag_0113'],\n",
       " [70, 'Rmag_0116'],\n",
       " [71, 'Rmag_0117'],\n",
       " [72, 'Rmag_0118'],\n",
       " [73, 'Rmag_0120'],\n",
       " [74, 'Rmag_0121'],\n",
       " [75, 'Rmag_0122'],\n",
       " [76, 'Rmag_0123'],\n",
       " [77, 'Rmag_0124'],\n",
       " [78, 'Rmag_0125'],\n",
       " [79, 'Rmag_0126'],\n",
       " [80, 'Rmag_0127'],\n",
       " [81, 'Rmag_0132'],\n",
       " [82, 'Rmag_0133'],\n",
       " [83, 'Rmag_0134'],\n",
       " [84, 'Rmag_0135'],\n",
       " [85, 'Rmag_0136'],\n",
       " [86, 'Rmag_0137'],\n",
       " [87, 'Rmag_0138'],\n",
       " [88, 'Rmag_0139'],\n",
       " [89, 'Rmag_0140'],\n",
       " [90, 'Rmag_0141'],\n",
       " [91, 'Rmag_0142'],\n",
       " [92, 'Rmag_0143'],\n",
       " [93, 'Rmag_0144'],\n",
       " [94, 'Rmag_0146'],\n",
       " [95, 'Rmag_0147'],\n",
       " [96, 'Rmag_0151'],\n",
       " [97, 'Rmag_0152'],\n",
       " [98, 'Rmag_0153'],\n",
       " [99, 'Rmag_0156'],\n",
       " [100, 'Rmag_0160'],\n",
       " [101, 'Rmag_0161'],\n",
       " [102, 'Rmag_0164'],\n",
       " [103, 'Rmag_0165'],\n",
       " [104, 'Rmag_0166'],\n",
       " [105, 'Rmag_0167'],\n",
       " [106, 'Rmag_0168'],\n",
       " [107, 'Rmag_0169'],\n",
       " [108, 'Rmag_0170'],\n",
       " [109, 'Rmag_0171'],\n",
       " [110, 'Rmag_0172'],\n",
       " [111, 'Rmag_0173'],\n",
       " [112, 'Rmag_0174'],\n",
       " [113, 'Rmag_0175'],\n",
       " [114, 'Rmag_0176'],\n",
       " [115, 'Rmag_0177'],\n",
       " [116, 'Rmag_0178'],\n",
       " [117, 'Rmag_0179'],\n",
       " [118, 'Rmag_0180'],\n",
       " [119, 'Rmag_0181'],\n",
       " [120, 'Rmag_0182'],\n",
       " [121, 'Rmag_0183'],\n",
       " [122, 'Rmag_0184'],\n",
       " [123, 'Rmag_0185'],\n",
       " [124, 'Rmag_0187'],\n",
       " [125, 'Rmag_0188'],\n",
       " [126, 'Rmag_0189'],\n",
       " [127, 'Rmag_0190'],\n",
       " [128, 'Rmag_0191'],\n",
       " [129, 'Rmag_0192'],\n",
       " [130, 'Rmag_0197'],\n",
       " [131, 'Rmag_0199'],\n",
       " [132, 'Rmag_0200'],\n",
       " [133, 'Rmag_0201'],\n",
       " [134, 'Rmag_0203'],\n",
       " [135, 'Rmag_0204'],\n",
       " [136, 'Rmag_0205'],\n",
       " [137, 'Rmag_0206'],\n",
       " [138, 'Rmag_0208'],\n",
       " [139, 'Rmag_0211'],\n",
       " [140, 'Rmag_0213'],\n",
       " [141, 'Rmag_0214'],\n",
       " [142, 'Rmag_0215'],\n",
       " [143, 'Rmag_0216'],\n",
       " [144, 'Rmag_0217'],\n",
       " [145, 'Rmag_0221'],\n",
       " [146, 'Rmag_0222'],\n",
       " [147, 'Rmag_0223'],\n",
       " [148, 'Rmag_0224'],\n",
       " [149, 'Rmag_0225'],\n",
       " [150, 'Rmag_0227'],\n",
       " [151, 'Rmag_0232'],\n",
       " [152, 'Rmag_0233'],\n",
       " [153, 'Rmag_0234'],\n",
       " [154, 'Rmag_0236'],\n",
       " [155, 'Rmag_0237'],\n",
       " [156, 'Rmag_0239'],\n",
       " [157, 'Rmag_0240'],\n",
       " [158, 'Rmag_0241'],\n",
       " [159, 'Rmag_0242'],\n",
       " [160, 'Rmag_0243'],\n",
       " [161, 'Rmag_0244'],\n",
       " [162, 'Rmag_0245'],\n",
       " [163, 'Rmag_0246'],\n",
       " [164, 'Rmag_0247'],\n",
       " [165, 'Rmag_0248'],\n",
       " [166, 'Rmag_0249'],\n",
       " [167, 'Rmag_0250'],\n",
       " [168, 'Rmag_0257'],\n",
       " [169, 'Rmag_0258'],\n",
       " [170, 'Rmag_0260'],\n",
       " [171, 'Rmag_0261'],\n",
       " [172, 'Rmag_0262'],\n",
       " [173, 'Rmag_0263'],\n",
       " [174, 'Rmag_0265'],\n",
       " [175, 'Rmag_0266'],\n",
       " [176, 'Rmag_0267'],\n",
       " [177, 'Rmag_0269'],\n",
       " [178, 'Rmag_0270'],\n",
       " [179, 'Rmag_0271'],\n",
       " [180, 'Rmag_0272'],\n",
       " [181, 'Rmag_0273'],\n",
       " [182, 'Rmag_0274'],\n",
       " [183, 'Rmag_0275'],\n",
       " [184, 'Rmag_0276'],\n",
       " [185, 'Rmag_0278'],\n",
       " [186, 'Rmag_0279'],\n",
       " [187, 'Rmag_0280'],\n",
       " [188, 'Rmag_0281'],\n",
       " [189, 'Rmag_0284'],\n",
       " [190, 'Rmag_0285'],\n",
       " [191, 'Rmag_0286'],\n",
       " [192, 'Rmag_0287'],\n",
       " [193, 'Rmag_0288'],\n",
       " [194, 'Rmag_0289'],\n",
       " [195, 'Rmag_0290'],\n",
       " [196, 'Rmag_0292'],\n",
       " [197, 'Rmag_0293'],\n",
       " [198, 'Rmag_0294'],\n",
       " [199, 'Rmag_0295'],\n",
       " [200, 'Rmag_0298'],\n",
       " [201, 'Rmag_0302'],\n",
       " [202, 'Rmag_0305'],\n",
       " [203, 'Rmag_0306'],\n",
       " [204, 'Rmag_0307'],\n",
       " [205, 'Rmag_0308'],\n",
       " [206, 'Rmag_0311'],\n",
       " [207, 'Rmag_0312'],\n",
       " [208, 'Rmag_0314'],\n",
       " [209, 'Rmag_0319'],\n",
       " [210, 'Rmag_0320'],\n",
       " [211, 'Rmag_0321'],\n",
       " [212, 'Rmag_0323'],\n",
       " [213, 'Rmag_0325'],\n",
       " [214, 'Rmag_0327'],\n",
       " [215, 'Rmag_0328'],\n",
       " [216, 'Rmag_0329'],\n",
       " [217, 'Rmag_0331'],\n",
       " [218, 'Rmag_0333'],\n",
       " [219, 'Rmag_0336'],\n",
       " [220, 'Rmag_0338'],\n",
       " [221, 'Rmag_0339'],\n",
       " [222, 'Rmag_0340'],\n",
       " [223, 'Rmag_0341'],\n",
       " [224, 'Rmag_0342'],\n",
       " [225, 'Rmag_0343'],\n",
       " [226, 'Rmag_0344'],\n",
       " [227, 'Rmag_0345'],\n",
       " [228, 'Rmag_0347'],\n",
       " [229, 'Rmag_0348'],\n",
       " [230, 'Rmag_0349'],\n",
       " [231, 'Rmag_0350'],\n",
       " [232, 'Rmag_0351'],\n",
       " [233, 'Rmag_0352'],\n",
       " [234, 'Rmag_0353'],\n",
       " [235, 'Rmag_0354'],\n",
       " [236, 'Rmag_0356'],\n",
       " [237, 'Rmag_0361'],\n",
       " [238, 'Rmag_0362'],\n",
       " [239, 'Rmag_0363'],\n",
       " [240, 'Rmag_0364'],\n",
       " [241, 'Rmag_0365'],\n",
       " [242, 'Rmag_0366'],\n",
       " [243, 'Rmag_0367'],\n",
       " [244, 'Rmag_0368'],\n",
       " [245, 'Rmag_0371'],\n",
       " [246, 'Rmag_0372'],\n",
       " [247, 'Rmag_0373'],\n",
       " [248, 'Rmag_0374'],\n",
       " [249, 'Rmag_0378'],\n",
       " [250, 'Rmag_0379'],\n",
       " [251, 'Rmag_0382'],\n",
       " [252, 'Rmag_0383'],\n",
       " [253, 'Rmag_0384'],\n",
       " [254, 'Rmag_0385'],\n",
       " [255, 'Rmag_0386'],\n",
       " [256, 'Rmag_0388'],\n",
       " [257, 'Rmag_0389'],\n",
       " [258, 'Rmag_0394'],\n",
       " [259, 'Rmag_0395'],\n",
       " [260, 'Rmag_0396'],\n",
       " [261, 'Rmag_0397'],\n",
       " [262, 'Rmag_0398'],\n",
       " [263, 'Rmag_0404'],\n",
       " [264, 'Rmag_0405'],\n",
       " [265, 'Rmag_0409'],\n",
       " [266, 'Rmag_0410'],\n",
       " [267, 'Rmag_0411'],\n",
       " [268, 'Rmag_0412'],\n",
       " [269, 'Rmag_0413'],\n",
       " [270, 'Rmag_0414'],\n",
       " [271, 'Rmag_0415'],\n",
       " [272, 'Rmag_0416'],\n",
       " [273, 'Rmag_0417'],\n",
       " [274, 'Rmag_0419'],\n",
       " [275, 'Rmag_0421'],\n",
       " [276, 'Rmag_0423'],\n",
       " [277, 'Rmag_0425'],\n",
       " [278, 'Rmag_0430'],\n",
       " [279, 'Rmag_0432'],\n",
       " [280, 'Rmag_0433'],\n",
       " [281, 'Rmag_0434'],\n",
       " [282, 'Rmag_0435'],\n",
       " [283, 'Rmag_0437'],\n",
       " [284, 'Rmag_0438'],\n",
       " [285, 'Rmag_0444'],\n",
       " [286, 'Rmag_0445'],\n",
       " [287, 'Rmag_0447'],\n",
       " [288, 'Rmag_0448'],\n",
       " [289, 'Rmag_0449'],\n",
       " [290, 'Rmag_0450'],\n",
       " [291, 'Rmag_0452'],\n",
       " [292, 'Rmag_0453'],\n",
       " [293, 'Rmag_0454'],\n",
       " [294, 'Rmag_0455'],\n",
       " [295, 'Rmag_0456'],\n",
       " [296, 'Rmag_0457'],\n",
       " [297, 'Rmag_0458'],\n",
       " [298, 'Rmag_0462'],\n",
       " [299, 'Rmag_0463'],\n",
       " [300, 'Rmag_0464'],\n",
       " [301, 'Rmag_0466'],\n",
       " [302, 'Rmag_0469'],\n",
       " [303, 'Rmag_0470'],\n",
       " [304, 'Rmag_0472'],\n",
       " [305, 'Rmag_0473'],\n",
       " [306, 'Rmag_0474'],\n",
       " [307, 'Rmag_0475'],\n",
       " [308, 'Rmag_0476'],\n",
       " [309, 'Rmag_0477'],\n",
       " [310, 'Rmag_0478'],\n",
       " [311, 'Rmag_0481'],\n",
       " [312, 'Rmag_0482'],\n",
       " [313, 'Rmag_0484'],\n",
       " [314, 'Rmag_0485'],\n",
       " [315, 'Rmag_0486'],\n",
       " [316, 'Rmag_0487'],\n",
       " [317, 'Rmag_0490'],\n",
       " [318, 'Rmag_0491'],\n",
       " [319, 'Rmag_0492'],\n",
       " [320, 'Rmag_0493'],\n",
       " [321, 'Rmag_0495'],\n",
       " [322, 'Rmag_0496'],\n",
       " [323, 'Rmag_0497'],\n",
       " [324, 'Rmag_0498'],\n",
       " [325, 'Rmag_0499'],\n",
       " [326, 'Rmag_0502'],\n",
       " [327, 'Rmag_0504'],\n",
       " [328, 'Rmag_0505'],\n",
       " [329, 'Rmag_0506'],\n",
       " [330, 'Rmag_0507'],\n",
       " [331, 'Rmag_0510'],\n",
       " [332, 'Rmag_0511'],\n",
       " [333, 'Rmag_0512'],\n",
       " [334, 'Rmag_0513'],\n",
       " [335, 'Rmag_0514'],\n",
       " [336, 'Rmag_0518'],\n",
       " [337, 'Rmag_0520'],\n",
       " [338, 'Rmag_0522'],\n",
       " [339, 'Rmag_0523'],\n",
       " [340, 'Rmag_0524'],\n",
       " [341, 'Rmag_0525'],\n",
       " [342, 'Rmag_0526'],\n",
       " [343, 'Rmag_0528'],\n",
       " [344, 'Rmag_0529'],\n",
       " [345, 'Rmag_0531'],\n",
       " [346, 'Rmag_0532'],\n",
       " [347, 'Rmag_0533'],\n",
       " [348, 'Rmag_0534'],\n",
       " [349, 'Rmag_0535'],\n",
       " [350, 'Rmag_0542'],\n",
       " [351, 'Rmag_0543'],\n",
       " [352, 'Rmag_0545'],\n",
       " [353, 'Rmag_0546'],\n",
       " [354, 'Rmag_0547'],\n",
       " [355, 'Rmag_0548'],\n",
       " [356, 'Rmag_0549'],\n",
       " [357, 'Rmag_0550'],\n",
       " [358, 'Rmag_0553'],\n",
       " [359, 'Rmag_0554'],\n",
       " [360, 'Rmag_0555'],\n",
       " [361, 'Rmag_0556'],\n",
       " [362, 'Rmag_0558'],\n",
       " [363, 'Rmag_0559'],\n",
       " [364, 'Rmag_0560'],\n",
       " [365, 'Rmag_0561'],\n",
       " [366, 'Rmag_0563'],\n",
       " [367, 'Rmag_0565'],\n",
       " [368, 'Rmag_0566'],\n",
       " [369, 'Rmag_0567'],\n",
       " [370, 'Rmag_0568'],\n",
       " [371, 'Rmag_0571'],\n",
       " [372, 'Rmag_0572'],\n",
       " [373, 'Rmag_0573'],\n",
       " [374, 'Rmag_0574'],\n",
       " [375, 'Rmag_0575'],\n",
       " [376, 'Rmag_0576'],\n",
       " [377, 'Rmag_0577'],\n",
       " [378, 'Rmag_0578'],\n",
       " [379, 'Rmag_0579'],\n",
       " [380, 'Rmag_0581'],\n",
       " [381, 'Rmag_0583'],\n",
       " [382, 'Rmag_0584'],\n",
       " [383, 'Rmag_0585'],\n",
       " [384, 'Rmag_0587'],\n",
       " [385, 'Rmag_0588'],\n",
       " [386, 'Rmag_0589'],\n",
       " [387, 'Rmag_0591'],\n",
       " [388, 'Rmag_0592'],\n",
       " [389, 'Rmag_0593'],\n",
       " [390, 'Rmag_0594'],\n",
       " [391, 'Rmag_0595'],\n",
       " [392, 'Rmag_0597'],\n",
       " [393, 'Rmag_0599'],\n",
       " [394, 'Rmag_0601'],\n",
       " [395, 'Rmag_0602'],\n",
       " [396, 'Rmag_0604'],\n",
       " [397, 'Rmag_0605'],\n",
       " [398, 'Rmag_0606'],\n",
       " [399, 'Rmag_0607'],\n",
       " [400, 'Rmag_0608'],\n",
       " [401, 'Rmag_0609'],\n",
       " [402, 'Rmag_0611'],\n",
       " [403, 'Rmag_0613'],\n",
       " [404, 'Rmag_0614'],\n",
       " [405, 'Rmag_0615'],\n",
       " [406, 'Rmag_0617'],\n",
       " [407, 'Rmag_0618'],\n",
       " [408, 'Rmag_0619'],\n",
       " [409, 'Rmag_0620'],\n",
       " [410, 'Rmag_0621'],\n",
       " [411, 'Rmag_0622'],\n",
       " [412, 'Rmag_0626'],\n",
       " [413, 'Rmag_0628'],\n",
       " [414, 'Rmag_0632'],\n",
       " [415, 'Rmag_0635'],\n",
       " [416, 'Rmag_0640'],\n",
       " [417, 'Rmag_0642'],\n",
       " [418, 'Rmag_0643'],\n",
       " [419, 'Rmag_0645'],\n",
       " [420, 'Rmag_0647'],\n",
       " [421, 'Rmag_0648'],\n",
       " [422, 'Rmag_0649'],\n",
       " [423, 'Rmag_0651'],\n",
       " [424, 'Rmag_0652'],\n",
       " [425, 'Rmag_0654'],\n",
       " [426, 'Rmag_0656'],\n",
       " [427, 'Rmag_0659'],\n",
       " [428, 'Rmag_0661'],\n",
       " [429, 'Rmag_0662'],\n",
       " [430, 'Rmag_0664'],\n",
       " [431, 'Rmag_0665'],\n",
       " [432, 'Rmag_0666'],\n",
       " [433, 'Rmag_0667'],\n",
       " [434, 'Rmag_0668'],\n",
       " [435, 'Rmag_0669'],\n",
       " [436, 'Rmag_0671'],\n",
       " [437, 'Rmag_0672'],\n",
       " [438, 'Rmag_0673'],\n",
       " [439, 'Rmag_0674'],\n",
       " [440, 'Rmag_0676'],\n",
       " [441, 'Rmag_0677'],\n",
       " [442, 'Rmag_0678'],\n",
       " [443, 'Rmag_0679'],\n",
       " [444, 'Rmag_0680'],\n",
       " [445, 'Rmag_0683'],\n",
       " [446, 'Rmag_0684'],\n",
       " [447, 'Rmag_0689'],\n",
       " [448, 'Rmag_0690'],\n",
       " [449, 'Rmag_0691'],\n",
       " [450, 'Rmag_0692'],\n",
       " [451, 'Rmag_0693'],\n",
       " [452, 'Rmag_0694'],\n",
       " [453, 'Rmag_0698'],\n",
       " [454, 'Rmag_0700'],\n",
       " [455, 'Rmag_0704'],\n",
       " [456, 'Rmag_0714'],\n",
       " [457, 'Rmag_0717'],\n",
       " [458, 'Rmag_0719'],\n",
       " [459, 'Rmag_0721'],\n",
       " [460, 'Rmag_0730'],\n",
       " [461, 'Rmag_0731'],\n",
       " [462, 'Rmag_0732'],\n",
       " [463, 'Rmag_0733'],\n",
       " [464, 'Rmag_0736'],\n",
       " [465, 'Rmag_0737'],\n",
       " [466, 'Rmag_0738'],\n",
       " [467, 'Rmag_0739'],\n",
       " [468, 'Rmag_0740'],\n",
       " [469, 'Rmag_0741'],\n",
       " [470, 'Rmag_0743'],\n",
       " [471, 'Rmag_0744'],\n",
       " [472, 'Rmag_0745'],\n",
       " [473, 'Rmag_0747'],\n",
       " [474, 'Rmag_0749'],\n",
       " [475, 'Rmag_0755'],\n",
       " [476, 'Rmag_0756'],\n",
       " [477, 'Rmag_0757'],\n",
       " [478, 'Rmag_0758'],\n",
       " [479, 'Rmag_0759'],\n",
       " [480, 'Rmag_0763'],\n",
       " [481, 'Rmag_0765'],\n",
       " [482, 'Rmag_0766'],\n",
       " [483, 'Rmag_0767'],\n",
       " [484, 'Rmag_0769'],\n",
       " [485, 'Rmag_0770'],\n",
       " [486, 'Rmag_0772'],\n",
       " [487, 'Rmag_0774'],\n",
       " [488, 'Rmag_0781'],\n",
       " [489, 'Rmag_0783'],\n",
       " [490, 'Rmag_0784'],\n",
       " [491, 'Rmag_0787'],\n",
       " [492, 'Rmag_0788'],\n",
       " [493, 'Rmag_0790'],\n",
       " [494, 'Rmag_0793'],\n",
       " [495, 'Rmag_0794'],\n",
       " [496, 'Rmag_0795'],\n",
       " [497, 'Rmag_0800'],\n",
       " [498, 'Rmag_0801'],\n",
       " [499, 'Rmag_0804'],\n",
       " [500, 'Rmag_0805'],\n",
       " [501, 'Rmag_0806'],\n",
       " [502, 'Rmag_0807'],\n",
       " [503, 'Rmag_0808'],\n",
       " [504, 'Rmag_0809'],\n",
       " [505, 'Rmag_0812'],\n",
       " [506, 'Rmag_0813'],\n",
       " [507, 'Rmag_0814'],\n",
       " [508, 'Rmag_0815'],\n",
       " [509, 'Rmag_0816'],\n",
       " [510, 'Rmag_0819'],\n",
       " [511, 'Rmag_0822'],\n",
       " [512, 'Rmag_0824'],\n",
       " [513, 'Rmag_0825'],\n",
       " [514, 'Rmag_0828'],\n",
       " [515, 'Rmag_0831'],\n",
       " [516, 'Rmag_0832'],\n",
       " [517, 'Rmag_0833'],\n",
       " [518, 'Rmag_0834'],\n",
       " [519, 'Rmag_0835'],\n",
       " [520, 'Rmag_0836'],\n",
       " [521, 'Rmag_0837'],\n",
       " [522, 'Rmag_0838'],\n",
       " [523, 'Rmag_0839'],\n",
       " [524, 'Rmag_0840'],\n",
       " [525, 'Rmag_0841'],\n",
       " [526, 'Rmag_0843'],\n",
       " [527, 'Rmag_0844'],\n",
       " [528, 'Rmag_0845'],\n",
       " [529, 'Rmag_0847'],\n",
       " [530, 'Rmag_0848'],\n",
       " [531, 'Rmag_0850'],\n",
       " [532, 'Rmag_0851'],\n",
       " [533, 'Rmag_0852'],\n",
       " [534, 'Rmag_0853'],\n",
       " [535, 'Rmag_0854'],\n",
       " [536, 'Rmag_0855'],\n",
       " [537, 'Rmag_0856'],\n",
       " [538, 'Rmag_0858'],\n",
       " [539, 'Rmag_0859'],\n",
       " [540, 'Rmag_0860'],\n",
       " [541, 'Rmag_0861'],\n",
       " [542, 'Rmag_0862'],\n",
       " [543, 'Rmag_0863'],\n",
       " [544, 'Rmag_0864'],\n",
       " [545, 'Rmag_0865'],\n",
       " [546, 'Rmag_0866'],\n",
       " [547, 'Rmag_0867'],\n",
       " [548, 'Rmag_0868'],\n",
       " [549, 'Rmag_0869'],\n",
       " [550, 'Rmag_0870'],\n",
       " [551, 'Rmag_0871'],\n",
       " [552, 'Rmag_0872'],\n",
       " [553, 'Rmag_0874'],\n",
       " [554, 'Rmag_0875'],\n",
       " [555, 'Rmag_0876'],\n",
       " [556, 'Rmag_0877'],\n",
       " [557, 'Rmag_0878'],\n",
       " [558, 'Rmag_0879'],\n",
       " [559, 'Rmag_0880'],\n",
       " [560, 'Rmag_0881'],\n",
       " [561, 'Rmag_0882'],\n",
       " [562, 'Rmag_0883'],\n",
       " [563, 'Rmag_0884'],\n",
       " [564, 'Rmag_0885'],\n",
       " [565, 'Rmag_0910'],\n",
       " [566, 'Rmag_0911'],\n",
       " [567, 'Rmag_0914'],\n",
       " [568, 'Rmag_0915'],\n",
       " [569, 'Rmag_0916'],\n",
       " [570, 'Rmag_0917'],\n",
       " [571, 'Rmag_0918'],\n",
       " [572, 'Rmag_0919'],\n",
       " [573, 'Rmag_0921'],\n",
       " [574, 'Rmag_0922'],\n",
       " [575, 'Rmag_0923'],\n",
       " [576, 'Rmag_0924'],\n",
       " [577, 'Rmag_0925'],\n",
       " [578, 'Rmag_0926'],\n",
       " [579, 'Rmag_0927'],\n",
       " [580, 'Rmag_0928'],\n",
       " [581, 'Rmag_0929'],\n",
       " [582, 'Rmag_0931'],\n",
       " [583, 'Rmag_0932'],\n",
       " [584, 'Rmag_0933'],\n",
       " [585, 'Rmag_0934'],\n",
       " [586, 'Rmag_0935'],\n",
       " [587, 'Rmag_0936'],\n",
       " [588, 'Rmag_0942'],\n",
       " [589, 'Rmag_0944'],\n",
       " [590, 'Rmag_0946'],\n",
       " [591, 'Rmag_0947'],\n",
       " [592, 'Rmag_0949'],\n",
       " [593, 'Rmag_0950'],\n",
       " [594, 'Rmag_0951'],\n",
       " [595, 'Rmag_0952'],\n",
       " [596, 'Rmag_0953'],\n",
       " [597, 'Rmag_0954'],\n",
       " [598, 'Rmag_0955'],\n",
       " [599, 'Rmag_0956'],\n",
       " [600, 'Rmag_0957'],\n",
       " [601, 'Rmag_0958'],\n",
       " [602, 'Rmag_0962'],\n",
       " [603, 'Rmag_0963'],\n",
       " [604, 'Rmag_0964'],\n",
       " [605, 'Rmag_0965'],\n",
       " [606, 'Rmag_0966'],\n",
       " [607, 'Rmag_0967'],\n",
       " [608, 'Rmag_0969'],\n",
       " [609, 'Rmag_0971'],\n",
       " [610, 'Rmag_0972'],\n",
       " [611, 'Rmag_0973'],\n",
       " [612, 'Rmag_0975'],\n",
       " [613, 'Rmag_0976'],\n",
       " [614, 'Rmag_0977'],\n",
       " [615, 'Rmag_0980'],\n",
       " [616, 'Rmag_0981'],\n",
       " [617, 'Rmag_0982'],\n",
       " [618, 'Rmag_0984'],\n",
       " [619, 'Rmag_0987'],\n",
       " [620, 'Rmag_0989'],\n",
       " [621, 'Rmag_0990'],\n",
       " [622, 'Rmag_0991'],\n",
       " [623, 'Rmag_0992'],\n",
       " [624, 'Rmag_0993'],\n",
       " [625, 'Rmag_0994'],\n",
       " [626, 'Rmag_0995'],\n",
       " [627, 'Rmag_0996'],\n",
       " [628, 'Rmag_0997'],\n",
       " [629, 'Rmag_0998'],\n",
       " [630, 'Rmag_0999'],\n",
       " [631, 'Rmag_1000'],\n",
       " [632, 'Rmag_1001'],\n",
       " [633, 'Rmag_1002'],\n",
       " [634, 'Rmag_1003'],\n",
       " [635, 'Rmag_1004'],\n",
       " [636, 'Rmag_1005'],\n",
       " [637, 'Rmag_1006'],\n",
       " [638, 'Rmag_1007'],\n",
       " [639, 'Rmag_1008'],\n",
       " [640, 'Rmag_1009'],\n",
       " [641, 'Rmag_1011'],\n",
       " [642, 'Rmag_1012'],\n",
       " [643, 'Rmag_1013'],\n",
       " [644, 'Rmag_1014'],\n",
       " [645, 'Rmag_1015'],\n",
       " [646, 'Rmag_1016'],\n",
       " [647, 'Rmag_1018'],\n",
       " [648, 'Rmag_1019'],\n",
       " [649, 'Rmag_1020'],\n",
       " [650, 'Rmag_1022'],\n",
       " [651, 'Rmag_1023'],\n",
       " [652, 'Rmag_1024'],\n",
       " [653, 'Rmag_1025'],\n",
       " [654, 'Rmag_1026'],\n",
       " [655, 'Rmag_1027'],\n",
       " [656, 'Rmag_1030'],\n",
       " [657, 'Rmag_1032'],\n",
       " [658, 'Rmag_1033'],\n",
       " [659, 'Rmag_1035'],\n",
       " [660, 'Rmag_1037'],\n",
       " [661, 'Rmag_1038'],\n",
       " [662, 'Rmag_1040'],\n",
       " [663, 'Rmag_1041'],\n",
       " [664, 'Rmag_1043'],\n",
       " [665, 'Rmag_1044'],\n",
       " [666, 'Rmag_1045'],\n",
       " [667, 'Rmag_1046'],\n",
       " [668, 'Rmag_1047'],\n",
       " [669, 'Rmag_1048'],\n",
       " [670, 'Rmag_1049'],\n",
       " [671, 'Rmag_1050'],\n",
       " [672, 'Rmag_1051'],\n",
       " [673, 'Rmag_1052'],\n",
       " [674, 'Rmag_1054'],\n",
       " [675, 'Rmag_1055'],\n",
       " [676, 'Rmag_1058'],\n",
       " [677, 'Rmag_1059'],\n",
       " [678, 'Rmag_1060'],\n",
       " [679, 'Rmag_1061'],\n",
       " [680, 'Rmag_1062'],\n",
       " [681, 'Rmag_1063'],\n",
       " [682, 'Rmag_1064'],\n",
       " [683, 'Rmag_1066'],\n",
       " [684, 'Rmag_1067'],\n",
       " [685, 'Rmag_1068'],\n",
       " [686, 'Rmag_1069'],\n",
       " [687, 'Rmag_1071'],\n",
       " [688, 'Rmag_1072'],\n",
       " [689, 'Rmag_1073'],\n",
       " [690, 'Rmag_1076']]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[[old.index(item)+1, item] for item in old]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SUP05</th>\n",
       "      <th>Bathy</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.fausta</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pliocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.marissinica</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>SP60_01005</td>\n",
       "      <td>MS2017_1444</td>\n",
       "      <td>Rmag_0392</td>\n",
       "      <td>HUE58_RS01445</td>\n",
       "      <td>Rpac_peg_1455</td>\n",
       "      <td>Rpha_peg_515</td>\n",
       "      <td>Rpli_peg_1614</td>\n",
       "      <td>Rrec_peg_986</td>\n",
       "      <td>Rsou_peg_1716</td>\n",
       "      <td>Vdia_peg_613</td>\n",
       "      <td>Vext_peg_601</td>\n",
       "      <td>Vsoy1_peg_389</td>\n",
       "      <td>Vsoy2_peg_592</td>\n",
       "      <td>Vgig1_peg_568</td>\n",
       "      <td>Vgig2_peg_453</td>\n",
       "      <td>COSY_0364</td>\n",
       "      <td>Vmar_0364</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>SP60_03595</td>\n",
       "      <td>MS2017_2181</td>\n",
       "      <td>Rmag_1021</td>\n",
       "      <td>HUE58_RS05285</td>\n",
       "      <td>Rpac_peg_816</td>\n",
       "      <td>Rpha_peg_864</td>\n",
       "      <td>Rpli_peg_1060</td>\n",
       "      <td>Rrec_peg_620</td>\n",
       "      <td>Rsou_peg_411</td>\n",
       "      <td>Vdia_peg_37</td>\n",
       "      <td>Vext_peg_34</td>\n",
       "      <td>Vsoy1_peg_958</td>\n",
       "      <td>Vsoy2_peg_173</td>\n",
       "      <td>Vgig1_peg_939</td>\n",
       "      <td>Vgig2_peg_824</td>\n",
       "      <td>COSY_0922</td>\n",
       "      <td>Vmar_0926</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>SP60_06030</td>\n",
       "      <td>MS2017_0455</td>\n",
       "      <td>Rmag_0811</td>\n",
       "      <td>HUE58_RS00050</td>\n",
       "      <td>Rpac_peg_245</td>\n",
       "      <td>Rpha_peg_2004</td>\n",
       "      <td>Rpli_peg_411</td>\n",
       "      <td>Rrec_peg_336</td>\n",
       "      <td>Rsou_peg_1177</td>\n",
       "      <td>Vdia_peg_215</td>\n",
       "      <td>Vext_peg_210</td>\n",
       "      <td>Vsoy1_peg_777</td>\n",
       "      <td>Vsoy2_peg_978</td>\n",
       "      <td>Vgig1_peg_178</td>\n",
       "      <td>Vgig2_peg_68</td>\n",
       "      <td>COSY_0736</td>\n",
       "      <td>Vmar_0750</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>SP60_06035</td>\n",
       "      <td>MS2017_0456</td>\n",
       "      <td>Rmag_0810</td>\n",
       "      <td>HUE58_RS00055</td>\n",
       "      <td>Rpac_peg_246</td>\n",
       "      <td>Rpha_peg_2005</td>\n",
       "      <td>Rpli_peg_410</td>\n",
       "      <td>Rrec_peg_335</td>\n",
       "      <td>Rsou_peg_1176</td>\n",
       "      <td>Vdia_peg_216</td>\n",
       "      <td>Vext_peg_211</td>\n",
       "      <td>Vsoy1_peg_776</td>\n",
       "      <td>Vsoy2_peg_977</td>\n",
       "      <td>Vgig1_peg_179</td>\n",
       "      <td>Vgig2_peg_69</td>\n",
       "      <td>COSY_0735</td>\n",
       "      <td>Vmar_0749</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>SP60_06070</td>\n",
       "      <td>MS2017_0468</td>\n",
       "      <td>Rmag_0803</td>\n",
       "      <td>HUE58_RS00090</td>\n",
       "      <td>Rpac_peg_254</td>\n",
       "      <td>Rpha_peg_2012</td>\n",
       "      <td>Rpli_peg_403</td>\n",
       "      <td>Rrec_peg_328</td>\n",
       "      <td>Rsou_peg_1169</td>\n",
       "      <td>Vdia_peg_223</td>\n",
       "      <td>Vext_peg_218</td>\n",
       "      <td>Vsoy1_peg_769</td>\n",
       "      <td>Vsoy2_peg_970</td>\n",
       "      <td>Vgig1_peg_186</td>\n",
       "      <td>Vgig2_peg_76</td>\n",
       "      <td>COSY_0728</td>\n",
       "      <td>Vmar_0742</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>799</th>\n",
       "      <td>SP60_06995</td>\n",
       "      <td>MS2017_0703</td>\n",
       "      <td>Rmag_0674</td>\n",
       "      <td>HUE58_RS03310</td>\n",
       "      <td>Rpac_peg_1170</td>\n",
       "      <td>Rpha_peg_735</td>\n",
       "      <td>Rpli_peg_452</td>\n",
       "      <td>Rrec_peg_1414</td>\n",
       "      <td>Rsou_peg_131</td>\n",
       "      <td>Vdia_peg_350</td>\n",
       "      <td>Vext_peg_341</td>\n",
       "      <td>Vsoy1_peg_656</td>\n",
       "      <td>Vsoy2_peg_856</td>\n",
       "      <td>Vgig1_peg_304</td>\n",
       "      <td>Vgig2_peg_191</td>\n",
       "      <td>COSY_0621</td>\n",
       "      <td>Vmar_0629</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>923</th>\n",
       "      <td>SP60_07945</td>\n",
       "      <td>MS2017_1026</td>\n",
       "      <td>Rmag_0544</td>\n",
       "      <td>HUE58_RS02530</td>\n",
       "      <td>Rpac_peg_1335</td>\n",
       "      <td>Rpha_peg_168</td>\n",
       "      <td>Rpli_peg_1171</td>\n",
       "      <td>Rrec_peg_1234</td>\n",
       "      <td>Rsou_peg_895</td>\n",
       "      <td>Vdia_peg_478</td>\n",
       "      <td>Vext_peg_465</td>\n",
       "      <td>Vsoy1_peg_529</td>\n",
       "      <td>Vsoy2_peg_728</td>\n",
       "      <td>Vgig1_peg_429</td>\n",
       "      <td>Vgig2_peg_314</td>\n",
       "      <td>COSY_0499</td>\n",
       "      <td>Vmar_0505</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          SUP05        Bathy R.magnifica       R.fausta     R.pacifica  \\\n",
       "115  SP60_01005  MS2017_1444   Rmag_0392  HUE58_RS01445  Rpac_peg_1455   \n",
       "374  SP60_03595  MS2017_2181   Rmag_1021  HUE58_RS05285   Rpac_peg_816   \n",
       "684  SP60_06030  MS2017_0455   Rmag_0811  HUE58_RS00050   Rpac_peg_245   \n",
       "685  SP60_06035  MS2017_0456   Rmag_0810  HUE58_RS00055   Rpac_peg_246   \n",
       "692  SP60_06070  MS2017_0468   Rmag_0803  HUE58_RS00090   Rpac_peg_254   \n",
       "799  SP60_06995  MS2017_0703   Rmag_0674  HUE58_RS03310  Rpac_peg_1170   \n",
       "923  SP60_07945  MS2017_1026   Rmag_0544  HUE58_RS02530  Rpac_peg_1335   \n",
       "\n",
       "    R.phaseoliformis   R.pliocardia   R.rectimargo  R.southwardae  \\\n",
       "115     Rpha_peg_515  Rpli_peg_1614   Rrec_peg_986  Rsou_peg_1716   \n",
       "374     Rpha_peg_864  Rpli_peg_1060   Rrec_peg_620   Rsou_peg_411   \n",
       "684    Rpha_peg_2004   Rpli_peg_411   Rrec_peg_336  Rsou_peg_1177   \n",
       "685    Rpha_peg_2005   Rpli_peg_410   Rrec_peg_335  Rsou_peg_1176   \n",
       "692    Rpha_peg_2012   Rpli_peg_403   Rrec_peg_328  Rsou_peg_1169   \n",
       "799     Rpha_peg_735   Rpli_peg_452  Rrec_peg_1414   Rsou_peg_131   \n",
       "923     Rpha_peg_168  Rpli_peg_1171  Rrec_peg_1234   Rsou_peg_895   \n",
       "\n",
       "     V.diagonalis     V.extenta      V.soyoae1      V.soyoae2       V.gigas1  \\\n",
       "115  Vdia_peg_613  Vext_peg_601  Vsoy1_peg_389  Vsoy2_peg_592  Vgig1_peg_568   \n",
       "374   Vdia_peg_37   Vext_peg_34  Vsoy1_peg_958  Vsoy2_peg_173  Vgig1_peg_939   \n",
       "684  Vdia_peg_215  Vext_peg_210  Vsoy1_peg_777  Vsoy2_peg_978  Vgig1_peg_178   \n",
       "685  Vdia_peg_216  Vext_peg_211  Vsoy1_peg_776  Vsoy2_peg_977  Vgig1_peg_179   \n",
       "692  Vdia_peg_223  Vext_peg_218  Vsoy1_peg_769  Vsoy2_peg_970  Vgig1_peg_186   \n",
       "799  Vdia_peg_350  Vext_peg_341  Vsoy1_peg_656  Vsoy2_peg_856  Vgig1_peg_304   \n",
       "923  Vdia_peg_478  Vext_peg_465  Vsoy1_peg_529  Vsoy2_peg_728  Vgig1_peg_429   \n",
       "\n",
       "          V.gigas2 V.okutanii V.marissinica  Count  \n",
       "115  Vgig2_peg_453  COSY_0364     Vmar_0364     17  \n",
       "374  Vgig2_peg_824  COSY_0922     Vmar_0926     17  \n",
       "684   Vgig2_peg_68  COSY_0736     Vmar_0750     17  \n",
       "685   Vgig2_peg_69  COSY_0735     Vmar_0749     17  \n",
       "692   Vgig2_peg_76  COSY_0728     Vmar_0742     17  \n",
       "799  Vgig2_peg_191  COSY_0621     Vmar_0629     17  \n",
       "923  Vgig2_peg_314  COSY_0499     Vmar_0505     17  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 81 Rmag_0392 issue will have to rerun\n",
    "# 280 Rmag_1021 issue will have to rerun\n",
    "# 525 Rmag_0811\n",
    "# 526 Rmag_0810\n",
    "# 533 Rmag_0803\n",
    "# 687 Rmag_0544\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',header=0,index_col=0,sep='\\t')\n",
    "ALL\n",
    "\n",
    "CORE=ALL[ALL['R.magnifica'].isin(['Rmag_0674','Rmag_0392', 'Rmag_0544', 'Rmag_0803', 'Rmag_0810', 'Rmag_0811', 'Rmag_1021'])]\n",
    "CORE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "81 Rmag_0392\n",
      "3825\n",
      "280 Rmag_1021\n",
      "3429\n",
      "525 Rmag_0811\n",
      "4083\n",
      "526 Rmag_0810\n",
      "4188\n",
      "533 Rmag_0803\n",
      "3435\n",
      "687 Rmag_0544\n",
      "3405\n"
     ]
    }
   ],
   "source": [
    "## extract core sequences and align them\n",
    "########### CORE: retreive sequences from GBK, align aa, backtranslate and output fna; 715 sequences ##########\n",
    "from Bio import AlignIO\n",
    "from Bio.codonalign import build\n",
    "from Bio.Alphabet import Gapped,IUPAC\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Alphabet import generic_dna, generic_protein\n",
    "from Bio.Align.Applications import MuscleCommandline,ClustalwCommandline\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "# os.mkdir('./fna')\n",
    "# os.mkdir('./faa')\n",
    "\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "goodfiles={'Bathy':'../genomes/Bathy_complete_with_CRISPRannot.gbk',\\\n",
    "'R.fausta':'../genomes/R.fausta.gbk',\\\n",
    "'R.magnifica':'../genomes/R.magnifica.gbk',\\\n",
    "'R.pacifica':'../genomes/R.pacifica_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.phaseoliformis':'../genomes/R.phaseoliformis_8_contigs_fully_annotated.gbk',\\\n",
    "'R.pliocardia':'../genomes/R.pliocardia_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.rectimargo':'../genomes/R.rectimargo_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.southwardae':'../genomes/R.southwardae_39_contigs_fully_annotated.gbk',\\\n",
    "'SUP05':'../genomes/SUP05.gbk',\\\n",
    "'V.diagonalis':'../genomes/V.diagonalis_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.extenta':'../genomes/V.extenta_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas1':'../genomes/V.gigas1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas2':'../genomes/V.gigas2_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.marissinica':'../genomes/V.marissinica_withlocustags.gbk',\\\n",
    "'V.okutanii':'../genomes/V.okutanii_with_locus_tags.gbk',\\\n",
    "'V.soyoae1':'../genomes/V.soyoae1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.soyoae2':'../genomes/V.soyoae2_one_contig_circular_fully_annotated.gbk'}\n",
    "\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',header=0,index_col=0,sep='\\t')\n",
    "ALL\n",
    "\n",
    "CORE=ALL[ALL['R.magnifica'].isin(['Rmag_0674','Rmag_0392', 'Rmag_0544', 'Rmag_0803', 'Rmag_0810', 'Rmag_0811', 'Rmag_1021'])]\n",
    "CORE\n",
    "\n",
    "# for i in range(688,len(CORE)):\n",
    "for i in [81,280,525,526,533,687]:\n",
    "    gene=CORE.iloc[i]\n",
    "    gene_name=CORE.iloc[i]['R.magnifica']\n",
    "    if gene_name.split('_')[1][0]=='R':\n",
    "        continue\n",
    "    fna=[]\n",
    "    faa=[]\n",
    "    print(i,gene_name)\n",
    "    for sample in sorted(gene.index):\n",
    "        for record in SeqIO.parse(goodfiles[sample],'genbank'):\n",
    "            for feature in record.features:\n",
    "                if feature.type=='CDS':\n",
    "                    locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                    if locus_tag == CORE.iloc[i][sample]:\n",
    "#                         if locus_tag == gene_name:\n",
    "#                             product= feature.qualifiers['product'][0]\n",
    "#                             print gene_name, product\n",
    "#                             df=df.append({'gene':gene_name,'product':product}, ignore_index=True)\n",
    "                        \n",
    "                        \n",
    "                        ### create SeqRecord object ####\n",
    "                        DNAseq=feature.location.extract(record).seq\n",
    "                        if locus_tag == gene_name:\n",
    "                            print(len(DNAseq))\n",
    "                        try:\n",
    "                            AAseq=Seq(feature.qualifiers['translation'][0],alphabet=Gapped(IUPAC.extended_protein))\n",
    "                        except KeyError:\n",
    "                            print(feature)\n",
    "#                             AAseq=feature.extract(record).translate(11)\n",
    "#                             AAseq.id=locus_tag\n",
    "#                             print(AAseq.seq)\n",
    "                        header='|'.join([sample,CORE.iloc[i][sample]])\n",
    "\n",
    "                        fna_seq = SeqRecord(DNAseq,id=header)\n",
    "                        faa_seq = SeqRecord(AAseq,id=header)\n",
    "                        \n",
    "                        fna.append(fna_seq)\n",
    "                        faa.append(faa_seq)                  \n",
    "\n",
    "    faa=[f for f in sorted(faa, key=lambda x : x.id)]\n",
    "    SeqIO.write(faa, path_to_files+'faa/'+gene_name+'.faa', \"fasta\")\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['SUP05', 'Bathy', 'R.magnifica', 'R.fausta', 'R.pacifica',\n",
      "       'R.phaseoliformis', 'R.pliocardia', 'R.rectimargo', 'R.southwardae',\n",
      "       'V.diagonalis', 'V.extenta', 'V.soyoae1', 'V.soyoae2', 'V.gigas1',\n",
      "       'V.gigas2', 'V.okutanii', 'V.marissinica'],\n",
      "      dtype='object')\n",
      "SUP05                  SP60_01005\n",
      "Bathy                 MS2017_1444\n",
      "R.magnifica             Rmag_0392\n",
      "R.fausta            HUE58_RS01445\n",
      "R.pacifica          Rpac_peg_1455\n",
      "R.phaseoliformis     Rpha_peg_515\n",
      "R.pliocardia        Rpli_peg_1614\n",
      "R.rectimargo         Rrec_peg_986\n",
      "R.southwardae       Rsou_peg_1716\n",
      "V.diagonalis         Vdia_peg_613\n",
      "V.extenta            Vext_peg_601\n",
      "V.soyoae1           Vsoy1_peg_389\n",
      "V.soyoae2           Vsoy2_peg_592\n",
      "V.gigas1            Vgig1_peg_568\n",
      "V.gigas2            Vgig2_peg_453\n",
      "V.okutanii              COSY_0364\n",
      "V.marissinica           Vmar_0364\n",
      "Name: 115, dtype: object\n",
      "0 Rmag_0392\n",
      "3825\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/GenBank/Scanner.py:215: BiopythonParserWarning: Over indented source feature?\n",
      "  BiopythonParserWarning,\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:1398: BiopythonParserWarning: Expected sequence length 1021329, found 1023923 (Contig_1).\n",
      "  BiopythonParserWarning,\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: SUP05|SP60_01005(M 0) does not correspond to SUP05|SP60_01005(ATT)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUP05                  SP60_03595\n",
      "Bathy                 MS2017_2181\n",
      "R.magnifica             Rmag_1021\n",
      "R.fausta            HUE58_RS05285\n",
      "R.pacifica           Rpac_peg_816\n",
      "R.phaseoliformis     Rpha_peg_864\n",
      "R.pliocardia        Rpli_peg_1060\n",
      "R.rectimargo         Rrec_peg_620\n",
      "R.southwardae        Rsou_peg_411\n",
      "V.diagonalis          Vdia_peg_37\n",
      "V.extenta             Vext_peg_34\n",
      "V.soyoae1           Vsoy1_peg_958\n",
      "V.soyoae2           Vsoy2_peg_173\n",
      "V.gigas1            Vgig1_peg_939\n",
      "V.gigas2            Vgig2_peg_824\n",
      "V.okutanii              COSY_0922\n",
      "V.marissinica           Vmar_0926\n",
      "Name: 374, dtype: object\n",
      "1 Rmag_1021\n",
      "3429\n",
      "SUP05                  SP60_06030\n",
      "Bathy                 MS2017_0455\n",
      "R.magnifica             Rmag_0811\n",
      "R.fausta            HUE58_RS00050\n",
      "R.pacifica           Rpac_peg_245\n",
      "R.phaseoliformis    Rpha_peg_2004\n",
      "R.pliocardia         Rpli_peg_411\n",
      "R.rectimargo         Rrec_peg_336\n",
      "R.southwardae       Rsou_peg_1177\n",
      "V.diagonalis         Vdia_peg_215\n",
      "V.extenta            Vext_peg_210\n",
      "V.soyoae1           Vsoy1_peg_777\n",
      "V.soyoae2           Vsoy2_peg_978\n",
      "V.gigas1            Vgig1_peg_178\n",
      "V.gigas2             Vgig2_peg_68\n",
      "V.okutanii              COSY_0736\n",
      "V.marissinica           Vmar_0750\n",
      "Name: 684, dtype: object\n",
      "2 Rmag_0811\n",
      "4083\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.pliocardia|Rpli_peg_411(M 0) does not correspond to R.pliocardia|Rpli_peg_411(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUP05                  SP60_06035\n",
      "Bathy                 MS2017_0456\n",
      "R.magnifica             Rmag_0810\n",
      "R.fausta            HUE58_RS00055\n",
      "R.pacifica           Rpac_peg_246\n",
      "R.phaseoliformis    Rpha_peg_2005\n",
      "R.pliocardia         Rpli_peg_410\n",
      "R.rectimargo         Rrec_peg_335\n",
      "R.southwardae       Rsou_peg_1176\n",
      "V.diagonalis         Vdia_peg_216\n",
      "V.extenta            Vext_peg_211\n",
      "V.soyoae1           Vsoy1_peg_776\n",
      "V.soyoae2           Vsoy2_peg_977\n",
      "V.gigas1            Vgig1_peg_179\n",
      "V.gigas2             Vgig2_peg_69\n",
      "V.okutanii              COSY_0735\n",
      "V.marissinica           Vmar_0749\n",
      "Name: 685, dtype: object\n",
      "3 Rmag_0810\n",
      "4188\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: SUP05|SP60_06035(M 0) does not correspond to SUP05|SP60_06035(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUP05                  SP60_06070\n",
      "Bathy                 MS2017_0468\n",
      "R.magnifica             Rmag_0803\n",
      "R.fausta            HUE58_RS00090\n",
      "R.pacifica           Rpac_peg_254\n",
      "R.phaseoliformis    Rpha_peg_2012\n",
      "R.pliocardia         Rpli_peg_403\n",
      "R.rectimargo         Rrec_peg_328\n",
      "R.southwardae       Rsou_peg_1169\n",
      "V.diagonalis         Vdia_peg_223\n",
      "V.extenta            Vext_peg_218\n",
      "V.soyoae1           Vsoy1_peg_769\n",
      "V.soyoae2           Vsoy2_peg_970\n",
      "V.gigas1            Vgig1_peg_186\n",
      "V.gigas2             Vgig2_peg_76\n",
      "V.okutanii              COSY_0728\n",
      "V.marissinica           Vmar_0742\n",
      "Name: 692, dtype: object\n",
      "4 Rmag_0803\n",
      "3435\n",
      "SUP05                  SP60_06995\n",
      "Bathy                 MS2017_0703\n",
      "R.magnifica             Rmag_0674\n",
      "R.fausta            HUE58_RS03310\n",
      "R.pacifica          Rpac_peg_1170\n",
      "R.phaseoliformis     Rpha_peg_735\n",
      "R.pliocardia         Rpli_peg_452\n",
      "R.rectimargo        Rrec_peg_1414\n",
      "R.southwardae        Rsou_peg_131\n",
      "V.diagonalis         Vdia_peg_350\n",
      "V.extenta            Vext_peg_341\n",
      "V.soyoae1           Vsoy1_peg_656\n",
      "V.soyoae2           Vsoy2_peg_856\n",
      "V.gigas1            Vgig1_peg_304\n",
      "V.gigas2            Vgig2_peg_191\n",
      "V.okutanii              COSY_0621\n",
      "V.marissinica           Vmar_0629\n",
      "Name: 799, dtype: object\n",
      "5 Rmag_0674\n",
      "1683\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.fausta|HUE58_RS03310(M 0) does not correspond to R.fausta|HUE58_RS03310(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.magnifica|Rmag_0674(M 0) does not correspond to R.magnifica|Rmag_0674(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.pacifica|Rpac_peg_1170(M 0) does not correspond to R.pacifica|Rpac_peg_1170(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.phaseoliformis|Rpha_peg_735(M 0) does not correspond to R.phaseoliformis|Rpha_peg_735(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.pliocardia|Rpli_peg_452(M 0) does not correspond to R.pliocardia|Rpli_peg_452(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.rectimargo|Rrec_peg_1414(M 0) does not correspond to R.rectimargo|Rrec_peg_1414(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: R.southwardae|Rsou_peg_131(M 0) does not correspond to R.southwardae|Rsou_peg_131(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.diagonalis|Vdia_peg_350(M 0) does not correspond to V.diagonalis|Vdia_peg_350(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.extenta|Vext_peg_341(M 0) does not correspond to V.extenta|Vext_peg_341(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.gigas1|Vgig1_peg_304(M 0) does not correspond to V.gigas1|Vgig1_peg_304(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.gigas2|Vgig2_peg_191(M 0) does not correspond to V.gigas2|Vgig2_peg_191(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.marissinica|Vmar_0629(M 0) does not correspond to V.marissinica|Vmar_0629(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.okutanii|COSY_0621(M 0) does not correspond to V.okutanii|COSY_0621(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.soyoae1|Vsoy1_peg_656(M 0) does not correspond to V.soyoae1|Vsoy1_peg_656(GTG)\n",
      "  BiopythonWarning)\n",
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.soyoae2|Vsoy2_peg_856(M 0) does not correspond to V.soyoae2|Vsoy2_peg_856(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUP05                  SP60_07945\n",
      "Bathy                 MS2017_1026\n",
      "R.magnifica             Rmag_0544\n",
      "R.fausta            HUE58_RS02530\n",
      "R.pacifica          Rpac_peg_1335\n",
      "R.phaseoliformis     Rpha_peg_168\n",
      "R.pliocardia        Rpli_peg_1171\n",
      "R.rectimargo        Rrec_peg_1234\n",
      "R.southwardae        Rsou_peg_895\n",
      "V.diagonalis         Vdia_peg_478\n",
      "V.extenta            Vext_peg_465\n",
      "V.soyoae1           Vsoy1_peg_529\n",
      "V.soyoae2           Vsoy2_peg_728\n",
      "V.gigas1            Vgig1_peg_429\n",
      "V.gigas2            Vgig2_peg_314\n",
      "V.okutanii              COSY_0499\n",
      "V.marissinica           Vmar_0505\n",
      "Name: 923, dtype: object\n",
      "6 Rmag_0544\n",
      "3405\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/maeperez/virtualenv/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: V.gigas2|Vgig2_peg_314(M 0) does not correspond to V.gigas2|Vgig2_peg_314(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    }
   ],
   "source": [
    "## extract core sequences and align them\n",
    "########### CORE: retreive sequences from GBK, align aa, backtranslate and output fna; 715 sequences ##########\n",
    "from Bio import AlignIO\n",
    "from Bio.codonalign import build\n",
    "from Bio.Alphabet import Gapped,IUPAC\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Alphabet import generic_dna, generic_protein\n",
    "from Bio.Align.Applications import MuscleCommandline,ClustalwCommandline\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "# os.mkdir('./fna')\n",
    "# os.mkdir('./faa')\n",
    "\n",
    "path_to_files='../mauve_alignments/'\n",
    "\n",
    "goodfiles={'Bathy':'../genomes/Bathy_complete_with_CRISPRannot.gbk',\\\n",
    "'R.fausta':'../genomes/R.fausta.gbk',\\\n",
    "'R.magnifica':'../genomes/R.magnifica.gbk',\\\n",
    "'R.pacifica':'../genomes/R.pacifica_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.phaseoliformis':'../genomes/R.phaseoliformis_8_contigs_fully_annotated.gbk',\\\n",
    "'R.pliocardia':'../genomes/R.pliocardia_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.rectimargo':'../genomes/R.rectimargo_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.southwardae':'../genomes/R.southwardae_39_contigs_fully_annotated.gbk',\\\n",
    "'SUP05':'../genomes/SUP05.gbk',\\\n",
    "'V.diagonalis':'../genomes/V.diagonalis_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.extenta':'../genomes/V.extenta_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas1':'../genomes/V.gigas1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas2':'../genomes/V.gigas2_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.marissinica':'../genomes/V.marissinica_withlocustags.gbk',\\\n",
    "'V.okutanii':'../genomes/V.okutanii_with_locus_tags.gbk',\\\n",
    "'V.soyoae1':'../genomes/V.soyoae1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.soyoae2':'../genomes/V.soyoae2_one_contig_circular_fully_annotated.gbk'}\n",
    "\n",
    "ALL=pd.read_csv(path_to_files+'Orthology_id35cov51_nopseudo.txt',header=0,index_col=0,sep='\\t')\n",
    "ALL\n",
    "\n",
    "CORE=ALL[ALL['R.magnifica'].isin(['Rmag_0674','Rmag_0392', 'Rmag_0544', 'Rmag_0803', 'Rmag_0810', 'Rmag_0811', 'Rmag_1021'])][['SUP05', 'Bathy', 'R.magnifica', 'R.fausta', 'R.pacifica',\n",
    "       'R.phaseoliformis', 'R.pliocardia', 'R.rectimargo', 'R.southwardae',\n",
    "       'V.diagonalis', 'V.extenta', 'V.soyoae1', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.gigas2', 'V.okutanii', 'V.marissinica']]\n",
    "print(CORE.columns)\n",
    "\n",
    "# for i in range(688,len(CORE)):\n",
    "for i in range(len(CORE)):\n",
    "    gene=CORE.iloc[i]\n",
    "    print(gene)\n",
    "    gene_name=CORE.iloc[i]['R.magnifica']\n",
    "    fna=[]\n",
    "    faa=[]\n",
    "    print(i,gene_name)\n",
    "    for sample in sorted(gene.index):\n",
    "        for record in SeqIO.parse(goodfiles[sample],'genbank'):\n",
    "            for feature in record.features:\n",
    "                if feature.type=='CDS':\n",
    "                    locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                    if locus_tag == CORE.iloc[i][sample]:\n",
    "#                         if locus_tag == gene_name:\n",
    "#                             product= feature.qualifiers['product'][0]\n",
    "#                             print gene_name, product\n",
    "#                             df=df.append({'gene':gene_name,'product':product}, ignore_index=True)\n",
    "                        \n",
    "                        \n",
    "                        ### create SeqRecord object ####\n",
    "                        DNAseq=feature.location.extract(record).seq\n",
    "                        if locus_tag == gene_name:\n",
    "                            print(len(DNAseq))\n",
    "                        try:\n",
    "                            AAseq=Seq(feature.qualifiers['translation'][0],alphabet=Gapped(IUPAC.extended_protein))\n",
    "                        except KeyError:\n",
    "                            print(feature)\n",
    "#                             AAseq=feature.extract(record).translate(11)\n",
    "#                             AAseq.id=locus_tag\n",
    "#                             print(AAseq.seq)\n",
    "                        header='|'.join([sample,CORE.iloc[i][sample]])\n",
    "\n",
    "                        fna_seq = SeqRecord(DNAseq,id=header)\n",
    "                        faa_seq = SeqRecord(AAseq,id=header)\n",
    "                        \n",
    "                        fna.append(fna_seq)\n",
    "                    \n",
    "    align = AlignIO.read(path_to_files+'faa/aligned_'+gene_name+'.faa', \"fasta\",alphabet=Gapped(IUPAC.extended_protein))\n",
    "    aln = MultipleSeqAlignment([f for f in sorted(align, key=lambda x : x.id)])\n",
    "    codon_aln = build(aln, fna)\n",
    "\n",
    "    SeqIO.write(codon_aln, path_to_files+'fna/aligned_'+gene_name+'.fna', \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "[]\n",
      "31\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Rmag_0004',\n",
       " 'Rmag_0019',\n",
       " 'Rmag_0022',\n",
       " 'Rmag_0023',\n",
       " 'Rmag_1057',\n",
       " 'Rmag_0235',\n",
       " 'Rmag_0238',\n",
       " 'Rmag_0785',\n",
       " 'Rmag_0701',\n",
       " 'Rmag_0699',\n",
       " 'Rmag_0446',\n",
       " 'Rmag_0624',\n",
       " 'Rmag_0623',\n",
       " 'Rmag_0600',\n",
       " 'Rmag_0570',\n",
       " 'Rmag_0562',\n",
       " 'Rmag_0537',\n",
       " 'Rmag_0483',\n",
       " 'Rmag_0420',\n",
       " 'Rmag_0337',\n",
       " 'Rmag_0304',\n",
       " 'Rmag_0303',\n",
       " 'Rmag_0909',\n",
       " 'Rmag_0162',\n",
       " 'Rmag_0158',\n",
       " 'Rmag_0145',\n",
       " 'Rmag_0074',\n",
       " 'Rmag_0058',\n",
       " 'Rmag_1039',\n",
       " 'Rmag_R0022',\n",
       " 'Rmag_1053']"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "985\n",
      "818\n",
      "5176\n",
      "6163 6084\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Bathy</th>\n",
       "      <th>SUP05</th>\n",
       "      <th>R.magnifica</th>\n",
       "      <th>R.pacifica</th>\n",
       "      <th>R.phaseoliformis</th>\n",
       "      <th>R.pleocardia</th>\n",
       "      <th>R.rectimargo</th>\n",
       "      <th>R.southwardae</th>\n",
       "      <th>V.diagonalis</th>\n",
       "      <th>V.extenta</th>\n",
       "      <th>V.okutanii</th>\n",
       "      <th>V.gigas2</th>\n",
       "      <th>V.soyoae2</th>\n",
       "      <th>V.gigas1</th>\n",
       "      <th>V.soyoae1</th>\n",
       "      <th>product</th>\n",
       "      <th>genome_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4548</th>\n",
       "      <td>MS2017_0003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recombination protein F</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Bathy SUP05 R.magnifica R.pacifica R.phaseoliformis R.pleocardia  \\\n",
       "4548  MS2017_0003   NaN         NaN        NaN              NaN          NaN   \n",
       "\n",
       "     R.rectimargo R.southwardae V.diagonalis V.extenta V.okutanii V.gigas2  \\\n",
       "4548          NaN           NaN          NaN       NaN        NaN      NaN   \n",
       "\n",
       "     V.soyoae2 V.gigas1 V.soyoae1                  product  genome_count  \n",
       "4548       NaN      NaN       NaN  recombination protein F             1  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x=ALL[ALL['Count']<17][['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values\n",
    "x=ALL[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pliocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values\n",
    "all_str=['-'.join(map(str,a)) for a in x]\n",
    "\n",
    "x=ORTH[ORTH['genome_count']<15][['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pleocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values\n",
    "x=ORTH[['Bathy', 'SUP05', 'R.magnifica', 'R.pacifica', 'R.phaseoliformis',\n",
    "       'R.pleocardia', 'R.rectimargo', 'R.southwardae', 'V.diagonalis',\n",
    "       'V.extenta', 'V.okutanii', 'V.gigas2', 'V.soyoae2', 'V.gigas1',\n",
    "       'V.soyoae1']].values\n",
    "\n",
    "\n",
    "orth_str=['-'.join(map(str,a)).replace('Rple','Rpli') for a in x]\n",
    "\n",
    "print(len(set(orth_str)-set(all_str)))\n",
    "print(len(set(all_str)-set(orth_str)))\n",
    "print(len(set.intersection(set(all_str), set(orth_str))))\n",
    "print(len(orth_str),len(all_str))\n",
    "\n",
    "set(all_str)-set(orth_str)\n",
    "ORTH[ORTH['Bathy']=='MS2017_0003']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Rmag_0048', 'Rmag_0335', 'Rmag_0436', 'Rmag_0530', 'Rmag_0610', 'Rmag_0657', 'Rmag_0658', 'Rmag_0734', 'Rmag_0748', 'Rmag_0970', 'Rmag_1010', 'Rmag_1034']\n",
      "12\n"
     ]
    }
   ],
   "source": [
    "pseudo=[]\n",
    "for file in listdir('./fna'):\n",
    "    if file[-4:]=='.fna':\n",
    "\n",
    "        aln=AlignIO.read('./fna/'+file,'fasta')\n",
    "        if len(aln) != 17:\n",
    "#             print(file[8:-4])\n",
    "            pseudo+=[file[8:-4]]\n",
    "#             print(len(aln))\n",
    "            faa=AlignIO.read('./faa/'+file[:-4]+'.faa','fasta')\n",
    "#             print(len(faa))\n",
    "#             print([record.id for record in aln])\n",
    "print(sorted(pseudo))\n",
    "print(len(pseudo))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/maeperez/scratch/Clams/relax/Rmag_0658_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0658_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0658_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0658.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0658_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0658.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0658.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0658.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0658.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0658.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0658.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0658.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0658.HKY.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0658.HKY.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0658.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0658.HKY.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0658.HKY.gard_finalout\n",
      "fna/aligned_Rmag_1010.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1010_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1010_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1010_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1010.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1010_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1010.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1010.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1010.tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1010.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1010.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1010.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1010.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1010.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_1010.HKY.gard\n",
      "fna/aligned_Rmag_0657.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0657_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0657_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0657_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0657.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0657_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0657.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0657.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0657.tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0657.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0657.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0657.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0657.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0657.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0657.HKY.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0657.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0657.HKY.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0657.HKY.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0657.HKY.gard_ga_details\n",
      "fna/aligned_Rmag_0734.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0734_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0734_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0734_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0734.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0734_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0734.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0734.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0734.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0734.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0734.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0734.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0734.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0734.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0734.HKY.gard\n",
      "fna/aligned_Rmag_0748.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0748_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0748_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0748.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0748_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0748_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0748.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0748.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0748.tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0748.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0748.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0748.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0748.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0748.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0748.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0748.HKY.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0748.HKY.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0748.HKY.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0748.HKY.gard.json\n",
      "fna/aligned_Rmag_0610.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0610.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0610_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0610_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0610_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0610_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0610.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0610.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0610.tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0610.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0610.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0610.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0610.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0610.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0610.HKY.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0610.HKY.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0610.HKY.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0610.HKY.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0610.HKY.gard_splits\n",
      "fna/aligned_Rmag_1034.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1034_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1034_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1034.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1034_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_1034_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1034.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1034.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_1034.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1034.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1034.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1034.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1034.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_1034.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_1034.HKY.gard\n",
      "fna/aligned_Rmag_0436.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0436_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0436.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0436_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0436_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0436_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0436.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0436.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0436.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0436.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0436.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0436.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0436.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0436.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0436.HKY.gard\n",
      "fna/aligned_Rmag_0335.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0335_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0335_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0335_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0335_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0335.relax\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0335.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0335.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0335.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0335.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0335.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0335.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0335.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0335.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0335.HKY.gard\n",
      "fna/aligned_Rmag_0048.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0048_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0048_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0048_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0048_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0048.relax\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0048.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0048.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0048.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0048.GTR.gard.json\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0048.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0048.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0048.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0048.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0048.HKY.gard\n",
      "fna/aligned_Rmag_0970.fna\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/maeperez/scratch/Clams/relax/Rmag_0970_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0970_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0970_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0970.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0970_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0970.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0970.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0970.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0970.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0970.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0970.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0970.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0970.HKY.gard\n",
      "fna/aligned_Rmag_0530.fna\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0530_GivsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0530_SymvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0530_GivsRu.RELAX.json\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0530.relax\n",
      "/home/maeperez/scratch/Clams/relax/Rmag_0530_RuvsFL.RELAX.json\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0530.tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0530.genera.labeled_tree\n",
      "/home/maeperez/scratch/Clams/init_trees/Rmag_0530.host.labeled_tree\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0530.GTR.gard\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0530.GTR.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0530.GTR.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0530.GTR.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0530.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0530.HKY.gard\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0530.HKY.gard_ga_details\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0530.HKY.gard_finalout\n",
      "/home/maeperez/scratch/Clams/gard/HKY/Rmag_0530.HKY.gard.json\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "for ps in pseudo:\n",
    "    for p in Path(\"./fna/\").glob(\"*\"+ps+\"*\"):\n",
    "        print(p)\n",
    "        p.unlink()\n",
    "    for p in Path(\"/home/maeperez/scratch/Clams/relax/\").glob(ps+\"*\"):\n",
    "        print(p)\n",
    "        p.unlink()\n",
    "    for p in Path(\"/home/maeperez/scratch/Clams/init_trees/\").glob(ps+\"*\"):\n",
    "        print(p)\n",
    "        p.unlink()\n",
    "    for p in Path(\"/home/maeperez/scratch/Clams/gard/GTR/\").glob(ps+\"*\"):\n",
    "        print(p)\n",
    "        p.unlink()\n",
    "    for p in Path(\"/home/maeperez/scratch/Clams/gard/HKY/\").glob(ps+\"*\"):\n",
    "        print(p)\n",
    "        p.unlink()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fna/aligned_Rmag_0052.fna\n",
      "1-462\n",
      "(((Bathy_MS2017_2110_unknown_description_:0.2008530969267975,SUP05_SP60_04320_unknown_description_:0.1423423128527979):0.1178855572750083,(((((R_fausta_HUE58_RS05990_unknown_description_:0,R_rectimargo_Rrec_peg_772_unknown_description_:0):0.01587169267889427,R_pacifica_Rpac_peg_163_unknown_description_:0.02388338993949496):0.0122337280548469,R_phaseoliformis_Rpha_peg_788_unknown_description_:0.032807421353688):0.01519821288257936,(R_pliocardia_Rpli_peg_1507_unknown_description_:0.03244915738047897,R_southwardae_Rsou_peg_584_unknown_description_:0.02869189576797013):0.008869576680349399):0.005028277226643493,R_magnifica_Rmag_0052_unknown_description_:0.05200120340165283):0.0482219171752319):0.05205890440064238,((V_diagonalis_Vdia_peg_928_unknown_description_:0.002056909775466394,V_extenta_Vext_peg_917_unknown_description_:0.0002311701997289527):0.02355623244974999,((V_gigas1_Vgig1_peg_64_unknown_description_:0,V_gigas2_Vgig2_peg_905_unknown_description_:0):0.03250926371639534,(V_okutanii_COSY_0057_unknown_description_:0.01706787438387957,(V_soyoae1_Vsoy1_peg_82_unknown_description_:0,V_soyoae2_Vsoy2_peg_285_unknown_description_:0):0.007155753710560406):0.009533087320527968):0.003899101866531786):0.002720378012407272,V_marissinica_Vmar_0055_unknown_description_:0.02385474236575754)\n",
      "fna/aligned_Rmag_0077.fna\n",
      "1-717\n",
      "(((Bathy_MS2017_2070_unknown_description_:0.121146837205593,SUP05_SP60_04540_unknown_description_:0.2212066386535295):0.06183621439880949,(((R_fausta_HUE58_RS06135_unknown_description_:0,R_rectimargo_Rrec_peg_805_unknown_description_:0):0.02345289644064539,R_pacifica_Rpac_peg_118_unknown_description_:0.02292846092519264):0.02733658591104969,(R_magnifica_Rmag_0077_unknown_description_:0.07146282516393766,(R_phaseoliformis_Rpha_peg_2123_unknown_description_:0.03009605782512371,(R_pliocardia_Rpli_peg_1470_unknown_description_:0.02739068698483446,R_southwardae_Rsou_peg_1657_unknown_description_:0.01772968253761717):0.01251744023773885):0.01787154163526124):0.00835527597675674):0.005840360353820129):0.05683082525132578,((V_diagonalis_Vdia_peg_901_unknown_description_:0,V_extenta_Vext_peg_890_unknown_description_:0):0.01779801409050335,(V_gigas1_Vgig1_peg_90_unknown_description_:0,V_gigas2_Vgig2_peg_880_unknown_description_:0):0.03115800483632857):0.002219560417441738,(V_marissinica_Vmar_0081_unknown_description_:0.02401088468728617,(V_okutanii_COSY_0083_unknown_description_:0.01492056741074627,(V_soyoae1_Vsoy1_peg_106_unknown_description_:0,V_soyoae2_Vsoy2_peg_310_unknown_description_:0):0.0157469943100321):0.005599851998673629):0.00257648711557717)\n",
      "fna/aligned_Rmag_0135.fna\n",
      "1-453\n",
      "(((((Bathy_MS2017_1936_unknown_description_:0.3616167922934704,SUP05_SP60_05045_unknown_description_:0.2808604962573376):0.1515393129939931,(((R_fausta_HUE58_RS06530_unknown_description_:0,R_rectimargo_Rrec_peg_886_unknown_description_:0):0.02910463149572735,R_pacifica_Rpac_peg_39_unknown_description_:0.03855717245972404):0.04636240099124817,(R_magnifica_Rmag_0135_unknown_description_:0.07799829676277614,((R_phaseoliformis_Rpha_peg_1946_unknown_description_:0.005623316361294697,R_southwardae_Rsou_peg_1765_unknown_description_:0.0049588865616808):0.0244043953123175,R_pliocardia_Rpli_peg_1393_unknown_description_:0.0798515008459466):0.01414945921933666):0.00755357206287665):0.05482460568610242):0.07879056102230265,V_marissinica_Vmar_0134_unknown_description_:0.03902186107708812):0.004007456188937553,(V_soyoae1_Vsoy1_peg_161_unknown_description_:0,V_soyoae2_Vsoy2_peg_366_unknown_description_:0):0.01406855050916027):0.001750987697542918,(V_diagonalis_Vdia_peg_846_unknown_description_:0,V_extenta_Vext_peg_835_unknown_description_:0):0.02982651432416581,((V_gigas1_Vgig1_peg_144_unknown_description_:0,V_gigas2_Vgig2_peg_33_unknown_description_:0):0.05124800549733641,V_okutanii_COSY_0137_unknown_description_:0.01640628502272534):0.003915242305279124)\n",
      "fna/aligned_Rmag_0160.fna\n",
      "1-459\n",
      "((((Bathy_MS2017_1856_unknown_description_:0.0911448876251866,SUP05_SP60_05215_unknown_description_:0.07169312743446829):0.02862846738309136,((((R_fausta_HUE58_RS06695_unknown_description_:0,R_rectimargo_Rrec_peg_921_unknown_description_:0):0.007734161715161418,R_pacifica_Rpac_peg_5_unknown_description_:0.01724119771769664):0.01054778344043535,R_phaseoliformis_Rpha_peg_1993_unknown_description_:0.01471945523295291):0.006411232420263482,(R_magnifica_Rmag_0160_unknown_description_:0.02517787192483667,(R_pliocardia_Rpli_peg_1342_unknown_description_:0.026265013957264,R_southwardae_Rsou_peg_1964_unknown_description_:0.01377472654764046):0.008730972022257168):0.00881048092046564):0.02686892204644087):0.04791190188218114,(V_soyoae1_Vsoy1_peg_185_unknown_description_:0,V_soyoae2_Vsoy2_peg_390_unknown_description_:0):0.009616308706018456):0.0009793550833414322,((V_diagonalis_Vdia_peg_821_unknown_description_:0,V_extenta_Vext_peg_810_unknown_description_:0):0.008369218447128143,V_okutanii_COSY_0164_unknown_description_:0.01343492494095239):0.002884432539111297,((V_gigas1_Vgig1_peg_168_unknown_description_:0,V_gigas2_Vgig2_peg_58_unknown_description_:0):0.0172056081945521,V_marissinica_Vmar_0160_unknown_description_:0.03733789600755692):0.002425630745126368)\n",
      "fna/aligned_Rmag_0178.fna\n",
      "1-303\n",
      "((((Bathy_MS2017_1838_unknown_description_:0.1259029119497999,SUP05_SP60_05305_unknown_description_:0.06980273993538733):0.03582344825688073,((((R_fausta_HUE58_RS03915_unknown_description_:0,R_rectimargo_Rrec_peg_939_unknown_description_:0):0.005571301009141337,R_pacifica_Rpac_peg_1038_unknown_description_:0.00806483377370202):0.01201733305902067,R_phaseoliformis_Rpha_peg_1594_unknown_description_:0.008610714593521718):0.006421369727573433,((R_magnifica_Rmag_0178_unknown_description_:0.03495835841888692,R_pliocardia_Rpli_peg_52_unknown_description_:0.01341045289188145):0.006235801143331361,R_southwardae_Rsou_peg_1236_unknown_description_:0.01463787668458015):0.002478364223659832):0.02772674013429762):0.05292913055034006,(V_gigas1_Vgig1_peg_750_unknown_description_:0,V_gigas2_Vgig2_peg_639_unknown_description_:0):0.01307064201202239):0.007408682230342443,((V_diagonalis_Vdia_peg_803_unknown_description_:0,V_extenta_Vext_peg_792_unknown_description_:0):0.01157035774730643,V_marissinica_Vmar_0178_unknown_description_:0.01334776671298801):0.002696956775381065,(V_okutanii_COSY_0182_unknown_description_:0.01527845908191296,(V_soyoae1_Vsoy1_peg_203_unknown_description_:0,V_soyoae2_Vsoy2_peg_408_unknown_description_:0):0.004970436034358023):0.006068589300421626)\n",
      "fna/aligned_Rmag_0450.fna\n",
      "1-420\n",
      "((((Bathy_MS2017_0877_unknown_description_:0.2729331446576078,SUP05_SP60_00490_unknown_description_:0.4099134960466694):0.09735515490691429,(((((R_fausta_HUE58_RS01830_unknown_description_:0,R_rectimargo_Rrec_peg_1076_unknown_description_:0):0.01909378381844552,R_pacifica_Rpac_peg_979_unknown_description_:0.03858558114644323):0.06357343675984549,R_southwardae_Rsou_peg_1400_unknown_description_:0.06600476128480297):0.00811124032021877,(R_phaseoliformis_Rpha_peg_392_unknown_description_:0.07235596492346759,R_pliocardia_Rpli_peg_1303_unknown_description_:0.07811420396670717):0.04994923222658136):0.01383036043390046,R_magnifica_Rmag_0450_unknown_description_:0.09777014148773727):0.01314228926731101):0.09166155113788851,((V_diagonalis_Vdia_peg_559_unknown_description_:0,V_extenta_Vext_peg_547_unknown_description_:0):0.02446156749334008,V_marissinica_Vmar_0419_unknown_description_:0.03533087650126775):0.008141879109532874):0.00573359177774569,(V_gigas1_Vgig1_peg_516_unknown_description_:0,V_gigas2_Vgig2_peg_403_unknown_description_:0):0.04142891604449978,(V_okutanii_COSY_0416_unknown_description_:0.02108310267308704,(V_soyoae1_Vsoy1_peg_443_unknown_description_:0,V_soyoae2_Vsoy2_peg_642_unknown_description_:0):0.0254740477008828):0.0128560697765925)\n",
      "fna/aligned_Rmag_0476.fna\n",
      "1-324\n",
      "((((Bathy_MS2017_1249_unknown_description_:0.1388743381219703,SUP05_SP60_00300_unknown_description_:0.1492279652915633):0.06738099788927701,((((R_fausta_HUE58_RS02070_unknown_description_:0,R_rectimargo_Rrec_peg_1129_unknown_description_:0):0.005383992192994286,R_pacifica_Rpac_peg_927_unknown_description_:0.02914754646207172):0.04283903213834191,((R_phaseoliformis_Rpha_peg_306_unknown_description_:0.0272011232812744,R_pliocardia_Rpli_peg_1255_unknown_description_:0.02034043743639736):0.0091051679299136,R_southwardae_Rsou_peg_1864_unknown_description_:0.02259015557710645):0.0009444485912144598):0.008833248964522186,R_magnifica_Rmag_0476_unknown_description_:0.06038440513359516):0.02817089950898038):0.07661282671109268,V_marissinica_Vmar_0443_unknown_description_:0.02930364728288919):0.003627525860754759,((V_diagonalis_Vdia_peg_537_unknown_description_:0.003082310519825022,V_extenta_Vext_peg_525_unknown_description_:0):0.01387230583497253,(V_gigas1_Vgig1_peg_492_unknown_description_:0,V_gigas2_Vgig2_peg_378_unknown_description_:0):0.008206384458326865):0.004742657370451025,(V_okutanii_COSY_0439_unknown_description_:0.00837538893842704,(V_soyoae1_Vsoy1_peg_468_unknown_description_:0,V_soyoae2_Vsoy2_peg_667_unknown_description_:0):0.004143446374751855):0.01348247785736929)\n",
      "fna/aligned_Rmag_0477.fna\n",
      "1-1284\n",
      "(((Bathy_MS2017_1248_unknown_description_:0.1292692186336235,SUP05_SP60_00295_unknown_description_:0.1050887265584546):0.04980537756578605,(((((R_fausta_HUE58_RS02075_unknown_description_:0,R_rectimargo_Rrec_peg_1130_unknown_description_:0):0.01660339781550975,R_pacifica_Rpac_peg_926_unknown_description_:0.02256722778709551):0.008362501835850682,R_phaseoliformis_Rpha_peg_305_unknown_description_:0.02857719528214413):0.01780261665372305,(R_pliocardia_Rpli_peg_1254_unknown_description_:0.02202574099914824,R_southwardae_Rsou_peg_1865_unknown_description_:0.02003987021697742):0.01629321353786955):0.00605151500648108,R_magnifica_Rmag_0477_unknown_description_:0.05078418552415457):0.01111227543707356):0.05368423931492566,(V_diagonalis_Vdia_peg_536_unknown_description_:0,V_extenta_Vext_peg_524_unknown_description_:0):0.01201477575026951,(((V_gigas1_Vgig1_peg_491_unknown_description_:0,V_gigas2_Vgig2_peg_377_unknown_description_:0):0.01681040024935064,(V_okutanii_COSY_0440_unknown_description_:0.005984609296938843,(V_soyoae1_Vsoy1_peg_469_unknown_description_:0,V_soyoae2_Vsoy2_peg_668_unknown_description_:0):0.006221324500380078):0.004130250214967143):0.002506255831350165,V_marissinica_Vmar_0444_unknown_description_:0.01408646764166143):0.0006765294273744851)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fna/aligned_Rmag_0478.fna\n",
      "1-1173\n",
      "((((((Bathy_MS2017_1247_unknown_description_:0.2493965640777943,SUP05_SP60_00290_unknown_description_:0.192509434657317):0.1245550074374939,(((((R_fausta_HUE58_RS02080_unknown_description_:0,R_rectimargo_Rrec_peg_1131_unknown_description_:0):0.01626607252909763,R_pacifica_Rpac_peg_925_unknown_description_:0.02246975869808962):0.03061125343229118,(R_pliocardia_Rpli_peg_1253_unknown_description_:0.04229205744419836,R_southwardae_Rsou_peg_1866_unknown_description_:0.03220384256188121):0.0128466201192823):0.001791208084261788,R_phaseoliformis_Rpha_peg_304_unknown_description_:0.04463978074822226):0.01564012050011,R_magnifica_Rmag_0478_unknown_description_:0.06963729041115731):0.009358291552381034):0.1226993754727846,(V_diagonalis_Vdia_peg_535_unknown_description_:0,V_extenta_Vext_peg_523_unknown_description_:0):0.02471913532701463):0.004644324960434619,(V_gigas1_Vgig1_peg_490_unknown_description_:0,V_gigas2_Vgig2_peg_376_unknown_description_:0.0009535743228443263):0.0334650941142212):0.00408245838386306,V_marissinica_Vmar_0445_unknown_description_:0.0232787314245764):0.0015117936447792,V_okutanii_COSY_0441_unknown_description_:0.01772230059349951,(V_soyoae1_Vsoy1_peg_470_unknown_description_:0,V_soyoae2_Vsoy2_peg_669_unknown_description_:0):0.01579774148549172)\n",
      "fna/aligned_Rmag_0595.fna\n",
      "1-486\n",
      "((((Bathy_MS2017_0789_unknown_description_:0.2152330887520284,SUP05_SP60_07460_unknown_description_:0.2624435073973205):0.06395777883224868,((((R_fausta_HUE58_RS02865_unknown_description_:0.001522776130775109,R_rectimargo_Rrec_peg_1309_unknown_description_:0.000774553805613672):0.02829126335891351,R_pacifica_Rpac_peg_1267_unknown_description_:0.02797072580616411):0.01222457729838505,R_phaseoliformis_Rpha_peg_99_unknown_description_:0.04032854025758616):0.03174857707128888,(R_magnifica_Rmag_0595_unknown_description_:0.05020852621936239,(R_pliocardia_Rpli_peg_560_unknown_description_:0.03666498414620915,R_southwardae_Rsou_peg_756_unknown_description_:0.02126351303498952):0.01467146360210456):0.01227852352832703):0.02169917381151529):0.1033629304805962,V_marissinica_Vmar_0556_unknown_description_:0.04154829168971928):0.004513251066260698,((V_diagonalis_Vdia_peg_425_unknown_description_:0,V_extenta_Vext_peg_413_unknown_description_:0):0.03237741062595838,(V_okutanii_COSY_0550_unknown_description_:0.01164731086827958,(V_soyoae1_Vsoy1_peg_581_unknown_description_:0,V_soyoae2_Vsoy2_peg_780_unknown_description_:0):0.01801163349545587):0.01589954133335674):0.001925899055055722,(V_gigas1_Vgig1_peg_379_unknown_description_:0,V_gigas2_Vgig2_peg_264_unknown_description_:0):0.04625066545840548)\n",
      "fna/aligned_Rmag_0647.fna\n",
      "1-531\n",
      "((((Bathy_MS2017_0765_unknown_description_:0.166761064987261,SUP05_SP60_07160_unknown_description_:0.09516220363671996):0.02714138509772384,(((((R_fausta_HUE58_RS03140_unknown_description_:0,R_rectimargo_Rrec_peg_1371_unknown_description_:0):0.02843205144259767,R_pacifica_Rpac_peg_1204_unknown_description_:0.01901332297151739):0.01393082228692229,(R_phaseoliformis_Rpha_peg_1177_unknown_description_:0.009765815478297777,R_pliocardia_Rpli_peg_490_unknown_description_:0.02142438894906764):0.006796106424351283):0.01841341586631098,R_southwardae_Rsou_peg_823_unknown_description_:0.00798819103065496):0.007143098080965273,R_magnifica_Rmag_0647_unknown_description_:0.04036106933269433):0.0375805227280851):0.05977807799129534,(V_soyoae1_Vsoy1_peg_629_unknown_description_:0,V_soyoae2_Vsoy2_peg_829_unknown_description_:0):0.0004528238137361043):0.01158793348847934,(((V_diagonalis_Vdia_peg_381_unknown_description_:0,V_extenta_Vext_peg_369_unknown_description_:0):0.01636669788360524,(V_gigas1_Vgig1_peg_332_unknown_description_:0,V_gigas2_Vgig2_peg_218_unknown_description_:0):0.01192160865643251):0.0007481410822604857,V_marissinica_Vmar_0600_unknown_description_:0.005591156278703937):0.000591286655275922,V_okutanii_COSY_0594_unknown_description_:0.01500531066589155)\n",
      "fna/aligned_Rmag_0767.fna\n",
      "1-684\n",
      "(((Bathy_MS2017_0555_unknown_description_:0.21737899696391,SUP05_SP60_06350_unknown_description_:0.104659423959585):0.5704109285342209,(((R_fausta_HUE58_RS00295_unknown_description_:0,R_rectimargo_Rrec_peg_277_unknown_description_:0):0.01222163269143605,R_pacifica_Rpac_peg_301_unknown_description_:0.01610832280598735):0.0338731157951256,(R_magnifica_Rmag_0767_unknown_description_:0.05915033405680468,((R_phaseoliformis_Rpha_peg_2084_unknown_description_:0.02734001780580557,R_pliocardia_Rpli_peg_341_unknown_description_:0.03247372772136932):0.000500359994940091,R_southwardae_Rsou_peg_1097_unknown_description_:0.03494688773798192):0.01578727888208742):0.001169209797674396):0.01978580387060536):0.05320417802376021,((V_diagonalis_Vdia_peg_251_unknown_description_:0,V_extenta_Vext_peg_246_unknown_description_:0):0.02405679042534641,V_marissinica_Vmar_0720_unknown_description_:0.02176302394992246):0.004570185370612382,((V_gigas1_Vgig1_peg_211_unknown_description_:0,V_gigas2_Vgig2_peg_100_unknown_description_:0):0.02088193625488521,(V_okutanii_COSY_0705_unknown_description_:0.008302906115783326,(V_soyoae1_Vsoy1_peg_745_unknown_description_:0,V_soyoae2_Vsoy2_peg_945_unknown_description_:0):0.004302375865592838):0.01484834469897568):0.002377588523025016)\n",
      "fna/aligned_Rmag_0828.fna\n",
      "1-747\n",
      "(((((Bathy_MS2017_1682_unknown_description_:0.1424549461087644,SUP05_SP60_01615_unknown_description_:0.1665269113028464):0.1573319194210986,(((R_fausta_HUE58_RS04090_unknown_description_:0,R_rectimargo_Rrec_peg_369_unknown_description_:0):0.02056626709051161,R_pacifica_Rpac_peg_577_unknown_description_:0.01960374888047388):0.04080928908676614,(R_phaseoliformis_Rpha_peg_1542_unknown_description_:0.02695881643530693,R_pliocardia_Rpli_peg_770_unknown_description_:0.02604023348747395):0.02317097025284871):0.03661456722600032):0.02436695384898416,(R_magnifica_Rmag_0828_unknown_description_:0.04933260306904422,R_southwardae_Rsou_peg_997_unknown_description_:0.05388243012266147):0.02739033333724847):0.07828433195573264,(V_okutanii_COSY_0754_unknown_description_:0.01611395132598462,(V_soyoae1_Vsoy1_peg_792_unknown_description_:0,V_soyoae2_Vsoy2_peg_8_unknown_description_:0):0.01658883070028315):0.01018528150955472):0.005265832363060587,(V_diagonalis_Vdia_peg_201_unknown_description_:0,V_extenta_Vext_peg_196_unknown_description_:0):0.0235464514149316,((V_gigas1_Vgig1_peg_773_unknown_description_:0,V_gigas2_Vgig2_peg_662_unknown_description_:0):0.01903768431293816,V_marissinica_Vmar_0765_unknown_description_:0.02670154140517718):0.008030448912926376)\n",
      "fna/aligned_Rmag_0884.fna\n",
      "1-1260\n",
      "((((Bathy_MS2017_1772_unknown_description_:0.1532473365016468,SUP05_SP60_01990_unknown_description_:0.1831648404935211):0.4896846694298508,((((R_fausta_HUE58_RS04510_unknown_description_:0,R_rectimargo_Rrec_peg_460_unknown_description_:0):0.009058104822039724,R_pacifica_Rpac_peg_664_unknown_description_:0.02624574238666681):0.01612437523825491,R_phaseoliformis_Rpha_peg_1444_unknown_description_:0.03575269348250507):0.02526342009607101,(R_magnifica_Rmag_0884_unknown_description_:0.05114523683346821,(R_pliocardia_Rpli_peg_864_unknown_description_:0.02841099470540751,R_southwardae_Rsou_peg_1989_unknown_description_:0.01915222422226134):0.01875899097246843):0.008360727586466291):0.01222463086839204):0.07871609984962265,((V_diagonalis_Vdia_peg_145_unknown_description_:0,V_extenta_Vext_peg_140_unknown_description_:0):0.01793754386018271,V_marissinica_Vmar_0820_unknown_description_:0.0172738431194898):0.002279067004489807):0.001249280560601362,(V_gigas1_Vgig1_peg_831_unknown_description_:0,V_gigas2_Vgig2_peg_719_unknown_description_:0):0.02037163901923247,(V_okutanii_COSY_0810_unknown_description_:0.01357112769507147,(V_soyoae1_Vsoy1_peg_849_unknown_description_:0.0006623748587062018,V_soyoae2_Vsoy2_peg_66_unknown_description_:0.0001530736853269399):0.007435997146748358):0.004057325803634451)\n",
      "fna/aligned_Rmag_0931.fna\n",
      "1-585\n",
      "(((((Bathy_MS2017_0283_unknown_description_:0.1533964300497312,SUP05_SP60_02960_unknown_description_:0.1331550445510311):0.06939415800248071,(((((R_fausta_HUE58_RS04660_unknown_description_:0,R_rectimargo_Rrec_peg_493_unknown_description_:0):0.02079686291645423,R_pacifica_Rpac_peg_696_unknown_description_:0.01264234337421326):0.02492137094084021,(R_pliocardia_Rpli_peg_908_unknown_description_:0.02237543501583309,R_southwardae_Rsou_peg_1273_unknown_description_:0.02338490124982114):0.01587285305728277):0.0108546507160141,R_phaseoliformis_Rpha_peg_1403_unknown_description_:0.02836633741323174):0.008338170673614597,R_magnifica_Rmag_0931_unknown_description_:0.04909216862311302):0.02537562294792943):0.08348841279808647,V_marissinica_Vmar_0846_unknown_description_:0.02460648952186702):0.003776136529583436,(V_okutanii_COSY_0837_unknown_description_:0.004735203169064032,(V_soyoae1_Vsoy1_peg_876_unknown_description_:0,V_soyoae2_Vsoy2_peg_92_unknown_description_:0):0.01318827027300549):0.0101567906053347):0.002116246621017817,(V_diagonalis_Vdia_peg_119_unknown_description_:0,V_extenta_Vext_peg_114_unknown_description_:0):0.02371395068266068,(V_gigas1_Vgig1_peg_858_unknown_description_:0,V_gigas2_Vgig2_peg_745_unknown_description_:0):0.02459763233086132)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fna/aligned_Rmag_0933.fna\n",
      "1-255\n",
      "((((((Bathy_MS2017_0281_unknown_description_:0.1627261048960603,SUP05_SP60_02970_unknown_description_:0.2116481284406847):0.082262252472181,R_southwardae_Rsou_peg_1276_unknown_description_:0.3575631637364876):0.02591311512707329,((((R_fausta_HUE58_RS04670_unknown_description_:0,R_rectimargo_Rrec_peg_495_unknown_description_:0):0.04240604903533937,R_pacifica_Rpac_peg_698_unknown_description_:0.02071787212632453):0.04398898342777671,R_phaseoliformis_Rpha_peg_1401_unknown_description_:0.008680821923918211):0.01126037310443607,(R_magnifica_Rmag_0933_unknown_description_:0.09561380006557973,R_pliocardia_Rpli_peg_910_unknown_description_:0.02133518644311967):0.004697505488240418):0.0152561375272568):0.1210706019889358,(V_diagonalis_Vdia_peg_117_unknown_description_:0,V_extenta_Vext_peg_112_unknown_description_:0):0.002899802266740539):0.01130897411859927,V_marissinica_Vmar_0848_unknown_description_:0.01313529154066525):0.001578059513437913,(V_gigas1_Vgig1_peg_860_unknown_description_:0,V_gigas2_Vgig2_peg_747_unknown_description_:0):0.01949382559649383,(V_okutanii_COSY_0839_unknown_description_:0.008126894533964736,(V_soyoae1_Vsoy1_peg_878_unknown_description_:0,V_soyoae2_Vsoy2_peg_94_unknown_description_:0):0.01267408788202588):0.00559734225644784)\n",
      "fna/aligned_Rmag_0935.fna\n",
      "1-393\n",
      "((((Bathy_MS2017_0277_unknown_description_:0.09130226552401452,SUP05_SP60_02990_unknown_description_:0.05044069093227802):0.06007968206985893,((((((R_fausta_HUE58_RS04690_unknown_description_:0,R_rectimargo_Rrec_peg_498_unknown_description_:0):0.01274687904678167,R_pacifica_Rpac_peg_701_unknown_description_:0.01028434373352228):0.01743401150565587,R_southwardae_Rsou_peg_1280_unknown_description_:0.01944676198695321):0.01053046138039342,R_pliocardia_Rpli_peg_916_unknown_description_:0.02028560296959341):0.004569232445682322,R_phaseoliformis_Rpha_peg_1395_unknown_description_:0.02636357776706253):0.01432354487992789,R_magnifica_Rmag_0935_unknown_description_:0.03820269457785578):0.03090980679840551):0.07437385970066396,V_marissinica_Vmar_0851_unknown_description_:0.01166818864729509):0.005293737209626257,((V_diagonalis_Vdia_peg_114_unknown_description_:0,V_extenta_Vext_peg_109_unknown_description_:0):0.005850879094967752,(V_gigas1_Vgig1_peg_863_unknown_description_:0,V_gigas2_Vgig2_peg_750_unknown_description_:0):0.01258243933601339):0.00502385254247956,(V_okutanii_COSY_0842_unknown_description_:0.006279425998863251,(V_soyoae1_Vsoy1_peg_881_unknown_description_:0,V_soyoae2_Vsoy2_peg_97_unknown_description_:0):0.009463678615077655):0.005882768777354056)\n",
      "fna/aligned_Rmag_0944.fna\n",
      "1-201\n",
      "(((((Bathy_MS2017_0261_unknown_description_:0.1122937016349718,SUP05_SP60_03020_unknown_description_:0.10078120816565):0.02111146221127964,((((R_fausta_HUE58_RS04725_unknown_description_:0,R_rectimargo_Rrec_peg_507_unknown_description_:0):0.02541926245350679,R_pacifica_Rpac_peg_713_unknown_description_:0.00534286693721096):0.01667515523119196,((R_phaseoliformis_Rpha_peg_1380_unknown_description_:0.0245988019442988,R_southwardae_Rsou_peg_1473_unknown_description_:0.00614610100993182):0.001529776409625809,R_pliocardia_Rpli_peg_933_unknown_description_:0.01430823327647634):0.0133355737565318):0.009848119561342898,R_magnifica_Rmag_0944_unknown_description_:0.04908019789937129):0.01739643247474265):0.02165229934128721,(V_gigas1_Vgig1_peg_888_unknown_description_:0,V_gigas2_Vgig2_peg_775_unknown_description_:0):0.0224131658227928):0.006165250186463607,(V_marissinica_Vmar_0854_unknown_description_:0.03057089820437546,V_okutanii_COSY_0845_unknown_description_:0.01059596691062531):0.00743732891677334):0.00323629296957471,(V_diagonalis_Vdia_peg_110_unknown_description_:0,V_extenta_Vext_peg_106_unknown_description_:0):0.02064466372446565,(V_soyoae1_Vsoy1_peg_884_unknown_description_:0,V_soyoae2_Vsoy2_peg_100_unknown_description_:0):0.01510597270454432)\n",
      "fna/aligned_Rmag_0969.fna\n",
      "1-1539\n",
      "((((Bathy_MS2017_0220_unknown_description_:0.1460337932314276,SUP05_SP60_03140_unknown_description_:0.1065964562346304):0.06466605286390202,(((((R_fausta_HUE58_RS04855_unknown_description_:0.001265382519197544,R_rectimargo_Rrec_peg_534_unknown_description_:0):0.01668042538508603,R_pacifica_Rpac_peg_745_unknown_description_:0.02253680176531525):0.020693194632941,R_phaseoliformis_Rpha_peg_1337_unknown_description_:0.02757618364873666):0.003583481823912804,(R_pliocardia_Rpli_peg_975_unknown_description_:0.0324492645530825,R_southwardae_Rsou_peg_1508_unknown_description_:0.02563375312566215):0.004584943606269565):0.01799069362089131,R_magnifica_Rmag_0969_unknown_description_:0.04859493104669967):0.03660050132519187):0.1144762523148176,((V_diagonalis_Vdia_peg_86_unknown_description_:0,V_extenta_Vext_peg_82_unknown_description_:0):0.02699861362382566,V_marissinica_Vmar_0879_unknown_description_:0.03009518930931685):0.003365835889992966):0.0002346548737911548,(V_gigas1_Vgig1_peg_890_unknown_description_:0,V_gigas2_Vgig2_peg_777_unknown_description_:0):0.02689384635193755,(V_okutanii_COSY_0869_unknown_description_:0.0161622096500996,(V_soyoae1_Vsoy1_peg_909_unknown_description_:0,V_soyoae2_Vsoy2_peg_125_unknown_description_:0):0.01429824999548433):0.01059177244946765)\n",
      "fna/aligned_Rmag_0989.fna\n",
      "1-609\n",
      "((((Bathy_MS2017_0185_unknown_description_:0.1942810584866639,SUP05_SP60_03255_unknown_description_:0.2209187064987513):0.0806167490851013,((((((R_fausta_HUE58_RS05040_unknown_description_:0,R_rectimargo_Rrec_peg_574_unknown_description_:0):0.01561162586574836,R_pacifica_Rpac_peg_771_unknown_description_:0.01555295635564618):0.02652262321198623,R_phaseoliformis_Rpha_peg_1634_unknown_description_:0.02896115510568429):0.01286666054030849,R_southwardae_Rsou_peg_1741_unknown_description_:0.037409814175023):0.02154527746659513,R_magnifica_Rmag_0989_unknown_description_:0.07445932434856142):0.01022411418478325,R_pliocardia_Rpli_peg_1007_unknown_description_:0.04503347561054939):0.0205302091401916):0.1023049172963148,(V_diagonalis_Vdia_peg_71_unknown_description_:0,V_extenta_Vext_peg_67_unknown_description_:0):0.01362217480805641):0.00674267925593533,(V_gigas1_Vgig1_peg_905_unknown_description_:0,V_gigas2_Vgig2_peg_792_unknown_description_:0):0.03492338046921842,(V_marissinica_Vmar_0894_unknown_description_:0.03115478039625564,(V_okutanii_COSY_0884_unknown_description_:0.0270008230774193,(V_soyoae1_Vsoy1_peg_925_unknown_description_:0,V_soyoae2_Vsoy2_peg_141_unknown_description_:0):0.02316586536015733):0.004549353089335205):0.003261333504739165)\n"
     ]
    }
   ],
   "source": [
    "gtr_fail=[\"Rmag_0052\",\"Rmag_0077\",\"Rmag_0135\",\"Rmag_0160\",\"Rmag_0178\",\"Rmag_0450\",\"Rmag_0476\",\"Rmag_0477\",\"Rmag_0478\",\"Rmag_0595\",\"Rmag_0647\",\"Rmag_0767\",\"Rmag_0828\",\"Rmag_0884\",\"Rmag_0931\",\"Rmag_0933\",\"Rmag_0935\",\"Rmag_0944\",\"Rmag_0969\",\"Rmag_0989\"]\n",
    "\n",
    "# for p in Path(\"/home/maeperez/scratch/Clams/gard/GTR/\").glob(\"*HKY*\"):\n",
    "#     if str(p).split('/')[-1][:9] not in gtr_fail:\n",
    "#         print(p)\n",
    "#         p.unlink()\n",
    "        \n",
    "for ps in gtr_fail:\n",
    "    for p in Path(\"/home/maeperez/scratch/Clams/gard/GTR/\").glob(ps+\"*HKY.gard_splits\"):\n",
    "#         print(p)\n",
    "        continue\n",
    "    \n",
    "    for p in Path(\"./fna/\").glob(\"*\"+ps+\"*\"):\n",
    "            fna=AlignIO.read(p,'fasta')\n",
    "            str_len=\"1-\"+str(len(fna[0].seq))\n",
    "            with open('/home/maeperez/scratch/Clams/init_trees/'+ps+'.tree','r') as f:\n",
    "                line=f.read().splitlines()[0][:-1]\n",
    "            \n",
    "            with open('/home/maeperez/scratch/Clams/gard/GTR/'+ps+'.NJ.gard_splits','w') as fa:\n",
    "                entry=str_len+'\\n'+line\n",
    "                fa.write(entry)\n",
    "            print(p)\n",
    "            print(str_len)\n",
    "            print(line)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0450.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0944.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0178.NJ.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0969.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0478.HKY.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0931.NJ.gard_splits\n",
      "/home/maeperez/scratch/Clams/gard/GTR/Rmag_0135.NJ.gard_splits\n"
     ]
    }
   ],
   "source": [
    "pref=[]\n",
    "ps=[]\n",
    "for p in Path('/home/maeperez/scratch/Clams/gard/GTR/').glob('*.gard_splits'):\n",
    "    pr=str(p).split('/')[-1][:9]\n",
    "    if pr not in pref:\n",
    "        pref+=[pr]\n",
    "        ps+=[p]\n",
    "    else:\n",
    "        print(p)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "for pr in ['Rmag_0450','Rmag_0944','Rmag_0178','Rmag_0969','Rmag_0478','Rmag_0931','Rmag_0135']:\n",
    "    for p in Path('/home/maeperez/scratch/Clams/gard/GTR/').glob(pr+'*.gard_splits'):\n",
    "#         print(p)\n",
    "        if 'NJ' in str(p):\n",
    "            print(p)\n",
    "            p.unlink()\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.pacifica.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.phaseoliformis.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.pleocardia.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.rectimargo.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/V.diagonalis.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/V.gigas.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.southwardae.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/R.kilmeri.gbk\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/V.extenta.gbk\n",
      "ATGATGATGACTGACACACCACTGGTAATTGCAGGAAAAACTTATCACTCTCGCTTGCTGGTTGGCTCAGGAAAATACAAAGATTTAACGCAAACAAAACTAGCAACTGAAGCCGCTCAGGCTGATATTATTACCGTTGCCATTCGTCGAACCAACATCGGACAAGACAAAAACGAGCCAAATTTATTAGACGTTATCAGCCTCGATAAATACACTATTTTGCCTAATACTGCAGGTTGCTACACCGCTAAAGATGCCGTACGTACGTGCCAATTAGCACGTGAACTTTTAGGCGGACATAATTTGGTTAAATTAGAAGTATTAGGCGATGAAAAAATCCTATACCCCAATATTGTTGAAACCCTATCTGCTGCACAAACCCTAGTTAATGATGGTTTTGATGTCATGGTTTACACCAGTGATGACCCAATCATTGCTAAAGAATTAGAACGCATCGGCTGTTGCGCCATTATGCCACTTGCCTCACTCATTGGCTCAGGCCAAGGCATTGTCAACCCAGCAAACATTAAACTCATCAAAGAGCACGCTAATGTACCCGTACTGGTTGATGCTGGTATTGGTTGCGCTTCAGATGCTGCCAAGGCTATGGAGCTAGGCTGCGATGGCGTACTAATGAACTCAGCCATCGCCAATGCCACCAATCCAGTCCTAATGGCAAGTGCCATGAAACACGCTGTCATTGCAGGGCGTGAATCCTTCCTTGCAGGTAGAATGATGAAAAAAGCTTACGCTTCAGCCTCATCCCCAATGGAAAACTTAATTTAA\n",
      "UDP-N-acetylmuramate:L-alanyl-gamma-D-glutamyl- meso-diaminopimelate ligase (EC 6.3.2.-)\n"
     ]
    }
   ],
   "source": [
    "######## create dictionnaries locus_tag -> db_xref, locus_tag -> product  #########\n",
    "\n",
    "\n",
    "assemblies=['R.pacifica.gbk', 'R.phaseoliformis.gbk', 'R.pleocardia.gbk', 'R.rectimargo.gbk', 'R.southwardae.gbk', 'R.kilmeri.gbk','V.extenta.gbk','V.gigas.gbk','V.diagonalis.gbk']\n",
    "\n",
    "mypath='/Users/maeperez/Desktop/SYMBIONT_GENOMES/Clams_symb/'\n",
    "onlyfiles = [mypath+f for f in listdir(mypath) if f[-3:]=='gbk' and f.split('/')[-1] in assemblies]\n",
    "print '\\n'.join(onlyfiles)\n",
    "\n",
    "db_xref_dic={}\n",
    "product_dic={}\n",
    "for file in onlyfiles:\n",
    "    sample=file.split('/')[-1][:-4]\n",
    "    db_xref_dic[sample]={}\n",
    "    product_dic[sample]={}\n",
    "    for record in SeqIO.parse(file,'genbank'):\n",
    "        for feature in record.features:\n",
    "            if feature.type=='CDS':\n",
    "                locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                if locus_tag =='Rsou_peg_980':\n",
    "                    print feature.location.extract(record).seq\n",
    "                db_xref = feature.qualifiers['db_xref'][0]\n",
    "                product = feature.qualifiers['product'][0]\n",
    "                \n",
    "                db_xref_dic[sample][locus_tag]=db_xref\n",
    "                product_dic[sample][locus_tag]=product\n",
    "                \n",
    "\n",
    "print product_dic['R.pacifica']['Rpac_peg_1059']   \n",
    "\n",
    "##### create new df with db_xref and products corresponding to the locus_tags  #########\n",
    "\n",
    "orth_dbxref=orth.replace(db_xref_dic)\n",
    "orth_prod=orth.replace(product_dic)\n",
    "print orth[:4]\n",
    "print orth_prod[:4]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extract pseudogenes to fna and faa files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bathy\n",
      "R.fausta\n",
      "R.magnifica\n",
      "R.pacifica\n",
      "R.phaseoliformis\n",
      "R.pliocardia\n",
      "R.rectimargo\n",
      "R.southwardae\n",
      "SUP05\n",
      "V.diagonalis\n",
      "V.extenta\n",
      "V.gigas1\n",
      "V.gigas2\n",
      "V.marissinica\n",
      "V.okutanii\n",
      "V.soyoae1\n",
      "V.soyoae2\n"
     ]
    }
   ],
   "source": [
    "## remove pseudogenes\n",
    "goodfiles={'Bathy':'../genomes/Bathy_complete_with_CRISPRannot.gbk',\\\n",
    "'R.fausta':'../genomes/R.fausta.gbk',\\\n",
    "'R.magnifica':'../genomes/R.magnifica.gbk',\\\n",
    "'R.pacifica':'../genomes/R.pacifica_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.phaseoliformis':'../genomes/R.phaseoliformis_8_contigs_fully_annotated.gbk',\\\n",
    "'R.pliocardia':'../genomes/R.pliocardia_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.rectimargo':'../genomes/R.rectimargo_one_contig_circular_fully_annotated.gbk',\\\n",
    "'R.southwardae':'../genomes/R.southwardae_39_contigs_fully_annotated.gbk',\\\n",
    "'SUP05':'../genomes/SUP05.gbk',\\\n",
    "'V.diagonalis':'../genomes/V.diagonalis_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.extenta':'../genomes/V.extenta_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas1':'../genomes/V.gigas1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.gigas2':'../genomes/V.gigas2_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.marissinica':'../genomes/V.marissinica_withlocustags.gbk',\\\n",
    "'V.okutanii':'../genomes/V.okutanii_with_locus_tags.gbk',\\\n",
    "'V.soyoae1':'../genomes/V.soyoae1_one_contig_circular_fully_annotated.gbk',\\\n",
    "'V.soyoae2':'../genomes/V.soyoae2_one_contig_circular_fully_annotated.gbk'}\n",
    "\n",
    "\n",
    "for sample in goodfiles.keys():\n",
    "    print(sample)\n",
    "    pseudogenes_fna=[]\n",
    "    pseudogenes_faa=[]\n",
    "    pseudogenes_lc=[]\n",
    "    for record in SeqIO.parse(goodfiles[sample],'genbank'):\n",
    "            for feature in record.features:\n",
    "                if feature.type in ['CDS','gene']:\n",
    "                    locus_tag = feature.qualifiers['locus_tag'][0]\n",
    "                    if 'pseudo' in feature.qualifiers: #verify if pseudogene\n",
    "                        if locus_tag in pseudogenes_lc:\n",
    "                            continue\n",
    "                        pseudogenes_lc+=[locus_tag]\n",
    "                        fna=feature.location.extract(record)\n",
    "                        fna.id=locus_tag\n",
    "                        fna.name=''\n",
    "                        fna.description=''\n",
    "                        faa=fna.translate(11)\n",
    "                        faa.id=locus_tag\n",
    "                        faa.name=''\n",
    "                        faa.description=''\n",
    "                        if 'note' in feature.qualifiers:\n",
    "                            fna.description='\"'+feature.qualifiers['note'][0]+'\"'\n",
    "                            faa.description='\"'+feature.qualifiers['note'][0]+'\"'\n",
    "\n",
    "                        pseudogenes_fna+=[fna]\n",
    "                        pseudogenes_faa+=[faa]\n",
    "    if len(pseudogenes_fna)>0:\n",
    "        SeqIO.write(pseudogenes_fna,sample+'_'+'pseudogenes.fna','fasta')\n",
    "        SeqIO.write(pseudogenes_faa,sample+'_'+'pseudogenes.faa','fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_files='../mauve_alignments/'\n",
    "d=read_csv(path_to_files+Orth)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.6",
   "language": "python",
   "name": "py36"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
