{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import Bio\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.SeqFeature import SeqFeature, FeatureLocation\n",
    "from Bio import AlignIO\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make unique genbank files from multi-gbk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AP014555.1\n",
      "10_12_13_a\n",
      "4-3_5-3_6-\n",
      "C._Magnifi\n",
      "11_MAR-806\n",
      "1_2?3_3?3_\n",
      "4_5_14_kil\n",
      "1_23_33_di\n",
      "9-3_17_BR-\n",
      "A._phaseol\n",
      "A._mariana\n",
      "8_pac_mt_I\n",
      "9_pac_mt_I\n",
      "15_16_giga\n"
     ]
    }
   ],
   "source": [
    "ids=[]\n",
    "for record in SeqIO.parse('Cmag_ref_GLOBAL_multi-GenBank.gbff','genbank'):\n",
    "    print(record.id)\n",
    "    ids+=[record.id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic={'AP014555.1':'P.okutanii_ref',\n",
    " '10_12_13_a':'A.phaseoliformis',\n",
    " '4-3_5-3_6-':'P.extenta',\n",
    " 'C._Magnifi':'C.magnifica',\n",
    " '11_MAR-806':'A.southwardae',\n",
    " '1_2?3_3?3_':'A.diagonalis', # doublon to be eliminated\n",
    " '4_5_14_kil':'P.soyae',\n",
    " '1_23_33_di':'A.diagonalis',\n",
    " '9-3_17_BR-':'X.pliocardia',\n",
    " 'A._phaseol':'A.phaseoliformis_ref', # reference AP014557.1\n",
    " 'A._mariana':'A.mariana_ref',\n",
    " '8_pac_mt_I':'C.pacifica',\n",
    " '9_pac_mt_I':'C.rectimargo',\n",
    " '15_16_giga':'A.gigas'}\n",
    "\n",
    "cols=['A.phaseoliformis_ref', 'P.soyae','C.magnifica','C.pacifica','A.phaseoliformis', 'X.pliocardia','C.rectimargo', 'A.southwardae','A.mariana_ref','A.diagonalis','P.extenta','A.gigas','P.okutanii_ref']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "find_end_of_gene() missing 1 required positional argument: 'gene_start'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-71-88e03cc6a213>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m             \u001b[0;32mreturn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m             \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mfind_end_of_gene\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: find_end_of_gene() missing 1 required positional argument: 'gene_start'"
     ]
    }
   ],
   "source": [
    "seq1=record.seq[feature.location.start:]\n",
    "def find_end_of_gene(seq1,gene_start):\n",
    "    for i in range(gene_start, len(seq1), 3):\n",
    "        if seq1[i:i+3] in ['TAG','TAA']:\n",
    "            return(i+3)\n",
    "            break\n",
    "find_end_of_gene(seq1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type: CDS\n",
      "location: [1553:2585](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Poku_06\n",
      "\n",
      "1553 2585\n",
      "1032\n",
      "gene_end 2585\n",
      "1032\n",
      "MLIGWFVPMATIIGGYIMVGGIFVCVYSVGLLGMWVGMELSFFGAISVLGGSSVEEVEGVQKYFIVQVFSSIFLLIGFLLVVNFYSGGWLVEFILVVGLMVKLGLFPFYFWIPSVMGVVSWVGCFVVSVAQKFNPIWILVNLMLEGYLQQVIEFSACLTGLVGCIGGIGVLQFRVLLAFSSMVHLGYMVFMSLIGWDVFLSYMGLYFCLSFSLMMSLWVCGVYSFLDLMKSNSGGEMNLFMISLYMFSFAGVPPLSGCFMKAVFLMSCWELFPIGCVILFLSSGVSLFYYSSFFMYLNLYWGSVLKWEFVSGKFGDLKVFLFFVSVLLNVAMGFSLFLLIVLV\n",
      "type: CDS\n",
      "location: [13750:14782](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Apha_44\n",
      "    Key: translation, Value: ['MLIG*FVPMAVIVRGYIMVGGIFVCIYRVGLLGIWVGIELRFFGAVGVLGGSRVEEVEGVQKYFIIQVFRSIFLLIGFLLVINFYKGG*LVEFIIVTGLIMKLGLFPFYF*IPSVIRVVSWVRCFVVSVVQKFNPI*ILVNLILESYLQQIIEFSACLTGLVGCIGGIGVLQFRVLLAFSSMVHLGYIIFICLVGWDEFLTYIILYFCLSFRLIMSLWVCGIYRFLDLIKSNGGRDVDLFMISLYIFSFAGIPPLSGCFMKAVFLVSCWEVFPVGCVVLFLSSGVSLFYYSRFFTYLNLY*GRILKWQFVKRRLADLKVFLFFVRVLLNVVLGFGLFLLIVII']\n",
      "\n",
      "13750 14782\n",
      "1032\n",
      "gene_end 14782\n",
      "1032\n",
      "MLIGWFVPMAVIVSGYIMVGGIFVCIYSVGLLGMWVGMELSFFGAVGVLGGSSVEEVEGVQKYFIIQVFSSIFLLIGFLLVINFYKGGWLVEFIMVTGLMMKLGLFPFYFWIPSVMSVVSWVSCFVVSVVQKFNPIWILVNLMLESYLQQMIEFSACLTGLVGCIGGIGVLQFRVLLAFSSMVHLGYMIFMCLVGWDEFLTYMILYFCLSFSLMMSLWVCGIYSFLDLMKSNGGSDVDLFMISLYMFSFAGIPPLSGCFMKAVFLVSCWEVFPVGCVVLFLSSGVSLFYYSSFFTYLNLYWGSMLKWQFVKSSLADLKVFLFFVSVLLNVVLGFGLFLLIVII\n",
      "type: CDS\n",
      "location: [13355:14387](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Pext_38\n",
      "    Key: translation, Value: ['MLIG*FVPIAIIIGGYIMVGGIFVCVYRVGLLGIWVGMELSFFGAIGVLGGSRVEEVEGVQKYFIIQVFRSIFLLIGFLLVVNFYMGGWLVEFILVVGLIMKLGLFPFYF*IPSVIGVVSWVGCFVVSVVQKFNPI*VLVNLILERYLQQVIEFSACLTGLVGCIGGIGVLQFRILLAFSSIVHLGYIVFISLIG*GVFLSYIGVYFCLSFSLIIRL*VCGVYRFLDLVKSNVDGEINLFMISLYIFSFAGVPPLRGCFIKGVFLISC*ELFPIGCVILFLSSGVSLFYYSRFFVYLNLY*GRILK*EFVKGRLADLKVILFFVRVLLNVVMGFGLFLLIVLV']\n",
      "\n",
      "13355 14387\n",
      "1032\n",
      "gene_end 14387\n",
      "1032\n",
      "MLIGWFVPMAIIIGGYIMVGGIFVCVYSVGLLGMWVGMELSFFGAIGVLGGSSVEEVEGVQKYFIIQVFSSIFLLIGFLLVVNFYMGGWLVEFILVVGLMMKLGLFPFYFWIPSVMGVVSWVGCFVVSVVQKFNPIWVLVNLMLESYLQQVIEFSACLTGLVGCIGGIGVLQFRILLAFSSMVHLGYMVFMSLIGWGVFLSYMGVYFCLSFSLMMSLWVCGVYSFLDLVKSNVDGEMNLFMISLYMFSFAGVPPLSGCFMKGVFLMSCWELFPIGCVILFLSSGVSLFYYSSFFVYLNLYWGSMLKWEFVKGSLADLKVILFFVSVLLNVVMGFGLFLLIVLV\n",
      "type: CDS\n",
      "location: [16273:17305](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Cmag_35\n",
      "    Key: translation, Value: ['MLIG*FVPMAVIVGGYLLVGGIFVCIYRVGLLGMWVGMELRFFGAVGVLGGSRVEEVEGVQKYFIIQVFRSIFLLIGFLLVINFYKGG*LVELIIIVGLIMKLGLFPFYF*IPSVIGVVSWVGCFVVSVVQKFNPI*ILVNLMLESYLQQVIEFSACLTGLVGCIGGIGVLQFRVLLAFSSIVHLGYMVFICLIGWDVFLPYMALYFCLSFRLIMRL*ACGIYRFLDLTKCNESVDIDLFIISLYMFSFAGIPPLRGCFIKAIFLVSC*EVFPIGCVVLFLSSGVSLFYYSRFFTYLNLYWGRILK*QFIKRRVMDLKVFLFFVRVLLNVVVGFGLFLSIVII']\n",
      "\n",
      "16273 17305\n",
      "1032\n",
      "gene_end 17305\n",
      "1032\n",
      "MLIGWFVPMAVIVGGYLLVGGIFVCIYSVGLLGMWVGMELSFFGAVGVLGGSSVEEVEGVQKYFIIQVFSSIFLLIGFLLVINFYKGGWLVELIMMVGLMMKLGLFPFYFWIPSVMGVVSWVGCFVVSVVQKFNPIWILVNLMLESYLQQVIEFSACLTGLVGCIGGIGVLQFRVLLAFSSMVHLGYMVFMCLIGWDVFLPYMALYFCLSFSLMMSLWACGIYSFLDLTKCNESVDMDLFMISLYMFSFAGIPPLSGCFMKAIFLVSCWEVFPIGCVVLFLSSGVSLFYYSSFFTYLNLYWGSMLKWQFMKSSVMDLKVFLFFVSVLLNVVVGFGLFLSIVII\n",
      "type: CDS\n",
      "location: [13279:14311](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Asou_45\n",
      "    Key: translation, Value: ['MLIG*FVPMAVIVRGYIMVGGIFVCIYRVGLLGIWVGMELRFFGAVGVLGGSRVEEVEGVQKYFIIQVFRSIFLLIGFLLVINFYKGG*LVEFILVVGLIIKLGLFPFYF*IPSVIRVVSWVRCFVVSVVQKFNPI*ILVNLILESYLQQIIEFSACLTGLVGCIGGIGVLQFRVLLAFSSIVHLGYIIFICLVGWDEFLTYIVLYFCLSFRLIIRLWICGIYRFLDLIKSNGGRDVDLFMISLYIFSFAGIPPLRGCFMKAVFLVSCWEVFPVGCVVLFLSSGVSLFYYRRFFTYLNLY*GRILK*QFVKRRLADLKVFLFFVRILLNVVIGFGLFLLIVLI']\n",
      "\n",
      "13279 14311\n",
      "1032\n",
      "gene_end 14311\n",
      "1032\n",
      "MLIGWFVPMAVIVSGYIMVGGIFVCIYSVGLLGMWVGMELSFFGAVGVLGGSSVEEVEGVQKYFIIQVFSSIFLLIGFLLVINFYKGGWLVEFILVVGLMMKLGLFPFYFWIPSVMSVVSWVSCFVVSVVQKFNPIWILVNLMLESYLQQMIEFSACLTGLVGCIGGIGVLQFRVLLAFSSMVHLGYMIFMCLVGWDEFLTYMVLYFCLSFSLMMSLWICGIYSFLDLMKSNGGSDVDLFMISLYMFSFAGIPPLSGCFMKAVFLVSCWEVFPVGCVVLFLSSGVSLFYYSSFFTYLNLYWGSMLKWQFVKSSLADLKVFLFFVSILLNVVMGFGLFLLIVLI\n",
      "type: CDS\n",
      "location: [15026:16049](+)\n",
      "qualifiers:\n",
      "    Key: gene, Value: ['ND2']\n",
      "    Key: locus_tag, Value: Psoy_44\n",
      "\n",
      "15026 16049\n",
      "1023\n"
     ]
    },
    {
     "ename": "TranslationError",
     "evalue": "Final codon 'GTA' is not a stop codon",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTranslationError\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-82-c7818d6621b0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     34\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlocation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mend\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlocation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 36\u001b[0;31m                 \u001b[0mprot_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextract\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranslate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     37\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     38\u001b[0m                 \u001b[0mgene_start\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlocation\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36mtranslate\u001b[0;34m(self, table, stop_symbol, to_stop, cds, gap)\u001b[0m\n\u001b[1;32m   1276\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1277\u001b[0m         protein = _translate_str(\n\u001b[0;32m-> 1278\u001b[0;31m             \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcodon_table\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstop_symbol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_stop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgap\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1279\u001b[0m         )\n\u001b[1;32m   1280\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m_translate_str\u001b[0;34m(sequence, table, stop_symbol, to_stop, cds, pos_stop, gap)\u001b[0m\n\u001b[1;32m   2846\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mstop_codons\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2847\u001b[0m             raise CodonTable.TranslationError(\n\u001b[0;32m-> 2848\u001b[0;31m                 \u001b[0;34m\"Final codon '{0}' is not a stop codon\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2849\u001b[0m             )\n\u001b[1;32m   2850\u001b[0m         \u001b[0;31m# Don't translate the stop symbol, and manually translate the M\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTranslationError\u001b[0m: Final codon 'GTA' is not a stop codon"
     ]
    }
   ],
   "source": [
    "recs={}\n",
    "seqs={}\n",
    "# seqs={'ND2': {}, 'ND6': {}, 'ND4L': {}, 'COX3': {}, 'COX1': {}, 'COX2': {}, 'CYTB': {}, 'ATP8': {}, 'ND4': {}, 'ATP6': {}, 'ND3': {}, 'ND1': {}, 'ND5': {}}\n",
    "def find_end_of_gene(seq1,gene_start):\n",
    "    for i in range(gene_start, len(seq1), 3):\n",
    "        if seq1[i:i+3] in ['TAG','TAA']:\n",
    "            return(i+3)\n",
    "            break\n",
    "\n",
    "for record in SeqIO.parse('Cmag_ref_GLOBAL_multi-GenBank_manually_annotated.gbff','genbank'):\n",
    "    if record.id=='1_2?3_3?3_':\n",
    "        continue\n",
    "        \n",
    "    i=0\n",
    "    recs[dic[record.id]]={}\n",
    "    record.name=dic[record.id]\n",
    "    for feature in record.features:\n",
    "        if feature.type=='source':\n",
    "            continue\n",
    "        if feature.type=='gene':\n",
    "            i+=1\n",
    "            feature.qualifiers['locus_tag']=record.name[0]+record.name[2:5]+'_'+\"{:02d}\".format(i)\n",
    "            last_locus_tag=record.name[0]+record.name[2:5]+'_'+\"{:02d}\".format(i)\n",
    "        else:\n",
    "            feature.qualifiers['locus_tag']=last_locus_tag\n",
    "        if feature.type=='CDS':\n",
    "            if feature.qualifiers['gene'][0]=='ND2':\n",
    "                recs[record.name].update({feature.qualifiers['gene'][0]:feature.qualifiers['locus_tag']})\n",
    "#                 print(feature,len(feature.extract(record.seq))%3)\n",
    "#                 print(len(feature.extract(record.seq)))\n",
    "                print(feature)\n",
    "                print(feature.location.start,feature.location.end)\n",
    "                print(feature.location.end-feature.location.start)\n",
    " \n",
    "                prot_seq=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "\n",
    "                gene_start=feature.location.start\n",
    "                gene_end=find_end_of_gene(record.seq,gene_start)\n",
    "                print('gene_end',gene_end)\n",
    "                nucl_seq=record.seq[gene_start:gene_end]\n",
    "                print(len(nucl_seq))\n",
    "                prot_seq=nucl_seq.translate(table=5, cds=True)\n",
    "                print (prot_seq)\n",
    "#             prot_seq=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "\n",
    "#             seqs[feature.qualifiers['locus_tag']]={\"prot\":prot_seq,\"nucl\":nucl_seq}\n",
    "\n",
    "#     break\n",
    "#     SeqIO.write(record, dic[record.id]+'.gbk', \"genbank\")\n",
    "     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16058"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "find_end_of_gene(record.seq,feature.location.start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.DataFrame.from_dict(recs,orient='index')\n",
    "df.columns\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add locus_tags to geneious files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic={'AP014555.1':'P.okutanii_ref',\n",
    " '10_12_13_a':'A.phaseoliformis',\n",
    " '4-3_5-3_6-':'P.extenta',\n",
    " 'C._Magnifi':'C.magnifica',\n",
    " '11_MAR-806':'A.southwardae',\n",
    " '1_2?3_3?3_':'A.diagonalis', # doublon to be eliminated\n",
    " '4_5_14_kil':'P.soyae',\n",
    " '1_23_33_di':'A.diagonalis',\n",
    " '9-3_17_BR-':'X.pliocardia',\n",
    " 'A._phaseol':'A.phaseoliformis_ref', # reference AP014557.1\n",
    " 'A._mariana':'A.mariana_ref',\n",
    " '8_pac_mt_I':'C.pacifica',\n",
    " '9_pac_mt_I':'C.rectimargo',\n",
    " '15_16_giga':'A.gigas'}\n",
    "\n",
    "dic={'Pac_II_':'C.rectimargo',\n",
    " '11_MAR':'A.southwardae',\n",
    " 'extenta_combined':'P.extenta',\n",
    " 'A._phaseoliformi':'A.phaseoliformis_ref',\n",
    " 'R_magnifica':'C.magnifica',\n",
    " '8-3_Blake_Ridge':'X.pliocardia',\n",
    " 'A._mariana':'A.mariana_ref',\n",
    " 'Contig_54_11_MAR':'A.southwardae_c54',\n",
    " '13_Aleutian':'A.phaseoliformis',\n",
    " 'diagonalis':'A.diagonalis',\n",
    " 'Pac_I_':'C.pacifica',\n",
    " 'P._okutanii':'P.okutanii_ref',\n",
    " 'kilmeri':'P.soyae'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 479,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'str' object is not callable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-479-a21d51fa2ff2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0monlyfiles\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmypath\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0;34m'.gb'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mrecs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0monlyfiles\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mfile\u001b[0m \u001b[0;32min\u001b[0m \u001b[0monlyfiles\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmypath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'genbank'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: 'str' object is not callable"
     ]
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "mypath='./geneious_mitoch/'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-3:]=='.gb']\n",
    "recs={}\n",
    "print(onlyfiles)\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        i=0\n",
    "        record.id=dic[record.name]\n",
    "        recs[record.id]={}\n",
    "        for n in range(len(record.features)):\n",
    "            \n",
    "            feature=record.features[n]\n",
    "            if n==0:\n",
    "                if feature.type=='source':\n",
    "                    feature.qualifiers['organism']=dic[record.name]\n",
    "                    continue\n",
    "                else:\n",
    "                    print(len(record.seq))\n",
    "                    new_feature=SeqFeature(FeatureLocation(1, len(record.seq)), type=\"source\")\n",
    "                    new_feature.qualifiers['organism']= dic[record.name]\n",
    "                    continue\n",
    "            if feature.type=='misc_feature':\n",
    "                continue\n",
    "            if feature.type=='tRNA':\n",
    "                continue\n",
    "            if feature.type=='CDS':\n",
    "                i+=1\n",
    "                try:\n",
    "                    feature.qualifiers['product']=feature.qualifiers['standard_name']\n",
    "                except KeyError:\n",
    "                    pass\n",
    "                feature.qualifiers['locus_tag']=record.id[0]+record.id[2:5]+'_'+\"{:02d}\".format(i)\n",
    "                last_locus_tag=record.id[0]+record.id[2:5]+'_'+\"{:02d}\".format(i)\n",
    "#                 print(feature.qualifiers.keys())\n",
    "                recs[record.id].update({feature.qualifiers['product'][0].split(' ')[0]:feature.qualifiers['locus_tag']})\n",
    "            else:\n",
    "#                 print(feature.qualifiers.keys())\n",
    "                try:\n",
    "                    feature.qualifiers['product']=feature.qualifiers['standard_name']\n",
    "                except KeyError:\n",
    "                    pass\n",
    "#                 feature.qualifiers['locus_tag']=last_locus_tag\n",
    "        if new_feature:\n",
    "#             print(new_feature)\n",
    "            record.features=[new_feature]+record.features\n",
    "#         SeqIO.write(record, path+'locus_taged_'+record.name+'.gbk', \"genbank\")\n",
    "#     break\n",
    "recs    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ND6</th>\n",
       "      <th>ND4L</th>\n",
       "      <th>ND2</th>\n",
       "      <th>COX3</th>\n",
       "      <th>COX1</th>\n",
       "      <th>COX</th>\n",
       "      <th>CYTB</th>\n",
       "      <th>ATP8</th>\n",
       "      <th>ND4</th>\n",
       "      <th>ATP6</th>\n",
       "      <th>ND3</th>\n",
       "      <th>ND1</th>\n",
       "      <th>ND5</th>\n",
       "      <th>COX2</th>\n",
       "      <th>cytochrome</th>\n",
       "      <th>ATP</th>\n",
       "      <th>NADH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C.rectimargo</th>\n",
       "      <td>Crec_01</td>\n",
       "      <td>Crec_02</td>\n",
       "      <td>Crec_03</td>\n",
       "      <td>Crec_04</td>\n",
       "      <td>Crec_05</td>\n",
       "      <td>Crec_06</td>\n",
       "      <td>Crec_07</td>\n",
       "      <td>Crec_08</td>\n",
       "      <td>Crec_09</td>\n",
       "      <td>Crec_10</td>\n",
       "      <td>Crec_11</td>\n",
       "      <td>Crec_12</td>\n",
       "      <td>Crec_13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.southwardae</th>\n",
       "      <td>Asou_13</td>\n",
       "      <td>Asou_01</td>\n",
       "      <td>Asou_02</td>\n",
       "      <td>Asou_03</td>\n",
       "      <td>Asou_04</td>\n",
       "      <td>Asou_05</td>\n",
       "      <td>Asou_06</td>\n",
       "      <td>Asou_07</td>\n",
       "      <td>Asou_08</td>\n",
       "      <td>Asou_09</td>\n",
       "      <td>Asou_10</td>\n",
       "      <td>Asou_11</td>\n",
       "      <td>Asou_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.extenta</th>\n",
       "      <td>Pext_01</td>\n",
       "      <td>Pext_02</td>\n",
       "      <td>Pext_03</td>\n",
       "      <td>Pext_04</td>\n",
       "      <td>Pext_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pext_07</td>\n",
       "      <td>Pext_08</td>\n",
       "      <td>Pext_09</td>\n",
       "      <td>Pext_10</td>\n",
       "      <td>Pext_11</td>\n",
       "      <td>Pext_12</td>\n",
       "      <td>Pext_13</td>\n",
       "      <td>Pext_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.phaseoliformis_ref</th>\n",
       "      <td>Apha_10</td>\n",
       "      <td>Apha_11</td>\n",
       "      <td>Apha_12</td>\n",
       "      <td>Apha_13</td>\n",
       "      <td>Apha_01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Apha_03</td>\n",
       "      <td>Apha_04</td>\n",
       "      <td>Apha_05</td>\n",
       "      <td>Apha_06</td>\n",
       "      <td>Apha_07</td>\n",
       "      <td>Apha_08</td>\n",
       "      <td>Apha_09</td>\n",
       "      <td>Apha_02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X.pliocardia</th>\n",
       "      <td>Xpli_01</td>\n",
       "      <td>Xpli_02</td>\n",
       "      <td>Xpli_03</td>\n",
       "      <td>Xpli_04</td>\n",
       "      <td>Xpli_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Xpli_07</td>\n",
       "      <td>Xpli_08</td>\n",
       "      <td>Xpli_09</td>\n",
       "      <td>Xpli_10</td>\n",
       "      <td>Xpli_11</td>\n",
       "      <td>Xpli_12</td>\n",
       "      <td>Xpli_13</td>\n",
       "      <td>Xpli_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.mariana_ref</th>\n",
       "      <td>Amar_01</td>\n",
       "      <td>Amar_02</td>\n",
       "      <td>Amar_03</td>\n",
       "      <td>Amar_04</td>\n",
       "      <td>Amar_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Amar_07</td>\n",
       "      <td>Amar_08</td>\n",
       "      <td>Amar_09</td>\n",
       "      <td>Amar_10</td>\n",
       "      <td>Amar_11</td>\n",
       "      <td>Amar_12</td>\n",
       "      <td>Amar_13</td>\n",
       "      <td>Amar_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.southwardae_c54</th>\n",
       "      <td>Asou_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asou_01</td>\n",
       "      <td>Asou_02</td>\n",
       "      <td>Asou_03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asou_05</td>\n",
       "      <td>Asou_06</td>\n",
       "      <td>Asou_07</td>\n",
       "      <td>Asou_08</td>\n",
       "      <td>Asou_09</td>\n",
       "      <td>Asou_10</td>\n",
       "      <td>Asou_11</td>\n",
       "      <td>Asou_04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.phaseoliformis</th>\n",
       "      <td>Apha_13</td>\n",
       "      <td>Apha_12</td>\n",
       "      <td>Apha_11</td>\n",
       "      <td>Apha_10</td>\n",
       "      <td>Apha_09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Apha_07</td>\n",
       "      <td>Apha_06</td>\n",
       "      <td>Apha_05</td>\n",
       "      <td>Apha_04</td>\n",
       "      <td>Apha_03</td>\n",
       "      <td>Apha_02</td>\n",
       "      <td>Apha_01</td>\n",
       "      <td>Apha_08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.diagonalis</th>\n",
       "      <td>Adia_01</td>\n",
       "      <td>Adia_02</td>\n",
       "      <td>Adia_03</td>\n",
       "      <td>Adia_04</td>\n",
       "      <td>Adia_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adia_07</td>\n",
       "      <td>Adia_08</td>\n",
       "      <td>Adia_09</td>\n",
       "      <td>Adia_10</td>\n",
       "      <td>Adia_11</td>\n",
       "      <td>Adia_12</td>\n",
       "      <td>Adia_13</td>\n",
       "      <td>Adia_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.okutanii_ref</th>\n",
       "      <td>Poku_01</td>\n",
       "      <td>Poku_02</td>\n",
       "      <td>Poku_03</td>\n",
       "      <td>Poku_04</td>\n",
       "      <td>Poku_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Poku_07</td>\n",
       "      <td>Poku_08</td>\n",
       "      <td>Poku_09</td>\n",
       "      <td>Poku_10</td>\n",
       "      <td>Poku_11</td>\n",
       "      <td>Poku_12</td>\n",
       "      <td>Poku_13</td>\n",
       "      <td>Poku_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.soyae</th>\n",
       "      <td>Psoy_01</td>\n",
       "      <td>Psoy_02</td>\n",
       "      <td>Psoy_03</td>\n",
       "      <td>Psoy_04</td>\n",
       "      <td>Psoy_05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Psoy_07</td>\n",
       "      <td>Psoy_08</td>\n",
       "      <td>Psoy_09</td>\n",
       "      <td>Psoy_10</td>\n",
       "      <td>Psoy_11</td>\n",
       "      <td>Psoy_12</td>\n",
       "      <td>Psoy_13</td>\n",
       "      <td>Psoy_06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C.pacifica</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Cpac_01</td>\n",
       "      <td>Cpac_02</td>\n",
       "      <td>Cpac_03</td>\n",
       "      <td>Cpac_04</td>\n",
       "      <td>Cpac_05</td>\n",
       "      <td>Cpac_06</td>\n",
       "      <td>Cpac_07</td>\n",
       "      <td>Cpac_08</td>\n",
       "      <td>Cpac_09</td>\n",
       "      <td>Cpac_10</td>\n",
       "      <td>Cpac_11</td>\n",
       "      <td>Cpac_12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C.magnifica</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cmag_13</td>\n",
       "      <td>Cmag_06</td>\n",
       "      <td>Cmag_12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          ND6     ND4L      ND2     COX3     COX1      COX  \\\n",
       "C.rectimargo          Crec_01  Crec_02  Crec_03  Crec_04  Crec_05  Crec_06   \n",
       "A.southwardae         Asou_13  Asou_01  Asou_02  Asou_03  Asou_04  Asou_05   \n",
       "P.extenta             Pext_01  Pext_02  Pext_03  Pext_04  Pext_05      NaN   \n",
       "A.phaseoliformis_ref  Apha_10  Apha_11  Apha_12  Apha_13  Apha_01      NaN   \n",
       "X.pliocardia          Xpli_01  Xpli_02  Xpli_03  Xpli_04  Xpli_05      NaN   \n",
       "A.mariana_ref         Amar_01  Amar_02  Amar_03  Amar_04  Amar_05      NaN   \n",
       "A.southwardae_c54     Asou_12      NaN  Asou_01  Asou_02  Asou_03      NaN   \n",
       "A.phaseoliformis      Apha_13  Apha_12  Apha_11  Apha_10  Apha_09      NaN   \n",
       "A.diagonalis          Adia_01  Adia_02  Adia_03  Adia_04  Adia_05      NaN   \n",
       "P.okutanii_ref        Poku_01  Poku_02  Poku_03  Poku_04  Poku_05      NaN   \n",
       "P.soyae               Psoy_01  Psoy_02  Psoy_03  Psoy_04  Psoy_05      NaN   \n",
       "C.pacifica                NaN  Cpac_01  Cpac_02  Cpac_03  Cpac_04  Cpac_05   \n",
       "C.magnifica               NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "\n",
       "                         CYTB     ATP8      ND4     ATP6      ND3      ND1  \\\n",
       "C.rectimargo          Crec_07  Crec_08  Crec_09  Crec_10  Crec_11  Crec_12   \n",
       "A.southwardae         Asou_06  Asou_07  Asou_08  Asou_09  Asou_10  Asou_11   \n",
       "P.extenta             Pext_07  Pext_08  Pext_09  Pext_10  Pext_11  Pext_12   \n",
       "A.phaseoliformis_ref  Apha_03  Apha_04  Apha_05  Apha_06  Apha_07  Apha_08   \n",
       "X.pliocardia          Xpli_07  Xpli_08  Xpli_09  Xpli_10  Xpli_11  Xpli_12   \n",
       "A.mariana_ref         Amar_07  Amar_08  Amar_09  Amar_10  Amar_11  Amar_12   \n",
       "A.southwardae_c54     Asou_05  Asou_06  Asou_07  Asou_08  Asou_09  Asou_10   \n",
       "A.phaseoliformis      Apha_07  Apha_06  Apha_05  Apha_04  Apha_03  Apha_02   \n",
       "A.diagonalis          Adia_07  Adia_08  Adia_09  Adia_10  Adia_11  Adia_12   \n",
       "P.okutanii_ref        Poku_07  Poku_08  Poku_09  Poku_10  Poku_11  Poku_12   \n",
       "P.soyae               Psoy_07  Psoy_08  Psoy_09  Psoy_10  Psoy_11  Psoy_12   \n",
       "C.pacifica            Cpac_06  Cpac_07  Cpac_08  Cpac_09  Cpac_10  Cpac_11   \n",
       "C.magnifica               NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "\n",
       "                          ND5     COX2 cytochrome      ATP     NADH  \n",
       "C.rectimargo          Crec_13      NaN        NaN      NaN      NaN  \n",
       "A.southwardae         Asou_12      NaN        NaN      NaN      NaN  \n",
       "P.extenta             Pext_13  Pext_06        NaN      NaN      NaN  \n",
       "A.phaseoliformis_ref  Apha_09  Apha_02        NaN      NaN      NaN  \n",
       "X.pliocardia          Xpli_13  Xpli_06        NaN      NaN      NaN  \n",
       "A.mariana_ref         Amar_13  Amar_06        NaN      NaN      NaN  \n",
       "A.southwardae_c54     Asou_11  Asou_04        NaN      NaN      NaN  \n",
       "A.phaseoliformis      Apha_01  Apha_08        NaN      NaN      NaN  \n",
       "A.diagonalis          Adia_13  Adia_06        NaN      NaN      NaN  \n",
       "P.okutanii_ref        Poku_13  Poku_06        NaN      NaN      NaN  \n",
       "P.soyae               Psoy_13  Psoy_06        NaN      NaN      NaN  \n",
       "C.pacifica            Cpac_12      NaN        NaN      NaN      NaN  \n",
       "C.magnifica               NaN      NaN    Cmag_13  Cmag_06  Cmag_12  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df=pd.DataFrame.from_dict(recs,orient='index')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fix genes on geneious mitoch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['locus_taged_extenta_combined.gbk', 'locus_taged_A._phaseoliformi.gbk', 'locus_taged_P._okutanii.gbk', 'locus_taged_Pac_II_.gbk', 'locus_taged_diagonalis.gbk', 'locus_taged_11_MAR.gbk', 'locus_taged_R_magnifica.gbk', 'locus_taged_13_Aleutian.gbk', 'locus_taged_Contig_54_11_MAR.gbk', 'locus_taged_kilmeri.gbk', 'locus_taged_8-3_Blake_Ridge.gbk', 'locus_taged_A._mariana.gbk', 'locus_taged_Pac_I_.gbk']\n",
      "locus_taged_extenta_combined.gbk\n",
      "['ND6 CDS']\n",
      "2057 2537\n",
      "ATGGCAGAAGTATTTGTTTTGGTATTTTTGCTTTTATGTAGGAATTTTTCAATTTGGGTAGAGCACCCTCTTATTTTGGGGAGAGGGCTATTGGGGGTGAGAATTGTGAGATTGCCCCTAATAATTTCTGTTGGTAGACTTTTCGGGTTCAGGTTTTTTATTGTGGTGGTTGGAGGAGTTTTAATTGTTTTTGCTTACTCAGTATCTTTAATTTCTGTAACTAAGAGCGATCTGTTATTGAAATATTTTACTCGGGGAGTGGGTATAGTTGGGAAATTTTGAACGGTTTTTTTATATTTTAGGTTGGTTTTTTTATTATCGAGGTGGGTGGTGGGAAGGGGGGTATTTTGTTGAAGGGATGTCTTATATTTTAGGAAGGGGTGGGGCTTTGGGGTAGTTTTGCTTGGCTTTTTATTGTTAGCTATTATGGTGATGGCTATTGGGTTGGCGAGGAAATTCAAAGGGGCCTTAATTAAATAA\n",
      "locus_taged_extenta_combined.gbk\n",
      "['ND4L CDS']\n",
      "2555 2825\n",
      "GTGTTTGTTTTTTTTTCTTTAATTCATTTTTTAACTCATTTTAATTTTTTTATAAATTTATTATTAGTTTTTGAGTTGATGGGATTTTGCGTTTTTTTATTTTGTATAAATTTATCTTTTATTTCTTTTAGGGGAGCAGGTTTATATTTTAGAATTGTTATTTTAACTTTTAGGGTGTGCGAGACTGTTTTAGGAATGGCTCTATTAGTGAGGGCTGCGCGTTATACAGGTCGAATAAATGTGAAGTCTTTTACATTTTTAGGGTTTTAA\n",
      "locus_taged_extenta_combined.gbk\n",
      "['ND2 CDS']\n",
      "2976 3984\n",
      "ATAGCCATTATTATTGGGGGTTATATTATGGTTGGGGGGATTTTTGTTTGTGTTTATAGGGTTGGTTTGTTAGGGATATGGGTGGGAATGGAGTTAAGCTTTTTCGGGGCTATTGGGGTTTTAGGGGGCTCGAGGGTGGAGGAGGTTGAGGGTGTCCAAAAGTATTTTATTATCCAGGTTTTTAGGTCTATTTTTTTATTAATTGGATTTTTGTTGGTTGTTAATTTTTATATGGGGGGTTGGCTGGTTGAATTTATTTTAGTAGTGGGTTTAATAATGAAATTGGGGTTATTTCCTTTTTATTTTTGAATTCCTTCTGTTATAGGGGTTGTGTCTTGGGTGGGGTGTTTTGTTGTTTCTGTAGTTCAAAAATTTAATCCTATTTGAGTTTTGGTGAACTTAATATTGGAAAGGTATCTTCAACAGGTAATTGAGTTTTCGGCTTGTTTAACTGGTTTAGTCGGTTGCATTGGGGGGATTGGAGTTTTGCAGTTTCGAATTTTACTAGCTTTTTCTTCAATAGTTCATTTAGGGTACATAGTTTTTATAAGTTTAATTGGGTGAGGTGTATTTTTGTCTTACATAGGGGTATATTTTTGCCTCAGTTTTAGCTTAATAATAAGACTTTGAGTTTGTGGGGTTTATAGGTTTTTGGATTTGGTAAAATCTAATGTGGATGGGGAAATAAATTTGTTTATGATCTCCCTCTATATATTCTCTTTTGCTGGCGTTCCCCCCTTAAGGGGTTGCTTTATAAAAGGGGTTTTTTTAATATCTTGTTGAGAGCTTTTTCCTATTGGTTGTGTGATTTTATTTTTAAGTTCTGGGGTTAGTTTATTTTATTATAGTAGGTTTTTTGTGTATTTAAATTTATATTGAGGGAGGATATTGAAATGAGAATTTGTTAAGGGGAGACTGGCTGATTTAAAAGTAATTTTATTTTTTGTGAGGGTTTTATTAAATGTGGTTATGGGATTCGGGTTATTTTTGTTAATTGTGCTGGTTTAG\n",
      "locus_taged_extenta_combined.gbk\n",
      "['COX3 CDS']\n",
      "5274 6125\n",
      "ATGGCTCGAACGGGGTATAGTAAGGTTCAAATGAGGCCTTGACCGATTGTGATTTCTGCAAGGGTTTTTGGTGTAGCAATTGGTTTAATTTGGTATTTAACTGACGAGGTTTCTTTTAGAATGTATTCGTATTTAGGTGTAGTTTCTTTTTTTCTTGGGATGGCTCTAACAGGGTGATGAAGTGATATTATTGAGGAGAGGACTTTTTTGGGGAATCATACCTCACATGTAGTGTTAAATTTACGATGGGGATTTTATTTATTTATTTTATCTGAAGCTTGGTTTTTTTTTAGTCTTTTTTGAGCTTTTTTTACTGCTAGTGTGGGGGAAATTTCTATACAAGGTGTGGGAGCACAGTGGCCGCCGGTGGGGATGGATGGGATGTACCCTTGGGGTGTTCCTTTTTTAAATACTGTTGTGTTAGTTAGGTCTGGGTTATGGGCAACAACAGCTCATAAAGCTGTTAAGGTCCACAGCTCAATTGGTTTAGGTAACCCTGAAAGGGTTGCCCTAAGAGTAAAATGGTGAAAATTAGGGTTGCTATGTTTGGGTATTACTATTGTTTTAGGGGTGTATTTTACTTATTTACAAATCCACGAGTATTATTCTTCTTCTTTTTCTATATCGGATGGGGCTTATGGGAGAGTATTTTTTGTATTAACAGGTTTTCACGGGTTACATGTAATTATTGGGAGTGTTTATTTAACAGTATGTTGAATTCGGCTTTTAAAGAAGCATTTCACTTATAAACATCATCATTTTGGTTTGGATAGTTGCATATGGTATTGGCATTTTGTGGATGCAGTTTGAATTGTGGTGTTTCTTTCTGTTTATGTGTGGGGAAATCAATA\n"
     ]
    },
    {
     "ename": "TranslationError",
     "evalue": "Sequence length 851 is not a multiple of three",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTranslationError\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-107-de3620388c2c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     16\u001b[0m                 \u001b[0mnucl_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextract\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnucl_seq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 18\u001b[0;31m                 \u001b[0mprot_seq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextract\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranslate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36mtranslate\u001b[0;34m(self, table, stop_symbol, to_stop, cds, gap)\u001b[0m\n\u001b[1;32m   1276\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1277\u001b[0m         protein = _translate_str(\n\u001b[0;32m-> 1278\u001b[0;31m             \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcodon_table\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstop_symbol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_stop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgap\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1279\u001b[0m         )\n\u001b[1;32m   1280\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m_translate_str\u001b[0;34m(sequence, table, stop_symbol, to_stop, cds, pos_stop, gap)\u001b[0m\n\u001b[1;32m   2842\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;36m3\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2843\u001b[0m             raise CodonTable.TranslationError(\n\u001b[0;32m-> 2844\u001b[0;31m                 \u001b[0;34m\"Sequence length {0} is not a multiple of three\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2845\u001b[0m             )\n\u001b[1;32m   2846\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mstop_codons\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTranslationError\u001b[0m: Sequence length 851 is not a multiple of three"
     ]
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "mypath='./geneious_mitoch/'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-4:]=='.gbk']\n",
    "print(onlyfiles)\n",
    "recs={}\n",
    "\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        for feature in record.features:\n",
    "            if feature.type in ['CDS']:\n",
    "                print(file)\n",
    "                print(feature.qualifiers['product'])\n",
    "                print(feature.location.start,feature.location.end)\n",
    "                nucl_seq=feature.extract(record.seq)\n",
    "                print(nucl_seq)\n",
    "                prot_seq=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CLEAN mitochondrial genomes from geneious"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make fasta from geneious gbk and fix record name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic={'Pac_II_':'C.rectimargo',\n",
    " '11_MAR':'A.southwardae',\n",
    " 'extenta_combined':'P.extenta',\n",
    " 'A._phaseoliformi':'A.phaseoliformis_ref',\n",
    " 'R_magnifica':'C.magnifica',\n",
    " '8-3_Blake_Ridge':'X.pliocardia',\n",
    " 'A._mariana':'A.mariana_ref',\n",
    " 'Contig_54_11_MAR':'A.southwardae_c54',\n",
    " '13_Aleutian':'A.phaseoliformis',\n",
    " 'diagonalis':'A.diagonalis',\n",
    " 'Pac_I_':'C.pacifica',\n",
    " 'P._okutanii':'P.okutanii_ref',\n",
    " 'kilmeri':'P.soyae'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '18468..45' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '18507..45' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17868..193' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17878..155' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17884..206' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17934..106' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17942..114' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:366: BiopythonParserWarning: Attempting to fix invalid location '17966..279' as it looks like incorrect origin wrapping. Please fix input file, this could have unintended behavior.\n",
      "  BiopythonParserWarning,\n"
     ]
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "mypath='./geneious_mitoch/original_annotations/'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-3:]=='.gb']\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        record.id=dic[record.name]\n",
    "        SeqIO.write(record,'./geneious_mitoch/'+record.id+'.fasta','fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Redo the anotations using GeSeq - Annotation of Organellar Genomes and Rmag as reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parse multigbk file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dic={'ref_Pokuta':'P.okutanii_ref',\n",
    " 'Aphaseolif':'A.phaseoliformis',\n",
    " 'Pextenta-4':'P.extenta',\n",
    " 'ref_Cmagni':'C.magnifica',\n",
    " 'Asouthward':'A.southwardae',\n",
    " 'Psoyae-7%2':'P.soyae',\n",
    " 'Adiagonali':'A.diagonalis',\n",
    " 'Xpliocardi':'X.pliocardia',\n",
    " 'ref_Aphase':'A.phaseoliformis_ref', # reference AP014557.1\n",
    " 'ref_Amaria':'A.mariana_ref',\n",
    " 'Cpacifica-':'C.pacifica',\n",
    " 'Crectimarg':'C.rectimargo',\n",
    " '15_16_giga':'A.gigas'}\n",
    "\n",
    "dic2={'P.okutanii_ref': 'AP014555.1',\n",
    " 'A.phaseoliformis': '10_12_13_a',\n",
    " 'P.extenta': '4-3_5-3_6-',\n",
    " 'C.magnifica': 'C._Magnifi',\n",
    " 'A.southwardae': '11_MAR-806',\n",
    " 'A.diagonalis': '1_23_33_di',\n",
    " 'P.soyae': '4_5_14_kil',\n",
    " 'X.pliocardia': '9-3_17_BR-',\n",
    " 'A.phaseoliformis_ref': 'AP014557.1',\n",
    " 'A.mariana_ref': 'A._mariana',\n",
    " 'C.pacifica': '8_pac_mt_I',\n",
    " 'C.rectimargo': '9_pac_mt_I',\n",
    " 'A.gigas': '15_16_giga'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.southwardae mitochondrion\n",
      "P.okutanii_ref mitochondrion\n",
      "C.rectimargo mitochondrion\n",
      "P.soyae mitochondrion\n",
      "C.pacifica mitochondrion\n",
      "A.phaseoliformis mitochondrion\n",
      "C.magnifica mitochondrion\n",
      "P.extenta mitochondrion\n",
      "A.phaseoliformis_ref mitochondrion\n",
      "X.pliocardia mitochondrion\n",
      "A.mariana_ref mitochondrion\n",
      "A.diagonalis mitochondrion\n"
     ]
    }
   ],
   "source": [
    "ids=[]\n",
    "for record in SeqIO.parse('geneious_mitoch/Geneious_mitoch_GLOBAL_multi-GenBank.gbff','genbank'):\n",
    "    # fix header\n",
    "    record.name= dic2[dic[record.id]]\n",
    "    record.description=dic[record.id]+' mitochondrion'\n",
    "    print(record.description)\n",
    "    i=0\n",
    "    for feature in record.features:\n",
    "        if feature.type=='source':\n",
    "            continue\n",
    "        if feature.type=='gene':\n",
    "            i+=1\n",
    "            feature.qualifiers['locus_tag']=record.description[0]+record.description[2:5]+'_'+\"{:02d}\".format(i)\n",
    "            last_locus_tag=record.description[0]+record.description[2:5]+'_'+\"{:02d}\".format(i)\n",
    "        else:\n",
    "            feature.qualifiers['locus_tag']=last_locus_tag\n",
    "        if feature.type=='CDS':\n",
    "            try:\n",
    "                feature.qualifiers['translation']=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "            except:\n",
    "                pass\n",
    "    SeqIO.write(record, dic[record.id]+'.gbk', \"genbank\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## read gbk files and fix CDS\n",
    "This was done manually so the script is garbage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['A.gigas_shifted.gbk']\n",
      "15_16_giga mitochondrion\n"
     ]
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "\n",
    "mypath='./geneious_mitoch/'\n",
    "mypath='./'\n",
    "\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-4:]=='.gbk']\n",
    "print(onlyfiles)\n",
    "\n",
    "#########################\n",
    "def find_end_of_gene(seq,gene_start):\n",
    "    for i in range(gene_start, len(seq), 3):\n",
    "        if seq[i:i+3] in ['TAG','TAA']:\n",
    "            return(i+3)\n",
    "            break\n",
    "#####################           \n",
    "\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        features=[]\n",
    "        print (record.description)\n",
    "        for feature in record.features:\n",
    "            if feature.type=='CDS':\n",
    "                if feature.qualifiers['locus_tag'][0] == 'Rpha_51':\n",
    "                    print('>')\n",
    "                    print(feature.extract(record.seq))\n",
    "                \n",
    "                #-------\n",
    "                try:\n",
    "                    feature.qualifiers['translation']=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "                \n",
    "                except:\n",
    "#                     if feature.qualifiers['locus_tag'][0] in ['Apha_38','Cpac_16','Psoy_16','Asou_14','Crec_14']:\n",
    "#                         continue\n",
    "                    print(feature.qualifiers['gene'],feature.qualifiers['locus_tag'])\n",
    "                    nucl_seq=feature.extract(record.seq)\n",
    "                    gene_start=feature.location.start\n",
    "                    gene_end=feature.location.end\n",
    "                    print(gene_start,gene_end)\n",
    "                    print(record.seq[gene_start-3:gene_start+3],record.seq[gene_end-3:gene_end+3])\n",
    "                    print(find_end_of_gene(record.seq,gene_start))\n",
    "                    print(nucl_seq)\n",
    "                    prot_seq=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "                    print(prot_seq)\n",
    "                    \n",
    "    SeqIO.write(record, dic[record.id]+'.gbk', \"genbank\")    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "# record = SeqIO.read(mypath+file,'genbank')\n",
    "# record=record[-30:]+record[:-30]\n",
    "# SeqIO.write(record, mypath+file[:-4]+'_shifted.gbk', \"genbank\") "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 463,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['C.magnifica.gbk', 'P.okutanii_ref.gbk', 'A.diagonalis.gbk', 'A.phaseoliformis_ref.gbk', 'P.extenta.gbk', 'A.phaseoliformis.gbk', 'C.pacifica.gbk', 'A.phaseoliformis_rc.gbk']\n"
     ]
    }
   ],
   "source": [
    "### Get the reverse complement of Phaseoliphormis\n",
    "mypath='./'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-4:]=='.gbk']\n",
    "print(onlyfiles)\n",
    "\n",
    "#####################           \n",
    "\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        record=record.reverse_complement()\n",
    "        \n",
    "        SeqIO.write(record, file[:-4]+'_rc.gbk', \"genbank\")    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get codon alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['A.gigas.gbk', 'C.magnifica.gbk', 'P.okutanii_ref.gbk', 'A.diagonalis.gbk', 'A.phaseoliformis_ref.gbk', 'P.extenta.gbk', 'A.phaseoliformis.gbk', 'C.pacifica.gbk', 'P.soyae.gbk', 'A.mariana_ref.gbk', 'A.southwardae.gbk', 'X.pliocardia.gbk', 'C.rectimargo.gbk']\n"
     ]
    }
   ],
   "source": [
    "recs={}\n",
    "seqs={}\n",
    "# seqs={'ND2': {}, 'ND6': {}, 'ND4L': {}, 'COX3': {}, 'COX1': {}, 'COX2': {}, 'CYTB': {}, 'ATP8': {}, 'ND4': {}, 'ATP6': {}, 'ND3': {}, 'ND1': {}, 'ND5': {}}\n",
    "mypath='./geneious_mitoch/'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-4:]=='.gbk']\n",
    "print(onlyfiles)\n",
    "\n",
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "#         print(dic[record.id])\n",
    "        recs[dic[record.id]]={}\n",
    "        for feature in record.features:\n",
    "            if feature.type=='CDS':\n",
    "\n",
    "                recs[dic[record.id]].update({feature.qualifiers['gene'][0]:feature.qualifiers['locus_tag'][0]})\n",
    "\n",
    "df=pd.DataFrame.from_dict(recs,orient='index')\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "OrderedDict([('gene', ['COX1']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['cytochrome c oxidase subunit 1']), ('protein_id', ['AFG18166.1']), ('translation', ['MRWLFSTNHKDIGTLYILFAIWAGMVGTSLSLLIRAELGQPGALLGDDQLYNVIVTAHAFIMIFFLVMPMMMGGFGNWFVPMMLGAPDMAFPRMNNMSFWLLPPALTLLLGSAAVESGAGTGWTVYPPLSGNLAHAGASVDLTIFSLHLAGASSIMASINFITTAINMRSRGMRFERMPLFVWSIKITAVLLLLSLPVLAGAITMLLTGRNFNTSFFDPAGGGDPILYQHLFWFFGHPEVYILILPGFGMISHIIGHHAGKKEVFGTLGMIYAMMAIGILGFIVWAHHMFTVGMDVDTRAYFTAATMIIAVPTGIKVFSWLATIYGSRIKYETPMVWALGFIFLFTIGGLTGVVLANSSLDIMMHDTYYVVAHFHYVLSMGAVFALFAAFNYWFPLMTGLTLHPIWTKAHFFIMFIGVNLTFFPQHFLGLSGMPRRYSDYPDCFTKWNVLSSFGSMISFISVLYFMFIVWEAFSAQRAVIWSSYVGTDTEWDDLLPVDFHNLSETTALVRA'])])\n",
      "OrderedDict([('gene', ['COX2']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['cytochrome c oxidase subunit 2']), ('protein_id', ['AFG18167.1']), ('translation', ['MLFWGQLGLQDAVSPLMEQLVFFHDHAMLILVMVASLIGYAIVSLIMSSFTCRYILEGQQIETVWTIAPAIVLVFLALPSLRLLYLLDEVNDPVLTIKTVGHQWYWSYEYSDFNSLEIDSYMVSTDDLKEGEFRLLEVDYRMVVPFNATIRLLVTAADVIHSWTIPSLGVKVDGIPGRLNQISFFINQPGLYYGQCSEICGANHSFMPIVVEAVDISSFINWVATHAA'])])\n",
      "OrderedDict([('gene', ['ND4L']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 4L']), ('protein_id', ['AFG18168.1']), ('translation', ['MTIFPIMFLCGTICFWSAMITLMFQYKHLLSMLLSLEAMMLSLFIMLCSSSTISLETFFVLMLITLGACEASLGLSILVALIRTHGNDYVFNFNLMKC'])])\n",
      "OrderedDict([('gene', ['ND4']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 4']), ('protein_id', ['AFG18169.1']), ('translation', ['MNSLMFMFCKNKTGWWVTLWSLALSFMTSIPMMFSPLLSFSYFSNQVLIDGLSMPLIILTLWISAMMIMASYKIKILNNSPKYFLITVLLLNLSLILSFSMSNIIWFYIFFELSLIPTLMLILSWGYQPERLQAGMYMMMYTIMASLPLLLLLMYSAMEMGHLSLFMEKISLFKSSSWLFEIQWVFLLLAFLVKTPIFSLHLWLPKAHVEAPVAGSMVLAGVLLKLGGYGMLRMMQVFYFKFCTLSIFILVFSLMGGALTSMMCLRQVDIKSLVAYSSIGHMSLVVGGTMSTTIWGWQGSLTMMIAHGLCSSSLFALTNFSYEKMSSRSMFISKGSLMLVPAMAMWWFIACISNMAAPPTINLLSEILLFMSSLFMSFWLLIPLSMMTFLAAAYSLYLYSSTQHGGAPKFISPFTPYKTQNFMIMMLHLMPVNILIMKSDMICSWII'])])\n",
      "OrderedDict([('gene', ['ND5']), ('note', [\"TAA stop codon is completed by the addition of 3' A residues to the mRNA\"]), ('codon_start', ['1']), ('transl_except', ['(pos:5859..5860,aa:TERM)']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 5']), ('protein_id', ['AFG18170.1']), ('translation', ['MKFYMIKSSSLCSIILFLYSIMILPISIYFIVMNKWILLEWEFMSSSTSPMMFPLILDPLSLSFSNIVCLISACVLMFSTKYMDSEKFLSRFIWVVMLFVMSMNMLVFIPNMLSLLIGWDGLGIVSFALVIYYENQKSLAAGMLTALANRIGDVTLIIAITLSINQGFWSSLHFIYHPFSYVVMLGIMVAAMTKSAQIPFSSWLPAAMAAPTPVSALVHSSTLVTAGVFLLIRFYPFLSTFQDFLWISMLFSSMTLLMAGLGANFEYDLKKIIALSTLSQLGVMMLSISLKMPMLALFHLYTHAMFKALLFLCAGAIIYNNMHSQDIRTMGNLWKQMPLTISCMNTANLALCGAPFMAGFYSKDLILETMLSANSSMLIILVVFFATGLTATYSVRLSIMVLWKQNNHLPLSSYNDEQWQMTSPMIILTMGAIFGGLLLQLLIPNFNESLTLPLMMKMLTMIVTLLGVWTGITLWNLKLNSPMNPSINTYFLSSMWFLASTSTQKTIYGPLSLTSNLTKMVDQGWLELYGGQGTLKLASSISKTNQNYQNNLMPSFIMMMISSILIIIMMIMYLK'])])\n",
      "OrderedDict([('gene', ['ATP6']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['ATP synthase F0 subunit 6']), ('protein_id', ['AFG18171.1']), ('translation', ['MMVDIFSSFDDHNMVFLSFVSLLWAGTLWVMVLFSMDYWVIKSRWSILLDSPKQIIYSQVFRSFGKNLGGFVNLVVGLFSFLLLTNLLGLFPYVFSSTSHLAVTFSLAFPFWFSLILSGFMNNMYAFTAGLLPGGAPTALNPFLVLVESLSICMRPITLSVRLAANMGAGHVVLGLIGTYLSSSFFCYSLLITLSLIFVQVIYFMFEFGVSLIQGYVFSLLVTLYADEHTH'])])\n",
      "OrderedDict([('gene', ['ATP8']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['ATP synthase F0 subunit 8']), ('protein_id', ['AFG18172.1']), ('translation', ['MPQLSPLNWLFLFIMFWSVIFLMMCMVWWTSSKSFLVESGKDNMKVLSRVFNWK'])])\n",
      "OrderedDict([('gene', ['CYTB']), ('note', [\"TAA stop codon is completed by the addition of 3' A residues to the mRNA\"]), ('codon_start', ['1']), ('transl_except', ['(pos:complement(6915),aa:TERM)']), ('transl_table', ['5']), ('product', ['cytochrome b']), ('protein_id', ['AFG18173.1']), ('translation', ['MRKNHPVLKIVNGSLYDLPAPLNISIWWNFGSLLGMCLVIQILTGIFLAMHYTGNVELAFSSVAHISRDVNYGWFIRYLHANSGSWFFICLYLHVGRGIYYGSYMNIYTWNIGVIILLLTMMTAFLGYVLPWGQMSFWAATVITNLLSAIPYVGKMLVEWIWGGFAIDNATLSRFYAFHFLFPFVIAALSALHMLFLHEGGSNNPLGVNSDSEKIPFHIYYTVKDLVGFIVVLFFLVMLVLLSPHMLTDPENFIPANPLVTPVHIQPEWYFLFAYAILRSIPNKLGGVVALVMSVLILFIMPIVHLSMFRSMAFYPLSQILFWFLVGLMLILTWIGSCPVESPYDTIGQIFTFFYFCFYFVNPMCQWVWDKILC'])])\n",
      "OrderedDict([('gene', ['ND6']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 6']), ('protein_id', ['AFG18174.1']), ('translation', ['MMVLIIFSSLVSSIAFIMPSMMQPLSLGLNIMVLSVFSCLLVGFEMSSWYGYIMFLIYVGGLLVMFSYVAALTPNSFFSGVSNLLLFFLSLVVFFSTYMFNWFSSSTKFVSADMKMTFVNEGSSGSLMCSNYNFGVLVFLGVILFLALVAVVKICFSQKAPLRPFG'])])\n",
      "OrderedDict([('gene', ['ND1']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 1']), ('protein_id', ['AFG18175.1']), ('translation', ['MVLSFVSIIISYVLVLLGMAFFTLLERKGLGYFQIRKGPNKVGLAGLPQPLADAAKLLTKEMVSPTMINIFPYFLAPVFSLALALMLWQLYPSEFFMTSIFWGILLFLCISSLNVYGTLVAGWASNSKYSLLGSLRAVAQTISYEVSLSLIMLFSLCVSSSFSFFNINLSNEYMWVGMLMIPISAVWYVSTLAETNRAPFDFAEGESEIVSGFNIEYSAGGFALIFMAEYTNILFMSMMSALLFFGGSFLVCSSDIIMFFKVLSFSFIFVWARATLPRLRYDLLMNLTWKAFLPVSLGGLIFVSGMKFLIELI'])])\n",
      "OrderedDict([('gene', ['COX3']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['cytochrome c oxidase subunit 3']), ('protein_id', ['AFG18176.1']), ('translation', ['MTRSPFHLVEYSPWPFTGSMGALFMTVGSASWFHGHTSSLLVLGLFLTIITLIQWWRDVIREAAFQGHHTSNVSKGLRWGMILFITSEVLFFFAFFWAYFHSSLAPVTDLGSCWPPTGIYPLNPFQVPLLNTAVLLGSGVTVTWAHHSLMESNRNNSIMGLLATCTLGIYFTYLQAGEYMEASFTIADGIYGSTFFVATGFHGLHVLVGTSFLLICLGRAWSHQFSSGHHFGFEAAAWYWHFVDVVWLFLYVSIYWWGS'])])\n",
      "OrderedDict([('gene', ['ND3']), ('note', [\"TAA stop codon is completed by the addition of 3' A residues to the mRNA\"]), ('codon_start', ['1']), ('transl_except', ['(pos:14496..14497,aa:TERM)']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 3']), ('protein_id', ['AFG18177.1']), ('translation', ['MTVLLISSILACMIPVIVMALAWILSYRTVMDREKSSPFECGFDPLGSARLPFSLRFFLLAVIFLIFDVEIVLLFPIISILSMSMSMSALISAMVFLFILIIGLLHEWREGSLDWAK'])])\n",
      "OrderedDict([('gene', ['ND2']), ('codon_start', ['1']), ('transl_table', ['5']), ('product', ['NADH dehydrogenase subunit 2']), ('protein_id', ['AFG18178.1']), ('translation', ['MNVYIPYNFMFLLLTMLGTIFSISSSHWMGVWAGLEINLLGFIPLLVYGGSTLEIESGVKYFIIQSMGSGILLMGSLLISMAALSWTNLGPLLSTEISSMSFIQKDMSFYGALVMVSLLIKIGLTPFHFWLPSVMSGLSWFSVLILASWQKIAPLILILMVVSKYSLFLLMCCAGSSMVGGVGGLNQTQLRGLMAYSSINHLGWMTAGSISSTASMVIYFIIYFLISMFLFYYLWMKEKSQARQMSTSSFQSMNEKLIFSTLILSLGGLPPLLGFVGKWLIMTELFIIEPMILMFLILGSLISLFYYLLLFFSYFFSNNSKISELSTKNLMPLSTSSYFIVMSVSMNFSGPIMILWLPSLIL'])])\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'COX1': {'faa': SeqRecord(seq=Seq('MRWLFSTNHKDIGTLYILFAIWAGMVGTSLSLLIRAELGQPGALLGDDQLYNVI...VRA', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18166.1', name='Z.Svelum|AFG18166.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGCGTTGGTTATTTTCTACCAATCACAAAGATATTGGAACTTTATATATTTTA...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18166.1', name='Z.Svelum|AFG18166.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'COX2': {'faa': SeqRecord(seq=Seq('MLFWGQLGLQDAVSPLMEQLVFFHDHAMLILVMVASLIGYAIVSLIMSSFTCRY...HAA', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18167.1', name='Z.Svelum|AFG18167.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGCTCTTTTGAGGACAATTAGGACTACAAGATGCTGTATCTCCATTAATAGAA...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18167.1', name='Z.Svelum|AFG18167.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND4L': {'faa': SeqRecord(seq=Seq('MTIFPIMFLCGTICFWSAMITLMFQYKHLLSMLLSLEAMMLSLFIMLCSSSTIS...MKC', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18168.1', name='Z.Svelum|AFG18168.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGACAATTTTTCCTATTATATTTTTATGCGGCACTATTTGTTTTTGATCAGCT...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18168.1', name='Z.Svelum|AFG18168.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND4': {'faa': SeqRecord(seq=Seq('MNSLMFMFCKNKTGWWVTLWSLALSFMTSIPMMFSPLLSFSYFSNQVLIDGLSM...WII', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18169.1', name='Z.Svelum|AFG18169.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATAAACTCTCTAATGTTTATATTTTGTAAGAATAAAACAGGCTGATGAGTAACT...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18169.1', name='Z.Svelum|AFG18169.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND5': {'faa': SeqRecord(seq=Seq('MKFYMIKSSSLCSIILFLYSIMILPISIYFIVMNKWILLEWEFMSSSTSPMMFP...YLK', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18170.1', name='Z.Svelum|AFG18170.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGAAATTTTATATAATTAAATCATCATCTTTATGCTCTATTATTTTATTTCTA...ATA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18170.1', name='Z.Svelum|AFG18170.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ATP6': {'faa': SeqRecord(seq=Seq('MMVDIFSSFDDHNMVFLSFVSLLWAGTLWVMVLFSMDYWVIKSRWSILLDSPKQ...HTH', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18171.1', name='Z.Svelum|AFG18171.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGATGGTTGATATTTTTTCTTCTTTTGATGATCATAATATGGTATTTTTATCT...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18171.1', name='Z.Svelum|AFG18171.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ATP8': {'faa': SeqRecord(seq=Seq('MPQLSPLNWLFLFIMFWSVIFLMMCMVWWTSSKSFLVESGKDNMKVLSRVFNWK', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18172.1', name='Z.Svelum|AFG18172.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGCCTCAACTTAGTCCGTTAAATTGGTTGTTTTTGTTTATTATATTTTGATCT...TAG', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18172.1', name='Z.Svelum|AFG18172.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'CYTB': {'faa': SeqRecord(seq=Seq('MRKNHPVLKIVNGSLYDLPAPLNISIWWNFGSLLGMCLVIQILTGIFLAMHYTG...ILC', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18173.1', name='Z.Svelum|AFG18173.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATTCGTAAAAATCATCCGGTTTTGAAAATTGTTAATGGGTCTCTTTATGACTTA...GTT', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18173.1', name='Z.Svelum|AFG18173.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND6': {'faa': SeqRecord(seq=Seq('MMVLIIFSSLVSSIAFIMPSMMQPLSLGLNIMVLSVFSCLLVGFEMSSWYGYIM...PFG', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18174.1', name='Z.Svelum|AFG18174.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATAATGGTTTTGATTATTTTTAGAAGATTAGTGAGAAGTATTGCTTTTATTATG...TAG', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18174.1', name='Z.Svelum|AFG18174.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND1': {'faa': SeqRecord(seq=Seq('MVLSFVSIIISYVLVLLGMAFFTLLERKGLGYFQIRKGPNKVGLAGLPQPLADA...ELI', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18175.1', name='Z.Svelum|AFG18175.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGGTTTTGTCTTTTGTTTCTATTATTATTTCTTATGTTTTAGTTTTATTAGGT...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18175.1', name='Z.Svelum|AFG18175.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'COX3': {'faa': SeqRecord(seq=Seq('MTRSPFHLVEYSPWPFTGSMGALFMTVGSASWFHGHTSSLLVLGLFLTIITLIQ...WGS', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18176.1', name='Z.Svelum|AFG18176.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGACTCGAAGCCCATTTCATTTAGTTGAGTATAGCCCATGGCCTTTCACAGGC...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18176.1', name='Z.Svelum|AFG18176.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND3': {'faa': SeqRecord(seq=Seq('MTVLLISSILACMIPVIVMALAWILSYRTVMDREKSSPFECGFDPLGSARLPFS...WAK', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18177.1', name='Z.Svelum|AFG18177.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGACAGTACTATTAATCTCAAGAATTCTTGCCTGTATAATTCCTGTAATTGTA...ATA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18177.1', name='Z.Svelum|AFG18177.1', description='<unknown description>', dbxrefs=[])},\n",
       " 'ND2': {'faa': SeqRecord(seq=Seq('MNVYIPYNFMFLLLTMLGTIFSISSSHWMGVWAGLEINLLGFIPLLVYGGSTLE...LIL', Gapped(ExtendedIUPACProtein(), '-')), id='Z.Svelum|AFG18178.1', name='Z.Svelum|AFG18178.1', description='<unknown description>', dbxrefs=[]),\n",
       "  'fna': SeqRecord(seq=Seq('ATGAATGTTTATATCCCCTATAATTTTATATTTCTATTGCTAACAATGCTGGGT...TAA', IUPACAmbiguousDNA()), id='Z.Svelum|AFG18178.1', name='Z.Svelum|AFG18178.1', description='<unknown description>', dbxrefs=[])}}"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record= SeqIO.read(mypath+'Z.Svelum.gb', \"genbank\")\n",
    "from Bio.Alphabet import Gapped,IUPAC\n",
    "from Bio.Alphabet.IUPAC import ExtendedIUPACProtein\n",
    "\n",
    "Svel={}\n",
    "for feature in record.features:\n",
    "    if feature.type=='CDS':\n",
    "        gene=feature.qualifiers['gene'][0]\n",
    "        header='Z.Svelum|'+feature.qualifiers['protein_id'][0]\n",
    "        print (feature.qualifiers)\n",
    "        vel_aa=SeqRecord(Seq(feature.qualifiers['translation'][0],Gapped(ExtendedIUPACProtein(), '-')),id=header,name=header)\n",
    "        vel_nucl=SeqRecord(feature.location.extract(record).seq,id=header,name=header)\n",
    "        Svel[gene]={'faa':vel_aa,'fna':vel_nucl}\n",
    "\n",
    "                   \n",
    "Svel    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COX1\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1803 columns (601 codons)\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATATGTTTATAAATTTGTATTT...GTT A.diagonalis|Adia_22\n",
      "---------------------------------ATATCTGTTTATAAATTTGTATTT...GTT A.gigas|Agig_01\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.mariana_ref|Amar_19\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.phaseoliformis_ref|Apha_55\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.phaseoliformis|Apha_35\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.southwardae|Asou_18\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGGGTCTTATGTCTGTTTATAAGTTTGTGTTT...GTT C.magnifica|Cmag_01\n",
      "ATGTTTGTGGTTGAGCGTGTGATTAAAAGTTTTATACTTATTTACAAGTTTGTATTT...GTT C.pacifica|Cpac_22\n",
      "ATATTTGTGCCTGAGCGTGTGGCTAAAATTTTTATATTTGTTTATAAGTTTATGTTT...GTT C.rectimargo|Crec_21\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.extenta|Pext_21\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.okutanii_ref|Poku_20\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.soyae|Psoy_21\n",
      "ATATTTGTGATTGAGCGTTTATCTAAGAGTATTATACTAGTTTATAAATTTGTGTTT...GTT X.pliocardia|Xpli_20\n",
      "---------------------------------------------------------...--- Z.Svelum|AFG18166.1\n",
      "14\n",
      "COX2\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1542 columns (514 codons)\n",
      "---------ATGGGATCTTGGAATAAGTTATGGTTTTTTGATATAAACACTGTTGTT...--- A.diagonalis|Adia_24\n",
      "---------ATGGGATCTTGGAATAAGTTATGGTTTTTTGATATAAACACTGTTGTT...--- A.gigas|Agig_03\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.mariana_ref|Amar_21\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.phaseoliformis_ref|Apha_02\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.phaseoliformis|Apha_33\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.southwardae|Asou_20\n",
      "---------ATGGGATCTTGAAACAAATTATGGTTTTTTGATATAAATACTGTGGTT...--- C.magnifica|Cmag_03\n",
      "TTGTTTTGAATGGGATCTTGAGAGAAAATGTGATTTTATGATATAAATACTGTTGTC...GTC C.pacifica|Cpac_24\n",
      "---------ATGGGATCCTGAGAGAAATTTTGGTTTTATGATATAAATACTGTTGTC...GAC C.rectimargo|Crec_24\n",
      "---------ATGGGGTCTTGAAGTAAGTTATGGTTTTTTGATATAAATACTGTGGTT...--- P.extenta|Pext_23\n",
      "---------ATGGGGTCTTGAAATAAGTTATGGTTTTTTGATATAAATACTGTTGTT...--- P.okutanii_ref|Poku_22\n",
      "---------ATGGGGTCTTGGAATAAGTTATGGTTTTTTGATATAAATACTGTTGTT...--- P.soyae|Psoy_24\n",
      "---------ATGGGATCTTGGCATAAATTGTGGTTTTATGATATAAATACTGTTGTT...--- X.pliocardia|Xpli_22\n",
      "---------ATGCTCTTTTGAGGACAATTAGGACTACAAGATGCTGTATCTCCATTA...--- Z.Svelum|AFG18167.1\n",
      "14\n",
      "CYTB\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1131 columns (377 codons)\n",
      "ATGGGAAACCATGATGGGAAAGGAATTCTTTATAGGATAAAGAGAGTTCTTTATGAT...GGG A.diagonalis|Adia_26\n",
      "ATGGGAAACCATGATGGGAAAGGAATTCTTTATAGGATAAAGAGAGTTCTTTATGAT...GAG A.gigas|Agig_05\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAATTTTGTATGAC...AAA A.mariana_ref|Amar_24\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.phaseoliformis_ref|Apha_04\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.phaseoliformis|Apha_30\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.southwardae|Asou_22\n",
      "ATGGGAAACCACGATGGGAAAGGGATTCTTTATAGAATAAAAAGAGCTTTATATGAT...AAA C.magnifica|Cmag_05\n",
      "ATGGGGAACCACGATGGGAAAGGTATTCTTTACAGGATAAAGAGGGCTTTATATGAT...AAA C.pacifica|Cpac_27\n",
      "ATGGGAAACCACGATGGGAAAGGGATTCTTTACAGGATAAAGGGGGCTTTATATGAT...AAA C.rectimargo|Crec_27\n",
      "ATGGGGAACCACGATGGAAAAGGAATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.extenta|Pext_25\n",
      "ATGGGAAACCACGATGGAAAAGGAATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.okutanii_ref|Poku_26\n",
      "ATGGGAAACCATGATGGAAAAGGTATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.soyae|Psoy_27\n",
      "ATGGGAAATCACGATAGGAAAGGAATTCTTTATAGAATGAAAAGGGCTTTGTATGAT...AAG X.pliocardia|Xpli_25\n",
      "------ATTCGTAAAAATCATCCGGTTTTGAAAATTGTTAATGGGTCTCTTTATGAC...TGT Z.Svelum|AFG18173.1\n",
      "14\n",
      "ATP8\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 162 columns (54 codons)\n",
      "ATGCCTCAGTTTTCTCCTCTTTTTGTTGATTTTATTTTTTTGTTCTATTGGGCGGTG...--- A.diagonalis|Adia_28\n",
      "ATGCCTCAGTTTTCTCCTCTTTTTGTTGATTTTATTTTTTTGTTCTATTGGGCGGTG...--- A.gigas|Agig_07\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTCGATTTTATTTTTTTATTTTATTGGGCGGTG...--- A.mariana_ref|Amar_28\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTA...--- A.phaseoliformis_ref|Apha_09\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTA...--- A.phaseoliformis|Apha_26\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTG...--- A.southwardae|Asou_27\n",
      "ATGCCTCAGTTTTCACCTCTTTTTGTAGATTTTATTTTTTTATTTTATTGGGCGGTG...--- C.magnifica|Cmag_07\n",
      "ATGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- C.pacifica|Cpac_29\n",
      "ATGCCTCAGTTTTCACCTCTATTTGTGGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- C.rectimargo|Crec_30\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTATGTTTTATTGGGTGGTG...--- P.extenta|Pext_28\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...AAG P.okutanii_ref|Poku_29\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- P.soyae|Psoy_30\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGAATAGTG...--- X.pliocardia|Xpli_29\n",
      "ATGCCTCAACTTAGTCCGTTAAATTGGTTGTTTTTGTTTATTATATTTTGATCTGTT...AAA Z.Svelum|AFG18172.1\n",
      "14\n",
      "ND4\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1389 columns (463 codons)\n",
      "ATGGGG------------ACAGTTATCTCTTGCTCTGCAGCTTATTGTATTCCGAAT...GTG A.diagonalis|Adia_29\n",
      "ATGGGG------------ACGGTGATYTCTTGCTTTGCAGCTTACTGTATCCCGAAT...GTA A.gigas|Agig_08\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTATTTATTGTATTCCAAAT...ATA A.mariana_ref|Amar_29\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTGTTTATTGTATTCCAAAT...ATA A.phaseoliformis_ref|Apha_10\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTGTTTATTGTATTCCAAAT...ATA A.phaseoliformis|Apha_25\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTATTTATTGTATTCCAAAT...ATA A.southwardae|Asou_28\n",
      "ATGGGA------------ACTGTAATTTCTTGTTTTGCTGCTTATTGTATTCCGAAT...ATA C.magnifica|Cmag_08\n",
      "ATAGGA------------ACTGTAATTTCTTGTTTTGTTAGTTATTGTGTTCCAAAT...ATA C.pacifica|Cpac_30\n",
      "ATAGGA------------GCTGTGATTTCTTGTTTTGTTAGTTATTGTGTCCCGAAC...GTG C.rectimargo|Crec_31\n",
      "ATGGGG------------ACGGTGATTTCTTGTTCTGCTGCTTATTGTATCCCTAAT...ATG P.extenta|Pext_29\n",
      "ATGGGT------------ACAGTGATTTCTTGTTCTGCAGCTTATTGTATCCCTAAT...ATG P.okutanii_ref|Poku_30\n",
      "ATGGGG------------ACAGTGATCTCTTGTTCTGCAGCTTACTGTATCCCTAAT...ATG P.soyae|Psoy_31\n",
      "ATGGGAGGAATTATAAGAACTGTAATTTCTTGTTTTGTAAGTTATTGTGTCCCGAAT...ATA X.pliocardia|Xpli_30\n",
      "ATAAAC------------TCTCTAATGTTTATA---------TTTTGTAAGAATAAA...ATT Z.Svelum|AFG18169.1\n",
      "14\n",
      "ATP6\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 720 columns (240 codons)\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGGGGATTTAGGGTA...TTG A.diagonalis|Adia_36\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATGGGTGGATTTAGAGTT...TTA A.gigas|Agig_16\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG A.mariana_ref|Amar_35\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG A.phaseoliformis_ref|Apha_16\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG A.phaseoliformis|Apha_19\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG A.southwardae|Asou_34\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGCTTTAGGGTG...TTG C.magnifica|Cmag_13\n",
      "ATGGTTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGGGGATTTAGGGTA...TTA C.pacifica|Cpac_35\n",
      "ATGGTTAGGGATATTTTCACT---TTTATGGATTGTTTTATAGGGGGATTTAGGGTG...TTA C.rectimargo|Crec_38\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATGGGCGGGTTTAGAGTG...TTA P.extenta|Pext_36\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG P.okutanii_ref|Poku_37\n",
      "ATGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAGTG...TTG P.soyae|Psoy_38\n",
      "GTGACTAGGGATATTTTTACT---TTTATGGATTGTTTTATAGGTGGATTTAGAATT...TTA X.pliocardia|Xpli_36\n",
      "ATGATGGTTGATATTTTTTCTTCTTTTGATGATCATAATATGGTATTTTTATCTTTT...--- Z.Svelum|AFG18171.1\n",
      "14\n",
      "ND3\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 363 columns (121 codons)\n",
      "ATTGGTTTATCTATGTTGGGA---------------------------GTTGTCGTT...GGC A.diagonalis|Adia_38\n",
      "ATTGGTTTAGTTATTTTGGGA---------------------------GTTGTAGTT...--- A.gigas|Agig_17\n",
      "ATTGCTTTAGTTATTTTAGGG---------------------------GTAATAGTT...GGT A.mariana_ref|Amar_36\n",
      "ATTGCTTTAGTAATTTTAGGG---------------------------GTAATAGTT...GGT A.phaseoliformis_ref|Apha_17\n",
      "ATTGCTTTAGTAATTTTAGGG---------------------------GTAATAGTT...GGT A.phaseoliformis|Apha_18\n",
      "---------------------------------------------------ATAGTT...GGC A.southwardae|Asou_35\n",
      "---------------------------------------------------ATGGTT...GGC C.magnifica|Cmag_15\n",
      "---------------------------------------------------GTGGTT...GGT C.pacifica|Cpac_36\n",
      "---------------------------------------------------GTGGTT...GGC C.rectimargo|Crec_39\n",
      "ATTGGCTTAGTTATTCTGGGA---------------------------GTTATGATT...GGT P.extenta|Pext_37\n",
      "ATTGGCTTAGTTATTTTGGGA---------------------------GTTGTAGTT...GGT P.okutanii_ref|Poku_38\n",
      "ATTGGCTTAGTTATTTTGGGA---------------------------GTTGTAGTT...GGT P.soyae|Psoy_39\n",
      "ATTGCCCTAGTAATTTTGGGA---------------------------GTAATAGTT...GGT X.pliocardia|Xpli_37\n",
      "ATGACAGTACTATTAATCTCAAGAATTCTTGCCTGTATAATTCCTGTAATTGTAATA...--- Z.Svelum|AFG18177.1\n",
      "14\n",
      "ND1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 945 columns (315 codons)\n",
      "------------GTGATTAGGGGGATAATTGTGGTTGTTGTTATAATAATATCTGTA...TTA A.diagonalis|Adia_45\n",
      "------------GTGATCAGGGGGTTAATTGTAGTTATTGTTATAATAATATCTGTA...CTA A.gigas|Agig_24\n",
      "------------GTGATTAGGGGGATAATTGTAGTTATTATTATAATAATGGCTGTA...TTA A.mariana_ref|Amar_43\n",
      "------------GTGATTAGGGGGATAATTGTAGTTATTATTATAATAATGTCTGTA...TTA A.phaseoliformis_ref|Apha_24\n",
      "------------GTGATTAGGGGGATAATTGTAGTTATTATTATAATAATGTCTGTA...TTA A.phaseoliformis|Apha_11\n",
      "------------GTGGTTAGGGGAATAATTGTAGTTATTATTATAATAATGTCTGTA...TTA A.southwardae|Asou_43\n",
      "------------ATGATTAGAGGGATAATTGTAGTCATTGTTATAATAGTGTCTGTA...CTA C.magnifica|Cmag_22\n",
      "------------GTGATTAGGGGTATAATTGTAGTTATTGTTATAATAGTGTCTGTA...TTG C.pacifica|Cpac_42\n",
      "------------GTGATTAGGGGTATAATTGTAGTTATTGTTATAATAGTGTCTGTA...TTG C.rectimargo|Crec_47\n",
      "------------GTGATTAGGGGGATAATTGTAGTTGTTATTATAATAATATCTGTA...TTA P.extenta|Pext_44\n",
      "------------GTGGTTAGGGGGATAATTGTAGTTATTATTATAATAATATCTGTA...TTA P.okutanii_ref|Poku_45\n",
      "------------GTGGTTAGGGGGATAATTGTAGTTATTATTATAATAATATCTGTA...TTA P.soyae|Psoy_46\n",
      "------------ATGATTAGGGGGATGATTGTGGTTATTGTAATAATAGTAGCTGTA...TTA X.pliocardia|Xpli_44\n",
      "ATGGTTTTGTCTTTTGTTTCTATTATTATTTCTTATGTTTTAGTTTTATTAGGTATA...ATT Z.Svelum|AFG18175.1\n",
      "14\n",
      "ND5\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1749 columns (583 codons)\n",
      "------GTGAAAGAATATAAACTGGCGGGGACCATGAAAATTGTATCAATA------...--- A.diagonalis|Adia_48\n",
      "---------------------------------ATAAAAATTGTATTAATA------...--- A.gigas|Agig_27\n",
      "---------------------------------ATAAAAATTTCTTTAATA------...--- A.mariana_ref|Amar_48\n",
      "---------------------------------ATAAAAATTTCTTTAATA------...--- A.phaseoliformis_ref|Apha_28\n",
      "---------------------------------ATAAAAATTTCTTTAATA------...--- A.phaseoliformis|Apha_05\n",
      "---------------------------------ATAAAAATTTCTTTAATA------...--- A.southwardae|Asou_48\n",
      "---------------------------------ATAAAAATTGCGTTAATA------...--- C.magnifica|Cmag_26\n",
      "---------------------------------ATAAAGATTGTTTTAATG------...--- C.pacifica|Cpac_46\n",
      "---------------------------------ATAAAAATTGTTTTAATG------...--- C.rectimargo|Crec_52\n",
      "---------------------------------ATAAAAATTGTGTTAATA------...--- P.extenta|Pext_48\n",
      "---------------------------------ATAAAAATTGTGTTAATA------...--- P.okutanii_ref|Poku_50\n",
      "---------------------------------ATAAAAATTGTGTTAATA------...--- P.soyae|Psoy_51\n",
      "---------------------------------ATAAAAATTGCTTTAATA------...--- X.pliocardia|Xpli_49\n",
      "ATGAAATTTTATATAATTAAATCATCATCTTTATGCTCTATTATTTTATTTCTATAC...AAA Z.Svelum|AFG18170.1\n",
      "14\n",
      "ND6\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 531 columns (177 codons)\n",
      "ATGGCTGAAATATTTGTTTTGGTGTTTTTGCTTTTATGTAGAAATTTTTCGATTTGG...--- A.diagonalis|Adia_05\n",
      "ATGGCTGAAATATTTGTTTTGGTGTTTTTACTTTTATGTAGAAATTTTTCAATTTGG...--- A.gigas|Agig_29\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCAATTTGA...--- A.mariana_ref|Amar_01\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCAATTTGA...--- A.phaseoliformis_ref|Apha_37\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCAATTTGA...--- A.phaseoliformis|Apha_53\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCAATTTGA...--- A.southwardae|Asou_57\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCGATTTGG...--- C.magnifica|Cmag_32\n",
      "ATGGCTGAAATATTTGTTTTGGTATTTTTGTTTTTATGTAGAAATTTTTCAATTTGA...--- C.pacifica|Cpac_03\n",
      "ATGGTTGAAATATTTGTTTTGGTATTTTTGTTTTTATGTAGAAATTTTTCAATTTGA...--- C.rectimargo|Crec_01\n",
      "ATGGCAGAAGTATTTGTTTTGGTATTTTTGCTTTTATGTAGGAATTTTTCAATTTGG...--- P.extenta|Pext_02\n",
      "ATGGCTGAAGTATTTGTTTTGGTATTTTTGCTTTTATGTAGAAATTTTTCAATTTGG...--- P.okutanii_ref|Poku_01\n",
      "ATGGCTGAAGTATTTGTTTTGGTATTTTTGCTTTTATGTAGGAATTTTTCAATTTGG...--- P.soyae|Psoy_02\n",
      "ATGGCTGAAGTATTTGTTTTAACATTTTTACTTGTTTGTAGAAATTTTTCTATTTGA...--- X.pliocardia|Xpli_03\n",
      "------ATAATGGTTTTGATTATTTTTAGAAGATTAGTGAGAAGTATTGCTTTTATT...GGT Z.Svelum|AFG18174.1\n",
      "14\n",
      "ND4L\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 303 columns (101 codons)\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...CTT A.diagonalis|Adia_06\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...CTT A.gigas|Agig_30\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT A.mariana_ref|Amar_02\n",
      "---------------------------------GTGTTCGTTTTTTTTTCT---TTA...TTT A.phaseoliformis_ref|Apha_38\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT A.phaseoliformis|Apha_52\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT A.southwardae|Asou_01\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT C.magnifica|Cmag_33\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT C.pacifica|Cpac_04\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT C.rectimargo|Crec_02\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT P.extenta|Pext_03\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT P.okutanii_ref|Poku_02\n",
      "---------------------------------GTGTTTGTTTTTTTTTGT---TTA...TTT P.soyae|Psoy_03\n",
      "---------------------------------GTGTTTGTTTTTTTTTCT---TTA...TTT X.pliocardia|Xpli_04\n",
      "ATGACAATTTTTCCTATTATATTTTTATGCGGCACTATTTGTTTTTGATCAGCTATA...TGT Z.Svelum|AFG18168.1\n",
      "14\n",
      "ND2\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 1116 columns (372 codons)\n",
      "------------------------ATGTCCACTATTGTTGGGGGTTATATTATGGTG...GTT A.diagonalis|Adia_08\n",
      "ATGTTAATTGGGTGATTTGTGCCTATGGCGACTATTATCGGGGGTTATGTTATGATC...GTT A.gigas|Agig_32\n",
      "ATGTTAATTGGGTGATTTGTTCCTATGGCTGTTATTGTTAGGGGTTATATTATGGTT...ATT A.mariana_ref|Amar_06\n",
      "ATGTTAATTGGGTGATTTGTTCCTATGGCTGTTATTGTTAGGGGTTACATTATGGTT...ATT A.phaseoliformis_ref|Apha_42\n",
      "ATGTTAATTGGGTGATTTGTTCCTATGGCTGTTATTGTTAGGGGTTATATTATGGTT...ATT A.phaseoliformis|Apha_48\n",
      "ATGTTAATTGGGTGATTTGTACCTATGGCTGTTATTGTTAGGGGTTATATTATGGTT...ATT A.southwardae|Asou_05\n",
      "ATGTTAATTGGGTGATTTGTACCTATGGCGGTCATTGTTGGAGGTTATCTTTTGGTT...ATT C.magnifica|Cmag_35\n",
      "ATGATAATTGGATGGTTTGTGCCTATAGCTGTTGTTATTAGGGGGTACATTATGGTT...ATT C.pacifica|Cpac_07\n",
      "ATGATAATTGGATGGTTTGTACCTATAGCTGTTATTGTTAGGGGGTCTATTATAGTT...ATT C.rectimargo|Crec_04\n",
      "ATGTTAATTGGGTGATTTGTGCCTATAGCCATTATTATTGGGGGTTATATTATGGTT...GTT P.extenta|Pext_05\n",
      "ATGCTAATTGGGTGATTTGTGCCTATAGCCACCATTATTGGGGGTTATATTATGGTC...GTT P.okutanii_ref|Poku_06\n",
      "------------------------ATAGCCACCATTATTGGGGGTTATATCATGGTT...GTT P.soyae|Psoy_07\n",
      "ATGTTAATTGGGTGGTTTATCCCTATGGCTGTTATTATTAGCGGTTATATTATGGTT...ATT X.pliocardia|Xpli_06\n",
      "------ATGAATGTTTATATCCCCTATAATTTTATATTTCTATTGCTAACAATGCTG...TTA Z.Svelum|AFG18178.1\n",
      "14\n",
      "COX3\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 951 columns (317 codons)\n",
      "---------------------------------------------------------...CAA A.diagonalis|Adia_17\n",
      "------------------------------------------------------ATT...CAA A.gigas|Agig_43\n",
      "---------------------------------------------------------...CAA A.mariana_ref|Amar_15\n",
      "---ATAGGTATTTACCTCCTTTTTATTAAAGAGAAGCGATGGTTACATAGTGAGATT...CAA A.phaseoliformis_ref|Apha_51\n",
      "------------------------------------------------------ATT...CAA A.phaseoliformis|Apha_38\n",
      "------------------------------------------------------ATT...CAA A.southwardae|Asou_14\n",
      "ATAGGTATTTACCTCCTTTTTATTGAAAAGGAAAAACGGTGGTTACATAGTGAGATT...CAA C.magnifica|Cmag_45\n",
      "------------------------------------------------------ATT...CAA C.pacifica|Cpac_16\n",
      "---------------------------ATTGAGAAACGGTGGTTACATAGTGAGATT...CAG C.rectimargo|Crec_14\n",
      "---------------------------------------------------------...CAA P.extenta|Pext_16\n",
      "---------------------------------------------------------...CAA P.okutanii_ref|Poku_15\n",
      "---------------------------------------------------------...CAA P.soyae|Psoy_16\n",
      "---ATAGGTATTTACCTCCTTTTTAAAAAGGAGAAACGATGGTTACATAGTGAGATT...CAG X.pliocardia|Xpli_15\n",
      "---------------------------------------------------------...--- Z.Svelum|AFG18176.1\n",
      "14\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 14 rows and 12705 columns (4235 codons)\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATATGTTTATAAATTTGTATTT...CAA Adiagonalis\n",
      "---------------------------------ATATCTGTTTATAAATTTGTATTT...CAA Agigas\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...CAA Amariana_ref\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...CAA Aphaseoliformis_ref\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...CAA Aphaseoliformis\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...CAA Asouthwardae\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGGGTCTTATGTCTGTTTATAAGTTTGTGTTT...CAA Cmagnifica\n",
      "ATGTTTGTGGTTGAGCGTGTGATTAAAAGTTTTATACTTATTTACAAGTTTGTATTT...CAA Cpacifica\n",
      "ATATTTGTGCCTGAGCGTGTGGCTAAAATTTTTATATTTGTTTATAAGTTTATGTTT...CAG Crectimargo\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...CAA Pextenta\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...CAA Pokutanii_ref\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...CAA Psoyae\n",
      "ATATTTGTGATTGAGCGTTTATCTAAGAGTATTATACTAGTTTATAAATTTGTGTTT...CAG Xpliocardia\n",
      "---------------------------------------------------------...--- ZSvelum\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "14"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import Alphabet\n",
    "from Bio.Alphabet import IUPAC,Gapped\n",
    "from Bio import AlignIO\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Align.Applications import MuscleCommandline\n",
    "from Bio.codonalign import build\n",
    "from Bio.Data.CodonTable import generic_by_id,unambiguous_dna_by_id\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "i=0\n",
    "table5=unambiguous_dna_by_id[5]\n",
    "table5.nucleotide_alphabet=IUPAC.ambiguous_dna\n",
    "table5.protein_alphabet=IUPAC.extended_protein\n",
    "table5.forward_table['AAN'] = \"K\"\n",
    "table5.forward_table['GAY'] = \"D\"\n",
    "table5.forward_table['YTA'] = \"L\"\n",
    "table5.forward_table['ATY'] = \"I\"\n",
    "table5.forward_table['TCY'] = \"S\"\n",
    "table5.forward_table['GTY'] = \"V\"\n",
    "table5.forward_table['GGY'] = \"G\"\n",
    "table5.forward_table['CCY'] = \"P\"\n",
    "table5.forward_table['GGR'] = \"G\"\n",
    "for gene in df.columns.values:\n",
    "    i+=1\n",
    "    j=0\n",
    "    print(gene)\n",
    "    fna=[]\n",
    "    faa=[]\n",
    "    for index in df.index.values:\n",
    "        j+=1\n",
    "        locus_tag=df[gene][index]\n",
    "        record = SeqIO.read(mypath+index+'.gbk', \"genbank\")\n",
    "        for feature in record.features:\n",
    "            if feature.type != 'CDS':\n",
    "                continue\n",
    "            if feature.qualifiers['locus_tag'][0]==locus_tag:\n",
    "                \n",
    "                ### create SeqRecord object ####\n",
    "                DNAseq=feature.location.extract(record).seq\n",
    "                AAseq=Seq(feature.qualifiers['translation'][0],alphabet=Gapped(IUPAC.extended_protein))\n",
    "                header='|'.join([index,locus_tag])\n",
    "\n",
    "                fna_seq = SeqRecord(DNAseq,id=header)\n",
    "                faa_seq = SeqRecord(AAseq,id=header)\n",
    "\n",
    "                fna.append(fna_seq)\n",
    "                faa.append(faa_seq)\n",
    "                if j==1:\n",
    "                    fna.append(Svel[gene]['fna'])\n",
    "                    faa.append(Svel[gene]['faa'])\n",
    "\n",
    "    faa=[f for f in sorted(faa, key=lambda x : x.id)]\n",
    "    fna=[f for f in sorted(fna, key=lambda x : x.id)]\n",
    "#     for rec in fna:\n",
    "#         print(rec)\n",
    "    #### run MUSCLE ####\n",
    "    muscle_cline = MuscleCommandline(clwstrict=True)\n",
    "    child = subprocess.Popen(str(muscle_cline),\n",
    "                     stdin=subprocess.PIPE,\n",
    "                     stdout=subprocess.PIPE,\n",
    "                     stderr=subprocess.PIPE,\n",
    "                     universal_newlines=True,\n",
    "                     shell=(sys.platform!=\"win32\"))\n",
    "    SeqIO.write(faa, child.stdin, \"fasta\")\n",
    "    child.stdin.close()\n",
    "    align = AlignIO.read(child.stdout, \"clustal\",alphabet=Gapped(IUPAC.extended_protein))\n",
    "    aln = MultipleSeqAlignment([f for f in sorted(align, key=lambda x : x.id)])\n",
    "    codon_aln = build(aln, fna,complete_protein=False,max_score=30,codon_table=table5)\n",
    "    print(codon_aln)\n",
    "    print(len(codon_aln))\n",
    "\n",
    "#     SeqIO.write(codon_aln, './geneious_mitoch/Codons_aln/'+gene+'.fna', \"fasta\")\n",
    "    SeqIO.write(faa, './geneious_mitoch/Codons_aln/'+gene+'.faa', \"fasta\")\n",
    "\n",
    "    n=0\n",
    "    names={}\n",
    "    for seq in codon_aln:\n",
    "        seq.id=seq.id.split('|')[0]\n",
    "        seq.name=seq.id\n",
    "        names[n]=seq.name.replace('.','')\n",
    "        n+=1\n",
    "    if i==1:\n",
    "        cat=codon_aln\n",
    "    if i>1:\n",
    "        cat.sort()\n",
    "        codon_aln.sort()\n",
    "        cat+=codon_aln\n",
    "for i in range(len(cat)):\n",
    "    cat[i].id=names[i]\n",
    "print(cat)\n",
    "\n",
    "SeqIO.write(cat, './geneious_mitoch/Codons_aln/cat_mitoch_genes.fna', \"fasta\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table 5 Invertebrate Mitochondrial, SGC4\n",
      "\n",
      "  |  T      |  C      |  A      |  G      |\n",
      "--+---------+---------+---------+---------+--\n",
      "T | TTT F   | TCT S   | TAT Y   | TGT C   | T\n",
      "T | TTC F   | TCC S   | TAC Y   | TGC C   | C\n",
      "T | TTA L   | TCA S   | TAA Stop| TGA W   | A\n",
      "T | TTG L(s)| TCG S   | TAG Stop| TGG W   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "C | CTT L   | CCT P   | CAT H   | CGT R   | T\n",
      "C | CTC L   | CCC P   | CAC H   | CGC R   | C\n",
      "C | CTA L   | CCA P   | CAA Q   | CGA R   | A\n",
      "C | CTG L   | CCG P   | CAG Q   | CGG R   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "A | ATT I(s)| ACT T   | AAT N   | AGT S   | T\n",
      "A | ATC I(s)| ACC T   | AAC N   | AGC S   | C\n",
      "A | ATA M(s)| ACA T   | AAA K   | AGA S   | A\n",
      "A | ATG M(s)| ACG T   | AAG K   | AGG S   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "G | GTT V   | GCT A   | GAT D   | GGT G   | T\n",
      "G | GTC V   | GCC A   | GAC D   | GGC G   | C\n",
      "G | GTA V   | GCA A   | GAA E   | GGA G   | A\n",
      "G | GTG V(s)| GCG A   | GAG E   | GGG G   | G\n",
      "--+---------+---------+---------+---------+--\n",
      "GATCRYWSMKHBVDN\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "register_ncbi_table() missing 5 required positional arguments: 'alt_name', 'id', 'table', 'start_codons', and 'stop_codons'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-83-bee5883d54be>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0mtable5\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprotein_alphabet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mIUPAC\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextended_protein\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     20\u001b[0m \u001b[0;31m# print(table5.forward_table)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 21\u001b[0;31m \u001b[0mregister_ncbi_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtable5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: register_ncbi_table() missing 5 required positional arguments: 'alt_name', 'id', 'table', 'start_codons', and 'stop_codons'"
     ]
    }
   ],
   "source": [
    "from Bio import Alphabet\n",
    "from Bio.Alphabet import IUPAC,Gapped\n",
    "from Bio import AlignIO\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Align.Applications import MuscleCommandline\n",
    "from Bio.codonalign import build\n",
    "from Bio.Data.CodonTable import generic_by_id,unambiguous_dna_by_id,register_ncbi_table\n",
    "import subprocess\n",
    "import sys\n",
    "import json\n",
    "\n",
    "pd.options.display.max_rows = 4000\n",
    "\n",
    "table5=generic_by_id[5]\n",
    "table5.nucleotide_alphabet=IUPAC.ambiguous_dna\n",
    "table5.protein_alphabet=IUPAC.extended_protein\n",
    "table5.forward_table['AAN'] = \"N\"\n",
    "codon_aln = codonalign.build(aln, sequence, codon_table=default_codon_table)\n",
    "\n",
    "# print(table5.forward_table)\n",
    "register_ncbi_table(table5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fix gbk file from GeSeq (Mitobim)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['mitobim_assemblies_PeNtenta-4_bb_bb_bb_GenBank.gb', 'mitobim_assemblies_Aphaseolif_bb_bb_bb_GenBank.gb', 'mitobim_all_Consensus_Vgig2Mitobim_Node2Spades_GenBank.gb', 'mitobim_assemblies_Crectimarg_bb_bb_bb_bb_GenBank.gb', 'mitobim_assemblies_Psoyae1-7%252_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb', 'mitobim_all_Consensus_Vgig1Mitobim_Node2Spades_GenBank.gb', 'mitobim_assemblies_Adiagonali_bb_bb_bb_bb_bb_bb_bb_GenBank.gb', 'mitobim_assemblies_Xpliocardi_bb_bb_bb_bb_bb_GenBank.gb', 'Rsou_shifted_Asouthward_bb_bb_bb_GenBank.gb', 'mitobim_assemblies_Cpacifica-_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb', 'mitobim_assemblies_Psoyae2-7%252_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb']\n",
      "{'Aphaseolif_bb': 'A.phaseoliformis', 'PeNtenta-4_bb': 'P.extenta', 'Consensus_Vgig2Mitobim': 'A.gigas2', 'Crectimarg_bb': 'C.rectimargo', 'Asouthward_bb': 'A.southwardae', 'Psoyae1-7%252_bb': 'P.soyae1', 'Consensus_Vgig1Mitobim': 'A.gigas1', 'Adiagonali_bb': 'A.diagonalis', 'Xpliocardi_bb': 'X.pliocardia', 'Cpacifica-_bb': 'C.pacifica', 'Psoyae2-7%252_bb': 'P.soyae2'}\n",
      "{'A.phaseoliformis': 'Aphaseolif_bb', 'P.extenta': 'PeNtenta-4_bb', 'A.gigas2': 'Consensus_Vgig2Mitobim', 'C.rectimargo': 'Crectimarg_bb', 'A.southwardae': 'Asouthward_bb', 'P.soyae1': 'Psoyae1-7%252_bb', 'A.gigas1': 'Consensus_Vgig1Mitobim', 'A.diagonalis': 'Adiagonali_bb', 'X.pliocardia': 'Xpliocardi_bb', 'C.pacifica': 'Cpacifica-_bb', 'P.soyae2': 'Psoyae2-7%252_bb'}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'P.extenta'"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "mypath='./mitobim_mitoch/'\n",
    "path=mypath\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-3:]=='.gb']\n",
    "print(onlyfiles)\n",
    "sampleid=['_'.join(file.split('_')[2:4]) for file in onlyfiles]\n",
    "\n",
    "dic=dict(zip(['A.phaseoliformis', 'P.extenta',  'A.gigas2','C.rectimargo','A.southwardae', 'P.soyae1','A.gigas1', 'A.diagonalis', 'X.pliocardia', 'C.pacifica',  'P.soyae2'],['Aphaseolif_bb','PeNtenta-4_bb', 'Consensus_Vgig2Mitobim', 'Crectimarg_bb', 'Asouthward_bb', 'Psoyae1-7%252_bb', 'Consensus_Vgig1Mitobim', 'Adiagonali_bb', 'Xpliocardi_bb', 'Cpacifica-_bb', 'Psoyae2-7%252_bb']))\n",
    "\n",
    "dic2=dict([v,k] for k,v in dic.items())\n",
    "print(dic2)\n",
    "print(dic)\n",
    "dic2['PeNtenta-4_bb']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P.extenta\n",
      "mitobim_assemblies_PeNtenta-4_bb_bb_bb_GenBank.gb\n",
      "A.phaseoliformis\n",
      "mitobim_assemblies_Aphaseolif_bb_bb_bb_GenBank.gb\n",
      "A.gigas2\n",
      "mitobim_all_Consensus_Vgig2Mitobim_Node2Spades_GenBank.gb\n",
      "!!! issue with cds\n",
      "C.rectimargo\n",
      "mitobim_assemblies_Crectimarg_bb_bb_bb_bb_GenBank.gb\n",
      "P.soyae1\n",
      "mitobim_assemblies_Psoyae1-7%252_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb\n",
      "A.gigas1\n",
      "mitobim_all_Consensus_Vgig1Mitobim_Node2Spades_GenBank.gb\n",
      "!!! issue with cds\n",
      "!!! issue with cds\n",
      "A.diagonalis\n",
      "mitobim_assemblies_Adiagonali_bb_bb_bb_bb_bb_bb_bb_GenBank.gb\n",
      "X.pliocardia\n",
      "mitobim_assemblies_Xpliocardi_bb_bb_bb_bb_bb_GenBank.gb\n",
      "A.southwardae\n",
      "Rsou_shifted_Asouthward_bb_bb_bb_GenBank.gb\n",
      "C.pacifica\n",
      "mitobim_assemblies_Cpacifica-_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb\n",
      "P.soyae2\n",
      "mitobim_assemblies_Psoyae2-7%252_bb_bb_bb_bb_bb_bb_bb_bb_bb_GenBank.gb\n"
     ]
    }
   ],
   "source": [
    "def find_end_of_gene(seq,gene_start):\n",
    "    for i in range(gene_start, len(seq), 3):\n",
    "        if seq[i:i+3] in ['TAG','TAA']:\n",
    "            return(i+3)\n",
    "            break\n",
    "def find_end_of_partial_cds(seq,gene_start):\n",
    "    for i in range(gene_start, len(seq), 3):\n",
    "        if seq[i:i+3] in ['NNN']:\n",
    "            return(i+3)\n",
    "            break\n",
    "            \n",
    "for file in onlyfiles:\n",
    "    \n",
    "    for record in SeqIO.parse(path+file,'genbank'):\n",
    "        # fix header\n",
    "        record.id='_'.join(file.split('_')[2:4])\n",
    "#         print(record.id)\n",
    "        record.name= dic2[record.id]\n",
    "        print(record.name)\n",
    "        record.description=record.name+' mitochondrion'\n",
    "        print(file)\n",
    "        i=0\n",
    "        for feature in record.features:\n",
    "            if feature.type=='source':\n",
    "                continue\n",
    "            if feature.type=='gene':\n",
    "                i+=1\n",
    "                feature.qualifiers['locus_tag']=record.description[0]+record.description[2:5]+'_'+\"{:02d}\".format(i)\n",
    "                last_locus_tag=record.description[0]+record.description[2:5]+'_'+\"{:02d}\".format(i)\n",
    "#                 print(last_locus_tag)\n",
    "            else:\n",
    "                feature.qualifiers['locus_tag']=last_locus_tag\n",
    "            if feature.type=='CDS':\n",
    "                try:\n",
    "                    feature.qualifiers['translation']=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "                except:\n",
    "                    print('!!! issue with cds')\n",
    "#                     print(feature)\n",
    "                    nucl_seq=feature.extract(record.seq)\n",
    "                    gene_start=feature.location.start\n",
    "                    gene_end=feature.location.end\n",
    "\n",
    "                    if feature.qualifiers['locus_tag']in ['Agig_30','Agig_33','Agig_55']:\n",
    "\n",
    "                        gene_end=find_end_of_partial_cds(record.seq,gene_start)\n",
    "                        feature.location=FeatureLocation(gene_start, gene_end, feature.location.strand)\n",
    "                        nucl_seq=record.seq[gene_start:gene_end-4]\n",
    "                        nucl_seq=nucl_seq[:-(len(nucl_seq)%3)]\n",
    "                        prot_seq=nucl_seq.translate(table=5, cds=False)\n",
    "                        feature.qualifiers['translation']=prot_seq\n",
    "                        feature.qualifiers['note']=['Partial coding sequence resolved due to low depth']\n",
    "#                         print(feature)\n",
    "\n",
    "\n",
    "\n",
    "#                     print(find_end_of_gene(record.seq,gene_start))\n",
    "#                     print(nucl_seq)\n",
    "#                     prot_seq=record.seq[gene_start,gene_end-3].translate(table=5, cds=True)\n",
    "#                     print(prot_seq)\n",
    "                    \n",
    "                    \n",
    "                    \n",
    "#         SeqIO.write(record, path+record.name+'_annotated'+'.gbk', \"genbank\")\n",
    "        if record.name=='A.gigas2':\n",
    "            record=record[:3111]+record[3118:]\n",
    "            SeqIO.write(record, path+record.name+'_annotated'+'.gbk', \"genbank\")\n",
    "\n",
    "            SeqIO.write(record, path+record.name+'.fasta', \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['./mitobim_mitoch/A.gigas1_annotated.gbk', './mitobim_mitoch/A.gigas2_annotated.gbk', './mitobim_mitoch/P.extenta_annotated.gbk', './mitobim_mitoch/A.phaseoliformis_annotated.gbk', './mitobim_mitoch/P.soyae2_annotated.gbk', './mitobim_mitoch/C.pacifica_annotated.gbk', './mitobim_mitoch/P.soyae1_annotated.gbk', './mitobim_mitoch/X.pliocardia_annotated.gbk', './mitobim_mitoch/C.rectimargo_annotated.gbk', './mitobim_mitoch/A.southwardae_annotated.gbk', './mitobim_mitoch/A.diagonalis_annotated.gbk', './geneious_mitoch/C.magnifica.gbk', './geneious_mitoch/P.okutanii_ref.gbk', './geneious_mitoch/A.phaseoliformis_ref.gbk', './geneious_mitoch/A.mariana_ref.gbk']\n",
      "./mitobim_mitoch/A.gigas1_annotated.gbk\n",
      "A.gigas1\n",
      "./mitobim_mitoch/A.gigas2_annotated.gbk\n",
      "A.gigas2\n",
      "./mitobim_mitoch/P.extenta_annotated.gbk\n",
      "P.extenta\n",
      "./mitobim_mitoch/A.phaseoliformis_annotated.gbk\n",
      "A.phaseoliformis\n",
      "./mitobim_mitoch/P.soyae2_annotated.gbk\n",
      "P.soyae2\n",
      "./mitobim_mitoch/C.pacifica_annotated.gbk\n",
      "C.pacifica\n",
      "./mitobim_mitoch/P.soyae1_annotated.gbk\n",
      "P.soyae1\n",
      "./mitobim_mitoch/X.pliocardia_annotated.gbk\n",
      "X.pliocardia\n",
      "./mitobim_mitoch/C.rectimargo_annotated.gbk\n",
      "C.rectimargo\n",
      "./mitobim_mitoch/A.southwardae_annotated.gbk\n",
      "A.southwardae\n",
      "./mitobim_mitoch/A.diagonalis_annotated.gbk\n",
      "A.diagonalis\n",
      "./geneious_mitoch/C.magnifica.gbk\n",
      "C.magnifica\n",
      "./geneious_mitoch/P.okutanii_ref.gbk\n",
      "P.okutanii_ref\n",
      "./geneious_mitoch/A.phaseoliformis_ref.gbk\n",
      "A.phaseoliformis_ref\n",
      "./geneious_mitoch/A.mariana_ref.gbk\n",
      "A.mariana_ref\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>COX1</th>\n",
       "      <th>COX2</th>\n",
       "      <th>CYTB</th>\n",
       "      <th>ATP8</th>\n",
       "      <th>ND4</th>\n",
       "      <th>ATP6</th>\n",
       "      <th>ND3</th>\n",
       "      <th>ND1</th>\n",
       "      <th>ND5</th>\n",
       "      <th>ND6</th>\n",
       "      <th>ND4L</th>\n",
       "      <th>ND2</th>\n",
       "      <th>COX3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A.gigas1</th>\n",
       "      <td>Agig_01</td>\n",
       "      <td>Agig_04</td>\n",
       "      <td>Agig_07</td>\n",
       "      <td>Agig_11</td>\n",
       "      <td>Agig_12</td>\n",
       "      <td>Agig_20</td>\n",
       "      <td>Agig_21</td>\n",
       "      <td>Agig_28</td>\n",
       "      <td>Agig_33</td>\n",
       "      <td>Agig_36</td>\n",
       "      <td>Agig_37</td>\n",
       "      <td>Agig_41</td>\n",
       "      <td>Agig_55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.gigas2</th>\n",
       "      <td>Agig_03</td>\n",
       "      <td>Agig_06</td>\n",
       "      <td>Agig_09</td>\n",
       "      <td>Agig_13</td>\n",
       "      <td>Agig_14</td>\n",
       "      <td>Agig_22</td>\n",
       "      <td>Agig_23</td>\n",
       "      <td>Agig_30</td>\n",
       "      <td>Agig_35</td>\n",
       "      <td>Agig_38</td>\n",
       "      <td>Agig_39</td>\n",
       "      <td>Agig_43</td>\n",
       "      <td>Agig_57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.extenta</th>\n",
       "      <td>Pext_24</td>\n",
       "      <td>Pext_27</td>\n",
       "      <td>Pext_31</td>\n",
       "      <td>Pext_34</td>\n",
       "      <td>Pext_35</td>\n",
       "      <td>Pext_42</td>\n",
       "      <td>Pext_43</td>\n",
       "      <td>Pext_51</td>\n",
       "      <td>Pext_56</td>\n",
       "      <td>Pext_02</td>\n",
       "      <td>Pext_03</td>\n",
       "      <td>Pext_07</td>\n",
       "      <td>Pext_20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.phaseoliformis</th>\n",
       "      <td>Apha_24</td>\n",
       "      <td>Apha_27</td>\n",
       "      <td>Apha_29</td>\n",
       "      <td>Apha_35</td>\n",
       "      <td>Apha_36</td>\n",
       "      <td>Apha_43</td>\n",
       "      <td>Apha_44</td>\n",
       "      <td>Apha_53</td>\n",
       "      <td>Apha_58</td>\n",
       "      <td>Apha_04</td>\n",
       "      <td>Apha_05</td>\n",
       "      <td>Apha_09</td>\n",
       "      <td>Apha_20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.soyae2</th>\n",
       "      <td>Psoy_23</td>\n",
       "      <td>Psoy_26</td>\n",
       "      <td>Psoy_31</td>\n",
       "      <td>Psoy_34</td>\n",
       "      <td>Psoy_35</td>\n",
       "      <td>Psoy_42</td>\n",
       "      <td>Psoy_43</td>\n",
       "      <td>Psoy_50</td>\n",
       "      <td>Psoy_55</td>\n",
       "      <td>Psoy_01</td>\n",
       "      <td>Psoy_02</td>\n",
       "      <td>Psoy_06</td>\n",
       "      <td>Psoy_18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C.pacifica</th>\n",
       "      <td>Cpac_27</td>\n",
       "      <td>Cpac_29</td>\n",
       "      <td>Cpac_32</td>\n",
       "      <td>Cpac_36</td>\n",
       "      <td>Cpac_37</td>\n",
       "      <td>Cpac_43</td>\n",
       "      <td>Cpac_44</td>\n",
       "      <td>Cpac_52</td>\n",
       "      <td>Cpac_57</td>\n",
       "      <td>Cpac_04</td>\n",
       "      <td>Cpac_05</td>\n",
       "      <td>Cpac_08</td>\n",
       "      <td>Cpac_20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.soyae1</th>\n",
       "      <td>Psoy_23</td>\n",
       "      <td>Psoy_26</td>\n",
       "      <td>Psoy_31</td>\n",
       "      <td>Psoy_34</td>\n",
       "      <td>Psoy_35</td>\n",
       "      <td>Psoy_42</td>\n",
       "      <td>Psoy_43</td>\n",
       "      <td>Psoy_50</td>\n",
       "      <td>Psoy_55</td>\n",
       "      <td>Psoy_01</td>\n",
       "      <td>Psoy_02</td>\n",
       "      <td>Psoy_06</td>\n",
       "      <td>Psoy_18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X.pliocardia</th>\n",
       "      <td>Xpli_32</td>\n",
       "      <td>Xpli_34</td>\n",
       "      <td>Xpli_38</td>\n",
       "      <td>Xpli_42</td>\n",
       "      <td>Xpli_43</td>\n",
       "      <td>Xpli_50</td>\n",
       "      <td>Xpli_51</td>\n",
       "      <td>Xpli_58</td>\n",
       "      <td>Xpli_63</td>\n",
       "      <td>Xpli_10</td>\n",
       "      <td>Xpli_11</td>\n",
       "      <td>Xpli_14</td>\n",
       "      <td>Xpli_26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C.rectimargo</th>\n",
       "      <td>Crec_21</td>\n",
       "      <td>Crec_24</td>\n",
       "      <td>Crec_27</td>\n",
       "      <td>Crec_30</td>\n",
       "      <td>Crec_31</td>\n",
       "      <td>Crec_38</td>\n",
       "      <td>Crec_39</td>\n",
       "      <td>Crec_48</td>\n",
       "      <td>Crec_53</td>\n",
       "      <td>Crec_01</td>\n",
       "      <td>Crec_02</td>\n",
       "      <td>Crec_04</td>\n",
       "      <td>Crec_14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.southwardae</th>\n",
       "      <td>Asou_23</td>\n",
       "      <td>Asou_26</td>\n",
       "      <td>Asou_28</td>\n",
       "      <td>Asou_33</td>\n",
       "      <td>Asou_34</td>\n",
       "      <td>Asou_40</td>\n",
       "      <td>Asou_41</td>\n",
       "      <td>Asou_51</td>\n",
       "      <td>Asou_56</td>\n",
       "      <td>Asou_01</td>\n",
       "      <td>Asou_02</td>\n",
       "      <td>Asou_06</td>\n",
       "      <td>Asou_18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.diagonalis</th>\n",
       "      <td>Adia_48</td>\n",
       "      <td>Adia_50</td>\n",
       "      <td>Adia_53</td>\n",
       "      <td>Adia_57</td>\n",
       "      <td>Adia_58</td>\n",
       "      <td>Adia_65</td>\n",
       "      <td>Adia_67</td>\n",
       "      <td>Adia_74</td>\n",
       "      <td>Adia_78</td>\n",
       "      <td>Adia_29</td>\n",
       "      <td>Adia_30</td>\n",
       "      <td>Adia_33</td>\n",
       "      <td>Adia_44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C.magnifica</th>\n",
       "      <td>Cmag_01</td>\n",
       "      <td>Cmag_03</td>\n",
       "      <td>Cmag_05</td>\n",
       "      <td>Cmag_07</td>\n",
       "      <td>Cmag_08</td>\n",
       "      <td>Cmag_13</td>\n",
       "      <td>Cmag_15</td>\n",
       "      <td>Cmag_22</td>\n",
       "      <td>Cmag_26</td>\n",
       "      <td>Cmag_32</td>\n",
       "      <td>Cmag_33</td>\n",
       "      <td>Cmag_35</td>\n",
       "      <td>Cmag_45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P.okutanii_ref</th>\n",
       "      <td>Poku_20</td>\n",
       "      <td>Poku_22</td>\n",
       "      <td>Poku_26</td>\n",
       "      <td>Poku_29</td>\n",
       "      <td>Poku_30</td>\n",
       "      <td>Poku_37</td>\n",
       "      <td>Poku_38</td>\n",
       "      <td>Poku_45</td>\n",
       "      <td>Poku_50</td>\n",
       "      <td>Poku_01</td>\n",
       "      <td>Poku_02</td>\n",
       "      <td>Poku_06</td>\n",
       "      <td>Poku_15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.phaseoliformis_ref</th>\n",
       "      <td>Apha_55</td>\n",
       "      <td>Apha_02</td>\n",
       "      <td>Apha_04</td>\n",
       "      <td>Apha_09</td>\n",
       "      <td>Apha_10</td>\n",
       "      <td>Apha_16</td>\n",
       "      <td>Apha_17</td>\n",
       "      <td>Apha_24</td>\n",
       "      <td>Apha_28</td>\n",
       "      <td>Apha_37</td>\n",
       "      <td>Apha_38</td>\n",
       "      <td>Apha_42</td>\n",
       "      <td>Apha_51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A.mariana_ref</th>\n",
       "      <td>Amar_19</td>\n",
       "      <td>Amar_21</td>\n",
       "      <td>Amar_24</td>\n",
       "      <td>Amar_28</td>\n",
       "      <td>Amar_29</td>\n",
       "      <td>Amar_35</td>\n",
       "      <td>Amar_36</td>\n",
       "      <td>Amar_43</td>\n",
       "      <td>Amar_48</td>\n",
       "      <td>Amar_01</td>\n",
       "      <td>Amar_02</td>\n",
       "      <td>Amar_06</td>\n",
       "      <td>Amar_15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         COX1     COX2     CYTB     ATP8      ND4     ATP6  \\\n",
       "A.gigas1              Agig_01  Agig_04  Agig_07  Agig_11  Agig_12  Agig_20   \n",
       "A.gigas2              Agig_03  Agig_06  Agig_09  Agig_13  Agig_14  Agig_22   \n",
       "P.extenta             Pext_24  Pext_27  Pext_31  Pext_34  Pext_35  Pext_42   \n",
       "A.phaseoliformis      Apha_24  Apha_27  Apha_29  Apha_35  Apha_36  Apha_43   \n",
       "P.soyae2              Psoy_23  Psoy_26  Psoy_31  Psoy_34  Psoy_35  Psoy_42   \n",
       "C.pacifica            Cpac_27  Cpac_29  Cpac_32  Cpac_36  Cpac_37  Cpac_43   \n",
       "P.soyae1              Psoy_23  Psoy_26  Psoy_31  Psoy_34  Psoy_35  Psoy_42   \n",
       "X.pliocardia          Xpli_32  Xpli_34  Xpli_38  Xpli_42  Xpli_43  Xpli_50   \n",
       "C.rectimargo          Crec_21  Crec_24  Crec_27  Crec_30  Crec_31  Crec_38   \n",
       "A.southwardae         Asou_23  Asou_26  Asou_28  Asou_33  Asou_34  Asou_40   \n",
       "A.diagonalis          Adia_48  Adia_50  Adia_53  Adia_57  Adia_58  Adia_65   \n",
       "C.magnifica           Cmag_01  Cmag_03  Cmag_05  Cmag_07  Cmag_08  Cmag_13   \n",
       "P.okutanii_ref        Poku_20  Poku_22  Poku_26  Poku_29  Poku_30  Poku_37   \n",
       "A.phaseoliformis_ref  Apha_55  Apha_02  Apha_04  Apha_09  Apha_10  Apha_16   \n",
       "A.mariana_ref         Amar_19  Amar_21  Amar_24  Amar_28  Amar_29  Amar_35   \n",
       "\n",
       "                          ND3      ND1      ND5      ND6     ND4L      ND2  \\\n",
       "A.gigas1              Agig_21  Agig_28  Agig_33  Agig_36  Agig_37  Agig_41   \n",
       "A.gigas2              Agig_23  Agig_30  Agig_35  Agig_38  Agig_39  Agig_43   \n",
       "P.extenta             Pext_43  Pext_51  Pext_56  Pext_02  Pext_03  Pext_07   \n",
       "A.phaseoliformis      Apha_44  Apha_53  Apha_58  Apha_04  Apha_05  Apha_09   \n",
       "P.soyae2              Psoy_43  Psoy_50  Psoy_55  Psoy_01  Psoy_02  Psoy_06   \n",
       "C.pacifica            Cpac_44  Cpac_52  Cpac_57  Cpac_04  Cpac_05  Cpac_08   \n",
       "P.soyae1              Psoy_43  Psoy_50  Psoy_55  Psoy_01  Psoy_02  Psoy_06   \n",
       "X.pliocardia          Xpli_51  Xpli_58  Xpli_63  Xpli_10  Xpli_11  Xpli_14   \n",
       "C.rectimargo          Crec_39  Crec_48  Crec_53  Crec_01  Crec_02  Crec_04   \n",
       "A.southwardae         Asou_41  Asou_51  Asou_56  Asou_01  Asou_02  Asou_06   \n",
       "A.diagonalis          Adia_67  Adia_74  Adia_78  Adia_29  Adia_30  Adia_33   \n",
       "C.magnifica           Cmag_15  Cmag_22  Cmag_26  Cmag_32  Cmag_33  Cmag_35   \n",
       "P.okutanii_ref        Poku_38  Poku_45  Poku_50  Poku_01  Poku_02  Poku_06   \n",
       "A.phaseoliformis_ref  Apha_17  Apha_24  Apha_28  Apha_37  Apha_38  Apha_42   \n",
       "A.mariana_ref         Amar_36  Amar_43  Amar_48  Amar_01  Amar_02  Amar_06   \n",
       "\n",
       "                         COX3  \n",
       "A.gigas1              Agig_55  \n",
       "A.gigas2              Agig_57  \n",
       "P.extenta             Pext_20  \n",
       "A.phaseoliformis      Apha_20  \n",
       "P.soyae2              Psoy_18  \n",
       "C.pacifica            Cpac_20  \n",
       "P.soyae1              Psoy_18  \n",
       "X.pliocardia          Xpli_26  \n",
       "C.rectimargo          Crec_14  \n",
       "A.southwardae         Asou_18  \n",
       "A.diagonalis          Adia_44  \n",
       "C.magnifica           Cmag_45  \n",
       "P.okutanii_ref        Poku_15  \n",
       "A.phaseoliformis_ref  Apha_51  \n",
       "A.mariana_ref         Amar_15  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mypath='./geneious_mitoch/'\n",
    "path='./mitobim_mitoch/'\n",
    "filedic={}\n",
    "recs={}\n",
    "seqs={}\n",
    "# seqs={'ND2': {}, 'ND6': {}, 'ND4L': {}, 'COX3': {}, 'COX1': {}, 'COX2': {}, 'CYTB': {}, 'ATP8': {}, 'ND4': {}, 'ATP6': {}, 'ND3': {}, 'ND1': {}, 'ND5': {}}\n",
    "allfiles = [path+f for f in listdir(path) if f[-4:]=='.gbk']+[mypath+'C.magnifica.gbk',\n",
    " mypath+'P.okutanii_ref.gbk', mypath+'A.phaseoliformis_ref.gbk',  mypath+'A.mariana_ref.gbk']\n",
    "print(allfiles)\n",
    "for file in allfiles:\n",
    "    print(file)\n",
    "    print(file.split('/')[2].replace('_annotated','').replace('.gbk',''))\n",
    "    for record in SeqIO.parse(file,'genbank'):\n",
    "        record.id=file.split('/')[2].replace('_annotated','').replace('.gbk','')\n",
    "        filedic[record.id]=file\n",
    "        recs[record.id]={}\n",
    "        for feature in record.features:\n",
    "            if feature.type=='CDS':\n",
    "\n",
    "                recs[record.id].update({feature.qualifiers['gene'][0]:feature.qualifiers['locus_tag'][0]})\n",
    "\n",
    "df=pd.DataFrame.from_dict(recs,orient='index')\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'A.gigas1': './mitobim_mitoch/A.gigas1_annotated.gbk',\n",
       " 'A.gigas2': './mitobim_mitoch/A.gigas2_annotated.gbk',\n",
       " 'P.extenta': './mitobim_mitoch/P.extenta_annotated.gbk',\n",
       " 'A.phaseoliformis': './mitobim_mitoch/A.phaseoliformis_annotated.gbk',\n",
       " 'P.soyae2': './mitobim_mitoch/P.soyae2_annotated.gbk',\n",
       " 'C.pacifica': './mitobim_mitoch/C.pacifica_annotated.gbk',\n",
       " 'P.soyae1': './mitobim_mitoch/P.soyae1_annotated.gbk',\n",
       " 'X.pliocardia': './mitobim_mitoch/X.pliocardia_annotated.gbk',\n",
       " 'C.rectimargo': './mitobim_mitoch/C.rectimargo_annotated.gbk',\n",
       " 'A.southwardae': './mitobim_mitoch/A.southwardae_annotated.gbk',\n",
       " 'A.diagonalis': './mitobim_mitoch/A.diagonalis_annotated.gbk',\n",
       " 'C.magnifica': './geneious_mitoch/C.magnifica.gbk',\n",
       " 'P.okutanii_ref': './geneious_mitoch/P.okutanii_ref.gbk',\n",
       " 'A.phaseoliformis_ref': './geneious_mitoch/A.phaseoliformis_ref.gbk',\n",
       " 'A.mariana_ref': './geneious_mitoch/A.mariana_ref.gbk'}"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filedic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COX1\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 1803 columns (601 codons)\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATATGTTTATAAATTTGTATTT...GTT A.diagonalis|Adia_48\n",
      "ATGTTTGTGGTTGAGCGTGTGGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT A.gigas1|Agig_01\n",
      "ATGTTTGTGGTTGAGCGTGTGGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT A.gigas2|Agig_03\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.mariana_ref|Amar_19\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.phaseoliformis_ref|Apha_55\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.phaseoliformis|Apha_24\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGAGTCTTATAGCTGTTTATAAATTTGTGTTT...GTT A.southwardae|Asou_23\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAGGGTCTTATGTCTGTTTATAAGTTTGTGTTT...GTT C.magnifica|Cmag_01\n",
      "ATGTTTGTGGTTGAGCGTGTGATTAAAAGTTTTATACTTATTTACAAGTTTGTATTT...GTT C.pacifica|Cpac_27\n",
      "ATATTTGTGCCTGAGCGTGTGGCTAAAATTTTTATATTTGTTTATAAGTTTATGTTT...GTT C.rectimargo|Crec_21\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.extenta|Pext_24\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.okutanii_ref|Poku_20\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.soyae1|Psoy_23\n",
      "ATATTTGTGGTTGAGCGTGTAGCTAAAAGTCTTATATCTGTTTATAAATTTGTATTT...GTT P.soyae2|Psoy_23\n",
      "ATATTTGTGATTGAGCGTTTATCTAAGAGTATTATACTAGTTTATAAATTTGTGTTT...GTT X.pliocardia|Xpli_32\n",
      "15\n",
      "COX2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: C.pacifica|Cpac_29(M 0) does not correspond to C.pacifica|Cpac_29(TTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 1485 columns (495 codons)\n",
      "---------ATGGGATCTTGGAATAAGTTATGGTTTTTTGATATAAACACTGTTGTT...--- A.diagonalis|Adia_50\n",
      "---------ATGGGATCTTGGAATAAGTTATGGTTTTTTGATATAAACACTGTTGTT...--- A.gigas1|Agig_04\n",
      "---------ATGGGATCTTGGAATAAGTTATGGTTTTTTGATATAAACACTGTTGTT...--- A.gigas2|Agig_06\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.mariana_ref|Amar_21\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.phaseoliformis_ref|Apha_02\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.phaseoliformis|Apha_27\n",
      "---------ATGGGGTCTTGAAATAAGTTATGATTTTATGATATAAATACTGTTGTT...--- A.southwardae|Asou_26\n",
      "---------ATGGGATCTTGAAACAAATTATGGTTTTTTGATATAAATACTGTGGTT...--- C.magnifica|Cmag_03\n",
      "TTGTTTTGAATGGGATCTTGAGAGAAAATGTGATTTTATGATATAAATACTGTTGTC...GTC C.pacifica|Cpac_29\n",
      "---------ATGGGATCCTGAGAGAAATTTTGGTTTTATGATATAAATACTGTTGTC...GAC C.rectimargo|Crec_24\n",
      "---------ATGGGGTCTTGAAGTAAGTTATGGTTTTTTGATATAAATACTGTGGTT...--- P.extenta|Pext_27\n",
      "---------ATGGGGTCTTGAAATAAGTTATGGTTTTTTGATATAAATACTGTTGTT...--- P.okutanii_ref|Poku_22\n",
      "---------ATGGGGTCTTGGAATAAGTTATGGTTTTTTGATATAAATACTGTTGTT...--- P.soyae1|Psoy_26\n",
      "---------ATGGGGTCTTGGAATAAGTTATGGTTTTTTGATATAAATACTGTTGTT...--- P.soyae2|Psoy_26\n",
      "---------ATGGGATCTTGGCATAAATTGTGGTTTTATGATATAAATACTGTTGTT...--- X.pliocardia|Xpli_34\n",
      "15\n",
      "CYTB\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 1131 columns (377 codons)\n",
      "ATGGGAAACCATGATGGGAAAGGAATTCTTTATAGGATAAAGAGAGTTCTTTATGAT...GGG A.diagonalis|Adia_53\n",
      "ATGGGAAACCATGATGGGAAAGGAATTCTTTATAGGATAAAGAGAGTTCTTTATGAT...GAG A.gigas1|Agig_07\n",
      "ATGGGAAACCATGATGGGAAAGGAATTCTTTATAGGATAAAGAGAGTTCTTTATGAT...GAG A.gigas2|Agig_09\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAATTTTGTATGAC...AAA A.mariana_ref|Amar_24\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.phaseoliformis_ref|Apha_04\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.phaseoliformis|Apha_29\n",
      "ATGGGAAATCATGATGGAAAAGGAATTTTTTATAGAATAAAAAGAGTTTTGTATGAC...AAA A.southwardae|Asou_28\n",
      "ATGGGAAACCACGATGGGAAAGGGATTCTTTATAGAATAAAAAGAGCTTTATATGAT...AAA C.magnifica|Cmag_05\n",
      "ATGGGGAACCACGATGGGAAAGGTATTCTTTACAGGATAAAGAGGGCTTTATATGAT...AAA C.pacifica|Cpac_32\n",
      "ATGGGAAACCACGATGGGAAAGGGATTCTTTACAGGATAAAGGGGGCTTTATATGAT...AAA C.rectimargo|Crec_27\n",
      "ATGGGGAACCACGATGGAAAAGGAATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.extenta|Pext_31\n",
      "ATGGGAAACCACGATGGAAAAGGAATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.okutanii_ref|Poku_26\n",
      "ATGGGAAACCATGATGGAAAAGGTATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.soyae1|Psoy_31\n",
      "ATGGGAAACCATGATGGAAAAGGTATTCTTTATAGGATAAAGAGAGCTTTATATGAT...AAG P.soyae2|Psoy_31\n",
      "ATGGGAAATCACGATAGGAAAGGAATTCTTTATAGAATGAAAAGGGCTTTGTATGAT...AAG X.pliocardia|Xpli_38\n",
      "15\n",
      "ATP8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: P.extenta|Pext_34(M 0) does not correspond to P.extenta|Pext_34(GTG)\n",
      "  BiopythonWarning)\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: P.okutanii_ref|Poku_29(M 0) does not correspond to P.okutanii_ref|Poku_29(GTG)\n",
      "  BiopythonWarning)\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: P.soyae1|Psoy_34(M 0) does not correspond to P.soyae1|Psoy_34(GTG)\n",
      "  BiopythonWarning)\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: P.soyae2|Psoy_34(M 0) does not correspond to P.soyae2|Psoy_34(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 114 columns (38 codons)\n",
      "ATGCCTCAGTTTTCTCCTCTTTTTGTTGATTTTATTTTTTTGTTCTATTGGGCGGTG...--- A.diagonalis|Adia_57\n",
      "ATGCCTCAGTTTTCTCCTCTTTTTGTTGATTTTATTTTTTTGTTCTATTGGGCGGTG...--- A.gigas1|Agig_11\n",
      "ATGCCTCAGTTTTCTCCTCTTTTTGTTGATTTTATTTTTTTGTTCTATTGGGCGGTG...--- A.gigas2|Agig_13\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTCGATTTTATTTTTTTATTTTATTGGGCGGTG...--- A.mariana_ref|Amar_28\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTA...--- A.phaseoliformis_ref|Apha_09\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTA...--- A.phaseoliformis|Apha_35\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTATTTTATTGGGCGGTG...--- A.southwardae|Asou_33\n",
      "ATGCCTCAGTTTTCACCTCTTTTTGTAGATTTTATTTTTTTATTTTATTGGGCGGTG...--- C.magnifica|Cmag_07\n",
      "ATGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- C.pacifica|Cpac_36\n",
      "ATGCCTCAGTTTTCACCTCTATTTGTGGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- C.rectimargo|Crec_30\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTATGTTTTATTGGGTGGTG...--- P.extenta|Pext_34\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...AAG P.okutanii_ref|Poku_29\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- P.soyae1|Psoy_34\n",
      "GTGCCTCAGTTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGGGCGGTG...--- P.soyae2|Psoy_34\n",
      "ATGCCTCAATTTTCACCTCTTTTTGTTGATTTTATTTTTTTGTTTTATTGAATAGTG...--- X.pliocardia|Xpli_42\n",
      "15\n",
      "ND4\n",
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 1356 columns (452 codons)\n",
      "ATGGGG------------ACAGTTATCTCTTGCTCTGCAGCTTATTGTATTCCGAAT...GTG A.diagonalis|Adia_58\n",
      "ATGGGG------------ACGGTGATCTCTTGCTTTGCAGCTTACTGTATCCCGAAT...GTA A.gigas1|Agig_12\n",
      "ATGGGG------------ACGGTGATTTCTTGCTTTGCAGCTTACTGTATCCCGAAT...GTA A.gigas2|Agig_14\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTATTTATTGTATTCCAAAT...ATA A.mariana_ref|Amar_29\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTGTTTATTGTATTCCAAAT...ATA A.phaseoliformis_ref|Apha_10\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTGTTTATTGTATTCCAAAT...ATA A.phaseoliformis|Apha_36\n",
      "ATGGGG------------ACTGTAATTTCTTGTTTTGTTATTTATTGTATTCCAAAT...ATA A.southwardae|Asou_34\n",
      "ATGGGA------------ACTGTAATTTCTTGTTTTGCTGCTTATTGTATTCCGAAT...ATA C.magnifica|Cmag_08\n",
      "ATAGGA------------ACTGTAATTTCTTGTTTTGTTAGTTATTGTGTTCCAAAT...ATA C.pacifica|Cpac_37\n",
      "ATAGGA------------GCTGTGATTTCTTGTTTTGTTAGTTATTGTGTCCCGAAC...GTG C.rectimargo|Crec_31\n",
      "ATGGGG------------ACGGTGATTTCTTGTTCTGCTGCTTATTGTATCCCTAAT...ATG P.extenta|Pext_35\n",
      "ATGGGT------------ACAGTGATTTCTTGTTCTGCAGCTTATTGTATCCCTAAT...ATG P.okutanii_ref|Poku_30\n",
      "ATGGGG------------ACAGTGATCTCTTGTTCTGCAGCTTACTGTATCCCTAAT...ATG P.soyae1|Psoy_35\n",
      "ATGGGG------------ACAGTGATCTCTTGTTCTGCAGCTTACTGTATCCCTAAT...ATG P.soyae2|Psoy_35\n",
      "ATGGGAGGAATTATAAGAACTGTAATTTCTTGTTTTGTAAGTTATTGTGTCCCGAAT...ATA X.pliocardia|Xpli_43\n",
      "15\n",
      "ATP6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: X.pliocardia|Xpli_50(M 0) does not correspond to X.pliocardia|Xpli_50(GTG)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CodonAlphabet(Invertebrate Mitochondrial) CodonAlignment with 15 rows and 711 columns (237 codons)\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGGGGATTTAGGGTACTT...TTG A.diagonalis|Adia_65\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATGGGTGGATTTAGAGTTCTT...TTA A.gigas1|Agig_20\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATGGGTGGATTTAGAGTTCTT...TTA A.gigas2|Agig_22\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGTTG...TTG A.mariana_ref|Amar_35\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGTTG...TTG A.phaseoliformis_ref|Apha_16\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGTTG...TTG A.phaseoliformis|Apha_43\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGTTA...TTG A.southwardae|Asou_40\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGCTTTAGGGTGCTA...TTG C.magnifica|Cmag_13\n",
      "ATGGTTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGGGGATTTAGGGTATTA...TTA C.pacifica|Cpac_43\n",
      "ATGGTTAGGGATATTTTCACTTTTATGGATTGTTTTATAGGGGGATTTAGGGTGTTA...TTA C.rectimargo|Crec_38\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATGGGCGGGTTTAGAGTGCTT...TTA P.extenta|Pext_42\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGCTT...TTG P.okutanii_ref|Poku_37\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGCTT...TTG P.soyae1|Psoy_42\n",
      "ATGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAGTGCTT...TTG P.soyae2|Psoy_42\n",
      "GTGACTAGGGATATTTTTACTTTTATGGATTGTTTTATAGGTGGATTTAGAATTTTA...TTA X.pliocardia|Xpli_50\n",
      "15\n",
      "ND3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: A.diagonalis|Adia_67(M 0) does not correspond to A.diagonalis|Adia_67(ATT)\n",
      "  BiopythonWarning)\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: A.gigas1|Agig_21(M 0) does not correspond to A.gigas1|Agig_21(ATT)\n",
      "  BiopythonWarning)\n",
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py:579: BiopythonWarning: A.gigas2|Agig_23(M 0) does not correspond to A.gigas2|Agig_23(ATT)\n",
      "  BiopythonWarning)\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Sequence contain codon not in the alphabet (NNN)! ",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-138-616e584adfd0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     67\u001b[0m     \u001b[0malign\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mAlignIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchild\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"clustal\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0malphabet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mGapped\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mIUPAC\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextended_protein\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m     \u001b[0maln\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mMultipleSeqAlignment\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0msorted\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malign\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 69\u001b[0;31m     \u001b[0mcodon_aln\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbuild\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maln\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfna\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcomplete_protein\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmax_score\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m30\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcodon_table\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtable5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     70\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodon_aln\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodon_aln\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py\u001b[0m in \u001b[0;36mbuild\u001b[0;34m(pro_align, nucl_seqs, corr_dict, gap_char, unknown, codon_table, alphabet, complete_protein, anchor_len, max_score)\u001b[0m\n\u001b[1;32m    175\u001b[0m                                        \u001b[0mcomplete_protein\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    176\u001b[0m                                        \u001b[0mcodon_table\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcodon_table\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 177\u001b[0;31m                                        max_score=max_score)\n\u001b[0m\u001b[1;32m    178\u001b[0m             \u001b[0mcodon_aln\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodon_rec\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    179\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mcorr_span\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/__init__.py\u001b[0m in \u001b[0;36m_get_codon_rec\u001b[0;34m(pro, nucl, span_mode, alphabet, gap_char, codon_table, complete_protein, max_score)\u001b[0m\n\u001b[1;32m    584\u001b[0m                 \u001b[0mcodon_seq\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mthis_codon\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m                 \u001b[0maa_num\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mSeqRecord\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mCodonSeq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcodon_seq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malphabet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0malphabet\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnucl\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mmode\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m         \u001b[0;32mfrom\u001b[0m \u001b[0mcollections\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mdeque\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/codonalign/codonseq.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, alphabet, gap_char, rf_table)\u001b[0m\n\u001b[1;32m     89\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mi\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0malphabet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mletters\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     90\u001b[0m                     raise ValueError(\"Sequence contain codon not in the alphabet \"\n\u001b[0;32m---> 91\u001b[0;31m                                      \"({0})! \".format(self._data[i:i + 3]))\n\u001b[0m\u001b[1;32m     92\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m             \u001b[0;31m# if gap_char in self._data:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Sequence contain codon not in the alphabet (NNN)! "
     ]
    }
   ],
   "source": [
    "from Bio import Alphabet\n",
    "from Bio.Alphabet import IUPAC,Gapped\n",
    "from Bio import AlignIO\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Align.Applications import MuscleCommandline\n",
    "from Bio.codonalign import build\n",
    "from Bio.Data.CodonTable import generic_by_id,unambiguous_dna_by_id\n",
    "import subprocess\n",
    "import sys\n",
    "\n",
    "i=0\n",
    "table5=unambiguous_dna_by_id[5]\n",
    "table5.nucleotide_alphabet=IUPAC.ambiguous_dna\n",
    "table5.protein_alphabet=IUPAC.extended_protein\n",
    "table5.forward_table['AAN'] = \"K\"\n",
    "table5.forward_table['GAY'] = \"D\"\n",
    "table5.forward_table['YTA'] = \"L\"\n",
    "table5.forward_table['ATY'] = \"I\"\n",
    "table5.forward_table['TCY'] = \"S\"\n",
    "table5.forward_table['GTY'] = \"V\"\n",
    "table5.forward_table['GGY'] = \"G\"\n",
    "table5.forward_table['CCY'] = \"P\"\n",
    "table5.forward_table['GGR'] = \"G\"\n",
    "for gene in df.columns.values:\n",
    "    i+=1\n",
    "    j=0\n",
    "    print(gene)\n",
    "    fna=[]\n",
    "    faa=[]\n",
    "    for index in df.index.values:\n",
    "        j+=1\n",
    "        locus_tag=df[gene][index]\n",
    "        record = SeqIO.read(filedic[index], \"genbank\")\n",
    "        for feature in record.features:\n",
    "            if feature.type != 'CDS':\n",
    "                continue\n",
    "            if feature.qualifiers['locus_tag'][0]==locus_tag:\n",
    "                \n",
    "                ### create SeqRecord object ####\n",
    "                DNAseq=feature.location.extract(record).seq\n",
    "                AAseq=Seq(feature.qualifiers['translation'][0],alphabet=Gapped(IUPAC.extended_protein))\n",
    "                header='|'.join([index,locus_tag])\n",
    "\n",
    "                fna_seq = SeqRecord(DNAseq,id=header)\n",
    "                faa_seq = SeqRecord(AAseq,id=header)\n",
    "\n",
    "                fna.append(fna_seq)\n",
    "                faa.append(faa_seq)\n",
    "#                 if j==1:\n",
    "#                     fna.append(Svel[gene]['fna'])\n",
    "#                     faa.append(Svel[gene]['faa'])\n",
    "\n",
    "    faa=[f for f in sorted(faa, key=lambda x : x.id)]\n",
    "    fna=[f for f in sorted(fna, key=lambda x : x.id)]\n",
    "#     for rec in fna:\n",
    "#         print(rec)\n",
    "    #### run MUSCLE ####\n",
    "    muscle_cline = MuscleCommandline(clwstrict=True)\n",
    "    child = subprocess.Popen(str(muscle_cline),\n",
    "                     stdin=subprocess.PIPE,\n",
    "                     stdout=subprocess.PIPE,\n",
    "                     stderr=subprocess.PIPE,\n",
    "                     universal_newlines=True,\n",
    "                     shell=(sys.platform!=\"win32\"))\n",
    "    SeqIO.write(faa, child.stdin, \"fasta\")\n",
    "    child.stdin.close()\n",
    "    align = AlignIO.read(child.stdout, \"clustal\",alphabet=Gapped(IUPAC.extended_protein))\n",
    "    aln = MultipleSeqAlignment([f for f in sorted(align, key=lambda x : x.id)])\n",
    "    codon_aln = build(aln, fna,complete_protein=False,max_score=30,codon_table=table5)\n",
    "    print(codon_aln)\n",
    "    print(len(codon_aln))\n",
    "\n",
    "    SeqIO.write(codon_aln, './mitobim_mitoch/Codon_aln/'+gene+'.fna', \"fasta\")\n",
    "    SeqIO.write(faa, './mitobim_mitoch/Codon_aln/'+gene+'.faa', \"fasta\")\n",
    "\n",
    "    n=0\n",
    "    names={}\n",
    "    for seq in codon_aln:\n",
    "        seq.id=seq.id.split('|')[0]\n",
    "        seq.name=seq.id\n",
    "        names[n]=seq.name.replace('.','')\n",
    "        n+=1\n",
    "    if i==1:\n",
    "        cat=codon_aln\n",
    "    if i>1:\n",
    "        cat.sort()\n",
    "        codon_aln.sort()\n",
    "        cat+=codon_aln\n",
    "for i in range(len(cat)):\n",
    "    cat[i].id=names[i]\n",
    "print(cat)\n",
    "\n",
    "SeqIO.write(cat, './mitobim_mitoch/Codon_aln/cat_mitoch_genes.fna', \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('TATTATCTTTAACATTTGCTTCTATAGCTTATATAAAGTGAAGAGCTTTTAACT...GGA', IUPACAmbiguousDNA())"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[gene_start:gene_end].seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clean up Geneious gbk files and generate Codon-based aln"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['X.pliocardia.gb',\n",
       " 'A.southwardae.gb',\n",
       " 'P.extenta.gb',\n",
       " 'A.gigas2.gb',\n",
       " 'A.diagonalis.gb',\n",
       " 'P.soyoae2.gb',\n",
       " 'C.rectimargo.gb',\n",
       " 'P.soyoae1.gb',\n",
       " 'C.pacifica.gb',\n",
       " 'A.gigas1.gb',\n",
       " 'A.phaseoliformis.gb']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "mypath='./mitobim_mitoch/batch_export/'\n",
    "onlyfiles = [f for f in listdir(mypath) if f[-3:]=='.gb']\n",
    "onlyfiles\n",
    "# SeqIO.convert(mypath+onlyfiles[0],'genbank',mypath+'test.gbk','genbank')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.gigas2.gb\n",
      "15759\n",
      "15759\n",
      "GTGTAGTTTT\n",
      "AAGTTTTTTN\n",
      "A.gigas1.gb\n",
      "15575\n",
      "15575\n",
      "TTTTTACTTC\n",
      "NNNNNNNNNN\n"
     ]
    }
   ],
   "source": [
    "for file in [onlyfiles[3],onlyfiles[9]]:\n",
    "    print(file)\n",
    "    if file=='A.gigas1.gb':\n",
    "        for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "            print(len(record))\n",
    "            newrecord=record[11173:]+record[:11173]\n",
    "            print(len(newrecord))\n",
    "            print(newrecord[:10].seq)\n",
    "            print(newrecord[-10:].seq)\n",
    "    if file=='A.gigas2.gb':\n",
    "        for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "            print(len(record))\n",
    "            newrecord=record[11559:]+record[:11559]\n",
    "            print(len(newrecord))\n",
    "            print(newrecord[:10].seq)\n",
    "            print(newrecord[-10:].seq)\n",
    "#     SeqIO.write(newrecord,file[:-3]+'.gbk','genbank')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A.gigas2.gb\n",
      "15759\n",
      "A.gigas1.gb\n",
      "14826\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/maeperez/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/GenBank/__init__.py:1398: BiopythonParserWarning: Expected sequence length 15426, found 14826 (Consensus_Vgig1Mitobim).\n",
      "  BiopythonParserWarning,\n"
     ]
    }
   ],
   "source": [
    "for file in [onlyfiles[3],onlyfiles[9]]:\n",
    "    print(file)\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        print(len(record))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pliocardia sp. isolate 17_S18 mitochondrial DNA, partial sequence\n",
      "Abyssogena southwardae isolate 11_S2 mitochondrion, complete genome\n",
      "Phreagena extenta isolate 18_S21 mitochondrial DNA, partial sequence\n",
      "Archivesica gigas isolate 15_S12 mitochondrial DNA, partial sequence\n",
      "!!! issue with cds\n",
      "Agig_14\n",
      "OrderedDict([('gene', ['ND3']), ('locus_tag', 'Agig_14'), ('note', [\"Derived using Geneious Prime 2020.2.2 'Annotate from Database' based on nucleotide similarity\"])])\n"
     ]
    },
    {
     "ename": "TranslationError",
     "evalue": "Sequence length 301 is not a multiple of three",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTranslationError\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-24-f2d81ad22ec3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     17\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 18\u001b[0;31m                     \u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'translation'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextract\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranslate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     19\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36mtranslate\u001b[0;34m(self, table, stop_symbol, to_stop, cds, gap)\u001b[0m\n\u001b[1;32m   1277\u001b[0m         protein = _translate_str(\n\u001b[0;32m-> 1278\u001b[0;31m             \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcodon_table\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstop_symbol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_stop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgap\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1279\u001b[0m         )\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m_translate_str\u001b[0;34m(sequence, table, stop_symbol, to_stop, cds, pos_stop, gap)\u001b[0m\n\u001b[1;32m   2843\u001b[0m             raise CodonTable.TranslationError(\n\u001b[0;32m-> 2844\u001b[0;31m                 \u001b[0;34m\"Sequence length {0} is not a multiple of three\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2845\u001b[0m             )\n",
      "\u001b[0;31mTranslationError\u001b[0m: Sequence length 301 is not a multiple of three",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mTranslationError\u001b[0m                          Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-24-f2d81ad22ec3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     21\u001b[0m                     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'locus_tag'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m                     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m                     \u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqualifiers\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'translation'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextract\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranslate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtable\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m                     \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36mtranslate\u001b[0;34m(self, table, stop_symbol, to_stop, cds, gap)\u001b[0m\n\u001b[1;32m   1276\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1277\u001b[0m         protein = _translate_str(\n\u001b[0;32m-> 1278\u001b[0;31m             \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcodon_table\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstop_symbol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_stop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgap\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgap\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1279\u001b[0m         )\n\u001b[1;32m   1280\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda2/envs/py36/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m_translate_str\u001b[0;34m(sequence, table, stop_symbol, to_stop, cds, pos_stop, gap)\u001b[0m\n\u001b[1;32m   2842\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;36m3\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2843\u001b[0m             raise CodonTable.TranslationError(\n\u001b[0;32m-> 2844\u001b[0;31m                 \u001b[0;34m\"Sequence length {0} is not a multiple of three\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2845\u001b[0m             )\n\u001b[1;32m   2846\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mstop_codons\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTranslationError\u001b[0m: Sequence length 301 is not a multiple of three"
     ]
    }
   ],
   "source": [
    "for file in onlyfiles:\n",
    "    for record in SeqIO.parse(mypath+file,'genbank'):\n",
    "        record.id=file[0]+file[1:-3]\n",
    "        print(record.description)\n",
    "        i=0\n",
    "        for feature in record.features:\n",
    "            if feature.type=='source':\n",
    "                continue\n",
    "            if feature.type!='gene':\n",
    "                i+=1\n",
    "                feature.qualifiers['locus_tag']=record.id[0]+record.id[2:5]+'_'+\"{:02d}\".format(i)\n",
    "                last_locus_tag=record.id[0]+record.id[2:5]+'_'+\"{:02d}\".format(i)\n",
    "#                 print(last_locus_tag)\n",
    "            else:\n",
    "                feature.qualifiers['locus_tag']=last_locus_tag\n",
    "            if feature.type=='CDS':\n",
    "                try:\n",
    "                    feature.qualifiers['translation']=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "                except:\n",
    "                    print('!!! issue with cds')\n",
    "                    print(feature.qualifiers['locus_tag'])\n",
    "                    print(feature.qualifiers)\n",
    "                    feature.qualifiers['translation']=feature.extract(record.seq).translate(table=5, cds=True)\n",
    "\n",
    "                    break\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
