{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.000\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/X.pliocardia.gbk 18886 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.001\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/P.soyoae2.gbk 19268 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.002\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/P.soyoae1.gbk 19298 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.003\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/P.extenta.gbk 18109 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.004\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/C.rectimargo.gbk 19326 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.005\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/C.pacifica.gbk 19910 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.006\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/A.southwardae.gbk 19082 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.007\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/A.phaseoliformis.gbk 17997 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.008\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/A.gigas2.gbk 15758 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.009\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/A.gigas1.gbk 14826 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.010\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export/A.diagonalis.gbk 20322 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.011\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/assemblies_I_picked/C.magnifica.gbk 19738 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.012\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/assemblies_I_picked/A.mariana_ref.gbk 15927 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.013\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/assemblies_I_picked/A.phaseoliformis_ref.gbk 19424 base pairs.\n",
      "Storing raw sequence at /var/folders/f1/fnq4czf942x8ctpl59lf6ntr0000gn/T/rawseq52423.014\n",
      "Sequence loaded successfully.\n",
      "/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/assemblies_I_picked/P.okutanii_ref.gbk 16336 base pairs.\n",
      "Using weight 11 mers for initial seeds\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "Sorted mer list loaded successfully\n",
      "3%..7%..10%..\n",
      "14%..18%..21%..\n",
      "25%..29%..32%..\n",
      "36%..40%..\n",
      "43%..47%..51%..\n",
      "54%..58%..61%..\n",
      "63%..65%..69%..72%..\n",
      "75%..78%..82%..\n",
      "86%..89%..92%..\n",
      "96%..98%..99%..done.\n",
      "using default bp penalty: 99106.1\n",
      "using default bp estimate min score: 297318\n",
      "Starting with 17130 multi-matches\n",
      "Computing genome content distance matrix...\n",
      "\n",
      "\n",
      "Genome conservation distance matrix: \n",
      "0\t0.262523\t0.260428\t0.260135\t0.253856\t0.244286\t0.20789\t0.204892\t0.232281\t0.2448\t0.298175\t0.243914\t0.179707\t0.214024\t0.242503\n",
      "0.262523\t0\t0.106156\t0.157359\t0.275411\t0.26016\t0.242104\t0.225412\t0.139186\t0.170185\t0.211157\t0.245829\t0.20306\t0.239718\t0.110201\n",
      "0.260428\t0.106156\t0\t0.153863\t0.276437\t0.262012\t0.239376\t0.224515\t0.139746\t0.170707\t0.211294\t0.245234\t0.20157\t0.238907\t0.109246\n",
      "0.260135\t0.157359\t0.153863\t0\t0.268164\t0.26016\t0.235108\t0.225208\t0.136267\t0.158219\t0.210244\t0.246668\t0.198317\t0.235665\t0.126802\n",
      "0.253856\t0.275411\t0.276437\t0.268164\t0\t0.180947\t0.221199\t0.219536\t0.24471\t0.258244\t0.303858\t0.253951\t0.19657\t0.222434\t0.251244\n",
      "0.244286\t0.26016\t0.262012\t0.26016\t0.180947\t0\t0.211273\t0.206825\t0.24469\t0.259576\t0.299202\t0.253477\t0.18238\t0.216637\t0.24402\n",
      "0.20789\t0.242104\t0.239376\t0.235108\t0.221199\t0.211273\t0\t0.0958634\t0.212614\t0.234381\t0.271889\t0.211496\t0.0897266\t0.0984249\t0.225835\n",
      "0.204892\t0.225412\t0.224515\t0.225208\t0.219536\t0.206825\t0.0958634\t0\t0.19382\t0.211777\t0.26517\t0.211955\t0.0617824\t0.052468\t0.204525\n",
      "0.232281\t0.139186\t0.139746\t0.136267\t0.24471\t0.24469\t0.212614\t0.19382\t0\t0.0575095\t0.167443\t0.220114\t0.167477\t0.20693\t0.109138\n",
      "0.2448\t0.170185\t0.170707\t0.158219\t0.258244\t0.259576\t0.234381\t0.211777\t0.0575095\t0\t0.190947\t0.233577\t0.179211\t0.224994\t0.122783\n",
      "0.298175\t0.211157\t0.211294\t0.210244\t0.303858\t0.299202\t0.271889\t0.26517\t0.167443\t0.190947\t0\t0.290175\t0.235472\t0.271247\t0.194128\n",
      "0.243914\t0.245829\t0.245234\t0.246668\t0.253951\t0.253477\t0.211496\t0.211955\t0.220114\t0.233577\t0.290175\t0\t0.177663\t0.208717\t0.222444\n",
      "0.179707\t0.20306\t0.20157\t0.198317\t0.19657\t0.18238\t0.0897266\t0.0617824\t0.167477\t0.179211\t0.235472\t0.177663\t0\t0.090653\t0.169843\n",
      "0.214024\t0.239718\t0.238907\t0.235665\t0.222434\t0.216637\t0.0984249\t0.052468\t0.20693\t0.224994\t0.271247\t0.208717\t0.090653\t0\t0.225544\n",
      "0.242503\t0.110201\t0.109246\t0.126802\t0.251244\t0.24402\t0.225835\t0.204525\t0.109138\t0.122783\t0.194128\t0.222444\t0.169843\t0.225544\t0\n",
      "\n",
      "Writing guide tree to /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/MITOCH.tree\n",
      "reading tree...\n",
      "initializing alignment tree...\n",
      "Constructing seed occurrence lists for repeat detection\n",
      "Calculating pairwise breakpoint distances\n",
      "Pair 0, 1 has 292 initial LCBs\n",
      "Using scaled bp penalty: 87945.7\n",
      "Pair (0,1) has 1 well-supported breakpoints\n",
      "Pair 0, 2 has 305 initial LCBs\n",
      "Using scaled bp penalty: 88949.4\n",
      "Pair (0,2) has 1 well-supported breakpoints\n",
      "Pair 0, 3 has 284 initial LCBs\n",
      "Using scaled bp penalty: 89090.4\n",
      "Pair (0,3) has 1 well-supported breakpoints\n",
      "Pair 0, 4 has 342 initial LCBs\n",
      "Using scaled bp penalty: 92153.8\n",
      "Pair (0,4) has 1 well-supported breakpoints\n",
      "Pair 0, 5 has 338 initial LCBs\n",
      "Using scaled bp penalty: 96973.4\n",
      "Pair (0,5) has 1 well-supported breakpoints\n",
      "Pair 0, 6 has 312 initial LCBs\n",
      "Using scaled bp penalty: 117048\n",
      "Pair (0,6) has 1 well-supported breakpoints\n",
      "Pair 0, 7 has 289 initial LCBs\n",
      "Using scaled bp penalty: 118830\n",
      "Pair (0,7) has 1 well-supported breakpoints\n",
      "Pair 0, 8 has 261 initial LCBs\n",
      "Using scaled bp penalty: 103283\n",
      "Pair (0,8) has 1 well-supported breakpoints\n",
      "Pair 0, 9 has 246 initial LCBs\n",
      "Using scaled bp penalty: 96709.4\n",
      "Pair (0,9) has 1 well-supported breakpoints\n",
      "Pair 0, 10 has 304 initial LCBs\n",
      "Using scaled bp penalty: 72133.3\n",
      "Pair (0,10) has 1 well-supported breakpoints\n",
      "Pair 0, 11 has 280 initial LCBs\n",
      "Using scaled bp penalty: 97164\n",
      "Pair (0,11) has 2 well-supported breakpoints\n",
      "Pair 0, 12 has 266 initial LCBs\n",
      "Using scaled bp penalty: 134616\n",
      "Pair (0,12) has 1 well-supported breakpoints\n",
      "Pair 0, 13 has 306 initial LCBs\n",
      "Using scaled bp penalty: 113464\n",
      "Pair (0,13) has 2 well-supported breakpoints\n",
      "Pair 0, 14 has 270 initial LCBs\n",
      "Using scaled bp penalty: 97891.4\n",
      "Pair (0,14) has 1 well-supported breakpoints\n",
      "Pair 1, 2 has 1 initial LCBs\n",
      "Using scaled bp penalty: 189788\n",
      "Pair (1,2) has 1 well-supported breakpoints\n",
      "Pair 1, 3 has 78 initial LCBs\n",
      "Using scaled bp penalty: 149897\n",
      "Pair (1,3) has 1 well-supported breakpoints\n",
      "Pair 1, 4 has 274 initial LCBs\n",
      "Using scaled bp penalty: 81957.7\n",
      "Pair (1,4) has 1 well-supported breakpoints\n",
      "Pair 1, 5 has 262 initial LCBs\n",
      "Using scaled bp penalty: 89078.6\n",
      "Pair (1,5) has 1 well-supported breakpoints\n",
      "Pair 1, 6 has 247 initial LCBs\n",
      "Using scaled bp penalty: 98097.8\n",
      "Pair (1,6) has 1 well-supported breakpoints\n",
      "Pair 1, 7 has 270 initial LCBs\n",
      "Using scaled bp penalty: 107030\n",
      "Pair (1,7) has 1 well-supported breakpoints\n",
      "Pair 1, 8 has 89 initial LCBs\n",
      "Using scaled bp penalty: 163253\n",
      "Pair (1,8) has 1 well-supported breakpoints\n",
      "Pair 1, 9 has 93 initial LCBs\n",
      "Using scaled bp penalty: 140977\n",
      "Pair (1,9) has 1 well-supported breakpoints\n",
      "Pair 1, 10 has 107 initial LCBs\n",
      "Using scaled bp penalty: 115129\n",
      "Pair (1,10) has 1 well-supported breakpoints\n",
      "Pair 1, 11 has 255 initial LCBs\n",
      "Using scaled bp penalty: 96183.8\n",
      "Pair (1,11) has 2 well-supported breakpoints\n",
      "Pair 1, 12 has 226 initial LCBs\n",
      "Using scaled bp penalty: 119929\n",
      "Pair (1,12) has 1 well-supported breakpoints\n",
      "Pair 1, 13 has 278 initial LCBs\n",
      "Using scaled bp penalty: 99339.4\n",
      "Pair (1,13) has 2 well-supported breakpoints\n",
      "Pair 1, 14 has 23 initial LCBs\n",
      "Using scaled bp penalty: 186376\n",
      "Pair (1,14) has 1 well-supported breakpoints\n",
      "Pair 2, 3 has 80 initial LCBs\n",
      "Using scaled bp penalty: 152400\n",
      "Pair (2,3) has 1 well-supported breakpoints\n",
      "Pair 2, 4 has 271 initial LCBs\n",
      "Using scaled bp penalty: 81494.1\n",
      "Pair (2,4) has 1 well-supported breakpoints\n",
      "Pair 2, 5 has 268 initial LCBs\n",
      "Using scaled bp penalty: 88189.8\n",
      "Pair (2,5) has 1 well-supported breakpoints\n",
      "Pair 2, 6 has 253 initial LCBs\n",
      "Using scaled bp penalty: 99517.9\n",
      "Pair (2,6) has 1 well-supported breakpoints\n",
      "Pair 2, 7 has 275 initial LCBs\n",
      "Using scaled bp penalty: 107527\n",
      "Pair (2,7) has 1 well-supported breakpoints\n",
      "Pair 2, 8 has 89 initial LCBs\n",
      "Using scaled bp penalty: 162828\n",
      "Pair (2,8) has 1 well-supported breakpoints\n",
      "Pair 2, 9 has 89 initial LCBs\n",
      "Using scaled bp penalty: 140622\n",
      "Pair (2,9) has 1 well-supported breakpoints\n",
      "Pair 2, 10 has 110 initial LCBs\n",
      "Using scaled bp penalty: 115049\n",
      "Pair (2,10) has 1 well-supported breakpoints\n",
      "Pair 2, 11 has 252 initial LCBs\n",
      "Using scaled bp penalty: 96487.4\n",
      "Pair (2,11) has 2 well-supported breakpoints\n",
      "Pair 2, 12 has 230 initial LCBs\n",
      "Using scaled bp penalty: 120828\n",
      "Pair (2,12) has 1 well-supported breakpoints\n",
      "Pair 2, 13 has 281 initial LCBs\n",
      "Using scaled bp penalty: 99763.8\n",
      "Pair (2,13) has 2 well-supported breakpoints\n",
      "Pair 2, 14 has 25 initial LCBs\n",
      "Using scaled bp penalty: 187177\n",
      "Pair (2,14) has 1 well-supported breakpoints\n",
      "Pair 3, 4 has 251 initial LCBs\n",
      "Using scaled bp penalty: 85285.6\n",
      "Pair (3,4) has 1 well-supported breakpoints\n",
      "Pair 3, 5 has 302 initial LCBs\n",
      "Using scaled bp penalty: 89078.4\n",
      "Pair (3,5) has 1 well-supported breakpoints\n",
      "Pair 3, 6 has 276 initial LCBs\n",
      "Using scaled bp penalty: 101770\n",
      "Pair (3,6) has 1 well-supported breakpoints\n",
      "Pair 3, 7 has 274 initial LCBs\n",
      "Using scaled bp penalty: 107143\n",
      "Pair (3,7) has 1 well-supported breakpoints\n",
      "Pair 3, 8 has 147 initial LCBs\n",
      "Using scaled bp penalty: 165478\n",
      "Pair (3,8) has 1 well-supported breakpoints\n",
      "Pair 3, 9 has 134 initial LCBs\n",
      "Using scaled bp penalty: 149286\n",
      "Pair (3,9) has 1 well-supported breakpoints\n",
      "Pair 3, 10 has 145 initial LCBs\n",
      "Using scaled bp penalty: 115663\n",
      "Pair (3,10) has 1 well-supported breakpoints\n",
      "Pair 3, 11 has 242 initial LCBs\n",
      "Using scaled bp penalty: 95756.1\n",
      "Pair (3,11) has 2 well-supported breakpoints\n",
      "Pair 3, 12 has 238 initial LCBs\n",
      "Using scaled bp penalty: 122810\n",
      "Pair (3,12) has 1 well-supported breakpoints\n",
      "Pair 3, 13 has 291 initial LCBs\n",
      "Using scaled bp penalty: 101474\n",
      "Pair (3,13) has 2 well-supported breakpoints\n",
      "Pair 3, 14 has 62 initial LCBs\n",
      "Using scaled bp penalty: 172851\n",
      "Pair (3,14) has 1 well-supported breakpoints\n",
      "Pair 4, 5 has 197 initial LCBs\n",
      "Using scaled bp penalty: 133804\n",
      "Pair (4,5) has 1 well-supported breakpoints\n",
      "Pair 4, 6 has 285 initial LCBs\n",
      "Using scaled bp penalty: 109377\n",
      "Pair (4,6) has 1 well-supported breakpoints\n",
      "Pair 4, 7 has 283 initial LCBs\n",
      "Using scaled bp penalty: 110314\n",
      "Pair (4,7) has 1 well-supported breakpoints\n",
      "Pair 4, 8 has 247 initial LCBs\n",
      "Using scaled bp penalty: 96755.6\n",
      "Pair (4,8) has 1 well-supported breakpoints\n",
      "Pair 4, 9 has 225 initial LCBs\n",
      "Using scaled bp penalty: 90004.9\n",
      "Pair (4,9) has 1 well-supported breakpoints\n",
      "Pair 4, 10 has 303 initial LCBs\n",
      "Using scaled bp penalty: 69825.5\n",
      "Pair (4,10) has 1 well-supported breakpoints\n",
      "Pair 4, 11 has 252 initial LCBs\n",
      "Using scaled bp penalty: 92106.5\n",
      "Pair (4,11) has 2 well-supported breakpoints\n",
      "Pair 4, 12 has 252 initial LCBs\n",
      "Using scaled bp penalty: 123883\n",
      "Pair (4,12) has 1 well-supported breakpoints\n",
      "Pair 4, 13 has 272 initial LCBs\n",
      "Using scaled bp penalty: 108685\n",
      "Pair (4,13) has 2 well-supported breakpoints\n",
      "Pair 4, 14 has 251 initial LCBs\n",
      "Using scaled bp penalty: 93450.7\n",
      "Pair (4,14) has 1 well-supported breakpoints\n",
      "Pair 5, 6 has 292 initial LCBs\n",
      "Using scaled bp penalty: 115061\n",
      "Pair (5,6) has 1 well-supported breakpoints\n",
      "Pair 5, 7 has 285 initial LCBs\n",
      "Using scaled bp penalty: 117678\n",
      "Pair (5,7) has 1 well-supported breakpoints\n",
      "Pair 5, 8 has 231 initial LCBs\n",
      "Using scaled bp penalty: 96766.2\n",
      "Pair (5,8) has 1 well-supported breakpoints\n",
      "Pair 5, 9 has 216 initial LCBs\n",
      "Using scaled bp penalty: 89360.1\n",
      "Pair (5,9) has 1 well-supported breakpoints\n",
      "Pair 5, 10 has 288 initial LCBs\n",
      "Using scaled bp penalty: 71712.2\n",
      "Pair (5,10) has 1 well-supported breakpoints\n",
      "Pair 5, 11 has 239 initial LCBs\n",
      "Using scaled bp penalty: 92340.9\n",
      "Pair (5,11) has 2 well-supported breakpoints\n",
      "Pair 5, 12 has 241 initial LCBs\n",
      "Using scaled bp penalty: 132870\n",
      "Pair (5,12) has 1 well-supported breakpoints\n",
      "Pair 5, 13 has 291 initial LCBs\n",
      "Using scaled bp penalty: 111963\n",
      "Pair (5,13) has 2 well-supported breakpoints\n",
      "Pair 5, 14 has 262 initial LCBs\n",
      "Using scaled bp penalty: 97109.8\n",
      "Pair (5,14) has 1 well-supported breakpoints\n",
      "Pair 6, 7 has 49 initial LCBs\n",
      "Using scaled bp penalty: 198682\n",
      "Pair (6,7) has 1 well-supported breakpoints\n",
      "Pair 6, 8 has 214 initial LCBs\n",
      "Using scaled bp penalty: 114280\n",
      "Pair (6,8) has 1 well-supported breakpoints\n",
      "Pair 6, 9 has 213 initial LCBs\n",
      "Using scaled bp penalty: 102158\n",
      "Pair (6,9) has 1 well-supported breakpoints\n",
      "Pair 6, 10 has 303 initial LCBs\n",
      "Using scaled bp penalty: 83562.5\n",
      "Pair (6,10) has 1 well-supported breakpoints\n",
      "Pair 6, 11 has 227 initial LCBs\n",
      "Using scaled bp penalty: 114931\n",
      "Pair (6,11) has 2 well-supported breakpoints\n",
      "Pair 6, 12 has 31 initial LCBs\n",
      "Using scaled bp penalty: 204131\n",
      "Pair (6,12) has 1 well-supported breakpoints\n",
      "Pair 6, 13 has 38 initial LCBs\n",
      "Using scaled bp penalty: 196440\n",
      "Pair (6,13) has 2 well-supported breakpoints\n",
      "Pair 6, 14 has 221 initial LCBs\n",
      "Using scaled bp penalty: 106796\n",
      "Pair (6,14) has 1 well-supported breakpoints\n",
      "Pair 7, 8 has 225 initial LCBs\n",
      "Using scaled bp penalty: 125589\n",
      "Pair (7,8) has 1 well-supported breakpoints\n",
      "Pair 7, 9 has 208 initial LCBs\n",
      "Using scaled bp penalty: 114767\n",
      "Pair (7,9) has 1 well-supported breakpoints\n",
      "Pair 7, 10 has 267 initial LCBs\n",
      "Using scaled bp penalty: 86690.2\n",
      "Pair (7,10) has 1 well-supported breakpoints\n",
      "Pair 7, 11 has 211 initial LCBs\n",
      "Using scaled bp penalty: 114664\n",
      "Pair (7,11) has 2 well-supported breakpoints\n",
      "Pair 7, 12 has 28 initial LCBs\n",
      "Using scaled bp penalty: 230375\n",
      "Pair (7,12) has 1 well-supported breakpoints\n",
      "Pair 7, 13 has 20 initial LCBs\n",
      "Using scaled bp penalty: 239661\n",
      "Pair (7,13) has 2 well-supported breakpoints\n",
      "Pair 7, 14 has 199 initial LCBs\n",
      "Using scaled bp penalty: 119050\n",
      "Pair (7,14) has 1 well-supported breakpoints\n",
      "Pair 8, 9 has 9 initial LCBs\n",
      "Using scaled bp penalty: 234601\n",
      "Pair (8,9) has 1 well-supported breakpoints\n",
      "Pair 8, 10 has 82 initial LCBs\n",
      "Using scaled bp penalty: 142849\n",
      "Pair (8,10) has 1 well-supported breakpoints\n",
      "Pair 8, 11 has 167 initial LCBs\n",
      "Using scaled bp penalty: 109988\n",
      "Pair (8,11) has 2 well-supported breakpoints\n",
      "Pair 8, 12 has 215 initial LCBs\n",
      "Using scaled bp penalty: 142826\n",
      "Pair (8,12) has 1 well-supported breakpoints\n",
      "Pair 8, 13 has 236 initial LCBs\n",
      "Using scaled bp penalty: 117616\n",
      "Pair (8,13) has 2 well-supported breakpoints\n",
      "Pair 8, 14 has 87 initial LCBs\n",
      "Using scaled bp penalty: 187268\n",
      "Pair (8,14) has 1 well-supported breakpoints\n",
      "Pair 9, 10 has 67 initial LCBs\n",
      "Using scaled bp penalty: 127388\n",
      "Pair (9,10) has 1 well-supported breakpoints\n",
      "Pair 9, 11 has 146 initial LCBs\n",
      "Using scaled bp penalty: 102588\n",
      "Pair (9,11) has 2 well-supported breakpoints\n",
      "Pair 9, 12 has 203 initial LCBs\n",
      "Using scaled bp penalty: 134942\n",
      "Pair (9,12) has 1 well-supported breakpoints\n",
      "Pair 9, 13 has 202 initial LCBs\n",
      "Using scaled bp penalty: 107261\n",
      "Pair (9,13) has 2 well-supported breakpoints\n",
      "Pair 9, 14 has 62 initial LCBs\n",
      "Using scaled bp penalty: 176055\n",
      "Pair (9,14) has 1 well-supported breakpoints\n",
      "Pair 10, 11 has 242 initial LCBs\n",
      "Using scaled bp penalty: 75479.1\n",
      "Pair (10,11) has 2 well-supported breakpoints\n",
      "Pair 10, 12 has 267 initial LCBs\n",
      "Using scaled bp penalty: 101577\n",
      "Pair (10,12) has 1 well-supported breakpoints\n",
      "Pair 10, 13 has 281 initial LCBs\n",
      "Using scaled bp penalty: 83857.6\n",
      "Pair (10,13) has 2 well-supported breakpoints\n",
      "Pair 10, 14 has 119 initial LCBs\n",
      "Using scaled bp penalty: 125396\n",
      "Pair (10,14) has 1 well-supported breakpoints\n",
      "Pair 11, 12 has 198 initial LCBs\n",
      "Using scaled bp penalty: 135963\n",
      "Pair (11,12) has 2 well-supported breakpoints\n",
      "Pair 11, 13 has 229 initial LCBs\n",
      "Using scaled bp penalty: 116560\n",
      "Pair (11,13) has 1 well-supported breakpoints\n",
      "Pair 11, 14 has 222 initial LCBs\n",
      "Using scaled bp penalty: 108680\n",
      "Pair (11,14) has 2 well-supported breakpoints\n",
      "Pair 12, 13 has 27 initial LCBs\n",
      "Using scaled bp penalty: 203301\n",
      "Pair (12,13) has 2 well-supported breakpoints\n",
      "Pair 12, 14 has 178 initial LCBs\n",
      "Using scaled bp penalty: 141209\n",
      "Pair (12,14) has 1 well-supported breakpoints\n",
      "Pair 13, 14 has 215 initial LCBs\n",
      "Using scaled bp penalty: 106957\n",
      "Pair (13,14) has 2 well-supported breakpoints\n",
      "genome content distance matrix:\n",
      "0\t0.262523\t0.260428\t0.260135\t0.253856\t0.244286\t0.20789\t0.204892\t0.232281\t0.2448\t0.298175\t0.243914\t0.179707\t0.214024\t0.242503\n",
      "0.262523\t0\t0.106156\t0.157359\t0.275411\t0.26016\t0.242104\t0.225412\t0.139186\t0.170185\t0.211157\t0.245829\t0.20306\t0.239718\t0.110201\n",
      "0.260428\t0.106156\t0\t0.153863\t0.276437\t0.262012\t0.239376\t0.224515\t0.139746\t0.170707\t0.211294\t0.245234\t0.20157\t0.238907\t0.109246\n",
      "0.260135\t0.157359\t0.153863\t0\t0.268164\t0.26016\t0.235108\t0.225208\t0.136267\t0.158219\t0.210244\t0.246668\t0.198317\t0.235665\t0.126802\n",
      "0.253856\t0.275411\t0.276437\t0.268164\t0\t0.180947\t0.221199\t0.219536\t0.24471\t0.258244\t0.303858\t0.253951\t0.19657\t0.222434\t0.251244\n",
      "0.244286\t0.26016\t0.262012\t0.26016\t0.180947\t0\t0.211273\t0.206825\t0.24469\t0.259576\t0.299202\t0.253477\t0.18238\t0.216637\t0.24402\n",
      "0.20789\t0.242104\t0.239376\t0.235108\t0.221199\t0.211273\t0\t0.0958634\t0.212614\t0.234381\t0.271889\t0.211496\t0.0897266\t0.0984249\t0.225835\n",
      "0.204892\t0.225412\t0.224515\t0.225208\t0.219536\t0.206825\t0.0958634\t0\t0.19382\t0.211777\t0.26517\t0.211955\t0.0617824\t0.052468\t0.204525\n",
      "0.232281\t0.139186\t0.139746\t0.136267\t0.24471\t0.24469\t0.212614\t0.19382\t0\t0.0575095\t0.167443\t0.220114\t0.167477\t0.20693\t0.109138\n",
      "0.2448\t0.170185\t0.170707\t0.158219\t0.258244\t0.259576\t0.234381\t0.211777\t0.0575095\t0\t0.190947\t0.233577\t0.179211\t0.224994\t0.122783\n",
      "0.298175\t0.211157\t0.211294\t0.210244\t0.303858\t0.299202\t0.271889\t0.26517\t0.167443\t0.190947\t0\t0.290175\t0.235472\t0.271247\t0.194128\n",
      "0.243914\t0.245829\t0.245234\t0.246668\t0.253951\t0.253477\t0.211496\t0.211955\t0.220114\t0.233577\t0.290175\t0\t0.177663\t0.208717\t0.222444\n",
      "0.179707\t0.20306\t0.20157\t0.198317\t0.19657\t0.18238\t0.0897266\t0.0617824\t0.167477\t0.179211\t0.235472\t0.177663\t0\t0.090653\t0.169843\n",
      "0.214024\t0.239718\t0.238907\t0.235665\t0.222434\t0.216637\t0.0984249\t0.052468\t0.20693\t0.224994\t0.271247\t0.208717\t0.090653\t0\t0.225544\n",
      "0.242503\t0.110201\t0.109246\t0.126802\t0.251244\t0.24402\t0.225835\t0.204525\t0.109138\t0.122783\t0.194128\t0.222444\t0.169843\t0.225544\t0\n",
      "\n",
      "bp distance matrix:\n",
      "0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.45\t0.9\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.9\t0.45\t0.9\t0.45\n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.9\t0.9\t0.45\n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.45\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.9\t0.45\t0.9\t0.9\n",
      "\n",
      "Aligning...\n",
      "Aligning node 21 to 22 via 20!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 89 tracking matches\n",
      "There are 178 / 2581 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 22 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.052468\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "4%..9%..\n",
      "13%..18%..\n",
      "22%..27%..\n",
      "31%..36%..\n",
      "40%..45%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(21) has 1 intervals\n",
      "descendant 1(22) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 2 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.02579e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 10 new matches outside LCBs\n",
      "0,0 has an additional 21 matches\n",
      "Restoring backed up alignment tree...\n",
      "13,7 has 21 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 87 tracking matches\n",
      "There are 174 / 2523 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 15 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.052468\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "6%..\n",
      "13%..\n",
      "20%..26%..\n",
      "33%..\n",
      "40%..46%..\n",
      "53%..\n",
      "60%..66%..\n",
      "73%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(21) has 1 intervals\n",
      "descendant 1(22) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 2 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.02579e+06, new anchor score: 1.08854e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 10 new matches outside LCBs\n",
      "0,0 has an additional 19 matches\n",
      "Restoring backed up alignment tree...\n",
      "13,7 has 19 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 87 tracking matches\n",
      "There are 174 / 2523 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 15 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.052468\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "6%..\n",
      "13%..\n",
      "20%..26%..\n",
      "33%..\n",
      "40%..46%..\n",
      "53%..\n",
      "60%..66%..\n",
      "73%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(21) has 1 intervals\n",
      "descendant 1(22) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 2 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.08854e+06, new anchor score: 1.08854e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "40%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 7 to 8 via 6!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 43 tracking matches\n",
      "There are 86 / 1247 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 11 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0575095\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 23655.6\n",
      "9%..\n",
      "18%..\n",
      "27%..\n",
      "36%..\n",
      "45%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 4 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(7) has 1 intervals\n",
      "descendant 1(8) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.09108e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 6 new matches outside LCBs\n",
      "0,0 has an additional 6 matches\n",
      "Restoring backed up alignment tree...\n",
      "9,8 has 6 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 43 tracking matches\n",
      "There are 86 / 1247 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 5 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0575095\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 23655.6\n",
      "\n",
      "20%..\n",
      "40%..\n",
      "60%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 4 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(7) has 1 intervals\n",
      "descendant 1(8) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.09108e+06, new anchor score: 1.09108e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 20 to 23 via 19!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 454 tracking matches\n",
      "There are 1790 / 13166 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 127 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0958634\n",
      "0.0984249\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20033.7\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..16%..17%..18%..19%..\n",
      "20%..21%..22%..23%..24%..25%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(20) has 2 intervals\n",
      "descendant 1(23) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.72299e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 18 new matches outside LCBs\n",
      "1,0 have 6 new matches outside LCBs\n",
      "0,0 has an additional 36 matches\n",
      "1,0 has an additional 23 matches\n",
      "Restoring backed up alignment tree...\n",
      "7,6 has 36 pairwise matches\n",
      "13,6 has 23 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 439 tracking matches\n",
      "There are 1738 / 12731 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 52 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0958634\n",
      "0.0984249\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20033.7\n",
      "1%..3%..5%..7%..9%..\n",
      "11%..13%..15%..17%..19%..\n",
      "21%..23%..25%..26%..28%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(20) has 2 intervals\n",
      "descendant 1(23) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.72299e+06, new anchor score: 1.81454e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 18 new matches outside LCBs\n",
      "1,0 have 6 new matches outside LCBs\n",
      "0,0 has an additional 30 matches\n",
      "1,0 has an additional 14 matches\n",
      "Restoring backed up alignment tree...\n",
      "7,6 has 30 pairwise matches\n",
      "13,6 has 14 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 440 tracking matches\n",
      "There are 1742 / 12760 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 52 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0958634\n",
      "0.0984249\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20033.7\n",
      "1%..3%..5%..7%..9%..\n",
      "11%..13%..15%..17%..19%..\n",
      "21%..23%..25%..26%..28%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(20) has 2 intervals\n",
      "descendant 1(23) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.81454e+06, new anchor score: 1.79962e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "34%..\n",
      "94%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 19 to 24 via 18!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 355 tracking matches\n",
      "There are 2115 / 10295 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 111 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0897266\n",
      "0.0617824\n",
      "0.090653\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20583.2\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..16%..17%..18%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 4 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(19) has 3 intervals\n",
      "descendant 1(24) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 2.44706e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 2 new matches outside LCBs\n",
      "1,0 have 2 new matches outside LCBs\n",
      "2,0 have 1 new matches outside LCBs\n",
      "0,0 has an additional 5 matches\n",
      "1,0 has an additional 5 matches\n",
      "2,0 has an additional 5 matches\n",
      "Restoring backed up alignment tree...\n",
      "6,12 has 5 pairwise matches\n",
      "7,12 has 5 pairwise matches\n",
      "13,12 has 5 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 340 tracking matches\n",
      "There are 2036 / 9860 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 4 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0897266\n",
      "0.0617824\n",
      "0.090653\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20583.2\n",
      "done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 3 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(19) has 3 intervals\n",
      "descendant 1(24) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 2.44706e+06, new anchor score: 2.77146e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 0 new matches outside LCBs\n",
      "1,0 have 0 new matches outside LCBs\n",
      "2,0 have 0 new matches outside LCBs\n",
      "Restoring backed up alignment tree...\n",
      "6,12 has 0 pairwise matches\n",
      "7,12 has 0 pairwise matches\n",
      "13,12 has 0 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 340 tracking matches\n",
      "There are 2036 / 9860 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 4 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.0897266\n",
      "0.0617824\n",
      "0.090653\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 20583.2\n",
      "done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 3 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(19) has 3 intervals\n",
      "descendant 1(24) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 2.77146e+06, new anchor score: 2.77146e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "20%..\n",
      "37%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "performing iterative refinement\n",
      "\n",
      "0%..\n",
      "38%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 13 to 14 via 12!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 4 tracking matches\n",
      "There are 8 / 116 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 3 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.106156\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 19137\n",
      "\n",
      "33%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(13) has 1 intervals\n",
      "descendant 1(14) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.27654e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 2 new matches outside LCBs\n",
      "0,0 has an additional 3 matches\n",
      "Restoring backed up alignment tree...\n",
      "2,1 has 3 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 5 tracking matches\n",
      "There are 10 / 145 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 2 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.106156\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 19137\n",
      "\n",
      "50%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(13) has 1 intervals\n",
      "descendant 1(14) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.27654e+06, new anchor score: 1.15656e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 12 to 15 via 11!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 174 tracking matches\n",
      "There are 696 / 5046 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 50 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.110201\n",
      "0.109246\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 18792.9\n",
      "2%..4%..6%..8%..\n",
      "10%..12%..14%..16%..18%..\n",
      "20%..22%..24%..26%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(12) has 1 intervals\n",
      "descendant 1(15) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.79825e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 2 new matches outside LCBs\n",
      "1,0 have 2 new matches outside LCBs\n",
      "0,0 has an additional 9 matches\n",
      "1,0 has an additional 10 matches\n",
      "Restoring backed up alignment tree...\n",
      "1,14 has 9 pairwise matches\n",
      "2,14 has 10 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 171 tracking matches\n",
      "There are 684 / 4959 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 8 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.110201\n",
      "0.109246\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 18792.9\n",
      "\n",
      "12%..\n",
      "25%..\n",
      "37%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(12) has 1 intervals\n",
      "descendant 1(15) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.79825e+06, new anchor score: 2.1964e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 2 new matches outside LCBs\n",
      "1,0 have 2 new matches outside LCBs\n",
      "0,0 has an additional 2 matches\n",
      "1,0 has an additional 3 matches\n",
      "Restoring backed up alignment tree...\n",
      "1,14 has 2 pairwise matches\n",
      "2,14 has 3 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 171 tracking matches\n",
      "There are 684 / 4959 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 8 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.110201\n",
      "0.109246\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 18792.9\n",
      "\n",
      "12%..\n",
      "25%..\n",
      "37%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(12) has 1 intervals\n",
      "descendant 1(15) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 2.1964e+06, new anchor score: 2.1964e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 11 to 16 via 10!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 457 tracking matches\n",
      "There are 2711 / 13253 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 232 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.126802\n",
      "0.157359\n",
      "0.153863\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 17429.2\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(11) has 1 intervals\n",
      "descendant 1(16) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 2.55153e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 3 new matches outside LCBs\n",
      "1,0 have 10 new matches outside LCBs\n",
      "2,0 have 12 new matches outside LCBs\n",
      "0,0 has an additional 5 matches\n",
      "1,0 has an additional 12 matches\n",
      "2,0 has an additional 15 matches\n",
      "Restoring backed up alignment tree...\n",
      "14,3 has 5 pairwise matches\n",
      "1,3 has 12 pairwise matches\n",
      "2,3 has 15 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 416 tracking matches\n",
      "There are 2485 / 12064 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 18 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.126802\n",
      "0.157359\n",
      "0.153863\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 17429.2\n",
      "5%..\n",
      "11%..16%..\n",
      "22%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(11) has 1 intervals\n",
      "descendant 1(16) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 2.55153e+06, new anchor score: 2.55462e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "performing iterative refinement\n",
      "\n",
      "0%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 6 to 9 via 5!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 375 tracking matches\n",
      "There are 1468 / 10875 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 140 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.167443\n",
      "0.190947\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 14403.9\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..16%..17%..18%..19%..\n",
      "20%..21%..22%..23%..24%..25%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(6) has 1 intervals\n",
      "descendant 1(9) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.44573e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 7 new matches outside LCBs\n",
      "1,0 have 5 new matches outside LCBs\n",
      "0,0 has an additional 17 matches\n",
      "1,0 has an additional 7 matches\n",
      "Restoring backed up alignment tree...\n",
      "8,10 has 17 pairwise matches\n",
      "9,10 has 7 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 352 tracking matches\n",
      "There are 1376 / 10208 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 19 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.167443\n",
      "0.190947\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 14403.9\n",
      "5%..\n",
      "10%..15%..\n",
      "21%..26%..\n",
      "31%..36%..\n",
      "42%..47%..\n",
      "52%..57%..\n",
      "63%..68%..\n",
      "73%..78%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(6) has 1 intervals\n",
      "descendant 1(9) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.44573e+06, new anchor score: 1.44605e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "25%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 5 to 10 via 4!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 843 tracking matches\n",
      "There are 9788 / 24447 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 2103 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.210244\t0.194128\t0.211157\t0.211294\n",
      "0.136267\t0.109138\t0.139186\t0.139746\n",
      "0.158219\t0.122783\t0.170185\t0.170707\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 11662.7\n",
      "1%..2%..3%..4%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(5) has 1 intervals\n",
      "descendant 1(10) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 8.19382e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 12 new matches outside LCBs\n",
      "0,1 have 1 new matches outside LCBs\n",
      "0,2 have 7 new matches outside LCBs\n",
      "0,3 have 9 new matches outside LCBs\n",
      "1,0 have 7 new matches outside LCBs\n",
      "1,1 have 21 new matches outside LCBs\n",
      "1,2 have 0 new matches outside LCBs\n",
      "1,3 have 0 new matches outside LCBs\n",
      "2,0 have 5 new matches outside LCBs\n",
      "2,1 have 8 new matches outside LCBs\n",
      "2,2 have 5 new matches outside LCBs\n",
      "2,3 have 5 new matches outside LCBs\n",
      "0,0 has an additional 24 matches\n",
      "0,1 has an additional 2 matches\n",
      "0,2 has an additional 9 matches\n",
      "0,3 has an additional 13 matches\n",
      "1,0 has an additional 8 matches\n",
      "1,1 has an additional 40 matches\n",
      "1,2 has an additional 1 matches\n",
      "1,3 has an additional 1 matches\n",
      "2,0 has an additional 9 matches\n",
      "2,1 has an additional 10 matches\n",
      "2,2 has an additional 6 matches\n",
      "2,3 has an additional 6 matches\n",
      "Restoring backed up alignment tree...\n",
      "10,3 has 24 pairwise matches\n",
      "10,14 has 2 pairwise matches\n",
      "10,1 has 9 pairwise matches\n",
      "10,2 has 13 pairwise matches\n",
      "8,3 has 8 pairwise matches\n",
      "8,14 has 40 pairwise matches\n",
      "8,1 has 1 pairwise matches\n",
      "8,2 has 1 pairwise matches\n",
      "9,3 has 9 pairwise matches\n",
      "9,14 has 10 pairwise matches\n",
      "9,1 has 6 pairwise matches\n",
      "9,2 has 6 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 733 tracking matches\n",
      "There are 8621 / 21257 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 187 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.210244\t0.194128\t0.211157\t0.211294\n",
      "0.136267\t0.109138\t0.139186\t0.139746\n",
      "0.158219\t0.122783\t0.170185\t0.170707\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 11662.7\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(5) has 1 intervals\n",
      "descendant 1(10) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 8.19382e+06, new anchor score: 8.2741e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 17 new matches outside LCBs\n",
      "0,1 have 1 new matches outside LCBs\n",
      "0,2 have 5 new matches outside LCBs\n",
      "0,3 have 6 new matches outside LCBs\n",
      "1,0 have 15 new matches outside LCBs\n",
      "1,1 have 21 new matches outside LCBs\n",
      "1,2 have 12 new matches outside LCBs\n",
      "1,3 have 12 new matches outside LCBs\n",
      "2,0 have 5 new matches outside LCBs\n",
      "2,1 have 8 new matches outside LCBs\n",
      "2,2 have 5 new matches outside LCBs\n",
      "2,3 have 5 new matches outside LCBs\n",
      "0,0 has an additional 32 matches\n",
      "0,1 has an additional 1 matches\n",
      "0,2 has an additional 6 matches\n",
      "0,3 has an additional 6 matches\n",
      "1,0 has an additional 31 matches\n",
      "1,1 has an additional 39 matches\n",
      "1,2 has an additional 24 matches\n",
      "1,3 has an additional 24 matches\n",
      "2,0 has an additional 8 matches\n",
      "2,1 has an additional 9 matches\n",
      "2,2 has an additional 5 matches\n",
      "2,3 has an additional 5 matches\n",
      "Restoring backed up alignment tree...\n",
      "10,3 has 32 pairwise matches\n",
      "10,14 has 1 pairwise matches\n",
      "10,1 has 6 pairwise matches\n",
      "10,2 has 6 pairwise matches\n",
      "8,3 has 31 pairwise matches\n",
      "8,14 has 39 pairwise matches\n",
      "8,1 has 24 pairwise matches\n",
      "8,2 has 24 pairwise matches\n",
      "9,3 has 8 pairwise matches\n",
      "9,14 has 9 pairwise matches\n",
      "9,1 has 5 pairwise matches\n",
      "9,2 has 5 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 734 tracking matches\n",
      "There are 8632 / 21286 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 140 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.210244\t0.194128\t0.211157\t0.211294\n",
      "0.136267\t0.109138\t0.139186\t0.139746\n",
      "0.158219\t0.122783\t0.170185\t0.170707\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 11662.7\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(5) has 1 intervals\n",
      "descendant 1(10) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 8.2741e+06, new anchor score: 8.42381e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 0 new matches outside LCBs\n",
      "0,1 have 1 new matches outside LCBs\n",
      "0,2 have 5 new matches outside LCBs\n",
      "0,3 have 6 new matches outside LCBs\n",
      "1,0 have 7 new matches outside LCBs\n",
      "1,1 have 21 new matches outside LCBs\n",
      "1,2 have 5 new matches outside LCBs\n",
      "1,3 have 5 new matches outside LCBs\n",
      "2,0 have 6 new matches outside LCBs\n",
      "2,1 have 8 new matches outside LCBs\n",
      "2,2 have 5 new matches outside LCBs\n",
      "2,3 have 5 new matches outside LCBs\n",
      "0,0 has an additional 3 matches\n",
      "0,1 has an additional 1 matches\n",
      "0,2 has an additional 12 matches\n",
      "0,3 has an additional 12 matches\n",
      "1,0 has an additional 7 matches\n",
      "1,1 has an additional 39 matches\n",
      "1,2 has an additional 11 matches\n",
      "1,3 has an additional 12 matches\n",
      "2,0 has an additional 6 matches\n",
      "2,1 has an additional 9 matches\n",
      "2,2 has an additional 5 matches\n",
      "2,3 has an additional 5 matches\n",
      "Restoring backed up alignment tree...\n",
      "10,3 has 3 pairwise matches\n",
      "10,14 has 1 pairwise matches\n",
      "10,1 has 12 pairwise matches\n",
      "10,2 has 12 pairwise matches\n",
      "8,3 has 7 pairwise matches\n",
      "8,14 has 39 pairwise matches\n",
      "8,1 has 11 pairwise matches\n",
      "8,2 has 12 pairwise matches\n",
      "9,3 has 6 pairwise matches\n",
      "9,14 has 9 pairwise matches\n",
      "9,1 has 5 pairwise matches\n",
      "9,2 has 5 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 730 tracking matches\n",
      "There are 8594 / 21170 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 118 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.210244\t0.194128\t0.211157\t0.211294\n",
      "0.136267\t0.109138\t0.139186\t0.139746\n",
      "0.158219\t0.122783\t0.170185\t0.170707\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 11662.7\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(5) has 1 intervals\n",
      "descendant 1(10) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 8.42381e+06, new anchor score: 8.42516e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "12%..\n",
      "25%..\n",
      "37%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "performing iterative refinement\n",
      "\n",
      "0%..\n",
      "25%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 26 to 27 via 25!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 446 tracking matches\n",
      "There are 892 / 12934 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 190 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.180947\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13491.9\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..16%..17%..18%..19%..\n",
      "20%..21%..22%..23%..24%..25%..26%..27%..28%..29%..\n",
      "30%..31%..32%..33%..34%..35%..36%..37%..38%..39%..\n",
      "40%..41%..42%..43%..44%..45%..46%..47%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(26) has 1 intervals\n",
      "descendant 1(27) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 677564\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 9 new matches outside LCBs\n",
      "0,0 has an additional 26 matches\n",
      "Restoring backed up alignment tree...\n",
      "5,4 has 26 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 378 tracking matches\n",
      "There are 756 / 10962 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 21 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.180947\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13491.9\n",
      "4%..9%..\n",
      "14%..19%..\n",
      "23%..28%..\n",
      "33%..38%..\n",
      "42%..47%..\n",
      "52%..57%..\n",
      "61%..66%..\n",
      "71%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(26) has 1 intervals\n",
      "descendant 1(27) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 677564, new anchor score: 698354\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 9 new matches outside LCBs\n",
      "0,0 has an additional 11 matches\n",
      "Restoring backed up alignment tree...\n",
      "5,4 has 11 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 370 tracking matches\n",
      "There are 740 / 10730 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 10 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.180947\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13491.9\n",
      "\n",
      "10%..\n",
      "20%..\n",
      "30%..\n",
      "40%..\n",
      "50%..\n",
      "60%..\n",
      "70%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(26) has 1 intervals\n",
      "descendant 1(27) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 698354, new anchor score: 698354\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 25 to 28 via 1!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 733 tracking matches\n",
      "There are 2917 / 21257 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 820 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.253856\n",
      "0.244286\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 9292.17\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..16%..17%..18%..19%..\n",
      "20%..21%..22%..23%..24%..25%..26%..27%..28%..29%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(25) has 1 intervals\n",
      "descendant 1(28) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 846324\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 4 new matches outside LCBs\n",
      "1,0 have 8 new matches outside LCBs\n",
      "0,0 has an additional 36 matches\n",
      "1,0 has an additional 44 matches\n",
      "Restoring backed up alignment tree...\n",
      "4,0 has 36 pairwise matches\n",
      "5,0 has 44 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 507 tracking matches\n",
      "There are 2019 / 14703 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 84 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.253856\n",
      "0.244286\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 9292.17\n",
      "1%..2%..3%..4%..5%..7%..8%..9%..\n",
      "10%..11%..13%..14%..15%..16%..17%..19%..\n",
      "20%..21%..22%..23%..25%..26%..27%..28%..29%..\n",
      "30%..32%..33%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(25) has 1 intervals\n",
      "descendant 1(28) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 846324, new anchor score: 874337\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 1 new matches outside LCBs\n",
      "1,0 have 1 new matches outside LCBs\n",
      "0,0 has an additional 17 matches\n",
      "1,0 has an additional 17 matches\n",
      "Restoring backed up alignment tree...\n",
      "4,0 has 17 pairwise matches\n",
      "5,0 has 17 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 493 tracking matches\n",
      "There are 1967 / 14297 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 43 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.253856\n",
      "0.244286\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\n",
      "0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 9292.17\n",
      "2%..4%..6%..9%..\n",
      "11%..13%..16%..18%..\n",
      "20%..23%..25%..27%..\n",
      "30%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(25) has 1 intervals\n",
      "descendant 1(28) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 1 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 874337, new anchor score: 870914\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "50%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 1 to 18 via 2!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 1170 tracking matches\n",
      "There are 13805 / 33930 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 6796 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.179707\t0.20789\t0.204892\t0.214024\n",
      "0.19657\t0.221199\t0.219536\t0.222434\n",
      "0.18238\t0.211273\t0.206825\t0.216637\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13573.8\n",
      "1%..2%..3%..4%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(1) has 1 intervals\n",
      "descendant 1(18) has 3 intervals\n",
      "propagateDescendantBreakpoints yields 4 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 5.04906e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 12 new matches outside LCBs\n",
      "0,1 have 15 new matches outside LCBs\n",
      "0,2 have 19 new matches outside LCBs\n",
      "0,3 have 9 new matches outside LCBs\n",
      "1,0 have 10 new matches outside LCBs\n",
      "1,1 have 11 new matches outside LCBs\n",
      "1,2 have 13 new matches outside LCBs\n",
      "1,3 have 6 new matches outside LCBs\n",
      "2,0 have 11 new matches outside LCBs\n",
      "2,1 have 24 new matches outside LCBs\n",
      "2,2 have 18 new matches outside LCBs\n",
      "2,3 have 14 new matches outside LCBs\n",
      "0,0 has an additional 20 matches\n",
      "0,1 has an additional 38 matches\n",
      "0,2 has an additional 36 matches\n",
      "0,3 has an additional 32 matches\n",
      "1,0 has an additional 19 matches\n",
      "1,1 has an additional 32 matches\n",
      "1,2 has an additional 30 matches\n",
      "1,3 has an additional 42 matches\n",
      "2,0 has an additional 17 matches\n",
      "2,1 has an additional 48 matches\n",
      "2,2 has an additional 38 matches\n",
      "2,3 has an additional 45 matches\n",
      "Restoring backed up alignment tree...\n",
      "0,12 has 20 pairwise matches\n",
      "0,6 has 38 pairwise matches\n",
      "0,7 has 36 pairwise matches\n",
      "0,13 has 32 pairwise matches\n",
      "4,12 has 19 pairwise matches\n",
      "4,6 has 32 pairwise matches\n",
      "4,7 has 30 pairwise matches\n",
      "4,13 has 42 pairwise matches\n",
      "5,12 has 17 pairwise matches\n",
      "5,6 has 48 pairwise matches\n",
      "5,7 has 38 pairwise matches\n",
      "5,13 has 45 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 807 tracking matches\n",
      "There are 9566 / 23403 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 439 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.179707\t0.20789\t0.204892\t0.214024\n",
      "0.19657\t0.221199\t0.219536\t0.222434\n",
      "0.18238\t0.211273\t0.206825\t0.216637\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13573.8\n",
      "1%..2%..3%..4%..5%..6%..7%..done\n",
      "Arrived at 1 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(1) has 1 intervals\n",
      "descendant 1(18) has 3 intervals\n",
      "propagateDescendantBreakpoints yields 3 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 5.04906e+06, new anchor score: 5.34162e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 0 new matches outside LCBs\n",
      "0,1 have 22 new matches outside LCBs\n",
      "0,2 have 11 new matches outside LCBs\n",
      "0,3 have 20 new matches outside LCBs\n",
      "1,0 have 0 new matches outside LCBs\n",
      "1,1 have 21 new matches outside LCBs\n",
      "1,2 have 12 new matches outside LCBs\n",
      "1,3 have 17 new matches outside LCBs\n",
      "2,0 have 0 new matches outside LCBs\n",
      "2,1 have 33 new matches outside LCBs\n",
      "2,2 have 12 new matches outside LCBs\n",
      "2,3 have 25 new matches outside LCBs\n",
      "0,1 has an additional 36 matches\n",
      "0,2 has an additional 14 matches\n",
      "0,3 has an additional 28 matches\n",
      "1,1 has an additional 39 matches\n",
      "1,2 has an additional 19 matches\n",
      "1,3 has an additional 35 matches\n",
      "2,1 has an additional 44 matches\n",
      "2,2 has an additional 19 matches\n",
      "2,3 has an additional 36 matches\n",
      "Restoring backed up alignment tree...\n",
      "0,12 has 0 pairwise matches\n",
      "0,6 has 36 pairwise matches\n",
      "0,7 has 14 pairwise matches\n",
      "0,13 has 28 pairwise matches\n",
      "4,12 has 0 pairwise matches\n",
      "4,6 has 39 pairwise matches\n",
      "4,7 has 19 pairwise matches\n",
      "4,13 has 35 pairwise matches\n",
      "5,12 has 0 pairwise matches\n",
      "5,6 has 44 pairwise matches\n",
      "5,7 has 19 pairwise matches\n",
      "5,13 has 36 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 799 tracking matches\n",
      "There are 9468 / 23171 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 326 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.179707\t0.20789\t0.204892\t0.214024\n",
      "0.19657\t0.221199\t0.219536\t0.222434\n",
      "0.18238\t0.211273\t0.206825\t0.216637\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "0.45\t0.45\t0.45\t0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 13573.8\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..done\n",
      "Arrived at 2 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 5 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(1) has 1 intervals\n",
      "descendant 1(18) has 3 intervals\n",
      "propagateDescendantBreakpoints yields 4 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 5.34162e+06, new anchor score: 5.34542e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "16%..\n",
      "32%..\n",
      "48%..\n",
      "64%..67%..\n",
      "83%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "performing iterative refinement\n",
      "\n",
      "0%..\n",
      "64%..67%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 2 to 17 via 3!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 955 tracking matches\n",
      "There are 12984 / 27695 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 3065 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.177663\n",
      "0.211496\n",
      "0.211955\n",
      "0.208717\n",
      "0.243914\n",
      "0.253951\n",
      "0.253477\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "0.45\n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..done\n",
      "Arrived at 5 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 16 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(2) has 3 intervals\n",
      "descendant 1(17) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 7 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 3.35893e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 7 new matches outside LCBs\n",
      "1,0 have 19 new matches outside LCBs\n",
      "2,0 have 17 new matches outside LCBs\n",
      "3,0 have 13 new matches outside LCBs\n",
      "4,0 have 9 new matches outside LCBs\n",
      "5,0 have 17 new matches outside LCBs\n",
      "6,0 have 16 new matches outside LCBs\n",
      "0,0 has an additional 16 matches\n",
      "1,0 has an additional 33 matches\n",
      "2,0 has an additional 34 matches\n",
      "3,0 has an additional 32 matches\n",
      "4,0 has an additional 25 matches\n",
      "5,0 has an additional 44 matches\n",
      "6,0 has an additional 37 matches\n",
      "Restoring backed up alignment tree...\n",
      "12,11 has 16 pairwise matches\n",
      "6,11 has 33 pairwise matches\n",
      "7,11 has 34 pairwise matches\n",
      "13,11 has 32 pairwise matches\n",
      "0,11 has 25 pairwise matches\n",
      "4,11 has 44 pairwise matches\n",
      "5,11 has 37 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 764 tracking matches\n",
      "There are 10426 / 22156 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 387 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.177663\n",
      "0.211496\n",
      "0.211955\n",
      "0.208717\n",
      "0.243914\n",
      "0.253951\n",
      "0.253477\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "0.45\n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..done\n",
      "Arrived at 5 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 16 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(2) has 3 intervals\n",
      "descendant 1(17) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 7 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 3.35893e+06, new anchor score: 3.52167e+06\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 2 new matches outside LCBs\n",
      "1,0 have 16 new matches outside LCBs\n",
      "2,0 have 7 new matches outside LCBs\n",
      "3,0 have 10 new matches outside LCBs\n",
      "4,0 have 11 new matches outside LCBs\n",
      "5,0 have 12 new matches outside LCBs\n",
      "6,0 have 14 new matches outside LCBs\n",
      "0,0 has an additional 2 matches\n",
      "1,0 has an additional 25 matches\n",
      "2,0 has an additional 16 matches\n",
      "3,0 has an additional 22 matches\n",
      "4,0 has an additional 20 matches\n",
      "5,0 has an additional 20 matches\n",
      "6,0 has an additional 26 matches\n",
      "Restoring backed up alignment tree...\n",
      "12,11 has 2 pairwise matches\n",
      "6,11 has 25 pairwise matches\n",
      "7,11 has 16 pairwise matches\n",
      "13,11 has 22 pairwise matches\n",
      "0,11 has 20 pairwise matches\n",
      "4,11 has 20 pairwise matches\n",
      "5,11 has 26 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 749 tracking matches\n",
      "There are 10163 / 21721 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 229 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.177663\n",
      "0.211496\n",
      "0.211955\n",
      "0.208717\n",
      "0.243914\n",
      "0.253951\n",
      "0.253477\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "0.45\n",
      "0.9\n",
      "0.9\n",
      "0.9\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..2%..3%..4%..5%..6%..7%..8%..9%..\n",
      "10%..11%..12%..13%..14%..15%..done\n",
      "Arrived at 11 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 31 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(2) has 3 intervals\n",
      "descendant 1(17) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 13 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 3.52167e+06, new anchor score: 3.50244e+06\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..\n",
      "58%..59%..\n",
      "63%..64%..\n",
      "70%..\n",
      "85%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Aligning node 3 to 4 via 0!\n",
      "get ancestral matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 1617 tracking matches\n",
      "There are 43594 / 46893 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 45425 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.246668\t0.222444\t0.245829\t0.245234\t0.290175\t0.220114\t0.233577\n",
      "0.198317\t0.169843\t0.20306\t0.20157\t0.235472\t0.167477\t0.179211\n",
      "0.235108\t0.225835\t0.242104\t0.239376\t0.271889\t0.212614\t0.234381\n",
      "0.225208\t0.204525\t0.225412\t0.224515\t0.26517\t0.19382\t0.211777\n",
      "0.235665\t0.225544\t0.239718\t0.238907\t0.271247\t0.20693\t0.224994\n",
      "0.260135\t0.242503\t0.262523\t0.260428\t0.298175\t0.232281\t0.2448\n",
      "0.268164\t0.251244\t0.275411\t0.276437\t0.303858\t0.24471\t0.258244\n",
      "0.26016\t0.24402\t0.26016\t0.262012\t0.299202\t0.24469\t0.259576\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..done\n",
      "Arrived at 6 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 20 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(3) has 7 intervals\n",
      "descendant 1(4) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 12 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: -1.79769e+308, new anchor score: 1.17305e+07\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 31 new matches outside LCBs\n",
      "0,1 have 29 new matches outside LCBs\n",
      "0,2 have 38 new matches outside LCBs\n",
      "0,3 have 38 new matches outside LCBs\n",
      "0,4 have 41 new matches outside LCBs\n",
      "0,5 have 26 new matches outside LCBs\n",
      "0,6 have 24 new matches outside LCBs\n",
      "1,0 have 22 new matches outside LCBs\n",
      "1,1 have 21 new matches outside LCBs\n",
      "1,2 have 26 new matches outside LCBs\n",
      "1,3 have 25 new matches outside LCBs\n",
      "1,4 have 27 new matches outside LCBs\n",
      "1,5 have 36 new matches outside LCBs\n",
      "1,6 have 25 new matches outside LCBs\n",
      "2,0 have 36 new matches outside LCBs\n",
      "2,1 have 34 new matches outside LCBs\n",
      "2,2 have 42 new matches outside LCBs\n",
      "2,3 have 45 new matches outside LCBs\n",
      "2,4 have 51 new matches outside LCBs\n",
      "2,5 have 30 new matches outside LCBs\n",
      "2,6 have 27 new matches outside LCBs\n",
      "3,0 have 25 new matches outside LCBs\n",
      "3,1 have 30 new matches outside LCBs\n",
      "3,2 have 35 new matches outside LCBs\n",
      "3,3 have 34 new matches outside LCBs\n",
      "3,4 have 35 new matches outside LCBs\n",
      "3,5 have 24 new matches outside LCBs\n",
      "3,6 have 28 new matches outside LCBs\n",
      "4,0 have 34 new matches outside LCBs\n",
      "4,1 have 31 new matches outside LCBs\n",
      "4,2 have 36 new matches outside LCBs\n",
      "4,3 have 40 new matches outside LCBs\n",
      "4,4 have 51 new matches outside LCBs\n",
      "4,5 have 24 new matches outside LCBs\n",
      "4,6 have 28 new matches outside LCBs\n",
      "5,0 have 27 new matches outside LCBs\n",
      "5,1 have 26 new matches outside LCBs\n",
      "5,2 have 39 new matches outside LCBs\n",
      "5,3 have 40 new matches outside LCBs\n",
      "5,4 have 39 new matches outside LCBs\n",
      "5,5 have 29 new matches outside LCBs\n",
      "5,6 have 29 new matches outside LCBs\n",
      "6,0 have 27 new matches outside LCBs\n",
      "6,1 have 26 new matches outside LCBs\n",
      "6,2 have 36 new matches outside LCBs\n",
      "6,3 have 37 new matches outside LCBs\n",
      "6,4 have 39 new matches outside LCBs\n",
      "6,5 have 20 new matches outside LCBs\n",
      "6,6 have 22 new matches outside LCBs\n",
      "7,0 have 34 new matches outside LCBs\n",
      "7,1 have 26 new matches outside LCBs\n",
      "7,2 have 39 new matches outside LCBs\n",
      "7,3 have 41 new matches outside LCBs\n",
      "7,4 have 44 new matches outside LCBs\n",
      "7,5 have 30 new matches outside LCBs\n",
      "7,6 have 25 new matches outside LCBs\n",
      "0,0 has an additional 134 matches\n",
      "0,1 has an additional 108 matches\n",
      "0,2 has an additional 122 matches\n",
      "0,3 has an additional 121 matches\n",
      "0,4 has an additional 145 matches\n",
      "0,5 has an additional 107 matches\n",
      "0,6 has an additional 96 matches\n",
      "1,0 has an additional 94 matches\n",
      "1,1 has an additional 92 matches\n",
      "1,2 has an additional 96 matches\n",
      "1,3 has an additional 95 matches\n",
      "1,4 has an additional 107 matches\n",
      "1,5 has an additional 111 matches\n",
      "1,6 has an additional 93 matches\n",
      "2,0 has an additional 117 matches\n",
      "2,1 has an additional 107 matches\n",
      "2,2 has an additional 124 matches\n",
      "2,3 has an additional 127 matches\n",
      "2,4 has an additional 138 matches\n",
      "2,5 has an additional 102 matches\n",
      "2,6 has an additional 96 matches\n",
      "3,0 has an additional 109 matches\n",
      "3,1 has an additional 107 matches\n",
      "3,2 has an additional 120 matches\n",
      "3,3 has an additional 119 matches\n",
      "3,4 has an additional 119 matches\n",
      "3,5 has an additional 92 matches\n",
      "3,6 has an additional 96 matches\n",
      "4,0 has an additional 119 matches\n",
      "4,1 has an additional 114 matches\n",
      "4,2 has an additional 123 matches\n",
      "4,3 has an additional 127 matches\n",
      "4,4 has an additional 136 matches\n",
      "4,5 has an additional 90 matches\n",
      "4,6 has an additional 98 matches\n",
      "5,0 has an additional 116 matches\n",
      "5,1 has an additional 122 matches\n",
      "5,2 has an additional 140 matches\n",
      "5,3 has an additional 141 matches\n",
      "5,4 has an additional 144 matches\n",
      "5,5 has an additional 115 matches\n",
      "5,6 has an additional 102 matches\n",
      "6,0 has an additional 137 matches\n",
      "6,1 has an additional 129 matches\n",
      "6,2 has an additional 151 matches\n",
      "6,3 has an additional 151 matches\n",
      "6,4 has an additional 152 matches\n",
      "6,5 has an additional 119 matches\n",
      "6,6 has an additional 113 matches\n",
      "7,0 has an additional 152 matches\n",
      "7,1 has an additional 120 matches\n",
      "7,2 has an additional 147 matches\n",
      "7,3 has an additional 149 matches\n",
      "7,4 has an additional 143 matches\n",
      "7,5 has an additional 121 matches\n",
      "7,6 has an additional 107 matches\n",
      "Restoring backed up alignment tree...\n",
      "11,3 has 134 pairwise matches\n",
      "11,14 has 108 pairwise matches\n",
      "11,1 has 122 pairwise matches\n",
      "11,2 has 121 pairwise matches\n",
      "11,10 has 145 pairwise matches\n",
      "11,8 has 107 pairwise matches\n",
      "11,9 has 96 pairwise matches\n",
      "12,3 has 94 pairwise matches\n",
      "12,14 has 92 pairwise matches\n",
      "12,1 has 96 pairwise matches\n",
      "12,2 has 95 pairwise matches\n",
      "12,10 has 107 pairwise matches\n",
      "12,8 has 111 pairwise matches\n",
      "12,9 has 93 pairwise matches\n",
      "6,3 has 117 pairwise matches\n",
      "6,14 has 107 pairwise matches\n",
      "6,1 has 124 pairwise matches\n",
      "6,2 has 127 pairwise matches\n",
      "6,10 has 138 pairwise matches\n",
      "6,8 has 102 pairwise matches\n",
      "6,9 has 96 pairwise matches\n",
      "7,3 has 109 pairwise matches\n",
      "7,14 has 107 pairwise matches\n",
      "7,1 has 120 pairwise matches\n",
      "7,2 has 119 pairwise matches\n",
      "7,10 has 119 pairwise matches\n",
      "7,8 has 92 pairwise matches\n",
      "7,9 has 96 pairwise matches\n",
      "13,3 has 119 pairwise matches\n",
      "13,14 has 114 pairwise matches\n",
      "13,1 has 123 pairwise matches\n",
      "13,2 has 127 pairwise matches\n",
      "13,10 has 136 pairwise matches\n",
      "13,8 has 90 pairwise matches\n",
      "13,9 has 98 pairwise matches\n",
      "0,3 has 116 pairwise matches\n",
      "0,14 has 122 pairwise matches\n",
      "0,1 has 140 pairwise matches\n",
      "0,2 has 141 pairwise matches\n",
      "0,10 has 144 pairwise matches\n",
      "0,8 has 115 pairwise matches\n",
      "0,9 has 102 pairwise matches\n",
      "4,3 has 137 pairwise matches\n",
      "4,14 has 129 pairwise matches\n",
      "4,1 has 151 pairwise matches\n",
      "4,2 has 151 pairwise matches\n",
      "4,10 has 152 pairwise matches\n",
      "4,8 has 119 pairwise matches\n",
      "4,9 has 113 pairwise matches\n",
      "5,3 has 152 pairwise matches\n",
      "5,14 has 120 pairwise matches\n",
      "5,1 has 147 pairwise matches\n",
      "5,2 has 149 pairwise matches\n",
      "5,10 has 143 pairwise matches\n",
      "5,8 has 121 pairwise matches\n",
      "5,9 has 107 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 766 tracking matches\n",
      "There are 20350 / 22214 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 2703 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.246668\t0.222444\t0.245829\t0.245234\t0.290175\t0.220114\t0.233577\n",
      "0.198317\t0.169843\t0.20306\t0.20157\t0.235472\t0.167477\t0.179211\n",
      "0.235108\t0.225835\t0.242104\t0.239376\t0.271889\t0.212614\t0.234381\n",
      "0.225208\t0.204525\t0.225412\t0.224515\t0.26517\t0.19382\t0.211777\n",
      "0.235665\t0.225544\t0.239718\t0.238907\t0.271247\t0.20693\t0.224994\n",
      "0.260135\t0.242503\t0.262523\t0.260428\t0.298175\t0.232281\t0.2448\n",
      "0.268164\t0.251244\t0.275411\t0.276437\t0.303858\t0.24471\t0.258244\n",
      "0.26016\t0.24402\t0.26016\t0.262012\t0.299202\t0.24469\t0.259576\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..2%..done\n",
      "Arrived at 6 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 17 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(3) has 7 intervals\n",
      "descendant 1(4) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 12 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 1.17305e+07, new anchor score: 3.061e+07\n",
      "Backing up alignment tree...\n",
      "propagating ancestral breakpoints\n",
      "recursive anchor search\n",
      "0,0 have 15 new matches outside LCBs\n",
      "0,1 have 3 new matches outside LCBs\n",
      "0,2 have 7 new matches outside LCBs\n",
      "0,3 have 10 new matches outside LCBs\n",
      "0,4 have 14 new matches outside LCBs\n",
      "0,5 have 3 new matches outside LCBs\n",
      "0,6 have 6 new matches outside LCBs\n",
      "1,0 have 0 new matches outside LCBs\n",
      "1,1 have 2 new matches outside LCBs\n",
      "1,2 have 0 new matches outside LCBs\n",
      "1,3 have 0 new matches outside LCBs\n",
      "1,4 have 1 new matches outside LCBs\n",
      "1,5 have 25 new matches outside LCBs\n",
      "1,6 have 9 new matches outside LCBs\n",
      "2,0 have 12 new matches outside LCBs\n",
      "2,1 have 3 new matches outside LCBs\n",
      "2,2 have 16 new matches outside LCBs\n",
      "2,3 have 19 new matches outside LCBs\n",
      "2,4 have 20 new matches outside LCBs\n",
      "2,5 have 7 new matches outside LCBs\n",
      "2,6 have 5 new matches outside LCBs\n",
      "3,0 have 8 new matches outside LCBs\n",
      "3,1 have 3 new matches outside LCBs\n",
      "3,2 have 12 new matches outside LCBs\n",
      "3,3 have 11 new matches outside LCBs\n",
      "3,4 have 12 new matches outside LCBs\n",
      "3,5 have 10 new matches outside LCBs\n",
      "3,6 have 7 new matches outside LCBs\n",
      "4,0 have 17 new matches outside LCBs\n",
      "4,1 have 2 new matches outside LCBs\n",
      "4,2 have 11 new matches outside LCBs\n",
      "4,3 have 11 new matches outside LCBs\n",
      "4,4 have 16 new matches outside LCBs\n",
      "4,5 have 6 new matches outside LCBs\n",
      "4,6 have 7 new matches outside LCBs\n",
      "5,0 have 17 new matches outside LCBs\n",
      "5,1 have 0 new matches outside LCBs\n",
      "5,2 have 7 new matches outside LCBs\n",
      "5,3 have 9 new matches outside LCBs\n",
      "5,4 have 7 new matches outside LCBs\n",
      "5,5 have 3 new matches outside LCBs\n",
      "5,6 have 6 new matches outside LCBs\n",
      "6,0 have 16 new matches outside LCBs\n",
      "6,1 have 3 new matches outside LCBs\n",
      "6,2 have 7 new matches outside LCBs\n",
      "6,3 have 11 new matches outside LCBs\n",
      "6,4 have 9 new matches outside LCBs\n",
      "6,5 have 3 new matches outside LCBs\n",
      "6,6 have 7 new matches outside LCBs\n",
      "7,0 have 12 new matches outside LCBs\n",
      "7,1 have 1 new matches outside LCBs\n",
      "7,2 have 7 new matches outside LCBs\n",
      "7,3 have 9 new matches outside LCBs\n",
      "7,4 have 14 new matches outside LCBs\n",
      "7,5 have 3 new matches outside LCBs\n",
      "7,6 have 7 new matches outside LCBs\n",
      "0,0 has an additional 20 matches\n",
      "0,1 has an additional 5 matches\n",
      "0,2 has an additional 7 matches\n",
      "0,3 has an additional 10 matches\n",
      "0,4 has an additional 17 matches\n",
      "0,5 has an additional 3 matches\n",
      "0,6 has an additional 6 matches\n",
      "1,1 has an additional 2 matches\n",
      "1,4 has an additional 1 matches\n",
      "1,5 has an additional 45 matches\n",
      "1,6 has an additional 9 matches\n",
      "2,0 has an additional 14 matches\n",
      "2,1 has an additional 4 matches\n",
      "2,2 has an additional 17 matches\n",
      "2,3 has an additional 19 matches\n",
      "2,4 has an additional 24 matches\n",
      "2,5 has an additional 7 matches\n",
      "2,6 has an additional 5 matches\n",
      "3,0 has an additional 16 matches\n",
      "3,1 has an additional 6 matches\n",
      "3,2 has an additional 14 matches\n",
      "3,3 has an additional 12 matches\n",
      "3,4 has an additional 17 matches\n",
      "3,5 has an additional 10 matches\n",
      "3,6 has an additional 7 matches\n",
      "4,0 has an additional 22 matches\n",
      "4,1 has an additional 4 matches\n",
      "4,2 has an additional 12 matches\n",
      "4,3 has an additional 11 matches\n",
      "4,4 has an additional 27 matches\n",
      "4,5 has an additional 6 matches\n",
      "4,6 has an additional 7 matches\n",
      "5,0 has an additional 20 matches\n",
      "5,2 has an additional 7 matches\n",
      "5,3 has an additional 9 matches\n",
      "5,4 has an additional 8 matches\n",
      "5,5 has an additional 3 matches\n",
      "5,6 has an additional 6 matches\n",
      "6,0 has an additional 18 matches\n",
      "6,1 has an additional 4 matches\n",
      "6,2 has an additional 7 matches\n",
      "6,3 has an additional 11 matches\n",
      "6,4 has an additional 12 matches\n",
      "6,5 has an additional 3 matches\n",
      "6,6 has an additional 7 matches\n",
      "7,0 has an additional 13 matches\n",
      "7,1 has an additional 2 matches\n",
      "7,2 has an additional 7 matches\n",
      "7,3 has an additional 9 matches\n",
      "7,4 has an additional 17 matches\n",
      "7,5 has an additional 3 matches\n",
      "7,6 has an additional 7 matches\n",
      "Restoring backed up alignment tree...\n",
      "11,3 has 20 pairwise matches\n",
      "11,14 has 5 pairwise matches\n",
      "11,1 has 7 pairwise matches\n",
      "11,2 has 10 pairwise matches\n",
      "11,10 has 17 pairwise matches\n",
      "11,8 has 3 pairwise matches\n",
      "11,9 has 6 pairwise matches\n",
      "12,3 has 0 pairwise matches\n",
      "12,14 has 2 pairwise matches\n",
      "12,1 has 0 pairwise matches\n",
      "12,2 has 0 pairwise matches\n",
      "12,10 has 1 pairwise matches\n",
      "12,8 has 45 pairwise matches\n",
      "12,9 has 9 pairwise matches\n",
      "6,3 has 14 pairwise matches\n",
      "6,14 has 4 pairwise matches\n",
      "6,1 has 17 pairwise matches\n",
      "6,2 has 19 pairwise matches\n",
      "6,10 has 24 pairwise matches\n",
      "6,8 has 7 pairwise matches\n",
      "6,9 has 5 pairwise matches\n",
      "7,3 has 16 pairwise matches\n",
      "7,14 has 6 pairwise matches\n",
      "7,1 has 14 pairwise matches\n",
      "7,2 has 12 pairwise matches\n",
      "7,10 has 17 pairwise matches\n",
      "7,8 has 10 pairwise matches\n",
      "7,9 has 7 pairwise matches\n",
      "13,3 has 22 pairwise matches\n",
      "13,14 has 4 pairwise matches\n",
      "13,1 has 12 pairwise matches\n",
      "13,2 has 11 pairwise matches\n",
      "13,10 has 27 pairwise matches\n",
      "13,8 has 6 pairwise matches\n",
      "13,9 has 7 pairwise matches\n",
      "0,3 has 20 pairwise matches\n",
      "0,14 has 0 pairwise matches\n",
      "0,1 has 7 pairwise matches\n",
      "0,2 has 9 pairwise matches\n",
      "0,10 has 8 pairwise matches\n",
      "0,8 has 3 pairwise matches\n",
      "0,9 has 6 pairwise matches\n",
      "4,3 has 18 pairwise matches\n",
      "4,14 has 4 pairwise matches\n",
      "4,1 has 7 pairwise matches\n",
      "4,2 has 11 pairwise matches\n",
      "4,10 has 12 pairwise matches\n",
      "4,8 has 3 pairwise matches\n",
      "4,9 has 7 pairwise matches\n",
      "5,3 has 13 pairwise matches\n",
      "5,14 has 2 pairwise matches\n",
      "5,1 has 7 pairwise matches\n",
      "5,2 has 9 pairwise matches\n",
      "5,10 has 17 pairwise matches\n",
      "5,8 has 3 pairwise matches\n",
      "5,9 has 7 pairwise matches\n",
      "Performing Sum-of-pairs Greedy Breakpoint Elimination\n",
      "construct LCB tracking matches\n",
      "There are 706 tracking matches\n",
      "There are 19105 / 20474 components used\n",
      "init tracking match LCB tracking\n",
      "pairwise score tracking matches\n",
      "get pairwise LCBs\n",
      "there are 1075 pairwise LCBs\n",
      "scaling bp penalty by conservation weight:\n",
      "0.246668\t0.222444\t0.245829\t0.245234\t0.290175\t0.220114\t0.233577\n",
      "0.198317\t0.169843\t0.20306\t0.20157\t0.235472\t0.167477\t0.179211\n",
      "0.235108\t0.225835\t0.242104\t0.239376\t0.271889\t0.212614\t0.234381\n",
      "0.225208\t0.204525\t0.225412\t0.224515\t0.26517\t0.19382\t0.211777\n",
      "0.235665\t0.225544\t0.239718\t0.238907\t0.271247\t0.20693\t0.224994\n",
      "0.260135\t0.242503\t0.262523\t0.260428\t0.298175\t0.232281\t0.2448\n",
      "0.268164\t0.251244\t0.275411\t0.276437\t0.303858\t0.24471\t0.258244\n",
      "0.26016\t0.24402\t0.26016\t0.262012\t0.299202\t0.24469\t0.259576\n",
      "\n",
      "\n",
      "scaling bp penalty by bp weight: \n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.9\t0.9\t0.9\t0.9\t0.9\t0.9\t0.9\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "0.45\t0.45\t0.45\t0.45\t0.45\t0.45\t0.45\n",
      "\n",
      "Greedy BPE\n",
      "Scoring with scaled breakpoint penalty: 4000\n",
      "1%..2%..3%..done\n",
      "Arrived at 8 intervals\n",
      "Adding unaligned intervals\n",
      "addUnalignedIntervals yields 23 intervals\n",
      "Merging unaligned intervals\n",
      "Marbling gaps\n",
      "Propagating descendant breakpoints\n",
      "descendant 0(3) has 7 intervals\n",
      "descendant 1(4) has 1 intervals\n",
      "propagateDescendantBreakpoints yields 14 intervals\n",
      "Creating ancestral ordering\n",
      "Previous anchoring score: 3.061e+07, new anchor score: 3.06401e+07\n",
      "propagating ancestral breakpoints\n",
      "performing a gapped alignment\n",
      "\n",
      "0%..5%..7%..\n",
      "24%..\n",
      "42%..43%..\n",
      "95%..96%..97%..98%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "Performing final pass iterative refinement\n",
      "\n",
      "0%..7%..8%..\n",
      "24%..\n",
      "41%..42%..\n",
      "94%..95%..96%..97%..\n",
      "100%..Fix left ends\n",
      "\n",
      "done.\n",
      "root alignment has 12 superintervals\n",
      "root alignment length: 24659\n",
      "Organisms have 31.8% GC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "outpath=/Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch\n",
    "refmitochpath=/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/assemblies_I_picked\n",
    "mitochpath=/Users/maeperez/Desktop/SYMBIONT_GENOMES/mitoch/mitobim_mitoch/batch_export\n",
    "\n",
    "#/Users/maeperez/Desktop/Mauve2.3.1.app/Contents/MacOS/progressiveMauve --output=${outpath}/MITOCH.xmfa --output-guide-tree=${outpath}/MITOCH.tree --backbone-output=${outpath}/MITOCH.backbone ${mitochpath}/X.pliocardia.gbk ${mitochpath}/P.soyoae2.gbk ${mitochpath}/P.soyoae1.gbk ${mitochpath}/P.extenta.gbk ${mitochpath}/C.rectimargo.gbk ${mitochpath}/C.pacifica.gbk ${mitochpath}/A.southwardae.gbk ${mitochpath}/A.phaseoliformis.gbk ${mitochpath}/A.gigas2.gbk ${mitochpath}/A.gigas1.gbk ${mitochpath}/A.diagonalis.gbk ${refmitochpath}/C.magnifica.gbk ${refmitochpath}/A.mariana_ref.gbk ${refmitochpath}/A.phaseoliformis_ref.gbk ${refmitochpath}/P.okutanii_ref.gbk\n",
    "\n",
    "/Applications/Mauve.app/Contents/MacOS/progressiveMauve --output=${outpath}/MITOCH.xmfa --output-guide-tree=${outpath}/MITOCH.tree --backbone-output=${outpath}/MITOCH.backbone ${mitochpath}/X.pliocardia.gbk ${mitochpath}/P.soyoae2.gbk ${mitochpath}/P.soyoae1.gbk ${mitochpath}/P.extenta.gbk ${mitochpath}/C.rectimargo.gbk ${mitochpath}/C.pacifica.gbk ${mitochpath}/A.southwardae.gbk ${mitochpath}/A.phaseoliformis.gbk ${mitochpath}/A.gigas2.gbk ${mitochpath}/A.gigas1.gbk ${mitochpath}/A.diagonalis.gbk ${refmitochpath}/C.magnifica.gbk ${refmitochpath}/A.mariana_ref.gbk ${refmitochpath}/A.phaseoliformis_ref.gbk ${refmitochpath}/P.okutanii_ref.gbk\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "outpath=/Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch\n",
    "\n",
    "#Extract colinear blocks found in all genomes\n",
    "\n",
    "/Users/maeperez/Desktop/Mauve/stripSubsetLCBs ${outpath}/MITOCH.xmfa ${outpath}/MITOCH.xmfa.bbcols ${outpath}/core_MITOCH.xmfa 50 15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converting /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/core_MITOCH.xmfa (xmfa) to a directory /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/core_MITOCH_alignments3 (fasta)\n",
      "Finished converting /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/core_MITOCH.xmfa to /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/core_MITOCH_alignments3/ (9 alignments)\n",
      "Output is now in /Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch/core_MITOCH_alignments3/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "outpath=/Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch\n",
    "\n",
    "# split into multiple fastas\n",
    "mkdir ${outpath}/core_MITOCH_alignments3\n",
    "\n",
    "perl /Users/maeperez/Desktop/Mauve/xmfa2fasta.pl -i ${outpath}/core_MITOCH.xmfa -o ${outpath}/core_MITOCH_alignments3/ -f fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 15 (thread    2)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    3) f\b\b\r",
      "STEP     4 / 14 (thread    1) f\b\b\r",
      "STEP     5 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    2) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    2) f\b\b\r",
      "STEP     8 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    2) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    0) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    3) f\b\b\r",
      "STEP     4 / 14 (thread    1) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    3) f\b\b\r",
      "STEP     8 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    2) f\b\b\r",
      "STEP    12 / 14 (thread    3) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    0) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 15 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     9 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    2) f\b\b\r",
      "STEP     8 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    1) f\b\b\r",
      "STEP    12 / 14 (thread    3) f\b\b\r",
      "STEP    13 / 14 (thread    2) d\b\b\r",
      "STEP    14 / 14 (thread    0) d\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    1) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    2) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    2) f\b\b\r",
      "STEP     8 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    2) f\b\b\r",
      "STEP    13 / 14 (thread    1) d\b\b\r",
      "STEP    14 / 14 (thread    0) d\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\n",
      "There are 121 ambiguous characters.\n",
      "\r",
      "    1 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP     9 / 14 (thread    2) f\b\b\r",
      "STEP    12 / 14 (thread    0) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    2) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0)\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b f\b\b\r",
      "STEP     6 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    0) f\b\b\r",
      "STEP     7 / 14 (thread    3) f\b\b\r",
      "STEP     9 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP    10 / 14 (thread    2) f\b\b\r",
      "STEP    12 / 14 (thread    0) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    3) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     5 / 14 (thread    1) d\b\b\r",
      "STEP     4 / 14 (thread    3) d\b\b\r",
      "STEP     6 / 14 (thread    0) d\b\b\r",
      "STEP     7 / 14 (thread    2) d\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    1) d\b\b\r",
      "STEP    11 / 14 (thread    2) d\b\b\r",
      "STEP     9 / 14 (thread    3) d\b\b\r",
      "STEP    12 / 14 (thread    0) d\b\b\r",
      "STEP    13 / 14 (thread    1) d\b\b\r",
      "STEP    14 / 14 (thread    2) d\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    1) f\b\b\r",
      "STEP     4 / 14 (thread    3) d\b\b\r",
      "STEP     6 / 14 (thread    1) d\b\b\r",
      "STEP     8 / 14 (thread    0) d\b\b\r",
      "STEP     5 / 14 (thread    2) d\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     7 / 14 (thread    3) d\b\b\r",
      "STEP     9 / 14 (thread    1) d\b\b\r",
      "STEP    11 / 14 (thread    2) d\b\b\r",
      "STEP    12 / 14 (thread    0) d\b\b\r",
      "STEP    13 / 14 (thread    3) d\b\b\r",
      "STEP    14 / 14 (thread    1) d\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\n",
      "There are 2 ambiguous characters.\n",
      "\r",
      "    1 / 15 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0)\r",
      "STEP     2 / 14 (thread    1) f\b\b f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP    10 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    0) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    1) f\b\b\r",
      "STEP    13 / 14 (thread    2) f\b\b\r",
      "STEP    14 / 14 (thread    0) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    1) f\b\b\r",
      "STEP     2 / 14 (thread    0) f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    2) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    1) f\b\b\r",
      "STEP    13 / 14 (thread    0) f\b\b\r",
      "STEP    14 / 14 (thread    3) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 15 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     2 / 14 (thread    1)\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     1 / 14 (thread    0) f\b\b f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    0) f\b\b\r",
      "STEP     7 / 14 (thread    1) f\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    3) f\b\b\r",
      "STEP    11 / 14 (thread    1) f\b\b\r",
      "STEP     9 / 14 (thread    2) f\b\b\r",
      "STEP    12 / 14 (thread    0) f\b\b\r",
      "STEP    13 / 14 (thread    3) f\b\b\r",
      "STEP    14 / 14 (thread    1) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1)\r",
      "STEP     3 / 14 (thread    2) f\b\b f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP    12 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    3) f\b\b\r",
      "STEP    10 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    0) f\b\b\r",
      "STEP    13 / 14 (thread    2) f\b\b\r",
      "STEP    14 / 14 (thread    3) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 15 (thread    2)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    1) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     2 / 14 (thread    0) f\b\b\r",
      "STEP     5 / 14 (thread    3) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP     7 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    1) f\b\b\r",
      "STEP    13 / 14 (thread    0) f\b\b\r",
      "STEP    14 / 14 (thread    2) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0)\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP     5 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    3) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    0) f\b\b\r",
      "STEP    13 / 14 (thread    2) f\b\b\r",
      "STEP    14 / 14 (thread    1) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\n",
      "There are 51 ambiguous characters.\n",
      "\r",
      "    1 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    1) f\b\b\r",
      "STEP     2 / 14 (thread    3) f\b\b\r",
      "STEP     3 / 14 (thread    0) f\b\b\r",
      "STEP     4 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP     7 / 14 (thread    3) f\b\b\r",
      "STEP    10 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    0) f\b\b\r",
      "STEP    12 / 14 (thread    3) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    2) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    1) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    1) f\b\b\r",
      "STEP     5 / 14 (thread    0) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP    10 / 14 (thread    1) f\b\b\r",
      "STEP     7 / 14 (thread    3) f\b\b\r",
      "STEP    11 / 14 (thread    2) f\b\b\r",
      "STEP     9 / 14 (thread    0) f\b\b\r",
      "STEP    12 / 14 (thread    1) f\b\b\r",
      "STEP    13 / 14 (thread    3) f\b\b\r",
      "STEP    14 / 14 (thread    2) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nthread = 4\n",
      "nthreadpair = 4\n",
      "nthreadtb = 4\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\n",
      "There are 1 ambiguous characters.\n",
      "\r",
      "    1 / 15 (thread    0)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    2) f\b\b\r",
      "STEP     3 / 14 (thread    3) f\b\b\r",
      "STEP     5 / 14 (thread    3) f\b\b\r",
      "STEP     4 / 14 (thread    1) f\b\b\r",
      "STEP     8 / 14 (thread    3) f\b\b\r",
      "STEP     6 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    1) f\b\b\r",
      "STEP     7 / 14 (thread    2) f\b\b\r",
      "STEP    11 / 14 (thread    1) f\b\b\r",
      "STEP    10 / 14 (thread    0) f\b\b\r",
      "STEP    12 / 14 (thread    3) f\b\b\r",
      "STEP    13 / 14 (thread    2) f\b\b\r",
      "STEP    14 / 14 (thread    1) f\b\b\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "\r",
      "    0 / 15 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "\r",
      "STEP     1 / 14 (thread    0) f\b\b\r",
      "STEP     2 / 14 (thread    1) f\b\b\r",
      "STEP     3 / 14 (thread    2) f\b\b\r",
      "STEP     5 / 14 (thread    2) f\b\b\r",
      "STEP     4 / 14 (thread    3) f\b\b\r",
      "STEP     8 / 14 (thread    2) f\b\b\r",
      "STEP     6 / 14 (thread    0) f\b\b\r",
      "STEP     9 / 14 (thread    3) f\b\b\r",
      "STEP     7 / 14 (thread    1) f\b\b\r",
      "STEP    11 / 14 (thread    0) f\b\b\r",
      "STEP    10 / 14 (thread    3) f\b\b\r",
      "STEP    12 / 14 (thread    2) f\b\b\r",
      "STEP    13 / 14 (thread    1) f\b\b\r",
      "STEP    14 / 14 (thread    0) f\b\b\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "4 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "outpath=/Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch\n",
    "\n",
    "#align with mafft and muscle\n",
    "mkdir ${outpath}/core_MITOCH_mafft3\n",
    "\n",
    "\n",
    "for file in `ls ${outpath}/core_MITOCH_alignments3/*fasta`\n",
    "do\n",
    "prefix=`echo $file`\n",
    "if [ -n \"$prefix\" ]\n",
    "then\n",
    "prefix=`echo $prefix |awk -F/ '{print $(NF)}'`\n",
    "\n",
    "mafft --thread 4 ${outpath}/core_MITOCH_alignments3/${prefix} > ${outpath}/core_MITOCH_mafft3/${prefix}_mafft\n",
    "\n",
    "fi\n",
    "done\n",
    "\n",
    "cat ${outpath}/core_MITOCH_mafft3/* > ${outpath}/CAT_MITOCH_mafft"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "outpath=/Users/maeperez/Desktop/Mauve/Clams/Clam_mitoch\n",
    "\n",
    "#Concatenate fasta\n",
    "\n",
    "python /Users/maeperez/Desktop/Mauve/Clams/Concactenate_fasta.py -i ${outpath}/CAT_MITOCH_mafft -o ${outpath}/CAT_MITOCH_fasta_conc -n 15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
