---
title: "Functional categories figures"
output: html_document
editor_options: 
  chunk_output_type: console
---
# Import packages and functions

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# setwd("/home/maeperez/projects/def-bacc/maeperez/Clams/VesicSymb_Evolution/dowstream_analyses_and_data/")
setwd("/Users/maeperez/Desktop/VesicSymb_Evolution/dowstream_analyses_and_data")
getwd()
library('ggplot2')
library('gridExtra')
library(RColorBrewer)
library(tidyr)
library(plyr) ##
library(dplyr)
library(reshape2) 
library(rsvg)
library(magick)
require('cowplot')
theme_set(theme_cowplot())
##
# install.packages(c('gridExtra','RColorBrewer','tidyr','plyr','dplyr','reshape2','vegan','ape'), dep = TRUE)
# install.packages(c('data.table','grid','gtable','scales'), dep = TRUE)



```


```{r load function to run Fisher tests}
# dat=d
# branch='All'
run_chisq_tests<- function(dat){
library("extraDistr")
  pval<-data.frame(p.value=as.numeric(numeric())) 
  for(br in colnames(dat)[-c(1,2)]){
    data<-cbind(dat[,2],dat[,br]) %>% replace(., is.na(.), 0)
    # p.value<-dmvhyper(x = data[,2], n = data[,1], k=sum(data[,2]))
    # print(br)
    p.value<-chisq.test(data[,2],data[,1])$p.value
    new_pval<-as.data.frame(p.value, row.names=br, col.names='p.value')
    pval<-rbind(pval,new_pval)
  }
  
  pval$comp<-row.names(pval)
  pval$symbol<-ifelse(pval$p.value<=0.01,'***',
                ifelse(pval$p.value<=0.05,'**',
                ifelse(pval$p.value<=0.1,'*','')))

return(pval)
}

run_fisher_tests<- function(dat){
library("extraDistr")
  pval<-data.frame(p.value=as.numeric(numeric()))
  for(br in colnames(dat)[-c(1,2)]){
    data<-cbind(dat[,2],dat[,br]) %>% replace(., is.na(.), 0)
    p.value<-fisher.test(data[,1],data[,2])$p.value
    new_pval<-as.data.frame(p.value, row.names=br, col.names='p.value')
    pval<-rbind(pval,new_pval)
  }

  pval$comp<-row.names(pval)
  pval$symbol<-ifelse(pval$p.value<=0.01,'***',
                ifelse(pval$p.value<=0.05,'**',
                ifelse(pval$p.value<=0.1,'*','')))

return(pval)
} ## not good

run_multihypergeometric_tests<- function(dat){
library("extraDistr")
  pval<-data.frame(p.value=as.numeric(numeric()))
  for(br in colnames(dat)[-c(1,2)]){
    data<-cbind(dat[,2],dat[,br]) %>% replace(., is.na(.), 0)
    dhyper(n=data[,1],x=data[,2],k=sum(data[,2]))
    ?dmvhyper
    rmvhyper(1,data[,1],sum(data[,2]))
    p.value<-dmvhyper(n=data[,1],x=data[,2],k=sum(data[,2]))
    new_pval<-as.data.frame(p.value, row.names=br, col.names='p.value')
    pval<-rbind(pval,new_pval)
  }

  pval$comp<-row.names(pval)
  pval$symbol<-ifelse(pval$p.value<=0.01,'***',
                ifelse(pval$p.value<=0.05,'**',
                ifelse(pval$p.value<=0.1,'*','')))

return(pval)
} ## Not good

montecarlo_test_pval<-function(df,branch,SEED_level){
  ecdf_fun <- function(x,perc) ecdf(x)(perc) # determine the quantile of a value against a distribution

  test=branch
  #generate random subsample with same number of genes as in test category
  df1=df[,c('gene_id','Core',test,SEED_level)]
  len_core=df1[df1[,'Core']==1,]$gene_id %>% unique %>% length
  len_selected=df1[df1[,test]==1,]$gene_id %>% unique %>% length
  
  pval_distrib=c()
  n=0
  while (n<100) {
    # pick len_selected genes from core at random
    random_gene_set=sample((subset(df1,Core==1)$gene_id %>% unique), len_selected)
    # add random column to 
    dd=cbind(df1,'random'=as.numeric(df1$gene_id %in% random_gene_set))
    formula=as.formula(paste0(".~ ",SEED_level))
    d<-aggregate(formula,dd,FUN = function(x) {sum(x > 0, na.rm = F)}, na.action = na.pass) 
    pval_distrib<-rbind(pval_distrib,c(dmvhyper(n=d[,c('Core')],x=d[,c('random')],k=sum(d[,c('random')]))))
    n=n+1}
  pval_test=dmvhyper(n=d[,c('Core')],x=d[,c(test)],k=sum(d[,c(test)]))
  
  final_pval<-ecdf_fun(sort(pval_distrib),pval_test)
  return(final_pval)} ## compare prop of distribution if test data to that of 100 trials of randomly sampled genes (same sample size as test data)

run_multihypergeometric_montecarlo_tests<- function(df,SEED_level){
library("extraDistr")
  branches_to_test=names(df)[which(sapply(df, is.numeric))]

  pval<-data.frame(p.value=as.numeric(numeric()))
  for(br in branches_to_test){
    p.value<-montecarlo_test_pval(df,br,SEED_level)
    new_pval<-as.data.frame(p.value, row.names=br, col.names='p.value')
    pval<-rbind(pval,new_pval)
  }

  pval$comp<-row.names(pval)
  pval$symbol<-ifelse(pval$p.value<=0.01,'***',
                ifelse(pval$p.value<=0.05,'**',
                ifelse(pval$p.value<=0.1,'*','')))

return(pval)
}

run_fisher_tests2<- function(dat){
fisher<-data.frame(matrix(ncol = length(colnames(dat)[-c(1,2)]), nrow = length(row.names(dat))),row.names=rownames(dat))
colnames(fisher) <- colnames(dat)[-c(1,2)]

differences<-data.frame(matrix(ncol = length(colnames(dat)[-c(1,2)]), nrow = length(row.names(dat))),row.names=rownames(dat))
colnames(differences) <- colnames(dat)[-c(1,2)]

for(br in colnames(dat)[-c(1,2)]){
  data<-dat[,c(colnames(dat)[2],br)] %>% replace(., is.na(.), 0)
  for (category in rownames(data)){
    success_in_sample<-data[rownames(data)==category,br]
    success_in_left_part<-data[rownames(data)==category,colnames(dat)[2]]-success_in_sample
    failure_in_sample<-sum(data[,br])-success_in_sample
    failure_in_left_part<-sum(data[rownames(data)!=category,colnames(dat)[2]])+success_in_sample
    p.value<-fisher.test(matrix(c(success_in_sample, success_in_left_part, failure_in_sample, failure_in_left_part), 2, 2),alternative = "two.sided")
    
    p.adjust(fisher[,br],
               method = "holm")

    fisher[category,br]=p.value
    differences[category,br]=(data[rownames(data)==category,br]/sum(data))-(sum(data[rownames(data)==category,])/sum(data))*(sum(data[,br])/sum(data))
  }
  }

fisher$categories<-rownames(fisher)
differences$categories<-rownames(differences)
breakdown<-cbind(melt(fisher, id.vars='categories',value.name='pvalue'), melt(differences, id.vars=c('categories'))[,3])
colnames(breakdown)[2]='comp'
colnames(breakdown)[4]='diff'
breakdown$pvalue<-as.numeric(breakdown$pvalue)
breakdown$diff<-as.numeric(breakdown$diff)*100 # expresses the difference in pobability of the event happening
breakdown$symbol<-ifelse(breakdown$pvalue<=0.01,'***',
              ifelse(breakdown$pvalue<=0.05,'**',
              ifelse(breakdown$pvalue<=0.1,'*','')))
breakdown$sign<-ifelse(breakdown$diff<0,'depletion' ,'enrichment')

return(breakdown)
} # runs fisher test for enrichment on each category


plot_breakdown_with_counts2<-function(breakdown,dat){
level<-colnames(dat)[1]
add_counts<-melt(dat,variable.name = 'comp',value.name = 'freq')
breakdown_w_counts<-merge(breakdown,add_counts, by.y=c('comp',level), by.x=c('comp','categories'),all=T)
breakdown_w_counts$freq[!(breakdown_w_counts$comp %in% c('Pan')) & breakdown_w_counts$pvalue>0.1] <- NA
tot<-aggregate(data=breakdown_w_counts,freq~comp,FUN='sum')
colnames(tot)[2]='tot'
breakdown_w_counts<-merge(breakdown_w_counts,tot,by='comp',all.x=T)
breakdown_w_counts$relfreq<-paste(round(breakdown_w_counts$freq/breakdown_w_counts$tot*100),'%')
breakdown_w_counts$relfreq[breakdown_w_counts$relfreq =='NA %'] <- NA
breakdown_w_counts$symbol[breakdown_w_counts$comp =='moderate'] <- NA
breakdown_w_counts$sign[breakdown_w_counts$comp =='moderate'] <- NA
breakdown_w_counts$diff[breakdown_w_counts$comp =='moderate'] <- NA

plot<-ggplot(data=subset(breakdown_w_counts,comp!='Pan'))+
    geom_point(aes(x=comp,y=categories,fill=sign,size=abs(diff),alpha=pvalue),shape=21,color='grey')+
    scale_x_discrete(limits = levels(dat$comp)) +
    scale_fill_manual(name="",breaks=c("depletion", "enrichment"), values =c( "lightblue", "#EC7063")) +
    scale_size_continuous(range=c(5,20),breaks=c(5,20,50,75,100))+
    scale_alpha_continuous(range=c(1,0.2), limits=c(0,0.1), na.value = 0,breaks=c(0,0.01,0.05,0.1,1),guide=FALSE)+
    geom_text(aes(x=comp,y=categories,label=symbol),vjust=-0.5)+
    geom_text(aes(x=comp,y=categories,label=freq),vjust=0.5)+
    geom_text(aes(x=comp,y=categories,label=relfreq),hjust=-1)+
    theme_bw() +
    theme(axis.title.x=element_blank(),
          axis.text.x = element_text(angle=60,hjust=1))
  return(plot)
}
```


# Selection: absREL
## Import data
```{r import data, echo=FALSE}
# Import list of genes with significant evidence for recombinaiton according to the AICc test.

recombining_genes= read.table('core_genes_with_recomb_AICc.txt',header=F)$V1
recombining_genes


# Import SEED annotations for all genes

seed<-read.csv('./SEED_annotations_all.tsv',header=T,sep='\t', na.strings = '',stringsAsFactors = T)
seed=seed[,c("locus_tag","SEED_level_1","SEED_level_2","SEED_level_3" ,"SEED_long_name")]
colnames(seed)[1]='gene_id'
seed = seed[!duplicated(seed),]

pval_thresh=0.05

core_sel=read.table(paste0('./absrel_results_pval_',pval_thresh,'.txt'),header=T,sep='\t',row.names = 1)

colSums(core_sel)

# core_sel[core_sel>0]=1 # presence/absence of selection in that gene (not counting multiple episodes of diversifying selection within a particular group)

core_sel$gene_id=rownames(core_sel)
core_sel=subset(core_sel, !(gene_id %in% recombining_genes)) # remove recombining genes

colSums(core_sel[,-ncol(core_sel)])

core_sel=merge(core_sel,seed,by = 'gene_id',all.x=T)
# core_sel$All<-1
head(core_sel)
nrow(core_sel)

core_orth=read.table('./Orthology_id35cov51_nopseudo.txt',header=T,sep='\t',row.names=1)
core_orth %>% head
core_orth=core_orth[core_orth$Count==17,] # Keep only Core orthologs
core_orth=core_orth[!(grepl('_R', core_orth$R.magnifica)),] # remove core RNA genes
nrow(core_orth) # # of non -pseudo core genes 
core_orth$gene_id<-core_orth$R.magnifica
core_orth= subset(core_orth, !(gene_id %in% recombining_genes)) # remove recombining genes

core_orth %>% nrow
core_orth$Core=1
core_orth=core_orth[,c('gene_id','Core')]
core_orth
core_orth[setdiff(colnames(core_sel),colnames(core_orth))[1:38]]<-0

seed %>% nrow
core_orth %>% nrow
core_orth %>% head
seed %>% head

core_orth=merge(core_orth,seed,by = 'gene_id',all.x=T) %>% unique(.)

core_orth %>% tail()
colnames(core_orth)
colnames(core_sel)
core_orth[,c("gene_id","Core","SEED_level_1","SEED_level_2","SEED_level_3" ,"SEED_long_name")] %>% colnames()

df = merge(core_orth[,c("gene_id","Core","SEED_level_1","SEED_level_2","SEED_level_3" ,"SEED_long_name")],core_sel,by=c('gene_id','SEED_level_1', 'SEED_level_2','SEED_level_3','SEED_long_name'),all=T)

df %>% colnames()

colnames(df)[9]='Sym'
colnames(df)[12]='FL'
df$allbranches=rowSums(df[,c('a','b','c')])
# colnames(df)[13]='allbranches'

df=df[,c('gene_id',"Core", "All", "All.groups", "Sym", "Gigas", "Ruthia", 
"FL", "allbranches", "a", "b", "c", "d", "e", "f", "g", "h", 
"i", "j", 'k','l','m','n', "R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo","R.fausta", "R.southwardae", "V.gigas1","V.gigas2", "V.okutanii",  "V.soyoae1", "V.soyoae2","V.marissinica","Bathy","SUP05", "SEED_level_1","SEED_level_2","SEED_level_3","SEED_long_name")]



# df[df>0]=1 # presence/absence of selection in that gene (not counting multiple episodes of diversifying selection within a particular group)

count_cols=c("Core", "All", "All.groups", "Sym", "Gigas", "Ruthia", 
"FL", "allbranches", "a", "b", "c", "d", "e", "f", "g", "h", 
"i", "j", 'k','l','m','n', "R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", "R.rectimargo","R.fausta", "R.southwardae", "V.gigas1","V.gigas2", "V.okutanii",  "V.soyoae1", "V.soyoae2","V.marissinica","Bathy","SUP05")

df[count_cols][is.na(df[count_cols])] <-0
df

df_events=df
df=df_events

df[count_cols][df[count_cols]>0] <-1 # reduce df to presence/absence

# df[!is.na(df$'SEED_level_1') & df[, 'SEED_level_1']=='',c('SEED_level_1')]<-
df %>% head()

# colSums(df[,c("Core", "All", "All.groups", "Sym", "Gigas", "Ruthia",
# "FL", "allbranches", "a", "b", "c", "d", "e", "f", "g", "h",
# "i", "j", 'k','l','m','n', "R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia",
# "R.rectimargo","R.fausta", "R.southwardae", "V.gigas1","V.gigas2", "V.okutanii",  "V.soyoae1", "V.soyoae2","V.marissinica","Bathy","SUP05")])

 
# colSums(df_events[,c("Core", "All", "All.groups", "Sym", "Gigas", "Ruthia", 
# "FL", "allbranches", "a", "b", "c", "d", "e", "f", "g", "h", 
# "i", "j", 'k','l','m','n', "R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
# "R.rectimargo","R.fausta", "R.southwardae", "V.gigas1","V.gigas2", "V.okutanii",  "V.soyoae1", "V.soyoae2","V.marissinica","Bathy","SUP05")])

print("Selection gene count summary")
summary=df[,c('gene_id',"Core", "All", "All.groups", "Gigas", "Ruthia", 
"FL", "allbranches", "a", "b", "c",'Bathy','SUP05')] %>% unique 
print(colSums(summary[,-1]))

print("Selection events count summary")
summary=df_events[,c("gene_id", "a", "b", "c", "d", "e", "f", "g", "h",
"i", "j", 'k','l','m','n', "R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia","R.rectimargo","R.fausta", "R.southwardae", "V.gigas1","V.gigas2", "V.okutanii",  "V.soyoae1", "V.soyoae2","V.marissinica","Bathy","SUP05")] %>% unique
print(colSums(summary[,-1]))

### Genes selected multiple times throughout the phylogeny
rep_sel=subset(df_events,All>1)
rep_sel$gene_id %>% unique %>% length
### Genes selected multiple times throughout the symbiont phylogeny
rep_sel=subset(df_events,Sym>1 )
rep_sel$gene_id %>% unique %>% length

df$SymMulti<-as.numeric(df$gene_id %in% (rep_sel$gene_id %>% unique))


get_non_empty_column <- function(x){
    if(is.numeric(x)){
        sum(x) >0
    } else {
        TRUE
    }
}

df =df[, sapply(df,  get_non_empty_column)]

 
```
## Run enrichment tests
### SEED_level_1
```{r run tests and format data frame for figure}
# #### 1) run hypergeometric montecarlo test ####
pval<-run_multihypergeometric_montecarlo_tests(df,'SEED_level_1')

#### 2) fisher test for over-representation of certain categories ####

dd = df[!duplicated(df[,c("gene_id",'Core',"SEED_level_1")]),]

dd$SEED_level_1 <- factor(dd$SEED_level_1, levels = levels(addNA(dd$SEED_level_1)), labels = c(levels(dd$SEED_level_1), 'NA'), exclude = NULL)


dd[,c(-1,-length(colnames(dd)))][is.na(dd[,c(-1,-length(colnames(dd)))])] <-0

d<-aggregate(.~SEED_level_1,dd,FUN = function(x) {sum(x > 0, na.rm = F)}, na.action = na.pass) 
colSums(d[,-1])

row.names(d)<-d$SEED_level_1
d=d[setdiff(colnames(d), c("gene_id",'SEED_level_2',"SEED_level_3","SEED_long_name"))]

colnames(d) %in% list(colSums(d[,-1])<2)
d=cbind('SEED_level_1'=d[,1],d[,2:ncol(d)][,colSums(d[,2:ncol(d)]) > 1])

breakdown<-run_fisher_tests2(d)
# bk<-plot_breakdown_with_counts2(breakdown,d)
# bk

## Comparing different test assessing if distribution of genes in branch is different from that in core

# pval2<-run_chisq_tests(d)
# pval3<-run_fisher_tests(d)
# cbind(pval,pval2,pval3)

#### Create dataframe for figure construction ####

core_lvl_order<-d[order(d$Core),]$SEED_level_1
core_lvl_order<-union(core_lvl_order[core_lvl_order!='NA'],'NA')


d$SEED_level_1 <- factor(d$SEED_level_1,levels = core_lvl_order)
# d$SEED_level_1 <- factor(d$SEED_level_1,levels = )

dat<-melt(d)
colnames(dat)[1]='categories'
colnames(dat)[2]='comp'
colnames(dat)[3]='count'
dat<-merge(breakdown,dat,by=c('categories','comp'),all=T)

dat$categories <- factor(dat$categories,levels =core_lvl_order)
dat$sign[dat$pvalue>=0.05] <- 'NS'
dat$symbol[is.na(dat$symbol)] <- ''
tot<-aggregate(count~comp,dat, function(x) sum(x))
colnames(tot)[2]='tot'
dat<-merge(dat,tot, by=c('comp'),all=T)
head(dat)
dput(levels(dat$comp))




dat$comp <- factor(dat$comp,levels = c( "SymMulti","All", "Sym",  "FL", "Ruthia","Gigas", 
 "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", 
"SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core"))

Group_labels=c("SymMulti","All", "Sym", "FL", "Clade II", "Clade I", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", "SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core")

names(Group_labels)=c( "All", "Sym",  "FL", "Ruthia","Gigas", 
 "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", 
"SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core")

dat
```
### SEED_level_2

```{r run tests and format data frame for figure}
# #### 1) run hypergeometric montecarlo test ####
pval<-run_multihypergeometric_montecarlo_tests(df,'SEED_level_2')

#### 2) fisher test for over-representation of certain categories ####

dd = df[!duplicated(df[,c("gene_id",'Core',"SEED_level_2")]),]

dd$SEED_level_2 <- factor(dd$SEED_level_2, levels = levels(addNA(dd$SEED_level_2)), labels = c(levels(dd$SEED_level_2), 'NA'), exclude = NULL)


dd[,c(-1,-length(colnames(dd)))][is.na(dd[,c(-1,-length(colnames(dd)))])] <-0

d<-aggregate(.~SEED_level_2,dd,FUN = function(x) {sum(x > 0, na.rm = F)}, na.action = na.pass) 
colSums(d[,-1])

row.names(d)<-d$SEED_level_2
d=d[setdiff(colnames(d), c("gene_id",'SEED_level_1',"SEED_level_3","SEED_long_name"))]

colnames(d) %in% list(colSums(d[,-1])<2)
d=cbind('SEED_level_2'=d[,1],d[,2:ncol(d)][,colSums(d[,2:ncol(d)]) > 1])

breakdown<-run_fisher_tests2(d)
bk<-plot_breakdown_with_counts2(breakdown,d)
bk

## Comparing different test assessing if distribution of genes in branch is different from that in core

# pval2<-run_chisq_tests(d)
# pval3<-run_fisher_tests(d)
# cbind(pval,pval2,pval3)

#### Create dataframe for figure construction ####

core_lvl_order<-d[order(d$Core),]$SEED_level_2
core_lvl_order<-union(core_lvl_order[core_lvl_order!='NA'],'NA')


d$SEED_level_2 <- factor(d$SEED_level_2,levels = core_lvl_order)
# d$SEED_level_2 <- factor(d$SEED_level_2,levels = )

dat<-melt(d)
colnames(dat)[1]='categories'
colnames(dat)[2]='comp'
colnames(dat)[3]='count'
dat<-merge(breakdown,dat,by=c('categories','comp'),all=T)

dat$categories <- factor(dat$categories,levels =core_lvl_order)
dat$sign[dat$pvalue>=0.05] <- 'NS'
dat$symbol[is.na(dat$symbol)] <- ''
tot<-aggregate(count~comp,dat, function(x) sum(x))
colnames(tot)[2]='tot'
dat<-merge(dat,tot, by=c('comp'),all=T)
head(dat)
dput(levels(dat$comp))




dat$comp <- factor(dat$comp,levels = c( "SymMulti","All", "Sym",  "FL", "Ruthia","Gigas", 
 "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", 
"SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core"))

Group_labels=c("SymMulti","All", "Sym", "FL", "Clade II", "Clade I", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", "SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core")

names(Group_labels)=c( "All", "Sym",  "FL", "Ruthia","Gigas", 
 "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", 'k','l','m','n',"Bathy", 
"SUP05","R.magnifica", "R.pacifica", "R.phaseoliformis", "R.pliocardia", 
"R.rectimargo", "R.fausta", "R.southwardae", "V.gigas1", "V.gigas2", 
"V.okutanii", "V.soyoae1", "V.soyoae2", "V.marissinica","All.groups","allbranches","Core")

dat
```
## Create figure
```{r  create Selection Figure: Core vs groups}

plot_Core<-ggplot(subset(dat,(comp %in% c('Core'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories),fill='grey20',stat='identity',color='black')+
  geom_bar(data=subset(dat,(comp %in% c('All.groups'))), aes(y=count,x=categories),fill='grey50',stat='identity',color='black')+
  scale_y_continuous(limits=c(0,122),
                     breaks=c(seq(0,50,25),seq(50,125,25)),
                     position='left',
                     sec.axis = sec_axis(~./subset(dat,(comp %in% c('Core')))$tot*100, name = 'Relative abundance (%)',breaks=c(seq(0,75,5))))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in core genome')+
  theme(axis.title.x = element_text(size=8)
        # axis.text.x.bottom = element_text(angle=50,vjust=1,hjust = 1)
        )
plot_Core

plot_multi<-ggplot(subset(dat,(comp %in% c('SymMulti'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('SymMulti')))$count)),
                     breaks=c(seq(0,10,2),seq(10,20,5)),
                     minor_breaks = c(seq(0,10,1),seq(10,20,2.5)),
                     position='left',
                     sec.axis = sec_axis(~./subset(dat,(comp %in% c('a')))$tot*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in a')+
  theme(axis.title.x = element_text(size=8))
plot_multi


plot_FL<-ggplot(subset(dat,(comp %in% c('FL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('FL')))$count)),
                     breaks=c(seq(0,max(subset(dat,(comp %in% c('FL')))$count),2)),
                     minor_breaks = c(seq(0,10,1)),
                     position='left',
                     sec.axis = sec_axis(~./subset(dat,(comp %in% c('FL')))$tot*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50','NA'='white'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in Free-living')+
  theme(axis.title.x = element_text(size=8))
plot_FL

plot_Ruthia<-ggplot(subset(dat,(comp %in% c('Ruthia'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('Ruthia')))$count)),
                     breaks=c(seq(0,20,2)),
                     position='left',
                     sec.axis = sec_axis(~./unique(subset(dat,(comp %in% c('Ruthia')))$tot)*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in Clade II')+
  theme(axis.title.x = element_text(size=8))
plot_Ruthia

plot_Gigas<-ggplot(subset(dat,(comp %in% c('Gigas'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('Gigas')))$count)),
                     breaks=c(seq(0,max(subset(dat,(comp %in% c('Gigas')))$count),1)),
                     position='left',
                     sec.axis = sec_axis(~./unique(subset(dat,(comp %in% c('Gigas')))$tot)*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in Clade I')+
  theme(axis.title.x = element_text(size=8))

plot_Gigas

#### here ####
plot_Sym<-ggplot(subset(dat,(comp %in% c('Sym'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('Sym')))$count)),
                     breaks=c(seq(0,10,1),seq(10,25,5)),
                     position='left',
                     sec.axis = sec_axis(~./unique(subset(dat,(comp %in% c('Sym')))$tot)*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in Sym')+
  theme(axis.title.x = element_text(size=8))
plot_Sym



plot1<-grid.arrange(arrangeGrob(plot_Core+theme(plot.margin=unit(c(1,0.5,0,0), "cm")),
                                plot_FL+theme(legend.position="none",
                                              axis.text.y = element_blank(),
                                              axis.title.y= element_blank(),
                                             plot.margin=unit(c(1,0,0,0), "cm")),
                                plot_Ruthia+theme(legend.position="none",
                                              axis.text.y = element_blank(),
                                              axis.title.y= element_blank(),
                                             plot.margin=unit(c(1,0,0,0), "cm")),
                                plot_Gigas+theme(legend.position="none",
                                              axis.text.y = element_blank(),
                                              axis.title.y= element_blank(),
                                             plot.margin=unit(c(1,0.5,0,0), "cm")),
                                nrow=1,widths=c(3.6,1,1,1)))
                    

plot1
# ggsave(file='plot_Selection_Core_vs_groups.svg',plot=plot1,width=40,units='cm',dpi=600,device='svg')
```

```{r  create Selection Figure: Core vs branches }

plot_Core<-ggplot(subset(dat,(comp %in% c('Core'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories),fill='grey20',stat='identity',color='black')+
  geom_bar(data=subset(dat,(comp %in% c('allbranches'))),aes(y=count,x=categories),fill='grey50',stat='identity',color='black')+
  scale_y_continuous(limits=c(0,122),
                     breaks=c(seq(0,125,25)),
                     minor_breaks = c(seq(0,50,10)),
                     position='left',
                     sec.axis = sec_axis(~./subset(dat,(comp %in% c('Core')))$tot*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in core genome')+
  theme(axis.title.x = element_text(size=8)
        # axis.text.x.bottom = element_text(angle=50,vjust=1,hjust = 1)
        )
plot_Core


plot_a<-ggplot(subset(dat,(comp %in% c('a'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('a')))$count)),
                     breaks=c(seq(0,10,2),seq(10,20,5)),
                     minor_breaks = c(seq(0,10,1),seq(10,20,2.5)),
                     position='left',
                     sec.axis = sec_axis(~./subset(dat,(comp %in% c('a')))$tot*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in a')+
  theme(axis.title.x = element_text(size=8))
plot_a

plot_b<-ggplot(subset(dat,(comp %in% c('b'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('b')))$count)),
                     breaks=c(seq(0,10,2),seq(10,25,5)),
                     position='left',
                     sec.axis = sec_axis(~./unique(subset(dat,(comp %in% c('b')))$tot)*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in b')+
  theme(axis.title.x = element_text(size=8))
plot_b

plot_c<-ggplot(subset(dat,(comp %in% c('c'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,max(subset(dat,(comp %in% c('c')))$count)),
                     breaks=c(seq(0,5,1)),
                     position='left',
                     sec.axis = sec_axis(~./unique(subset(dat,(comp %in% c('c')))$tot)*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  # scale_fill_gradient2(low = "blue", high = "red", mid = 'grey20',na.value= 'grey20',limits=c(-0.5, 0.5), oob=squish)+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in c')+
  theme(axis.title.x = element_text(size=8))
plot_c



plot2<-grid.arrange(arrangeGrob(plot_Core+theme(plot.margin=unit(c(0.5,0.5,1,0), "cm")),
                                arrangeGrob(
                                  plot_a+theme(legend.position="none",
                                               axis.text.y = element_blank(),
                                               axis.title.y= element_blank(),
                                               plot.margin=unit(c(0.5,0,1,0), "cm")),
                                  plot_b+theme(legend.position="none",
                                               axis.text.y = element_blank(),
                                               axis.title.y= element_blank(),
                                               plot.margin=unit(c(0.5,0,1,0), "cm")),
                                  plot_c+theme(legend.position="none",
                                               axis.text.y = element_blank(),
                                               axis.title.y= element_blank(),
                                               plot.margin=unit(c(0.5,0.5,1,0), "cm")),
                                  nrow=1,widths=c(1,1,1)),nrow=1,widths=c(1.2,0.99)))
                    

plot2
```

```{r mount Selection composite figure}
plot_Selection<-grid.arrange(plot1,plot2,nrow=2,ncol=1)
plot_Selection



subset(df,All==1 & !(is.na(SEED_level_1)))$gene_id %>% unique %>% length

subset(df,All==1 & !(is.na(SEED_level_1)))$gene_id %>% unique %>% length /sum(summary['All']) # of selected genes for which functional annotation was present
  
print(colSums(summary[,-1]))
```

```{r save figure}

# ggsave(file=paste0('plot_Selection',pval_thresh,'.svg'),plot=plot_Selection,width=8.53,height=8.36,units='in',dpi=600,device='svg')
filename=paste0('./figures/','plot_Selection_',pval_thresh,format(Sys.time(), "%F-%H%M"))
                
ggsave(file=paste0(filename,'.svg'),plot=plot_Selection,width=10.53,height=8.36,units='in',dpi=600,device='svg')

# p5<-ggdraw() +
#   draw_plot(plot_Selection)+
#   draw_image(magick::image_read_svg("./figures/plot_Selection_extralayer.svg"))
# p5
# 
# 
# img<-draw_image(magick::image_read_svg("./figures/plot_Selection_grouplayer.svg"))
# img<-png::readPNG('./figures/g12577.png') 
# ggdraw()+img
# library(ggpubr)
# 
# ggplot() +
#   background_image(img)
# 
# library('grImport2')
# install.packages('grConvert')
# 
# simple <- readPicture("./figures/plot_Selection_extralayer.svg")
# 
# library(grid)
# simpleGrob <- gTree(children=gList(pictureGrob(simple)))

# ggsave(file=paste0(filename,'.png'),plot=p5,width=10.53,height=8.36,units='in',dpi=600)

```

## Inspect enriched categories
```{r get genes for specific categories}
t1<-subset(df,SEED_level_1 %in% c('Nucleosides and Nucleotides') & b>0 ) %>% .[!duplicated(.[,c('gene_id')]),]
t1

t2<-subset(df,SEED_level_1 %in% c('Virulence, Disease and Defense') & c>0 ) %>% .[!duplicated(.[,c('gene_id')]),]
t2

t3<-subset(df_events,Sym>1) %>% .[!duplicated(.[,c('gene_id')]),]
t3

t3<-subset(df, Sym>0 ) %>% .[!duplicated(.[,c('gene_id','SEED_level_1','SEED_level_2')]),]
t3

t4<-subset(df,SEED_level_1 %in% c('Sulfur Metabolism') & All>0 ) %>% .[!duplicated(.[,c('gene_id')]),]
t4
```


Rmag_0156   
474 Rmag_0607    
579 Rmag_0807    
628 Rmag_0858  
631 Rmag_0862   
637 Rmag_0867  
641 Rmag_0870 
# Relaxation: RELAX

```{r import data, echo=FALSE}
# Import list of genes with significant evidence for recombinaiton according to the AICc test.

recombining_genes= read.table('core_genes_with_recomb_AICc.txt',header=F)$V1
recombining_genes


# Import SEED annotations for all genes

seed<-read.csv('./SEED_annotations_all.txt',header=T,sep='\t', na.strings = '',stringsAsFactors = T)
seed=seed[,c("locus_tag","SEED_level_1","SEED_level_2","SEED_level_3" ,"SEED_long_name")]
colnames(seed)[1]='gene_id'
seed = seed[!duplicated(seed),]

pval_thresh=0.05

relax=read.table('./relax_results.txt',header=T,sep='\t')
relax=data.table::as.data.table(relax)
relax=subset(relax,!(gene %in% recombining_genes)) # remove genes with evidence of recombination
nrow(relax)
core_relax=relax[,c(1,2)] %>% unique
nrow(core_relax)

core_relax$branches='Core'
core_relax$k=999
core_relax$pval=0

long=data.table::melt(relax,id.vars=c('gene','ncodons'),
            measure.vars = patterns("_k", "_pval"),
            variable.name = 'branches',
            value.name=c('k','pval'),
            variable.factor = T)
levels(long$branches)<-c("SymvsFL","RuvsFL","GivsFL","GivsRu")

long=rbind(core_relax,long)
colnames(long)[1]='gene_id'
df_rel=merge(long,seed,by = 'gene_id',all.x=T) %>% unique(.)
df_rel<-subset(df_rel, pval<pval_thresh)
```

## Relaxation of selection
```{r get data Relaxation}
## append all symb relax
new_rows<-subset(df_rel,branches != 'Core' & pval<0.05 & k<1) %>% .[!duplicated(.[,c('gene_id','SEED_level_1','SEED_level_2','SEED_level_3','SEED_long_name','k')]),]
new_rows$branches<-'SymbAll'
new_rows$k<-999
new_rows$pval<-0
nrow(new_rows)

df_relax = rbind(df_rel,new_rows)

d<-dcast(subset(df_relax,(!is.na(SEED_level_1) | is.na(SEED_long_name)) & pval<0.05 & (k<1 | k==999)),SEED_level_1~branches, value.var= 'gene_id', function(x) length(unique(x)))
d 
colSums(d[,-1])

print("Relaxation count summary")
summary=dcast(subset(df_relax,(pval<0.05 & (k<1 | k==999)))[,c('gene_id','branches')],gene_id~branches,value.var='gene_id',function(x) length(unique(x)) )
print(colSums(summary[,-1]))

df=dcast(subset(df_relax,(pval<0.05 & (k<1 | k==999)))[,c('gene_id','branches')],gene_id~branches,value.var='gene_id')
df=merge(df,)
```

```{r run hypergeometric tests and format data frame for figure}

d$SEED_level_1 <- factor(d$SEED_level_1, levels = levels(addNA(d$SEED_level_1)), labels = c(levels(d$SEED_level_1), 'NA'), exclude = NULL)

row.names(d)<-d$SEED_level_1
d=d[setdiff(colnames(d), c("gene_id",'SEED_level_2',"SEED_level_3","SEED_long_name"))]

#### 1) run hypergeometric test DEPRECATED Needs to be updated to include MonteCarlo test####
# pval<-run_multihypergeometric_tests(d)

#### 2) fisher test for overrepresentationof certain categories ####
breakdown<-run_fisher_tests2(d)
# bk<-plot_breakdown_with_counts2(breakdown,d)


#### Create dataframe for figure construction ####

core_lvl_order<-d[order(d$Core),]$SEED_level_1
core_lvl_order<-union(core_lvl_order[core_lvl_order!='NA'],'NA')


d$SEED_level_1 <- factor(d$SEED_level_1,levels = core_lvl_order)
# d$SEED_level_1 <- factor(d$SEED_level_1,levels = )

dat<-melt(d)
colnames(dat)[1]='categories'
colnames(dat)[2]='comp'
colnames(dat)[3]='count'
dat<-merge(breakdown,dat,by=c('categories','comp'),all=T)

dat$categories <- factor(dat$categories,levels =core_lvl_order)
dat$sign[dat$pvalue>=0.05] <- 'NS'
dat$symbol[is.na(dat$symbol)] <- ''
tot<-aggregate(count~comp,dat, function(x) sum(x))
colnames(tot)[2]='tot'
dat<-merge(dat,tot, by=c('comp'),all=T)
head(dat)
dput(levels(dat$comp))


dat$comp <- factor(dat$comp,levels = c( "Core", 'SymbAll','SymvsFL', 'RuvsFL','GivsFL', 'GivsRu'))

dat_relax<-dat
```

```{r plot relax}
thresh1=25

dat<-dat_relax
# tot_Core_relax<-subset(dat,(comp %in% c('Core')))$tot[1]
# plot_Core_relax<-ggplot(subset(dat,(comp %in% c('Core'))))+
#   theme_bw()+
#   geom_bar(aes(y=(count+count*0.5)/2,x=categories),fill='grey20',stat='identity',color='black')+
#   geom_bar(data=subset(dat,(comp %in% c('SymbAll'))), aes(y=count,x=categories),fill='grey50',stat='identity',color='black')+
#   # scale_x_discrete(position='top')+
#   scale_y_reverse(limits=c(),
#                   breaks=c(seq(0,50,10),seq(50,100,25)),
#                   position='left',
#                   sec.axis = sec_axis(~./tot_Core_relax*100, name = 'Relative abundance (%)',breaks=c(seq(0,10,2))))+
#   coord_flip()+
#   # scale_y_reverse()+
#   labs( x = "SEED categories", y='Gene count in core genome')+
#   theme(axis.title.x = element_text(size=8))
# plot_Core_relax

tot_SymvsFL_relax<-subset(dat,(comp %in% c('SymvsFL')))$tot[1]
plot_SymvsFL_relax<-ggplot(subset(dat,(comp %in% c('SymvsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_reverse(expand = c(0,0),
                  limits=c(thresh1/100*tot_SymvsFL_relax,0),
                     breaks=c(seq(0,thresh1/100*tot_SymvsFL_relax,20)),
                     minor_breaks = c(seq(0,thresh1/100*tot_SymvsFL_relax,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_SymvsFL_relax*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh1,10))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in SymvsFL')+
  theme(axis.title.x = element_text(size=8))
plot_SymvsFL_relax

tot_RuvsFL_relax<-subset(dat,(comp %in% c('RuvsFL')))$tot[1]
plot_RuvsFL_relax<-ggplot(subset(dat,(comp %in% c('RuvsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_reverse(limits=c(thresh1/100*tot_RuvsFL_relax,0),
                  expand=c(0,0),
                     # breaks=c(seq(0,10,2),seq(10,25,5)),
                     breaks=c(seq(0,thresh1/100*tot_RuvsFL_relax,20)),
                     minor_breaks = c(seq(0,thresh1/100*tot_SymvsFL_relax,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_RuvsFL_relax*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh1,10))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in RuvsFL')+
  theme(axis.title.x = element_text(size=8))
plot_RuvsFL_relax

tot_GivsFL_relax<-subset(dat,(comp %in% c('GivsFL')))$tot[1]
plot_GivsFL_relax<-ggplot(subset(dat,(comp %in% c('GivsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_reverse(limits=c(thresh1/100*tot_GivsFL_relax,0),
                  expand=c(0,0),
                     breaks=c(seq(0,thresh1/100*tot_GivsFL_relax,20)),
                     minor_breaks = c(seq(0,thresh1/100*tot_GivsFL_relax,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_GivsFL_relax*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh1,10))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in GivsFL')+
  theme(axis.title.x = element_text(size=8))
plot_GivsFL_relax

tot_GivsRu_relax<-subset(dat,(comp %in% c('GivsRu')))$tot[1]
plot_GivsRu_relax<-ggplot(subset(dat,(comp %in% c('GivsRu'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_reverse(limits=c(thresh1/100*tot_GivsRu_relax,0),
                  expand=c(0,0),
                     breaks=c(seq(0,thresh1/100*tot_GivsRu_relax,5)),
                     minor_breaks = c(seq(0,thresh1/100*tot_GivsRu_relax,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_GivsRu_relax*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh1,10))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey50'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in GivsRu')+
  theme(axis.title.x = element_text(size=8))
plot_GivsRu_relax

```

## Intensification of selection

```{r get data Intensification}
head(df_rel)
subset(df_rel, branches != 'Core' & pval<0.05 & k>1)$gene_id %>% unique

groups<-c('SymvsFL','RuvsFL','GivsFL')
new_rows<-subset(df_rel,branches != 'Core' & pval<0.05 & k>1) %>% .[!duplicated(.[,c('gene_id','SEED_level_1','SEED_level_2','SEED_level_3','SEED_long_name','k')]),]
new_rows$gene_id %>% unique
new_rows$branches<-'SymbAll'
new_rows$k<-999
new_rows$pval<-0
nrow(new_rows)

df_intens = rbind(df_rel,new_rows)


d<-dcast(subset(df_intens,(!is.na(SEED_level_1) | is.na(SEED_long_name)) & (k>1 | k==999)),SEED_level_1~branches, value.var= 'gene_id', function(x) length(unique(x)))
d 
colSums(d[,-1])

print("Intensification count summary")
summary=dcast(subset(df_intens,(pval<0.05 & (k>1 | k==999)))[,c('gene_id','branches')],gene_id~branches,value.var='gene_id',function(x) length(unique(x)) )
print(colSums(summary[,-1]))
```

```{r Run hypergeometric tests and format data frame for figure}


d$SEED_level_1 <- factor(d$SEED_level_1, levels = levels(addNA(d$SEED_level_1)), labels = c(levels(d$SEED_level_1), 'NA'), exclude = NULL)

row.names(d)<-d$SEED_level_1
d=d[setdiff(colnames(d), c("gene_id",'SEED_level_2',"SEED_level_3","SEED_long_name"))]

#### 1) run hypergeometric test DEPRECATED Needs to be updated to include MonteCarlo test####
# pval<-run_multihypergeometric_tests(d)

#### 2) fisher test for overrepresentationof certain categories ####
breakdown<-run_fisher_tests2(d)
# bk<-plot_breakdown_with_counts2(breakdown,d)


#### Create dataframe for figure construction ####

core_lvl_order<-d[order(d$Core),]$SEED_level_1
core_lvl_order<-union(core_lvl_order[core_lvl_order!='NA'],'NA')


d$SEED_level_1 <- factor(d$SEED_level_1,levels = core_lvl_order)
# d$SEED_level_1 <- factor(d$SEED_level_1,levels = )

dat<-melt(d)
colnames(dat)[1]='categories'
colnames(dat)[2]='comp'
colnames(dat)[3]='count'
dat<-merge(breakdown,dat,by=c('categories','comp'),all=T)

dat$categories <- factor(dat$categories,levels =core_lvl_order)
dat$sign[dat$pvalue>=0.05] <- 'NS'
dat$symbol[is.na(dat$symbol)] <- ''
tot<-aggregate(count~comp,dat, function(x) sum(x))
colnames(tot)[2]='tot'
dat<-merge(dat,tot, by=c('comp'),all=T)
head(dat)
dput(levels(dat$comp))


dat$comp <- factor(dat$comp,levels = c( "Core", 'SymbAll','SymvsFL', 'RuvsFL','GivsFL', 'GivsRu'))

dat_intens<-dat
```

```{r Plot intens}
dat<-dat_intens

merge(subset(dat_intens,comp=='SymvsFL'),subset(dat_relax,comp=='SymvsFL'),by=c('comp','categories'))

head(dat)

thresh2=50
# tot_Core_intens<-subset(dat,(comp %in% c('Core')))$tot[1]
# plot_Core_intens<-ggplot(subset(dat,(comp %in% c('Core'))))+
#   theme_bw()+
#   geom_bar(aes(y=(count-count*0.5)/2,x=categories),fill='grey20',stat='identity',color='black')+
#   geom_bar(data=subset(dat,(comp %in% c('SymbAll'))), aes(y=count,x=categories),fill='grey50',stat='identity',color='black')+
#   scale_y_continuous(limits=c(0,75),
#                   breaks=c(seq(0,50,10),seq(50,250,25)),
#                   position='left',
#                   sec.axis = sec_axis(~./tot_Core_intens*100, name = 'Relative abundance (%)',breaks=c(seq(0,10,2))))+
#   coord_flip()+
#   labs( x = "SEED categories", y='Gene count in core genome')+
#   theme(axis.title.x = element_text(size=8))
# plot_Core_intens

tot_SymvsFL_intens<-subset(dat,(comp %in% c('SymvsFL')))$tot[1]
plot_SymvsFL_intens<-ggplot(subset(dat,(comp %in% c('SymvsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,(thresh2+2)/100*tot_SymvsFL_intens),
                     expand = c(0,0),
                     breaks=c(seq(0,thresh2/100*tot_SymvsFL_intens,2)),
                     # minor_breaks = c(seq(0,10,1),seq(10,thresh2/100*tot_SymvsFL_intens,2.5)),
                     position='left',
                     sec.axis = sec_axis(~./tot_SymvsFL_intens*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh2,20))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey80'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in SymvsFL')+
  theme(axis.title.x = element_text(size=8))
plot_SymvsFL_intens

tot_RuvsFL_intens<-subset(dat,(comp %in% c('RuvsFL')))$tot[1]
plot_RuvsFL_intens<-ggplot(subset(dat,(comp %in% c('RuvsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,(thresh2+1)/100*tot_RuvsFL_intens),
                     expand=c(0,0),
                     breaks=c(seq(0,thresh2/100*tot_RuvsFL_intens,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_RuvsFL_intens*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh2,20))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey80'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in RuvsFL')+
  theme(axis.title.x = element_text(size=8))
plot_RuvsFL_intens


tot_GivsFL_intens<-subset(dat,(comp %in% c('GivsFL')))$tot[1]
plot_GivsFL_intens<-ggplot(subset(dat,(comp %in% c('GivsFL'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,(thresh2+1)/100*tot_GivsFL_intens),
                     expand=c(0,0),
                     breaks=c(seq(0,thresh2/100*tot_GivsFL_intens,2)),
                     position='left',
                     sec.axis = sec_axis(~./tot_GivsFL_intens*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh2,20))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey80'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in GivsFL')+
  theme(axis.title.x = element_text(size=8))
plot_GivsFL_intens

tot_GivsRu_intens<-subset(dat,(comp %in% c('GivsRu')))$tot[1]
plot_GivsRu_intens<-ggplot(subset(dat,(comp %in% c('GivsRu'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories,fill=sign),stat='identity',color='black')+
  scale_y_continuous(limits=c(0,(thresh2+1)/100*tot_GivsRu_intens),
                     expand=c(0,0),
                     breaks=c(seq(0,thresh2/100*tot_GivsRu_intens,10)),
                     position='left',
                     sec.axis = sec_axis(~./tot_GivsRu_intens*100, name = 'Relative abundance (%)',breaks=c(seq(0,thresh2,20))))+
  scale_fill_manual(values=c('depletion'='blue','enrichment'='red','NS'='grey80'))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in GivsRu')+
  theme(axis.title.x = element_text(size=8))
plot_GivsRu_intens
```


## Plot figure

```{r mount plots together}
# library('PtProcess')
library(ggplot2)
library(gtable)
library(grid)
library(gridExtra)

tot_Core<-subset(dat,(comp %in% c('Core')))$tot[1]
plot_Core<-ggplot(subset(dat_relax,(comp %in% c('Core'))))+
  theme_bw()+
  geom_bar(aes(y=count,x=categories),fill='grey20',stat='identity',color='black')+
  geom_bar(data=subset(dat_relax,(comp %in% c('SymbAll'))), aes(y=count,x=categories),fill='grey50',stat='identity',color='black')+
    geom_bar(data=subset(dat_intens,(comp %in% c('SymbAll'))), aes(y=count,x=categories),fill='grey80',stat='identity',color='black')+
  scale_y_continuous(limits=c(0,150),
                  breaks=c(seq(0,250,25)),
                  position='left',
                  sec.axis = sec_axis(~./tot_Core*100, name = 'Relative abundance (%)',breaks=c(seq(0,25,5))))+
  coord_flip()+
  labs( x = "SEED categories", y='Gene count in core genome')+
  theme(axis.title.x = element_text(size=8),
        plot.margin=unit(c(0,0.25,0,0), "cm"))
plot_Core

plot_SymvsFL_rela<-plot_SymvsFL_relax+theme(plot.margin=unit(c(0,0,0,0.25), "cm"),
                                            legend.position="none",
                                            axis.text.y = element_blank(),
                                            axis.title.y= element_blank())
plot_SymvsFL_inten<-plot_SymvsFL_intens+theme(legend.position="none",
                                         axis.text.y = element_blank(),
                                         axis.ticks.y = element_blank(),
                                         axis.title.y= element_blank(),
                                         plot.margin=unit(c(0,0.25,0,0), "cm"),
                                         panel.grid.minor.x = element_blank())

plot_RuvsFL_rela<-plot_RuvsFL_relax+theme(plot.margin=unit(c(0,0,0,0.25), "cm"),
                                            legend.position="none",
                                          axis.text.y = element_blank(),
                                          axis.title.y= element_blank())
plot_RuvsFL_inten<-plot_RuvsFL_intens+theme(legend.position="none",
                                         axis.text.y = element_blank(),
                                         axis.ticks.y = element_blank(),
                                         axis.title.y= element_blank(),
                                         plot.margin=unit(c(0,0.25,0,0), "cm"),
                                         panel.grid.minor.x = element_blank())

plot_GivsFL_rela<-plot_GivsFL_relax+theme(plot.margin=unit(c(0,0,0,0.25), "cm"),
                                          legend.position="none",
                                          axis.text.y = element_blank(),
                                          axis.title.y= element_blank())
plot_GivsFL_inten<-plot_GivsFL_intens+theme(legend.position="none",
                                         axis.text.y = element_blank(),
                                         axis.ticks.y = element_blank(),
                                         axis.title.y= element_blank(),
                                         plot.margin=unit(c(0,0.25,0,0), "cm"),
                                         panel.grid.minor.x = element_blank())

plot_GivsRu_rela<-plot_GivsRu_relax+theme(plot.margin=unit(c(0,0,0,0.25), "cm"),
                                          legend.position="none",
                                          axis.text.y = element_blank(),
                                          axis.title.y= element_blank())
plot_GivsRu_inten<-plot_GivsRu_intens+theme(legend.position="none",
                                         axis.text.y = element_blank(),
                                         axis.ticks.y = element_blank(),
                                         axis.title.y= element_blank(),
                                         plot.margin=unit(c(0,0.25,0,0), "cm"),
                                         panel.grid.minor.x = element_blank())


plot_RELAX<-grid.arrange(plot_Core,plot_SymvsFL_rela, plot_SymvsFL_inten, plot_RuvsFL_rela, plot_RuvsFL_inten,plot_GivsFL_rela, plot_GivsFL_inten,plot_GivsRu_rela,plot_GivsRu_inten,nrow=1,widths=c(6,1,1,1,1,1,1,1,1))
plot_RELAX


ggsave(file=paste0('./figures/plot_Relaxation_',pval_thresh,format(Sys.time(), "%F-%H%M"),'.svg'),plot=plot_RELAX,width=30.5,units='cm',dpi=600,device='svg')

```
## fetch data
```{r get genes for specific categories}
t1<-subset(df_rel,SEED_level_1 %in% c('Respiration') & branches %in% c('SymbvsFL','RuvsFL') & k>1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t1

t2<-subset(df_rel,SEED_level_1 %in% c('Sulfur Metabolism') & branches %in% c('SymbvsFL','GivsFL') & k>1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t2

t3<-subset(df_rel,SEED_level_1 %in% c('Protein Metabolism') & branches %in% c('SymbvsFL','GivsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t3

t4<-subset(df_rel,SEED_level_1 %in% c('Amino Acids and Derivatives') & branches %in% c('SymbvsFL','GivsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t4

t5<-subset(df_rel,SEED_level_1 %in% c('Carbohydrates') & branches %in% c('SymbvsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t5

t5<-subset(df_rel,SEED_level_1 %in% c('Nucleosides and Nucleotides') & branches %in% c('SymbvsFL','GivsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t5

t5<-subset(df_rel,SEED_level_1 %in% c('DNA Metabolism') & branches %in% c('SymbvsFL','GivsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t5

t5<-subset(df_rel,SEED_level_1 %in% c('Cell Division and Cell Cycle') & branches %in% c('SymbvsFL','GivsFL','RuvsFL') & k<1 & pval<0.05) %>% .[!duplicated(.[,c('gene_id')]),]
t5
```