---
title: "Codon_bias analyses"
output: html_document
editor_options: 
  chunk_output_type: console
---
#initi environment
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# setwd("~/Desktop/codon_usage/")
setwd("./")
setwd("~/Desktop/VesicSymb_Evolution/dowstream_analyses_and_data/")


library('ggplot2')
library('gridExtra')
library(RColorBrewer)
library(dplyr)
library(tidyr)
library(grid)
require(gtable)
library('scales')
# library(plyr)
# library(reshape2)
library(data.table)
```

# Import data
```{r}
recombining_genes=read.table('core_genes_with_recomb_AICc.txt')$V1
pseudogenes = ('') # to add pseudogene list
  
relax=read.table('./relax_results.txt',header=T,sep='\t')
relax=as.data.table(relax)
core_nopseudo=relax$gene
nrow(relax)
head(relax)

relax=subset(relax,!(gene %in% recombining_genes)) # remove genes with evidence of recombination
nrow(relax)
head(relax)

relax2<-data.table::melt(relax,id.vars=c('gene','ncodons'),
            measure.vars = patterns("_k", "_pval"),
            variable.name = 'comp',
            value.name=c('k','pval'),
            variable.factor = T)

levels(relax2$comp)<-c('SymvsFL','RuvsFL','GivsFL','GivsRu')


df_all=read.csv('codonbias_results.csv',sep='\t',row.names = 1,header=T,na.strings = '')
df_all$Group<-ifelse(df_all$Genome %in% c('Bathy','SUP05'),'FL',
                 ifelse(df_all$Genome %in% c('V.okutanii','V.soyoae1','V.soyoae2','V.diagonalis','V.extenta','V.gigas1','V.gigas2','V.marissinica'),'Gigas','Ruthia'))
colnames(df_all)

df=df_all[,-c(14:ncol(df_all)-3)]

head(df)

subset(df, !(coregene_id %in% recombining_genes ))$coregene_id %>% unique %>% length


```
# Plots

## Distribution of selection intensity parameter K from RELAX analysis
only the K parameter values with a p-value <0.05 are present in the table
```{r}
head(relax)

relax2$comp<-factor(relax2$comp, levels=rev(c("SymvsFL", "RuvsFL","GivsFL","GivsRu")))


Kdistrib<-ggplot(subset(relax2,comp %in% c('SymvsFL','GivsFL','RuvsFL','GivsRu') & pval<0.05))+
  theme_bw()+
  # geom_boxplot(aes(x=comp,y=k))+
  geom_violin(aes(x=comp,y=k),fill='grey90')+
  geom_jitter(aes(x=comp,y=k+0.001),width = 0.05,size=0.05)+
  scale_x_discrete(position='top',labels=c('SymvsFL'="Symbionts vs\nfree-living","RuvsFL"='Clade II vs\nfree-living',"GivsFL"='Clade I vs\nfree-living',"GivsRu"='Clade I vs\nClade II'))+
  geom_hline(yintercept = 1,color='red')+
  scale_y_log10(limits=c(0.001,50),breaks=c(0.1,seq(0,0.6,0.2),1,seq(5,10,5),seq(25,50,25)))+
  ylab('Selection coefficient (k)')+
  coord_flip()+
  theme(axis.title.y = element_blank(),
        axis.title.x = element_text(size=8),
        plot.margin=unit(c(0.90,0,0,0.25), "cm"))
Kdistrib
```

## Distribution of GC%
```{r distribution GC content}

# qplot(as.numeric(GC),data=df,geom='density',fill=Genome,alpha=0.05)+facet_grid(Genome~.)
# qplot(as.numeric(GC1),data=df,geom='density',fill=Genome,alpha=0.05)+facet_grid(Genome~.)
# qplot(as.numeric(GC2),data=df,geom='density',fill=Genome,alpha=0.05)+facet_grid(Genome~.)
# qplot(as.numeric(GC3),data=df,geom='density',fill=Genome,alpha=0.05)+facet_grid(Genome~.)

colors=list(Gi='#606E96',Ru='#68C47F',FL='#F6EFA1')
levels(df$Group) <- c("Free-living", "Gigas", "Ruthia")
group_names <- c(`FL` = "Free-living",
                    `Gigas` = "Clade I",
                    `Ruthia` = "Clade II")
colors_fill=c(Bathy = "#ffa800", SUP05 = "#ff006d", R.magnifica = "#287C8EFF", 
R.phaseoliformis = "#21908CFF", R.pacifica = "#20A486FF", R.rectimargo = "#35B779FF", 
R.pliocardia = "#5DC863FF", R.southwardae = "#8FD744FF", V.okutanii = "#440154FF", 
V.soyoae1 = "#331f70",V.soyoae2 = "#481F70FF", V.diagonalis = "#443A83FF", V.extenta = "#3B528BFF", 
V.gigas2 = "#31688EFF",V.gigas1 = "#308d8d",V.marissinica = "#33701f", R.fausta = "#68C47F")

df$Group<-factor(df$Group,levels = c('FL','Ruthia','Gigas'))
df$Genome<-factor(df$Genome,levels = c("Bathy", "SUP05", "R.magnifica", "R.fausta", "R.pacifica", 
"R.phaseoliformis", "R.pliocardia", "R.rectimargo", "R.southwardae", "V.diagonalis", "V.extenta", "V.gigas1", "V.gigas2", "V.marissinica", "V.soyoae1", "V.soyoae2","V.okutanii"))
dput(unique(df$Genome))

qplot(as.numeric(GC3),data=df,geom='density',fill=Genome,alpha=0.05)+facet_grid(Group~.)+scale_fill_manual(values=colors_fill)



ggplot(data=subset(df))+geom_hex(aes(x=GC3,y=GC1,fill=..count..),bins=30)+facet_wrap(~Genome)+xlim(c(0,0.6))+ylim(c(0,0.6))

```

## Distribution of CDCs

```{r}
colors_fill=c(Bathy = "#ffa800", SUP05 = "#ff006d", R.magnifica = "#287C8EFF", 
R.phaseoliformis = "#21908CFF", R.pacifica = "#20A486FF", R.rectimargo = "#35B779FF", 
R.pliocardia = "#5DC863FF", R.southwardae = "#8FD744FF", V.okutanii = "#440154FF", 
V.soyoae1 = "#331f70",V.soyoae2 = "#481F70FF", V.diagonalis = "#443A83FF", V.extenta = "#3B528BFF", 
V.gigas2 = "#31688EFF",V.gigas1 = "#308d8d",V.marissinica = "#33701f", R.fausta = "#68C47F")


df$Group<-factor(df$Group,levels = c('FL','Ruthia','Gigas'))

group_names <- c(`FL` = "Free-living",
                    `Gigas` = "Clade I",
                    `Ruthia` = "Clade II")
sup05cdc=subset(df,Genome=='SUP05' & P.CDC.<0.05 & ispseudo==0)$CDC
maxdensitysup05=density(sup05cdc)$x[which.max(density(sup05cdc)$y)]


CDC_spectra<-ggplot(data=subset(df,P.CDC.<0.05 & ispseudo==0) )+
  theme_bw()+
  geom_density(aes(x=CDC,color=Group,fill=Genome,y=..density..),color='black',alpha=0.3)+
  geom_vline(aes(xintercept = maxdensitysup05),linetype=2,color='grey60')+
  scale_x_continuous(limits=c(0,0.35),minor_breaks = seq(0.05, 0.25, by=0.01),breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3))+
  facet_grid(Group~.,labeller = as_labeller(group_names),switch = 'y')+
  scale_color_manual(values=c(colors$FL,colors$Ru,colors$Gi))+
  scale_fill_manual(values=colors_fill)+
  labs(x='Codon Deviation Coefficient (CDC)',y='Density')+
  theme(strip.background = element_rect(fill = 'white'),
    axis.title.x = element_text(),
    panel.grid.minor.y = element_blank(),
    strip.text.y = element_text(angle=-90)
    )

CDC_spectra

```

## Biplot: CDC in Symbiont as function of CSC in free-living

```{r}

bipl<-data.table::dcast(as.data.table(subset(df, coregene_id %in% core_nopseudo)), coregene_id ~ Group, value.var = 'CDC', fun=mean,subset=.(P.CDC. < 0.05 & coregene_id %in% core_nopseudo))

Bi_plot<-ggplot(bipl)+
  theme_bw()+
  geom_point(aes(x=FL,y=Gigas,color='2'),alpha=0.6,size=0.5)+
  geom_point(aes(x=FL,y=Ruthia,color='1',alpha=1-abs(1-(Ruthia/FL))),size=0.5)+
  geom_abline(slope=1,intercept = 0)+
  geom_smooth(aes(x=FL,y=Ruthia,color='1'),fill='grey90',method='lm',se=T,size=0.5)+
  geom_smooth(aes(x=FL,y=Gigas,color='2'),fill='grey90',method='lm',se=T,size=0.5)+
  scale_alpha_continuous(range = c(0.4, 0.6))+
  scale_y_continuous(limits=c(0.05,0.4))+
  scale_x_continuous(limits=c(0.05,0.36))+ expand_limits(x = 0)+
  xlab(label = 'CDC in Free-Living')+ylab(label= 'CDC in symbionts')+
  scale_color_manual(values=c(colors$Ru,colors$Gi),labels=c('Clade II','Clade I'),name='')+
  theme(
    legend.text=element_text(size=8))
Bi_plot

#### Arrange plots ###
p3<-grid.arrange(CDC_spectra + theme(legend.position="none"), Bi_plot+ theme(legend.position="none"), nrow=1, widths = unit(c(10,9), "cm"), heights = unit(rep(10, 1), "cm"))
p3

p4<-grid.arrange(p3,Kdistrib,nrow=2,heights = unit(c(10,7), "cm"), widths = unit(c(15), "cm"))
p4
ggsave(file="plot_CDCk.svg", plot=p4,width = 20,height = 20,units = 'cm')


```

## PCA with RSCU

```{R}

df2=cbind(df_all[,grepl( "RSCU" , names( df_all ) ) ],df_all[,c('Genome','ncodons','Group','ispseudo','coregene_id','locus_tag','ortho_id')])
df_all%>% colnames
df2=subset(df2,ispseudo==0 & !(is.na(coregene_id)))

mat<-df2[,c(1:64)][ , which(apply(df2[,c(1:64)], 2, var) != 0)] # remove unvariable columns (RSCU Codons)
ncol(mat)

df2.pca=prcomp(mat, center = T,scale. = T)
screeplot(df2.pca)
as.data.frame(df2.pca$sdev/sum(df2.pca$sdev))[1,]
as.data.frame(df2.pca$sdev/sum(df2.pca$sdev))[2,]

# biplot(df2.pca)

df_out <- cbind(as.data.frame(df2.pca$x),df2[,65:ncol(df2)])
head(df_out)


ggplot(subset(df_out,!(is.na(coregene_id) & ispseudo==0)))+
  geom_density2d(aes(x=PC1,y=PC2,color=Group))


plot<-ggplot(subset(df_out,!(is.na(coregene_id) & ispseudo==0)))+
  geom_point(aes(x=PC1,y=PC2,color=Group),alpha=0.5)

ggplot(subset(df_out,!(is.na(coregene_id) & ispseudo==0)))+
  geom_point(aes(x=coregene_id,y=PC1,color=Group),alpha=0.5)


datapc <- data.frame(varnames=rownames(df2.pca$rotation), df2.pca$rotation)
data <- data.frame(obsnames=row.names(df2.pca$x), df2.pca$x)
mult <- min(
    (max(data[,'PC2']) - min(data[,'PC2'])/(max(datapc[,'PC2'])-min(datapc[,'PC2']))),
    (max(data[,'PC1']) - min(data[,'PC1'])/(max(datapc[,'PC1'])-min(datapc[,'PC1'])))
    )
datapc <- transform(datapc,
        v1 = .7 * mult * (get('PC1')),
        v2 = .7 * mult * (get('PC2'))
        )
plot <- plot + coord_equal() + geom_text(data=datapc, aes(x=v1, y=v2, label=varnames), size = 5, vjust=1, color="red")
plot <- plot + geom_segment(data=datapc, aes(x=0, y=0, xend=v1, yend=v2), arrow=arrow(length=unit(0.2,"cm")), alpha=0.75, color="red")
plot
    
    

#################
df3<-data.table::melt(as.data.table(df2),id.vars=names(df2)[65:ncol(df2)],
            measure.vars = patterns("RSCU"),
            variable.name = 'Codon',
            value.name=c('RSCU'),
            variable.factor = T)

df3 %>% head

test<-as.data.frame(dcast(df3,formula = Codon~Genome,fun.aggregate = mean))
rownames(test)<-test$Codon

test.clust=hclust(dist(test[,-1]))
plot(test.clust)
t<-as.data.frame(cbind(test.clust$labels,as.numeric(test.clust$order))) %>% .[order(as.numeric(.$V2)),]
t$V1
str(test.clust)

df3$Codon<-factor(df3$Codon, levels=t$V1)
ggplot(subset(df3,ncodons>300 & Codon %in% c('RSCU.GCC.','RSCU.GTA.','RSCU.AGA.','RSCU.ACG.')))+
  geom_raster(aes(x=Genome,y=Codon,fill=log10(RSCU)),stat='identity')+
  scale_fill_gradient2(low='blue',mid='white',high='red',midpoint = 0,na.value = 'white')

ggplot(subset(df3,ncodons>300))+
  geom_raster(aes(x=Genome,y=Codon,fill=log10(RSCU)),stat='identity')+
  scale_fill_gradient2(low='blue',mid='white',high='red',midpoint = 0,na.value = 'white')

 

ggplot(subset(df3, !(is.na(coregene_id))))+
  geom_bin2d(aes(x=Group,y=Codon,fill=RSCU),stat='identity')

##########3
library(vegan)
# df.rda = rda(df2[,c(1:64)], df2[,c(65:ncol(df2))])
tree_pca$CA$eig
```



```{r}

CDC_codons<-ggplot(data=subset(df,P.CDC.<0.05))+
  theme_bw()+
  geom_density_2d(aes(x=CDC,y=ncodons,color=as.factor(ispseudo)))+
  facet_grid(Group~.,labeller = as_labeller(group_names),switch = 'y')+
  theme(strip.background = element_rect(fill = 'white'),
    axis.title.x = element_text(),
    panel.grid.minor.y = element_blank(),
    strip.text.y = element_text(angle=-90)
    )

CDC_codons

CDC_spectra<-ggplot(data=subset(df, P.CDC.<0.05 & ncodons>300))+
  theme_bw()+
  geom_violin(aes(x=CDC,color=Group,fill=Genome,y=Genome),color='black',alpha=0.3)+
  geom_vline(aes(xintercept = maxdensitysup05))+
  facet_grid(Group~factor(ispseudo),labeller = as_labeller(group_names),switch = 'y',scales='free')+
  scale_color_manual(values=c(colors$FL,colors$Ru,colors$Gi))+
  scale_fill_manual(values=colors_fill)+
  labs(x='Codon Deviation Coefficient (CDC)',y='Density')+
  theme(strip.background = element_rect(fill = 'white'),
    axis.title.x = element_text(),
    panel.grid.minor.y = element_blank(),
    strip.text.y = element_text(angle=-90)
    )

CDC_spectra


CDC_spectra<-ggplot(data=subset(df,P.CDC.<0.05 & ispseudo==0 & ncodons>300))+
  theme_bw()+
  geom_density(aes(x=CDC,color=Group,fill=Genome,y=..density..),color='black',alpha=0.3)+
  geom_vline(aes(xintercept = maxdensitysup05))+
  scale_x_continuous(limits=c(0,0.35),minor_breaks = seq(0.05, 0.25, by=0.01),breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3))+
  facet_grid(Group~.,labeller = as_labeller(group_names),switch = 'y')+
  scale_color_manual(values=c(colors$FL,colors$Ru,colors$Gi))+
  scale_fill_manual(values=colors_fill)+
  labs(x='Codon Deviation Coefficient (CDC)',y='Density')+
  theme(strip.background = element_rect(fill = 'white'),
    axis.title.x = element_text(),
    panel.grid.minor.y = element_blank(),
    strip.text.y = element_text(angle=-90)
    )

CDC_spectra
```


## Pseudo vs non pseudo

```{R}
df


ggplot(subset(df,ortho_id %in% intersect(df[df$ispseudo==1,]$ortho_id,df[df$ispseudo==0,]$ortho_id)))+
  geom_point(alpha=0.8,size=1,aes(x=ortho_id,y=CDC,color=factor(ispseudo)))+facet_grid(.~factor(Group))

ggplot(subset(df,ortho_id %in% intersect(df[df$ispseudo==1,]$ortho_id,df[df$ispseudo==0,]$ortho_id)))+
  geom_violin(alpha=0.8,size=1,aes(y=CDC,x=ispseudo,color=factor(ispseudo)))+facet_grid(.~factor(Group))

ggplot(subset(df,ortho_id %in% intersect(df[df$ispseudo==1,]$ortho_id,df[df$ispseudo==0,]$ortho_id)))+
  geom_density(alpha=0.5,aes(x=CDC,fill=factor(ispseudo)))+facet_grid(Group~.)

```
## Heatmaps

```{r}
#### heatmap ####
library("Hmisc")
library("Cairo")

bipl2<-dcast(df, coregene_id ~ Genome, value.var = 'CDC', fun=mean,subset=.(P.CDC. < 0.05))


res2 <- rcorr(as.matrix(bipl2[,c(-1)]),type='pearson')

res<-as.data.frame((res2$r)^2)
hm<-heatmap((res2$r^2),symm = TRUE,sideColors=c("darkgreen", "yellowgreen"), col=terrain.colors(12),legend=col)
hm

# svg(hm,filename='heatmap_pearsons_rsq_corr_cdc.svg',width=40, height=40)

library(gplots)
hm<-heatmap.2(as.matrix(res2$r^2), margins=c(5,8),col=(heat.colors(17)),
        key.xlab="r^2",
        key=TRUE, symkey=FALSE, density.info="none", trace="none")

hm

# pdf(hm,file='heatmap_pearsons_rsq_corr_cdc.pdf',width=40, height=40)
```


# Test correlation of CDC for genome pairs
Is there a good correlation of CDC values across pairs? Do genes that have a relatively small CDC values in one genome also have a relatively small CDC value in the other?
```{r}
#### For species pair ####
combi=combn(c(2:18), 2)
library("ggpubr")

for(i in c(1:2)){
  shap<-shapiro.test(bipl2[,combi[,i][1]])
  ggqqplot(bipl2[,combi[,i][1]])
  print(shap$p.value)
}

## genome pairs for which CDC values are not correlated across genomes
## pearson
for(i in c(1:136)){
cortest<-cor.test(bipl2[,combi[,i][1]],bipl2[,combi[,i][2]], method='pearson',use = "complete.obs")
if(cortest$p.value>0.01){print(colnames(bipl2)[c(combi[,i][1],combi[,i][2])])
  print(cortest$p.value)}}

## kendall
for(i in c(1:136)){
cortest<-cor.test(bipl2[,combi[,i][1]],bipl2[,combi[,i][2]], method='kendall', continuity = T)
if(cortest$p.value>0.01){print(colnames(bipl2)[c(combi[,i][1],combi[,i][2])])
  print(cortest$p.value)}}

## spearman
for(i in c(1:136)){
cortest<-cor.test(bipl2[,combi[,i][1]],bipl2[,combi[,i][2]], method='spearman', continuity = T)
if(cortest$p.value>0.01){print(colnames(bipl2)[c(combi[,i][1],combi[,i][2])])
  print(cortest$p.value)}}


#### For lineage pair ####
bipl
combi=combn(c(2:4), 2)

for(i in c(1:3)){
cortest<-cor.test(bipl[,combi[,i][1]],bipl[,combi[,i][2]], method='pearson')
if(cortest$p.value>0.0001){print(colnames(bipl)[c(combi[,i][1],combi[,i][2])])
  print(cortest$p.value)}}

for(i in c(1:3)){
cortest<-cor.test(bipl[,combi[,i][1]],bipl[,combi[,i][2]], method='kendall', continuity=T)
if(cortest$p.value>0.0001){print(colnames(bipl)[c(combi[,i][1],combi[,i][2])])
  print(cortest$p.value)}}

```
